{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "79b46d6a-5972-4afa-a21d-a2195611500d",
   "metadata": {},
   "source": [
    "# SCENIC+"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e74f852-c69e-4265-a760-0791dc7dc963",
   "metadata": {},
   "source": [
    "# Database\n",
    "\n",
    "All the steps were done in command lne, but I copy them here for posterity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7edd61df-49e1-4e1e-8fd4-b9d19a0b1fc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "git clone https://github.com/aertslab/create_cisTarget_databases"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7aaa71e3-0614-4f4a-9ff7-8b0af3a3dfe1",
   "metadata": {},
   "source": [
    "## Cluster buster download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3586709d-f691-47ff-80a4-0f2207639e1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "wget https://resources.aertslab.org/cistarget/programs/cbust\n",
    "chmod a+x cbust"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c38ca57d-52de-4d28-8be0-0e419afa53f1",
   "metadata": {},
   "source": [
    "## Motif collection download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40f0ecef-bbba-4764-82ea-64e7752cd774",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p aertslab_motif_colleciton\n",
    "wget -O aertslab_motif_colleciton/v10nr_clust_public.zip https://resources.aertslab.org/cistarget/motif_collections/v10nr_clust_public/v10nr_clust_public.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "066f5f33-c70b-4f18-b686-974d31942fb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd aertslab_motif_colleciton; unzip -q v10nr_clust_public.zip\n",
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "9087ce69-258f-4991-a303-6d18086e1595",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "motifs-v10-nr.chicken-m0.00001-o0.0.tbl  motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "motifs-v10-nr.flybase-m0.00001-o0.0.tbl  motifs-v10-nr.mgi-m0.00001-o0.0.tbl\n"
     ]
    }
   ],
   "source": [
    "!ls aertslab_motif_colleciton/v10nr_clust_public/snapshots/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "e2afd35e-fcef-4c1d-8d78-d09b9356f00f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "total 42412\n",
      "-rw-r--r-- 1 vsc33857 nogroup   163 Jan 27  2022 bergman__Adf1.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup    75 Jan 27  2022 bergman__Aef1.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup    75 Jan 27  2022 bergman__Hr46.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup   113 Jan 27  2022 bergman__Kr.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup    86 Jan 27  2022 bergman__Su_H_.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup    83 Jan 27  2022 bergman__TFAM.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup    77 Jan 27  2022 bergman__ap.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup   398 Jan 27  2022 bergman__bcd.cb\n",
      "-rw-r--r-- 1 vsc33857 nogroup    84 Jan 27  2022 bergman__bin.cb\n",
      "ls: write error: Broken pipe\n"
     ]
    }
   ],
   "source": [
    "!ls -l aertslab_motif_colleciton/v10nr_clust_public/singletons | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "161b2e80-2236-4339-b37e-e910a0dacf68",
   "metadata": {},
   "source": [
    "## Fasta from consensus regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912f3c88-f27c-4b07-8f70-dde5fc773cb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load BEDTools/2.31.1\n",
    "\n",
    "REGION_BED=\"../atac/outs/consensus_peak_calling/consensus_regions.bed\"\n",
    "GENOME_FASTA=\"/genomes/homo_sapiens/hg38_ucsc/fasta/hg38.fa\"\n",
    "CHROMSIZES=\"/genomes/homo_sapiens/hg38_ucsc/fasta/hg38.chrom.sizes\"\n",
    "DATABASE_PREFIX=\"10x_ONT_BALL\"\n",
    "SCRIPT_DIR=\"/grn/create_cisTarget_databases\"\n",
    "\n",
    "${SCRIPT_DIR}/create_fasta_with_padded_bg_from_bed.sh \\\n",
    "        ${GENOME_FASTA} \\\n",
    "        ${CHROMSIZES} \\\n",
    "        ${REGION_BED} \\\n",
    "        hg38.BALL.with_1kb_bg_padding.fa \\\n",
    "        1000 \\\n",
    "        yes\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0b7dd7c-aa22-4e1c-b37b-e07f50dcee22",
   "metadata": {},
   "source": [
    "## Create cistarget database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "8ac5ea09-e930-4f62-80ad-1e7922f0a1a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls aertslab_motif_colleciton/v10nr_clust_public/singletons > motifs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "720877cf",
   "metadata": {},
   "outputs": [],
   "source": [
    "OUT_DIR=\"/grn/\"\n",
    "CBDIR=\"${OUT_DIR}/aertslab_motif_colleciton/v10nr_clust_public/singletons\"\n",
    "FASTA_FILE=\"${OUT_DIR}/hg38.BALL.with_1kb_bg_padding.fa\"\n",
    "MOTIF_LIST=\"${OUT_DIR}/motifs.txt\"\n",
    "DATABASE_PREFIX=\"10x_ONT_BALL\"\n",
    "\n",
    "\"${OUT_DIR}/create_cisTarget_databases/create_cistarget_motif_databases.py\" \\\n",
    "    -f ${FASTA_FILE} \\\n",
    "    -M ${CBDIR} \\\n",
    "    -m ${MOTIF_LIST} \\\n",
    "    -o ${OUT_DIR}/${DATABASE_PREFIX} \\\n",
    "    --bgpadding 1000 \\\n",
    "    -t 20"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "093a0b3e-fae8-49a1-b80b-16ea0e87fef5",
   "metadata": {},
   "source": [
    "# RNA preprocessing\n",
    "\n",
    "Save rna counts in adata.raw and change RNA index barcode to ATAC one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e44f8610-986b-46d2-a7e9-4800c71e2a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fe07a7c-f455-4ef4-91a5-cbb81d612ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna = sc.read_h5ad(\"/path/to/rna/concatenated_adata_log_pca_celltypist_23012025.h5ad\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "397d69ed-b4db-417d-a448-34eb9ab227b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Layers with keys: counts, log_counts"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.layers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1d6a2b44-a39b-4991-a815-8182c30f671c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>log1p_n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>log1p_total_counts</th>\n",
       "      <th>pct_counts_in_top_20_genes</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "      <th>pct_counts_ribo</th>\n",
       "      <th>pct_counts_hb</th>\n",
       "      <th>outlier</th>\n",
       "      <th>...</th>\n",
       "      <th>leiden_1.25</th>\n",
       "      <th>leiden_1.5</th>\n",
       "      <th>leiden_2.0</th>\n",
       "      <th>leiden_5.0</th>\n",
       "      <th>leiden_10.0</th>\n",
       "      <th>leiden_20.0</th>\n",
       "      <th>celltypist_cell_label_coarse</th>\n",
       "      <th>celltypist_conf_score_coarse</th>\n",
       "      <th>celltypist_cell_label_fine</th>\n",
       "      <th>celltypist_conf_score_fine</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACAGCCAGTTTGTG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1891</td>\n",
       "      <td>7.545390</td>\n",
       "      <td>7641.0</td>\n",
       "      <td>8.941415</td>\n",
       "      <td>21.823060</td>\n",
       "      <td>13.584609</td>\n",
       "      <td>7.642979</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>39</td>\n",
       "      <td>294</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.978114</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.950723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACAGCCATGTTTGG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>2319</td>\n",
       "      <td>7.749322</td>\n",
       "      <td>10986.0</td>\n",
       "      <td>9.304468</td>\n",
       "      <td>21.354451</td>\n",
       "      <td>13.899508</td>\n",
       "      <td>9.111597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>73</td>\n",
       "      <td>51</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.603815</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.370262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACAGCCATTTAAGC-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1667</td>\n",
       "      <td>7.419381</td>\n",
       "      <td>6636.0</td>\n",
       "      <td>8.800415</td>\n",
       "      <td>21.503918</td>\n",
       "      <td>13.803495</td>\n",
       "      <td>6.268836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>29</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.951335</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.903238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACATGCATTGGGAG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1599</td>\n",
       "      <td>7.377759</td>\n",
       "      <td>6888.0</td>\n",
       "      <td>8.837681</td>\n",
       "      <td>24.811266</td>\n",
       "      <td>18.626598</td>\n",
       "      <td>5.894309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "      <td>69</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.121671</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.183371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCAACAATAACCT-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1403</td>\n",
       "      <td>7.247081</td>\n",
       "      <td>5495.0</td>\n",
       "      <td>8.611775</td>\n",
       "      <td>24.913558</td>\n",
       "      <td>18.817106</td>\n",
       "      <td>5.150136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>3</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.991872</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.996793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGAAGCGAGTAA-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1361</td>\n",
       "      <td>7.216709</td>\n",
       "      <td>5059.0</td>\n",
       "      <td>8.529121</td>\n",
       "      <td>23.739870</td>\n",
       "      <td>17.019173</td>\n",
       "      <td>8.381103</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>30</td>\n",
       "      <td>16</td>\n",
       "      <td>54</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.403587</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.973642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGAAGCTTTGGG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>2369</td>\n",
       "      <td>7.770645</td>\n",
       "      <td>8738.5</td>\n",
       "      <td>9.075608</td>\n",
       "      <td>15.860846</td>\n",
       "      <td>9.990273</td>\n",
       "      <td>4.691881</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>91</td>\n",
       "      <td>B cells</td>\n",
       "      <td>0.776546</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.532727</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGGCAGCCTGCA-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1536</td>\n",
       "      <td>7.337588</td>\n",
       "      <td>6272.0</td>\n",
       "      <td>8.744010</td>\n",
       "      <td>29.432398</td>\n",
       "      <td>23.852041</td>\n",
       "      <td>6.505102</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>91</td>\n",
       "      <td>B cells</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.177973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTAAACAAG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1279</td>\n",
       "      <td>7.154615</td>\n",
       "      <td>5301.0</td>\n",
       "      <td>8.575839</td>\n",
       "      <td>35.653650</td>\n",
       "      <td>28.730429</td>\n",
       "      <td>2.886248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>23</td>\n",
       "      <td>57</td>\n",
       "      <td>20</td>\n",
       "      <td>Monocytes</td>\n",
       "      <td>0.745200</td>\n",
       "      <td>Non-classical monocytes</td>\n",
       "      <td>0.041574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTTAAGCTG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1756</td>\n",
       "      <td>7.471363</td>\n",
       "      <td>7045.0</td>\n",
       "      <td>8.860215</td>\n",
       "      <td>24.442867</td>\n",
       "      <td>18.424414</td>\n",
       "      <td>6.018453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>36</td>\n",
       "      <td>124</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.122617</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.947241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11264 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample  n_genes_by_counts  log1p_n_genes_by_counts  \\\n",
       "AAACAGCCAGTTTGTG-D0     D0               1891                 7.545390   \n",
       "AAACAGCCATGTTTGG-D0     D0               2319                 7.749322   \n",
       "AAACAGCCATTTAAGC-D0     D0               1667                 7.419381   \n",
       "AAACATGCATTGGGAG-D0     D0               1599                 7.377759   \n",
       "AAACCAACAATAACCT-D0     D0               1403                 7.247081   \n",
       "...                    ...                ...                      ...   \n",
       "TTTGTGAAGCGAGTAA-Q3     Q3               1361                 7.216709   \n",
       "TTTGTGAAGCTTTGGG-Q3     Q3               2369                 7.770645   \n",
       "TTTGTGGCAGCCTGCA-Q3     Q3               1536                 7.337588   \n",
       "TTTGTTGGTAAACAAG-Q3     Q3               1279                 7.154615   \n",
       "TTTGTTGGTTAAGCTG-Q3     Q3               1756                 7.471363   \n",
       "\n",
       "                     total_counts  log1p_total_counts  \\\n",
       "AAACAGCCAGTTTGTG-D0        7641.0            8.941415   \n",
       "AAACAGCCATGTTTGG-D0       10986.0            9.304468   \n",
       "AAACAGCCATTTAAGC-D0        6636.0            8.800415   \n",
       "AAACATGCATTGGGAG-D0        6888.0            8.837681   \n",
       "AAACCAACAATAACCT-D0        5495.0            8.611775   \n",
       "...                           ...                 ...   \n",
       "TTTGTGAAGCGAGTAA-Q3        5059.0            8.529121   \n",
       "TTTGTGAAGCTTTGGG-Q3        8738.5            9.075608   \n",
       "TTTGTGGCAGCCTGCA-Q3        6272.0            8.744010   \n",
       "TTTGTTGGTAAACAAG-Q3        5301.0            8.575839   \n",
       "TTTGTTGGTTAAGCTG-Q3        7045.0            8.860215   \n",
       "\n",
       "                     pct_counts_in_top_20_genes  pct_counts_mt  \\\n",
       "AAACAGCCAGTTTGTG-D0                   21.823060      13.584609   \n",
       "AAACAGCCATGTTTGG-D0                   21.354451      13.899508   \n",
       "AAACAGCCATTTAAGC-D0                   21.503918      13.803495   \n",
       "AAACATGCATTGGGAG-D0                   24.811266      18.626598   \n",
       "AAACCAACAATAACCT-D0                   24.913558      18.817106   \n",
       "...                                         ...            ...   \n",
       "TTTGTGAAGCGAGTAA-Q3                   23.739870      17.019173   \n",
       "TTTGTGAAGCTTTGGG-Q3                   15.860846       9.990273   \n",
       "TTTGTGGCAGCCTGCA-Q3                   29.432398      23.852041   \n",
       "TTTGTTGGTAAACAAG-Q3                   35.653650      28.730429   \n",
       "TTTGTTGGTTAAGCTG-Q3                   24.442867      18.424414   \n",
       "\n",
       "                     pct_counts_ribo  pct_counts_hb  outlier  ...  \\\n",
       "AAACAGCCAGTTTGTG-D0         7.642979            0.0    False  ...   \n",
       "AAACAGCCATGTTTGG-D0         9.111597            0.0    False  ...   \n",
       "AAACAGCCATTTAAGC-D0         6.268836            0.0    False  ...   \n",
       "AAACATGCATTGGGAG-D0         5.894309            0.0    False  ...   \n",
       "AAACCAACAATAACCT-D0         5.150136            0.0    False  ...   \n",
       "...                              ...            ...      ...  ...   \n",
       "TTTGTGAAGCGAGTAA-Q3         8.381103            0.0    False  ...   \n",
       "TTTGTGAAGCTTTGGG-Q3         4.691881            0.0    False  ...   \n",
       "TTTGTGGCAGCCTGCA-Q3         6.505102            0.0    False  ...   \n",
       "TTTGTTGGTAAACAAG-Q3         2.886248            0.0    False  ...   \n",
       "TTTGTTGGTTAAGCTG-Q3         6.018453            0.0    False  ...   \n",
       "\n",
       "                     leiden_1.25  leiden_1.5  leiden_2.0  leiden_5.0  \\\n",
       "AAACAGCCAGTTTGTG-D0            3           4           2           7   \n",
       "AAACAGCCATGTTTGG-D0            7           7           8           6   \n",
       "AAACAGCCATTTAAGC-D0            3           4           2           2   \n",
       "AAACATGCATTGGGAG-D0            3           4           2           7   \n",
       "AAACCAACAATAACCT-D0            3           4           2           2   \n",
       "...                          ...         ...         ...         ...   \n",
       "TTTGTGAAGCGAGTAA-Q3            1           1          11          30   \n",
       "TTTGTGAAGCTTTGGG-Q3            1           1           0           0   \n",
       "TTTGTGGCAGCCTGCA-Q3            1           1           0           0   \n",
       "TTTGTTGGTAAACAAG-Q3           11          11          12          23   \n",
       "TTTGTTGGTTAAGCTG-Q3            1           1           0           3   \n",
       "\n",
       "                    leiden_10.0 leiden_20.0 celltypist_cell_label_coarse  \\\n",
       "AAACAGCCAGTTTGTG-D0          39         294               B-cell lineage   \n",
       "AAACAGCCATGTTTGG-D0          73          51               B-cell lineage   \n",
       "AAACAGCCATTTAAGC-D0          24          29               B-cell lineage   \n",
       "AAACATGCATTGGGAG-D0          31          69               B-cell lineage   \n",
       "AAACCAACAATAACCT-D0           4           9               B-cell lineage   \n",
       "...                         ...         ...                          ...   \n",
       "TTTGTGAAGCGAGTAA-Q3          16          54               B-cell lineage   \n",
       "TTTGTGAAGCTTTGGG-Q3          58          91                      B cells   \n",
       "TTTGTGGCAGCCTGCA-Q3          58          91                      B cells   \n",
       "TTTGTTGGTAAACAAG-Q3          57          20                    Monocytes   \n",
       "TTTGTTGGTTAAGCTG-Q3          36         124               B-cell lineage   \n",
       "\n",
       "                    celltypist_conf_score_coarse celltypist_cell_label_fine  \\\n",
       "AAACAGCCAGTTTGTG-D0                     0.978114                Pro-B cells   \n",
       "AAACAGCCATGTTTGG-D0                     0.603815                Pro-B cells   \n",
       "AAACAGCCATTTAAGC-D0                     0.951335                Pro-B cells   \n",
       "AAACATGCATTGGGAG-D0                     0.121671                Pro-B cells   \n",
       "AAACCAACAATAACCT-D0                     0.991872                Pro-B cells   \n",
       "...                                          ...                        ...   \n",
       "TTTGTGAAGCGAGTAA-Q3                     0.403587                Pro-B cells   \n",
       "TTTGTGAAGCTTTGGG-Q3                     0.776546                Pro-B cells   \n",
       "TTTGTGGCAGCCTGCA-Q3                     0.484664                Pro-B cells   \n",
       "TTTGTTGGTAAACAAG-Q3                     0.745200    Non-classical monocytes   \n",
       "TTTGTTGGTTAAGCTG-Q3                     0.122617                Pro-B cells   \n",
       "\n",
       "                    celltypist_conf_score_fine  \n",
       "AAACAGCCAGTTTGTG-D0                   0.950723  \n",
       "AAACAGCCATGTTTGG-D0                   0.370262  \n",
       "AAACAGCCATTTAAGC-D0                   0.903238  \n",
       "AAACATGCATTGGGAG-D0                   0.183371  \n",
       "AAACCAACAATAACCT-D0                   0.996793  \n",
       "...                                        ...  \n",
       "TTTGTGAAGCGAGTAA-Q3                   0.973642  \n",
       "TTTGTGAAGCTTTGGG-Q3                   0.532727  \n",
       "TTTGTGGCAGCCTGCA-Q3                   0.177973  \n",
       "TTTGTTGGTAAACAAG-Q3                   0.041574  \n",
       "TTTGTTGGTTAAGCTG-Q3                   0.947241  \n",
       "\n",
       "[11264 rows x 34 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14c445c1-27a7-4b84-9407-e751e1a16c40",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import multiome barcodes\n",
    "import pandas as pd\n",
    "atac_CB = pd.read_csv('/res_00001/barcodes/cellranger_arc_atac.737K-arc-v1.txt.gz', header = None)\n",
    "rna_CB = pd.read_csv('/res_00001/barcodes/cellranger_arc_rna.737K-arc-v1.txt.gz', header = None)\n",
    "rna_CB.rename(columns={0: 'rna_CB'}, inplace=True)\n",
    "atac_CB.rename(columns={0: 'atac_CB'}, inplace=True)\n",
    "df_merged =  pd.concat([rna_CB, atac_CB], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6a73088a-f226-4ce5-a76f-f698c66d8b37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>log1p_n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>log1p_total_counts</th>\n",
       "      <th>pct_counts_in_top_20_genes</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "      <th>pct_counts_ribo</th>\n",
       "      <th>pct_counts_hb</th>\n",
       "      <th>outlier</th>\n",
       "      <th>...</th>\n",
       "      <th>leiden_1.5</th>\n",
       "      <th>leiden_2.0</th>\n",
       "      <th>leiden_5.0</th>\n",
       "      <th>leiden_10.0</th>\n",
       "      <th>leiden_20.0</th>\n",
       "      <th>celltypist_cell_label_coarse</th>\n",
       "      <th>celltypist_conf_score_coarse</th>\n",
       "      <th>celltypist_cell_label_fine</th>\n",
       "      <th>celltypist_conf_score_fine</th>\n",
       "      <th>base_index</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>atac_CB</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTGTTACTA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1891</td>\n",
       "      <td>7.545390</td>\n",
       "      <td>7641.0</td>\n",
       "      <td>8.941415</td>\n",
       "      <td>21.823060</td>\n",
       "      <td>13.584609</td>\n",
       "      <td>7.642979</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>39</td>\n",
       "      <td>294</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.978114</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.950723</td>\n",
       "      <td>AAACAGCCAGTTTGTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTTTATCCT-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>2319</td>\n",
       "      <td>7.749322</td>\n",
       "      <td>10986.0</td>\n",
       "      <td>9.304468</td>\n",
       "      <td>21.354451</td>\n",
       "      <td>13.899508</td>\n",
       "      <td>9.111597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>8</td>\n",
       "      <td>6</td>\n",
       "      <td>73</td>\n",
       "      <td>51</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.603815</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.370262</td>\n",
       "      <td>AAACAGCCATGTTTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTTCACTTT-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1667</td>\n",
       "      <td>7.419381</td>\n",
       "      <td>6636.0</td>\n",
       "      <td>8.800415</td>\n",
       "      <td>21.503918</td>\n",
       "      <td>13.803495</td>\n",
       "      <td>6.268836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>24</td>\n",
       "      <td>29</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.951335</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.903238</td>\n",
       "      <td>AAACAGCCATTTAAGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CATTTAGGTGTCCACA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1599</td>\n",
       "      <td>7.377759</td>\n",
       "      <td>6888.0</td>\n",
       "      <td>8.837681</td>\n",
       "      <td>24.811266</td>\n",
       "      <td>18.626598</td>\n",
       "      <td>5.894309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>31</td>\n",
       "      <td>69</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.121671</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.183371</td>\n",
       "      <td>AAACATGCATTGGGAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTTTATCGTCGAGGCA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1403</td>\n",
       "      <td>7.247081</td>\n",
       "      <td>5495.0</td>\n",
       "      <td>8.611775</td>\n",
       "      <td>24.913558</td>\n",
       "      <td>18.817106</td>\n",
       "      <td>5.150136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.991872</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.996793</td>\n",
       "      <td>AAACCAACAATAACCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTAGGTGTCGGTCCAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1361</td>\n",
       "      <td>7.216709</td>\n",
       "      <td>5059.0</td>\n",
       "      <td>8.529121</td>\n",
       "      <td>23.739870</td>\n",
       "      <td>17.019173</td>\n",
       "      <td>8.381103</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>30</td>\n",
       "      <td>16</td>\n",
       "      <td>54</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.403587</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.973642</td>\n",
       "      <td>TTTGTGAAGCGAGTAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTAGGTGTCCCAATAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>2369</td>\n",
       "      <td>7.770645</td>\n",
       "      <td>8738.5</td>\n",
       "      <td>9.075608</td>\n",
       "      <td>15.860846</td>\n",
       "      <td>9.990273</td>\n",
       "      <td>4.691881</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>91</td>\n",
       "      <td>B cells</td>\n",
       "      <td>0.776546</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.532727</td>\n",
       "      <td>TTTGTGAAGCTTTGGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTTTGGTGTTTCTAAG-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1536</td>\n",
       "      <td>7.337588</td>\n",
       "      <td>6272.0</td>\n",
       "      <td>8.744010</td>\n",
       "      <td>29.432398</td>\n",
       "      <td>23.852041</td>\n",
       "      <td>6.505102</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>58</td>\n",
       "      <td>91</td>\n",
       "      <td>B cells</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.177973</td>\n",
       "      <td>TTTGTGGCAGCCTGCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTACTTCAAATATGC-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1279</td>\n",
       "      <td>7.154615</td>\n",
       "      <td>5301.0</td>\n",
       "      <td>8.575839</td>\n",
       "      <td>35.653650</td>\n",
       "      <td>28.730429</td>\n",
       "      <td>2.886248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>12</td>\n",
       "      <td>23</td>\n",
       "      <td>57</td>\n",
       "      <td>20</td>\n",
       "      <td>Monocytes</td>\n",
       "      <td>0.745200</td>\n",
       "      <td>Non-classical monocytes</td>\n",
       "      <td>0.041574</td>\n",
       "      <td>TTTGTTGGTAAACAAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTACTTCATGATGAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>1756</td>\n",
       "      <td>7.471363</td>\n",
       "      <td>7045.0</td>\n",
       "      <td>8.860215</td>\n",
       "      <td>24.442867</td>\n",
       "      <td>18.424414</td>\n",
       "      <td>6.018453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>36</td>\n",
       "      <td>124</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.122617</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.947241</td>\n",
       "      <td>TTTGTTGGTTAAGCTG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>11264 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         sample  n_genes_by_counts  log1p_n_genes_by_counts  \\\n",
       "atac_CB                                                                       \n",
       "ACAGCGGGTGTTACTA-D0___D0     D0               1891                 7.545390   \n",
       "ACAGCGGGTTTATCCT-D0___D0     D0               2319                 7.749322   \n",
       "ACAGCGGGTTCACTTT-D0___D0     D0               1667                 7.419381   \n",
       "CATTTAGGTGTCCACA-D0___D0     D0               1599                 7.377759   \n",
       "CTTTATCGTCGAGGCA-D0___D0     D0               1403                 7.247081   \n",
       "...                         ...                ...                      ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3     Q3               1361                 7.216709   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3     Q3               2369                 7.770645   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3     Q3               1536                 7.337588   \n",
       "CGTACTTCAAATATGC-Q3___Q3     Q3               1279                 7.154615   \n",
       "CGTACTTCATGATGAT-Q3___Q3     Q3               1756                 7.471363   \n",
       "\n",
       "                          total_counts  log1p_total_counts  \\\n",
       "atac_CB                                                      \n",
       "ACAGCGGGTGTTACTA-D0___D0        7641.0            8.941415   \n",
       "ACAGCGGGTTTATCCT-D0___D0       10986.0            9.304468   \n",
       "ACAGCGGGTTCACTTT-D0___D0        6636.0            8.800415   \n",
       "CATTTAGGTGTCCACA-D0___D0        6888.0            8.837681   \n",
       "CTTTATCGTCGAGGCA-D0___D0        5495.0            8.611775   \n",
       "...                                ...                 ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3        5059.0            8.529121   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3        8738.5            9.075608   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3        6272.0            8.744010   \n",
       "CGTACTTCAAATATGC-Q3___Q3        5301.0            8.575839   \n",
       "CGTACTTCATGATGAT-Q3___Q3        7045.0            8.860215   \n",
       "\n",
       "                          pct_counts_in_top_20_genes  pct_counts_mt  \\\n",
       "atac_CB                                                               \n",
       "ACAGCGGGTGTTACTA-D0___D0                   21.823060      13.584609   \n",
       "ACAGCGGGTTTATCCT-D0___D0                   21.354451      13.899508   \n",
       "ACAGCGGGTTCACTTT-D0___D0                   21.503918      13.803495   \n",
       "CATTTAGGTGTCCACA-D0___D0                   24.811266      18.626598   \n",
       "CTTTATCGTCGAGGCA-D0___D0                   24.913558      18.817106   \n",
       "...                                              ...            ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                   23.739870      17.019173   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3                   15.860846       9.990273   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                   29.432398      23.852041   \n",
       "CGTACTTCAAATATGC-Q3___Q3                   35.653650      28.730429   \n",
       "CGTACTTCATGATGAT-Q3___Q3                   24.442867      18.424414   \n",
       "\n",
       "                          pct_counts_ribo  pct_counts_hb  outlier  ...  \\\n",
       "atac_CB                                                            ...   \n",
       "ACAGCGGGTGTTACTA-D0___D0         7.642979            0.0    False  ...   \n",
       "ACAGCGGGTTTATCCT-D0___D0         9.111597            0.0    False  ...   \n",
       "ACAGCGGGTTCACTTT-D0___D0         6.268836            0.0    False  ...   \n",
       "CATTTAGGTGTCCACA-D0___D0         5.894309            0.0    False  ...   \n",
       "CTTTATCGTCGAGGCA-D0___D0         5.150136            0.0    False  ...   \n",
       "...                                   ...            ...      ...  ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3         8.381103            0.0    False  ...   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3         4.691881            0.0    False  ...   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3         6.505102            0.0    False  ...   \n",
       "CGTACTTCAAATATGC-Q3___Q3         2.886248            0.0    False  ...   \n",
       "CGTACTTCATGATGAT-Q3___Q3         6.018453            0.0    False  ...   \n",
       "\n",
       "                          leiden_1.5  leiden_2.0  leiden_5.0  leiden_10.0  \\\n",
       "atac_CB                                                                     \n",
       "ACAGCGGGTGTTACTA-D0___D0           4           2           7           39   \n",
       "ACAGCGGGTTTATCCT-D0___D0           7           8           6           73   \n",
       "ACAGCGGGTTCACTTT-D0___D0           4           2           2           24   \n",
       "CATTTAGGTGTCCACA-D0___D0           4           2           7           31   \n",
       "CTTTATCGTCGAGGCA-D0___D0           4           2           2            4   \n",
       "...                              ...         ...         ...          ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3           1          11          30           16   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3           1           0           0           58   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3           1           0           0           58   \n",
       "CGTACTTCAAATATGC-Q3___Q3          11          12          23           57   \n",
       "CGTACTTCATGATGAT-Q3___Q3           1           0           3           36   \n",
       "\n",
       "                         leiden_20.0 celltypist_cell_label_coarse  \\\n",
       "atac_CB                                                             \n",
       "ACAGCGGGTGTTACTA-D0___D0         294               B-cell lineage   \n",
       "ACAGCGGGTTTATCCT-D0___D0          51               B-cell lineage   \n",
       "ACAGCGGGTTCACTTT-D0___D0          29               B-cell lineage   \n",
       "CATTTAGGTGTCCACA-D0___D0          69               B-cell lineage   \n",
       "CTTTATCGTCGAGGCA-D0___D0           9               B-cell lineage   \n",
       "...                              ...                          ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3          54               B-cell lineage   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3          91                      B cells   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3          91                      B cells   \n",
       "CGTACTTCAAATATGC-Q3___Q3          20                    Monocytes   \n",
       "CGTACTTCATGATGAT-Q3___Q3         124               B-cell lineage   \n",
       "\n",
       "                         celltypist_conf_score_coarse  \\\n",
       "atac_CB                                                 \n",
       "ACAGCGGGTGTTACTA-D0___D0                     0.978114   \n",
       "ACAGCGGGTTTATCCT-D0___D0                     0.603815   \n",
       "ACAGCGGGTTCACTTT-D0___D0                     0.951335   \n",
       "CATTTAGGTGTCCACA-D0___D0                     0.121671   \n",
       "CTTTATCGTCGAGGCA-D0___D0                     0.991872   \n",
       "...                                               ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                     0.403587   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3                     0.776546   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                     0.484664   \n",
       "CGTACTTCAAATATGC-Q3___Q3                     0.745200   \n",
       "CGTACTTCATGATGAT-Q3___Q3                     0.122617   \n",
       "\n",
       "                         celltypist_cell_label_fine  \\\n",
       "atac_CB                                               \n",
       "ACAGCGGGTGTTACTA-D0___D0                Pro-B cells   \n",
       "ACAGCGGGTTTATCCT-D0___D0                Pro-B cells   \n",
       "ACAGCGGGTTCACTTT-D0___D0                Pro-B cells   \n",
       "CATTTAGGTGTCCACA-D0___D0                Pro-B cells   \n",
       "CTTTATCGTCGAGGCA-D0___D0                Pro-B cells   \n",
       "...                                             ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                Pro-B cells   \n",
       "GTAGGTGTCCCAATAT-Q3___Q3                Pro-B cells   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                Pro-B cells   \n",
       "CGTACTTCAAATATGC-Q3___Q3    Non-classical monocytes   \n",
       "CGTACTTCATGATGAT-Q3___Q3                Pro-B cells   \n",
       "\n",
       "                         celltypist_conf_score_fine        base_index  \n",
       "atac_CB                                                                \n",
       "ACAGCGGGTGTTACTA-D0___D0                   0.950723  AAACAGCCAGTTTGTG  \n",
       "ACAGCGGGTTTATCCT-D0___D0                   0.370262  AAACAGCCATGTTTGG  \n",
       "ACAGCGGGTTCACTTT-D0___D0                   0.903238  AAACAGCCATTTAAGC  \n",
       "CATTTAGGTGTCCACA-D0___D0                   0.183371  AAACATGCATTGGGAG  \n",
       "CTTTATCGTCGAGGCA-D0___D0                   0.996793  AAACCAACAATAACCT  \n",
       "...                                             ...               ...  \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                   0.973642  TTTGTGAAGCGAGTAA  \n",
       "GTAGGTGTCCCAATAT-Q3___Q3                   0.532727  TTTGTGAAGCTTTGGG  \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                   0.177973  TTTGTGGCAGCCTGCA  \n",
       "CGTACTTCAAATATGC-Q3___Q3                   0.041574  TTTGTTGGTAAACAAG  \n",
       "CGTACTTCATGATGAT-Q3___Q3                   0.947241  TTTGTTGGTTAAGCTG  \n",
       "\n",
       "[11264 rows x 35 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs['base_index'] = rna.obs.index.str.split('-').str[0]\n",
    "mapping_dict = pd.Series(df_merged['atac_CB'].values, index=df_merged['rna_CB']).to_dict()\n",
    "rna.obs['atac_CB'] = rna.obs['base_index'].map(mapping_dict)\n",
    "rna.obs['sample'] = rna.obs['sample'].str.split('_').str[0]\n",
    "rna.obs['atac_CB'] = rna.obs['atac_CB'].astype(str)+'-'+rna.obs['sample'].astype(str) + '___' + rna.obs['sample'].astype(str)\n",
    "rna.obs.set_index('atac_CB', inplace=True)\n",
    "rna.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "aa0c039c-9b2d-47d1-ba94-06a0915f7724",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna.X = rna.layers['counts'].copy()\n",
    "rna.raw = rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "344d8c9b-745d-46b0-8bfe-c7021f5be98d",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna.write_h5ad(\"/grn/concatenated_adata_log_pca_celltypist_23012025_with_raw_counts.BALL.h5ad\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d51006aa-fecc-45b8-8192-d9e78d598dc2",
   "metadata": {},
   "source": [
    "# run scenicplus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3b7a0cbe-2e68-4295-b716-2f8ecb47ffa1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "   ____   ____ _____ _   _ ___ ____      \n",
      "  / ___| / ___| ____| \\ | |_ _/ ___|\u001b[31;1m _ \u001b[0m\n",
      "  \\___ \\| |   |  _| |  \\| || | |  \u001b[31;1m _|.|_\u001b[0m\n",
      "   ___) | |___| |___| |\\  || | |__\u001b[31;1m|_..._|\u001b[0m\n",
      "  |____/ \\____|_____|_| \\_|___\\____|\u001b[31;1m|_|\u001b[0m \n",
      "\n",
      "\n",
      "scenicplus verions: 1.0a1\n",
      "usage: scenicplus [-h] {init_snakemake,prepare_data,grn_inference} ...\n",
      "\n",
      "Single-Cell Enhancer-driven gene regulatory Network Inference and Clustering\n",
      "\n",
      "positional arguments:\n",
      "  {init_snakemake,prepare_data,grn_inference}\n",
      "\n",
      "options:\n",
      "  -h, --help            show this help message and exit\n"
     ]
    }
   ],
   "source": [
    "! scenicplus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "498ce8e6-899a-460a-946a-8ff58572ba5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2025-04-10 13:46:01,688 SCENIC+      INFO     Creating snakemake folder in: scplus_pipeline\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!scenicplus init_snakemake --out_dir scplus_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "20b10d7a-e9bb-421e-8c0b-063fada2c316",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: tree: command not found\n"
     ]
    }
   ],
   "source": [
    "!tree scplus_pipeline/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cedf0a69-eab9-44ec-bbfc-e27c95967884",
   "metadata": {},
   "source": [
    "```\n",
    "tree -d scplus_pipeline/\n",
    "scplus_pipeline/\n",
    "└── Snakemake\n",
    "    ├── config\n",
    "    └── workflow\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "08bf6d99-9ae5-4bdf-b317-707f101bb886",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p outs\n",
    "!mkdir -p tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f08bcff7",
   "metadata": {},
   "outputs": [],
   "source": [
    "! ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "67ec337d-7008-4b29-8749-e67932773e7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd scplus_pipeline/Snakemake/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "47e6e5ce-e535-4898-b5b9-3f1478ce21c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "input_data:\n",
      "  cisTopic_obj_fname: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/cistopic_obj.pkl\"\n",
      "  GEX_anndata_fname: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/concatenated_adata_log_pca_celltypist_23012025_with_raw_counts.BALL.h5ad\"\n",
      "  region_set_folder: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\"\n",
      "  ctx_db_fname: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\"\n",
      "  dem_db_fname: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather\"\n",
      "  path_to_motif_annotations: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\"\n",
      "\n",
      "output_data:\n",
      "  # output for prepare_GEX_ACC .h5mu\n",
      "  combined_GEX_ACC_mudata: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\"\n",
      "  # output for motif enrichment results .hdf5\n",
      "  dem_result_fname: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\"\n",
      "  ctx_result_fname: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\"\n",
      "  # output html for motif enrichment results .html\n",
      "  output_fname_dem_html: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\"\n",
      "  output_fname_ctx_html: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\"\n",
      "  # output for prepare_menr .h5ad\n",
      "  cistromes_direct: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad\"\n",
      "  cistromes_extended: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\"\n",
      "  # output tf names .txt\n",
      "  tf_names: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\"\n",
      "  # output for download_genome_annotations .tsv\n",
      "  genome_annotation: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\"\n",
      "  chromsizes: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\"\n",
      "  # output for search_space .tsb\n",
      "  search_space: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\"\n",
      "  # output tf_to_gene .tsv\n",
      "  tf_to_gene_adjacencies: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\"\n",
      "  # output region_to_gene .tsv\n",
      "  region_to_gene_adjacencies: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\"\n",
      "  # output eGRN .tsv\n",
      "  eRegulons_direct: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\"\n",
      "  eRegulons_extended: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\"\n",
      "  # output AUCell .h5mu\n",
      "  AUCell_direct: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\"\n",
      "  AUCell_extended: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\"\n",
      "  # output scplus mudata .h5mu\n",
      "  scplus_mdata: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\"\n",
      "\n",
      "params_general:\n",
      "  temp_dir: \"/staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/tmp\"\n",
      "  n_cpu: 20\n",
      "  seed: 666\n",
      "\n",
      "params_data_preparation:\n",
      "  # Params for prepare_GEX_ACC\n",
      "  bc_transform_func: \"\\\"lambda x: f'{x}'\\\"\"\n",
      "  is_multiome: True\n",
      "  key_to_group_by: \"\"\n",
      "  nr_cells_per_metacells: 10\n",
      "  # Params for prepare_menr\n",
      "  direct_annotation: \"Direct_annot\"\n",
      "  extended_annotation: \"Orthology_annot\"\n",
      "  # Params for download_genome_annotations\n",
      "  species: \"hsapiens\"\n",
      "  biomart_host: \"http://www.ensembl.org\"\n",
      "  # Params for search_space\n",
      "  search_space_upstream: \"1000 150000\"\n",
      "  search_space_downstream: \"1000 150000\"\n",
      "  search_space_extend_tss: \"10 10\"\n",
      "\n",
      "params_motif_enrichment:\n",
      "  species: \"homo_sapiens\"\n",
      "  annotation_version: \"v10nr_clust\"\n",
      "  motif_similarity_fdr: 0.001\n",
      "  orthologous_identity_threshold: 0.0\n",
      "  annotations_to_use: \"Direct_annot Orthology_annot\"\n",
      "  fraction_overlap_w_dem_database: 0.4\n",
      "  dem_max_bg_regions: 500\n",
      "  dem_balance_number_of_promoters: True\n",
      "  dem_promoter_space: 1_000\n",
      "  dem_adj_pval_thr: 0.05\n",
      "  dem_log2fc_thr: 1.0\n",
      "  dem_mean_fg_thr: 0.0\n",
      "  dem_motif_hit_thr: 3.0\n",
      "  fraction_overlap_w_ctx_database: 0.4\n",
      "  ctx_auc_threshold: 0.005\n",
      "  ctx_nes_threshold: 3.0\n",
      "  ctx_rank_threshold: 0.05\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "params_inference:\n",
      "  # Params for tf_to_gene\n",
      "  tf_to_gene_importance_method: \"GBM\"\n",
      "  # Params regions_to_gene\n",
      "  region_to_gene_importance_method: \"GBM\"\n",
      "  region_to_gene_correlation_method: \"SR\"\n",
      "  # Params for eGRN inference\n",
      "  order_regions_to_genes_by: \"importance\"\n",
      "  order_TFs_to_genes_by: \"importance\"\n",
      "  gsea_n_perm: 1000\n",
      "  quantile_thresholds_region_to_gene: \"0.85 0.90 0.95\"\n",
      "  top_n_regionTogenes_per_gene: \"5 10 15\"\n",
      "  top_n_regionTogenes_per_region: \"\"\n",
      "  min_regions_per_gene: 0\n",
      "  rho_threshold: 0.05\n",
      "  min_target_genes: 10\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!cat scplus_pipeline/Snakemake/config/config.yaml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31f9d082",
   "metadata": {},
   "outputs": [],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "047dc5a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /scplus_pipeline/Snakemake/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "95b901cd-b6fd-47c0-b19e-69064b0d784e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mAssuming unrestricted shared filesystem usage for local execution.\u001b[0m\n",
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
      "\u001b[33mProvided cores: 24\u001b[0m\n",
      "\u001b[33mRules claiming more threads will be scaled down.\u001b[0m\n",
      "\u001b[33mJob stats:\n",
      "job                            count\n",
      "---------------------------  -------\n",
      "AUCell_direct                      1\n",
      "AUCell_extended                    1\n",
      "all                                1\n",
      "download_genome_annotations        1\n",
      "eGRN_direct                        1\n",
      "eGRN_extended                      1\n",
      "get_search_space                   1\n",
      "motif_enrichment_cistarget         1\n",
      "motif_enrichment_dem               1\n",
      "prepare_GEX_ACC_multiome           1\n",
      "prepare_menr                       1\n",
      "region_to_gene                     1\n",
      "scplus_mudata                      1\n",
      "tf_to_gene                         1\n",
      "total                             14\n",
      "\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 3 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:53:12 2025]\u001b[0m\n",
      "\u001b[32mlocalrule motif_enrichment_cistarget:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\n",
      "    jobid: 9\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:53:12 2025]\u001b[0m\n",
      "\u001b[32mlocalrule download_genome_annotations:\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "    jobid: 8\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:53:12 2025]\u001b[0m\n",
      "\u001b[32mlocalrule prepare_GEX_ACC_multiome:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/cistopic_obj.pkl, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/concatenated_adata_log_pca_celltypist_23012025_with_raw_counts.BALL.h5ad\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    jobid: 2\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 13:53:19,098 SCENIC+      INFO     Reading region sets from: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\n",
      "2025-04-10 13:53:19,099 SCENIC+      INFO     Reading all .bed files in: Topics_top_3k\n",
      "2025-04-10 13:53:19,156 SCENIC+      INFO     Reading cisTopic object.\n",
      "2025-04-10 13:53:19,879 SCENIC+      INFO     Reading gene expression AnnData.\n",
      "2025-04-10 13:53:21,864 Ingesting multiome data INFO     Found 8399 multiome cells.\n",
      "2025-04-10 13:53:21,961 cisTopic     INFO     Imputing region accessibility\n",
      "2025-04-10 13:53:21,961 cisTopic     INFO     Impute region accessibility for regions 0-20000\n",
      "2025-04-10 13:53:22,752 cisTopic     INFO     Impute region accessibility for regions 20000-40000\n",
      "2025-04-10 13:53:22,815 SCENIC+      INFO     Reading all .bed files in: DARs_cell_type\n",
      "2025-04-10 13:53:23,551 cisTopic     INFO     Impute region accessibility for regions 40000-60000\n",
      "2025-04-10 13:53:24,240 SCENIC+      INFO     Reading all .bed files in: Topics_otsu\n",
      "2025-04-10 13:53:24,345 cisTopic     INFO     Impute region accessibility for regions 60000-80000\n",
      "2025-04-10 13:53:25,140 cisTopic     INFO     Impute region accessibility for regions 80000-100000\n",
      "2025-04-10 13:53:25,928 cisTopic     INFO     Impute region accessibility for regions 100000-120000\n",
      "2025-04-10 13:53:26,724 cisTopic     INFO     Impute region accessibility for regions 120000-140000\n",
      "2025-04-10 13:53:27,523 cisTopic     INFO     Impute region accessibility for regions 140000-160000\n",
      "2025-04-10 13:53:28,327 cisTopic     INFO     Impute region accessibility for regions 160000-180000\n",
      "2025-04-10 13:53:29,529 cisTopic     INFO     Impute region accessibility for regions 180000-200000\n",
      "2025-04-10 13:53:30,735 cisTopic     INFO     Impute region accessibility for regions 200000-220000\n",
      "2025-04-10 13:53:31,644 cisTopic     INFO     Impute region accessibility for regions 220000-240000\n",
      "2025-04-10 13:53:31,829 cisTopic     INFO     Done!\n",
      "... storing 'barcode' as categorical\n",
      "... storing 'cell_type' as categorical\n",
      "... storing 'sample_id' as categorical\n",
      "... storing 'pycisTopic_leiden_10_0.2' as categorical\n",
      "... storing 'pycisTopic_leiden_10_0.3' as categorical\n",
      "... storing 'pycisTopic_leiden_10_0.6' as categorical\n",
      "... storing 'pycisTopic_leiden_10_1.2' as categorical\n",
      "... storing 'pycisTopic_leiden_10_3' as categorical\n",
      "... storing 'pycisTopic_leiden_10_6' as categorical\n",
      "... storing 'cell_type_all' as categorical\n",
      "... storing 'Chromosome' as categorical\n",
      "2025-04-10 13:53:35,356 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,427 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,437 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,463 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,468 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,510 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,553 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,557 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,576 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:35,932 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,001 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,003 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,003 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,011 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,026 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,026 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,033 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,035 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,037 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:36,074 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:39,105 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic16 which has 3000 regions\n",
      "2025-04-10 13:53:39,164 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic8 which has 3000 regions\n",
      "2025-04-10 13:53:39,208 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic25 which has 3000 regions\n",
      "2025-04-10 13:53:39,305 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic11 which has 3000 regions\n",
      "2025-04-10 13:53:39,423 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic14 which has 3000 regions\n",
      "2025-04-10 13:53:39,433 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic17 which has 3000 regions\n",
      "2025-04-10 13:53:39,552 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic5 which has 3000 regions\n",
      "2025-04-10 13:53:39,646 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic24 which has 3000 regions\n",
      "2025-04-10 13:53:39,661 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic21 which has 3000 regions\n",
      "2025-04-10 13:53:39,938 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic1 which has 3000 regions\n",
      "2025-04-10 13:53:40,010 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic23 which has 3000 regions\n",
      "2025-04-10 13:53:40,011 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic18 which has 3000 regions\n",
      "2025-04-10 13:53:40,016 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic20 which has 3000 regions\n",
      "2025-04-10 13:53:40,071 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic19 which has 3000 regions\n",
      "2025-04-10 13:53:40,129 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic9 which has 3000 regions\n",
      "2025-04-10 13:53:40,141 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic15 which has 3000 regions\n",
      "2025-04-10 13:53:40,168 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic10 which has 3000 regions\n",
      "2025-04-10 13:53:40,182 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic2 which has 3000 regions\n",
      "2025-04-10 13:53:40,259 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic12 which has 3000 regions\n",
      "2025-04-10 13:53:40,286 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic22 which has 3000 regions\n",
      "/opt/venv/lib/python3.11/site-packages/pybiomart/dataset.py:269: DtypeWarning: Columns (0) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  result = pd.read_csv(StringIO(response.text), sep='\\t')\n",
      "2025-04-10 13:53:42,931 Download gene annotation INFO     Using genome: GRCh38.p14\n",
      "2025-04-10 13:53:43,486 Download gene annotation INFO     Found corresponding genome Id 51 on NCBI\n",
      "2025-04-10 13:53:44,483 Download gene annotation INFO     Found corresponding assembly Id 11968211 on NCBI\n",
      "2025-04-10 13:53:44,775 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic5\n",
      "2025-04-10 13:53:44,776 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic16\n",
      "2025-04-10 13:53:45,024 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic8\n",
      "2025-04-10 13:53:45,054 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic11\n",
      "2025-04-10 13:53:45,062 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic25\n",
      "2025-04-10 13:53:45,182 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic24\n",
      "2025-04-10 13:53:45,432 Download gene annotation INFO     Downloading assembly information from: ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_assembly_report.txt\n",
      "2025-04-10 13:53:45,566 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic23\n",
      "2025-04-10 13:53:45,703 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic15\n",
      "2025-04-10 13:53:45,733 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic10\n",
      "2025-04-10 13:53:45,772 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic21\n",
      "2025-04-10 13:53:45,807 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic14\n",
      "2025-04-10 13:53:45,807 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic18\n",
      "2025-04-10 13:53:45,849 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic19\n",
      "2025-04-10 13:53:45,855 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic2\n",
      "2025-04-10 13:53:46,043 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic17\n",
      "2025-04-10 13:53:46,112 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic12\n",
      "2025-04-10 13:53:46,146 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic20\n",
      "2025-04-10 13:53:46,335 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic22\n",
      "2025-04-10 13:53:46,403 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic1\n",
      "2025-04-10 13:53:46,530 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic9\n",
      "2025-04-10 13:53:48,160 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic5\n",
      "2025-04-10 13:53:48,213 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic16\n",
      "2025-04-10 13:53:48,233 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic23\n",
      "2025-04-10 13:53:48,234 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic12\n",
      "2025-04-10 13:53:48,240 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic15\n",
      "2025-04-10 13:53:48,240 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic24\n",
      "2025-04-10 13:53:48,349 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic10\n",
      "2025-04-10 13:53:48,356 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic2\n",
      "2025-04-10 13:53:48,356 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic11\n",
      "2025-04-10 13:53:48,374 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic25\n",
      "2025-04-10 13:53:48,382 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic8\n",
      "2025-04-10 13:53:48,422 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic19\n",
      "2025-04-10 13:53:48,425 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic18\n",
      "2025-04-10 13:53:48,438 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:48,589 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:48,612 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:48,648 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:48,664 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic20\n",
      "2025-04-10 13:53:48,693 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:48,711 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:48,781 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic22\n",
      "2025-04-10 13:53:48,805 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic21\n",
      "2025-04-10 13:53:48,965 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:49,023 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:49,060 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:49,092 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic17\n",
      "2025-04-10 13:53:49,109 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:49,165 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic14\n",
      "2025-04-10 13:53:49,200 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:49,273 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:49,319 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-04-10 13:53:49,428 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic9\n",
      "2025-04-10 13:53:49,490 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic1\n",
      "2025-04-10 13:53:49,668 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:50,073 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:50,314 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:50,603 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:50,975 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[32m[Thu Apr 10 13:53:51 2025]\u001b[0m\n",
      "\u001b[32mFinished job 2.\u001b[0m\n",
      "\u001b[32m1 of 14 steps (7%) done\u001b[0m\n",
      "2025-04-10 13:53:51,604 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:51,792 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic3 which has 3000 regions\n",
      "2025-04-10 13:53:51,804 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic4 which has 3000 regions\n",
      "2025-04-10 13:53:51,852 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic7 which has 3000 regions\n",
      "2025-04-10 13:53:51,920 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic6 which has 3000 regions\n",
      "2025-04-10 13:53:52,018 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic13 which has 3000 regions\n",
      "2025-04-10 13:53:52,041 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:52,728 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q3 which has 3450 regions\n",
      "2025-04-10 13:53:52,888 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q1 which has 4526 regions\n",
      "2025-04-10 13:53:53,155 cisTarget    INFO     Running cisTarget for DARs_cell_type_Mid_erythroid which has 6680 regions\n",
      "2025-04-10 13:53:53,168 cisTarget    INFO     Running cisTarget for DARs_cell_type_CD16pos_NK_cells which has 10869 regions\n",
      "2025-04-10 13:53:53,519 cisTarget    INFO     Running cisTarget for DARs_cell_type_Non-classical_monocytes which has 9501 regions\n",
      "2025-04-10 13:53:53,526 cisTarget    INFO     Running cisTarget for DARs_cell_type_Classical_monocytes which has 9983 regions\n",
      "2025-04-10 13:53:53,711 cisTarget    INFO     Running cisTarget for DARs_cell_type_Naive_B_cells which has 8546 regions\n",
      "2025-04-10 13:53:53,984 cisTarget    INFO     Running cisTarget for DARs_cell_type_Tem_Trm_cytotoxic_T_cells which has 12058 regions\n",
      "2025-04-10 13:53:54,140 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q2 which has 9660 regions\n",
      "2025-04-10 13:53:54,368 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_D0 which has 4724 regions\n",
      "2025-04-10 13:53:54,776 cisTarget    INFO     Running cisTarget for DARs_cell_type_Tcm_Naive_helper_T_cells which has 11043 regions\n",
      "2025-04-10 13:53:55,092 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic3\n",
      "2025-04-10 13:53:55,123 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic21 which has 7856 regions\n",
      "2025-04-10 13:53:55,185 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic4\n",
      "2025-04-10 13:53:55,432 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic13\n",
      "2025-04-10 13:53:55,578 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic7\n",
      "2025-04-10 13:53:56,007 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q3\n",
      "2025-04-10 13:53:56,328 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic6\n",
      "2025-04-10 13:53:56,378 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic2 which has 8766 regions\n",
      "2025-04-10 13:53:56,650 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q1\n",
      "2025-04-10 13:53:56,677 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic15 which has 23264 regions\n",
      "2025-04-10 13:53:56,820 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic11 which has 11091 regions\n",
      "2025-04-10 13:53:56,970 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic3\n",
      "2025-04-10 13:53:57,090 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic4\n",
      "2025-04-10 13:53:57,274 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:57,357 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic13\n",
      "2025-04-10 13:53:57,491 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:57,635 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic7\n",
      "2025-04-10 13:53:57,796 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q3\n",
      "2025-04-10 13:53:57,870 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:57,966 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_D0\n",
      "2025-04-10 13:53:58,127 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:58,367 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:53:58,511 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q2\n",
      "2025-04-10 13:53:58,667 cisTarget    INFO     Annotating motifs for DARs_cell_type_Mid_erythroid\n",
      "2025-04-10 13:53:58,705 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q1\n",
      "2025-04-10 13:53:58,719 cisTarget    INFO     Annotating motifs for DARs_cell_type_Non-classical_monocytes\n",
      "2025-04-10 13:53:58,729 cisTarget    INFO     Annotating motifs for DARs_cell_type_Classical_monocytes\n",
      "2025-04-10 13:53:59,049 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic6\n",
      "2025-04-10 13:53:59,445 cisTarget    INFO     Annotating motifs for DARs_cell_type_Naive_B_cells\n",
      "\u001b[0m2025-04-10 13:53:59,810 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_D0\n",
      "2025-04-10 13:54:00,223 cisTarget    INFO     Annotating motifs for DARs_cell_type_CD16pos_NK_cells\n",
      "2025-04-10 13:54:00,661 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q2\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "2025-04-10 13:54:00,696 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic21\n",
      "2025-04-10 13:54:00,831 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-04-10 13:54:01,118 cisTarget    INFO     Getting cistromes for DARs_cell_type_Non-classical_monocytes\n",
      "2025-04-10 13:54:01,154 cisTarget    INFO     Getting cistromes for DARs_cell_type_Classical_monocytes\n",
      "2025-04-10 13:54:01,279 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic1 which has 6942 regions\n",
      "2025-04-10 13:54:01,392 cisTarget    INFO     Annotating motifs for DARs_cell_type_Tcm_Naive_helper_T_cells\n",
      "2025-04-10 13:54:01,509 cisTarget    INFO     Getting cistromes for DARs_cell_type_Mid_erythroid\n",
      "2025-04-10 13:54:01,510 cisTarget    INFO     Annotating motifs for DARs_cell_type_Tem_Trm_cytotoxic_T_cells\n",
      "2025-04-10 13:54:01,766 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic2\n",
      "2025-04-10 13:54:01,814 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic11\n",
      "2025-04-10 13:54:02,390 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic25 which has 11182 regions\n",
      "2025-04-10 13:54:02,540 cisTarget    INFO     Getting cistromes for DARs_cell_type_Naive_B_cells\n",
      "2025-04-10 13:54:02,658 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic5 which has 24486 regions\n",
      "2025-04-10 13:54:02,728 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic17 which has 12255 regions\n",
      "\u001b[0m2025-04-10 13:54:03,380 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic9 which has 23 regions\n",
      "\u001b[0m2025-04-10 13:54:03,836 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic15\n",
      "2025-04-10 13:54:04,159 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic12 which has 40107 regions\n",
      "2025-04-10 13:54:04,205 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic11\n",
      "2025-04-10 13:54:04,306 cisTarget    INFO     Getting cistromes for DARs_cell_type_CD16pos_NK_cells\n",
      "2025-04-10 13:54:04,754 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic2\n",
      "\u001b[0m2025-04-10 13:54:04,932 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic21\n",
      "2025-04-10 13:54:05,034 cisTarget    INFO     Getting cistromes for DARs_cell_type_Tcm_Naive_helper_T_cells\n",
      "2025-04-10 13:54:05,866 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic9\n",
      "2025-04-10 13:54:06,098 cisTarget    INFO     Getting cistromes for DARs_cell_type_Tem_Trm_cytotoxic_T_cells\n",
      "2025-04-10 13:54:06,202 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic15\n",
      "\u001b[0m2025-04-10 13:54:07,261 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic1\n",
      "2025-04-10 13:54:07,440 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic9\n",
      "\u001b[0m2025-04-10 13:54:08,026 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic25\n",
      "2025-04-10 13:54:08,397 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-04-10 13:54:09,383 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic5\n",
      "2025-04-10 13:54:09,461 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:10,219 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic17\n",
      "2025-04-10 13:54:10,332 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:11,044 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic25\n",
      "2025-04-10 13:54:11,199 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:11,537 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic5\n",
      "2025-04-10 13:54:12,041 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:12,138 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:12,325 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic20 which has 12226 regions\n",
      "\u001b[0m2025-04-10 13:54:13,204 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic1\n",
      "2025-04-10 13:54:13,491 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic22 which has 24 regions\n",
      "2025-04-10 13:54:13,972 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic14 which has 8649 regions\n",
      "\u001b[0m2025-04-10 13:54:14,119 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic24 which has 24115 regions\n",
      "2025-04-10 13:54:14,348 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:14,810 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-04-10 13:54:15,217 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic12\n",
      "2025-04-10 13:54:15,555 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic17\n",
      "2025-04-10 13:54:15,915 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic10 which has 10567 regions\n",
      "\u001b[0m2025-04-10 13:54:16,105 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic22\n",
      "2025-04-10 13:54:16,519 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic23 which has 20024 regions\n",
      "\u001b[0m2025-04-10 13:54:17,147 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:17,534 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic12\n",
      "2025-04-10 13:54:17,612 Download gene annotation INFO     Found following assembled molecules (chromosomes): \n",
      "\t1\n",
      "\t2\n",
      "\t3\n",
      "\t4\n",
      "\t5\n",
      "\t6\n",
      "\t7\n",
      "\t8\n",
      "\t9\n",
      "\t10\n",
      "\t11\n",
      "\t12\n",
      "\t13\n",
      "\t14\n",
      "\t15\n",
      "\t16\n",
      "\t17\n",
      "\t18\n",
      "\t19\n",
      "\t20\n",
      "\t21\n",
      "\t22\n",
      "\tX\n",
      "\tY\n",
      "\tMT\n",
      "2025-04-10 13:54:17,628 Download gene annotation INFO     Converting chromosomes names to UCSC style as follows: \n",
      "\tOriginal\tUCSC\n",
      "\t1\tchr1\n",
      "\t2\tchr2\n",
      "\t3\tchr3\n",
      "\t4\tchr4\n",
      "\t5\tchr5\n",
      "\t6\tchr6\n",
      "\t7\tchr7\n",
      "\t8\tchr8\n",
      "\t9\tchr9\n",
      "\t10\tchr10\n",
      "\t11\tchr11\n",
      "\t12\tchr12\n",
      "\t13\tchr13\n",
      "\t14\tchr14\n",
      "\t15\tchr15\n",
      "\t16\tchr16\n",
      "\t17\tchr17\n",
      "\t18\tchr18\n",
      "\t19\tchr19\n",
      "\t20\tchr20\n",
      "\t21\tchr21\n",
      "\t22\tchr22\n",
      "\tX\tchrX\n",
      "\tY\tchrY\n",
      "\tMT\tchrM\n",
      "2025-04-10 13:54:17,640 SCENIC+      INFO     Saving chromosome sizes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "2025-04-10 13:54:17,731 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic22\n",
      "2025-04-10 13:54:17,888 SCENIC+      INFO     Saving genome annotation to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "\u001b[0m2025-04-10 13:54:18,340 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic8 which has 11940 regions\n",
      "\u001b[0m\u001b[32m[Thu Apr 10 13:54:18 2025]\u001b[0m\n",
      "\u001b[32mFinished job 8.\u001b[0m\n",
      "\u001b[32m2 of 14 steps (14%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:54:18 2025]\u001b[0m\n",
      "\u001b[32mlocalrule get_search_space:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    jobid: 11\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 13:54:19,046 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic19 which has 15015 regions\n",
      "\u001b[0m2025-04-10 13:54:19,779 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:20,413 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic18 which has 2998 regions\n",
      "\u001b[0m2025-04-10 13:54:20,699 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:21,965 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:22,218 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic14\n",
      "2025-04-10 13:54:22,327 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic10\n",
      "2025-04-10 13:54:22,499 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic20\n",
      "2025-04-10 13:54:22,770 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-04-10 13:54:23,329 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:23,604 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic24\n",
      "2025-04-10 13:54:23,840 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic13 which has 12450 regions\n",
      "2025-04-10 13:54:24,193 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 13:54:24,464 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic10\n",
      "\u001b[0m2025-04-10 13:54:25,409 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic8\n",
      "2025-04-10 13:54:25,669 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic24\n",
      "2025-04-10 13:54:25,671 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic18\n",
      "2025-04-10 13:54:25,725 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic16 which has 29482 regions\n",
      "\u001b[0m2025-04-10 13:54:26,180 SCENIC+      INFO     Reading data\n",
      "2025-04-10 13:54:26,881 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic3 which has 22959 regions\n",
      "2025-04-10 13:54:26,959 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic23\n",
      "2025-04-10 13:54:27,046 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic4 which has 13375 regions\n",
      "2025-04-10 13:54:27,112 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic14\n",
      "2025-04-10 13:54:27,130 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic20\n",
      "2025-04-10 13:54:27,400 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic19\n",
      "2025-04-10 13:54:27,402 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic7 which has 9004 regions\n",
      "2025-04-10 13:54:27,586 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic8\n",
      "2025-04-10 13:54:27,595 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic18\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 13:54:28,500 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic6 which has 15848 regions\n",
      "2025-04-10 13:54:28,712 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic13\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 13:54:30,113 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic19\n",
      "2025-04-10 13:54:30,151 Get search space INFO     Extending promoter annotation to 10 bp upstream and 10 downstream\n",
      "2025-04-10 13:54:30,224 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic23\n",
      "2025-04-10 13:54:30,328 Get search space INFO     Extending search space to:\n",
      "            \t\t\t\t\t\t150000 bp downstream of the end of the gene.\n",
      "            \t\t\t\t\t\t150000 bp upstream of the start of the gene.\n",
      "2025-04-10 13:54:30,832 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic13\n",
      "2025-04-10 13:54:31,058 Get search space INFO     Intersecting with regions.\n",
      "2025-04-10 13:54:32,604 Get search space INFO     Calculating distances from region to gene\n",
      "\u001b[0m2025-04-10 13:54:34,643 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic16\n",
      "2025-04-10 13:54:34,698 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic7\n",
      "2025-04-10 13:54:35,131 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic4\n",
      "2025-04-10 13:54:35,536 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic3\n",
      "2025-04-10 13:54:36,282 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic16\n",
      "2025-04-10 13:54:37,632 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic3\n",
      "2025-04-10 13:54:37,703 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic4\n",
      "2025-04-10 13:54:37,930 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic7\n",
      "2025-04-10 13:54:38,406 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic6\n",
      "2025-04-10 13:54:43,376 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic6\n",
      "2025-04-10 13:54:48,381 SCENIC+      INFO     Writing html to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\n",
      "2025-04-10 13:54:48,866 SCENIC+      INFO     Writing output to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "2025-04-10 13:55:17,870 Get search space INFO     Imploding multiple entries per region and gene\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Apr 10 13:55:36 2025]\u001b[0m\n",
      "\u001b[32mFinished job 9.\u001b[0m\n",
      "\u001b[32m3 of 14 steps (21%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:55:36 2025]\u001b[0m\n",
      "\u001b[32mlocalrule motif_enrichment_dem:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\n",
      "    jobid: 7\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 13:55:44,006 SCENIC+      INFO     Reading region sets from: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\n",
      "2025-04-10 13:55:44,009 SCENIC+      INFO     Reading all .bed files in: Topics_top_3k\n",
      "2025-04-10 13:55:44,197 SCENIC+      INFO     Reading all .bed files in: DARs_cell_type\n",
      "2025-04-10 13:55:44,257 SCENIC+      INFO     Reading all .bed files in: Topics_otsu\n",
      "2025-04-10 13:55:54,304 DEM          INFO     Running DEM for Topics_top_3k_Topic5_vs_all\n",
      "2025-04-10 13:55:54,616 DEM          INFO     Running DEM for Topics_top_3k_Topic2_vs_all\n",
      "2025-04-10 13:55:54,618 DEM          INFO     Running DEM for Topics_top_3k_Topic1_vs_all\n",
      "2025-04-10 13:55:54,678 DEM          INFO     Running DEM for Topics_top_3k_Topic21_vs_all\n",
      "2025-04-10 13:55:54,687 DEM          INFO     Running DEM for Topics_top_3k_Topic12_vs_all\n",
      "2025-04-10 13:55:54,700 DEM          INFO     Running DEM for Topics_top_3k_Topic17_vs_all\n",
      "2025-04-10 13:55:54,709 DEM          INFO     Running DEM for Topics_top_3k_Topic25_vs_all\n",
      "2025-04-10 13:55:54,724 DEM          INFO     Running DEM for Topics_top_3k_Topic11_vs_all\n",
      "2025-04-10 13:55:54,726 DEM          INFO     Running DEM for Topics_top_3k_Topic9_vs_all\n",
      "2025-04-10 13:55:54,742 DEM          INFO     Running DEM for Topics_top_3k_Topic15_vs_all\n",
      "2025-04-10 13:55:54,746 DEM          INFO     Running DEM for Topics_top_3k_Topic23_vs_all\n",
      "2025-04-10 13:55:54,794 DEM          INFO     Running DEM for Topics_top_3k_Topic10_vs_all\n",
      "2025-04-10 13:55:54,800 DEM          INFO     Running DEM for Topics_top_3k_Topic20_vs_all\n",
      "2025-04-10 13:55:54,855 DEM          INFO     Running DEM for Topics_top_3k_Topic24_vs_all\n",
      "2025-04-10 13:55:54,921 DEM          INFO     Running DEM for Topics_top_3k_Topic19_vs_all\n",
      "2025-04-10 13:55:54,925 DEM          INFO     Running DEM for Topics_top_3k_Topic22_vs_all\n",
      "2025-04-10 13:55:54,929 DEM          INFO     Running DEM for Topics_top_3k_Topic14_vs_all\n",
      "2025-04-10 13:55:54,933 DEM          INFO     Running DEM for Topics_top_3k_Topic8_vs_all\n",
      "2025-04-10 13:55:54,984 DEM          INFO     Running DEM for Topics_top_3k_Topic16_vs_all\n",
      "2025-04-10 13:55:54,998 DEM          INFO     Running DEM for Topics_top_3k_Topic18_vs_all\n",
      "2025-04-10 13:56:07,353 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:07,547 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:07,628 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:07,782 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:07,802 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,007 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,015 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,034 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,045 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,073 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,106 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,116 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,133 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,187 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,293 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,324 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,371 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:08,446 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:09,691 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:09,896 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:11,429 DEM          INFO     Running DEM for Topics_top_3k_Topic3_vs_all\n",
      "2025-04-10 13:56:11,603 DEM          INFO     Running DEM for Topics_top_3k_Topic4_vs_all\n",
      "2025-04-10 13:56:11,874 DEM          INFO     Running DEM for Topics_top_3k_Topic7_vs_all\n",
      "2025-04-10 13:56:12,063 DEM          INFO     Running DEM for DARs_cell_type_CD16pos_NK_cells_vs_all\n",
      "2025-04-10 13:56:12,067 DEM          INFO     Running DEM for Topics_top_3k_Topic6_vs_all\n",
      "2025-04-10 13:56:12,146 DEM          INFO     Running DEM for DARs_cell_type_Classical_monocytes_vs_all\n",
      "2025-04-10 13:56:12,156 DEM          INFO     Running DEM for Topics_top_3k_Topic13_vs_all\n",
      "2025-04-10 13:56:12,188 DEM          INFO     Running DEM for DARs_cell_type_Non-classical_monocytes_vs_all\n",
      "2025-04-10 13:56:12,216 DEM          INFO     Running DEM for DARs_cell_type_Mid_erythroid_vs_all\n",
      "2025-04-10 13:56:12,291 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q1_vs_all\n",
      "2025-04-10 13:56:12,302 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q3_vs_all\n",
      "2025-04-10 13:56:12,323 DEM          INFO     Running DEM for DARs_cell_type_Tem_Trm_cytotoxic_T_cells_vs_all\n",
      "2025-04-10 13:56:12,467 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q2_vs_all\n",
      "2025-04-10 13:56:12,510 DEM          INFO     Running DEM for DARs_cell_type_Tcm_Naive_helper_T_cells_vs_all\n",
      "2025-04-10 13:56:12,512 DEM          INFO     Running DEM for DARs_cell_type_Naive_B_cells_vs_all\n",
      "2025-04-10 13:56:12,532 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_D0_vs_all\n",
      "2025-04-10 13:56:13,730 DEM          INFO     Running DEM for Topics_otsu_Topic21_vs_all\n",
      "2025-04-10 13:56:13,822 DEM          INFO     Running DEM for Topics_otsu_Topic15_vs_all\n",
      "2025-04-10 13:56:15,258 DEM          INFO     Running DEM for Topics_otsu_Topic2_vs_all\n",
      "2025-04-10 13:56:15,585 DEM          INFO     Running DEM for Topics_otsu_Topic11_vs_all\n",
      "2025-04-10 13:56:16,816 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:17,266 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:17,392 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:17,698 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:17,712 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:18,362 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:19,631 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:56:20,116 DEM          INFO     Adding motif-to-TF annotation\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m2025-04-10 13:56:21,403 DEM          INFO     Running DEM for Topics_otsu_Topic5_vs_all\n",
      "2025-04-10 13:56:21,848 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:21,863 DEM          INFO     Running DEM for Topics_otsu_Topic1_vs_all\n",
      "2025-04-10 13:56:22,006 DEM          INFO     Running DEM for Topics_otsu_Topic12_vs_all\n",
      "2025-04-10 13:56:22,090 DEM          INFO     Running DEM for Topics_otsu_Topic25_vs_all\n",
      "2025-04-10 13:56:22,257 DEM          INFO     Running DEM for Topics_otsu_Topic17_vs_all\n",
      "\u001b[0m\u001b[0m2025-04-10 13:56:24,484 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:24,514 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:25,449 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:26,457 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:26,490 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:56:26,799 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:27,253 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:56:27,859 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:28,436 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:56:29,358 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m2025-04-10 13:56:31,031 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:31,271 DEM          INFO     Running DEM for Topics_otsu_Topic9_vs_all\n",
      "\u001b[0m2025-04-10 13:56:32,559 DEM          INFO     Running DEM for Topics_otsu_Topic20_vs_all\n",
      "\u001b[0m2025-04-10 13:56:33,414 DEM          INFO     Running DEM for Topics_otsu_Topic24_vs_all\n",
      "\u001b[0m2025-04-10 13:56:34,252 DEM          INFO     Running DEM for Topics_otsu_Topic14_vs_all\n",
      "\u001b[0m\u001b[0m2025-04-10 13:56:36,104 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:56:37,531 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:37,741 DEM          INFO     Running DEM for Topics_otsu_Topic23_vs_all\n",
      "\u001b[0m2025-04-10 13:56:38,846 DEM          INFO     Running DEM for Topics_otsu_Topic10_vs_all\n",
      "2025-04-10 13:56:39,276 DEM          INFO     Running DEM for Topics_otsu_Topic8_vs_all\n",
      "\u001b[0m2025-04-10 13:56:40,618 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:41,193 DEM          INFO     Running DEM for Topics_otsu_Topic19_vs_all\n",
      "2025-04-10 13:56:41,560 DEM          INFO     Running DEM for Topics_otsu_Topic18_vs_all\n",
      "2025-04-10 13:56:42,686 DEM          INFO     Running DEM for Topics_otsu_Topic22_vs_all\n",
      "2025-04-10 13:56:44,110 DEM          INFO     Running DEM for Topics_otsu_Topic16_vs_all\n",
      "2025-04-10 13:56:44,896 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:44,927 DEM          INFO     Running DEM for Topics_otsu_Topic3_vs_all\n",
      "2025-04-10 13:56:46,097 DEM          INFO     Running DEM for Topics_otsu_Topic4_vs_all\n",
      "\u001b[0m2025-04-10 13:56:46,779 DEM          INFO     Running DEM for Topics_otsu_Topic7_vs_all\n",
      "2025-04-10 13:56:47,871 DEM          INFO     Running DEM for Topics_otsu_Topic6_vs_all\n",
      "2025-04-10 13:56:49,640 DEM          INFO     Running DEM for Topics_otsu_Topic13_vs_all\n",
      "2025-04-10 13:56:51,957 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:52,083 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:53,257 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:56:54,416 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:56:55,690 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:00,354 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:02,459 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:05,598 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:07,413 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:07,514 SCENIC+      INFO     Writing search space to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "\u001b[0m\u001b[32m[Thu Apr 10 13:57:09 2025]\u001b[0m\n",
      "\u001b[32mFinished job 11.\u001b[0m\n",
      "\u001b[32m4 of 14 steps (29%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "2025-04-10 13:57:10,119 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:10,294 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:11,463 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:11,697 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:12,248 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-04-10 13:57:14,023 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:20,131 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:25,436 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-04-10 13:57:26,806 SCENIC+      INFO     Writing html to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\n",
      "2025-04-10 13:57:26,952 SCENIC+      INFO     Writing output to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Apr 10 13:57:37 2025]\u001b[0m\n",
      "\u001b[32mFinished job 7.\u001b[0m\n",
      "\u001b[32m5 of 14 steps (36%) done\u001b[0m\n",
      "\u001b[33mExecute 2 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:57:37 2025]\u001b[0m\n",
      "\u001b[32mlocalrule prepare_menr:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\n",
      "    jobid: 6\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 13:57:37 2025]\u001b[0m\n",
      "\u001b[32mlocalrule region_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    jobid: 10\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 13:57:43,393 SCENIC+      INFO     Reading multiome MuData.\n",
      "2025-04-10 13:57:43,959 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 13:57:46,582 SCENIC+      INFO     Getting cistromes.\n",
      "2025-04-10 13:57:46,881 SCENIC+      INFO     Reading search space\n",
      "2025-04-10 13:57:47,283 R2G          INFO     Calculating region to gene importances, using GBM method\n",
      "Running using 20 cores:   2%|▎              | 340/16369 [00:25<13:31, 19.75it/s]/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "Running using 20 cores:   3%|▍              | 440/16369 [00:30<16:20, 16.25it/s]/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 13:58:23,550 SCENIC+      INFO     Found 857 TFs.\n",
      "2025-04-10 13:58:23,551 SCENIC+      INFO     Saving TF names to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "2025-04-10 13:58:23,576 SCENIC+      INFO     Writing direct cistromes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad\n",
      "... storing 'motifs' as categorical\n",
      "2025-04-10 13:58:23,718 SCENIC+      INFO     Writing extended cistromes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\n",
      "... storing 'motifs' as categorical\n",
      "Running using 20 cores:   3%|▍              | 460/16369 [00:32<16:16, 16.30it/s]\u001b[32m[Thu Apr 10 13:58:24 2025]\u001b[0m\n",
      "\u001b[32mFinished job 6.\u001b[0m\n",
      "\u001b[32m6 of 14 steps (43%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "Running using 20 cores: 100%|█████████████| 16369/16369 [14:38<00:00, 18.64it/s]\n",
      "2025-04-10 14:12:40,151 R2G          INFO     Calculating region to gene correlation, using SR method\n",
      "Running using 20 cores:   0%|                 | 8/16369 [00:00<03:29, 77.93it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   0%|               | 74/16369 [00:00<01:03, 258.14it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|              | 100/16369 [00:00<01:11, 228.94it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏             | 186/16369 [00:00<01:02, 260.52it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏             | 217/16369 [00:00<00:58, 275.37it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▏             | 259/16369 [00:00<00:50, 316.89it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▎             | 334/16369 [00:01<00:50, 318.46it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▎             | 367/16369 [00:01<01:03, 253.74it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▎             | 398/16369 [00:01<01:00, 262.29it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   3%|▎             | 427/16369 [00:01<01:02, 253.10it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   3%|▍             | 506/16369 [00:01<00:50, 313.91it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   3%|▍             | 539/16369 [00:02<01:15, 209.86it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores: 100%|████████████| 16369/16369 [01:08<00:00, 238.69it/s]\n",
      "2025-04-10 14:14:09,594 R2G          INFO     Done!\n",
      "2025-04-10 14:14:09,736 SCENIC+      INFO     Saving region to gene adjacencies to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Apr 10 14:14:15 2025]\u001b[0m\n",
      "\u001b[32mFinished job 10.\u001b[0m\n",
      "\u001b[32m7 of 14 steps (50%) done\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 14:14:15 2025]\u001b[0m\n",
      "\u001b[32mlocalrule tf_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    jobid: 5\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 14:14:24,083 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 14:14:26,948 SCENIC+      INFO     Using 858 TFs.\n",
      "2025-04-10 14:14:27,289 TF2G         INFO     Calculating TF-to-gene importance\n",
      "Running using 20 cores: 100%|█████████████| 36601/36601 [38:38<00:00, 15.78it/s]\n",
      "2025-04-10 14:53:14,137 TF2G         INFO     Adding correlation coefficients to adjacencies.\n",
      "2025-04-10 14:53:31,789 TF2G         INFO     Warning: adding TFs as their own target to adjecencies matrix. Importance values will be max + 1e-05\n",
      "2025-04-10 14:53:35,982 SCENIC+      INFO     Saving TF to gene adjacencies to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "\u001b[0m\u001b[32m[Thu Apr 10 14:53:42 2025]\u001b[0m\n",
      "\u001b[32mFinished job 5.\u001b[0m\n",
      "\u001b[32m8 of 14 steps (57%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 14:53:42 2025]\u001b[0m\n",
      "\u001b[32mlocalrule eGRN_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    jobid: 4\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 14:53:50,738 SCENIC+      INFO     Loading TF to gene adjacencies.\n",
      "2025-04-10 14:53:51,600 SCENIC+      INFO     Loading region to gene adjacencies.\n",
      "2025-04-10 14:53:52,499 SCENIC+      INFO     Loading cistromes.\n",
      "2025-04-10 14:53:52,560 GSEA         INFO     Thresholding region to gene relationships\n",
      "  0%|                                                    | 0/14 [00:00<?, ?it/s]\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   0%|          | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   4%| | 27/641 [00:00<00:02, 262.97it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   9%| | 56/641 [00:00<00:02, 274.89it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  13%|▏| 84/641 [00:00<00:03, 146.18it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  16%|▏| 104/641 [00:00<00:04, 111.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  20%|▏| 125/641 [00:00<00:03, 131.06it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  24%|▏| 153/641 [00:00<00:02, 163.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  29%|▎| 187/641 [00:01<00:02, 204.63it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  33%|▎| 212/641 [00:01<00:03, 125.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  36%|▎| 233/641 [00:01<00:02, 137.54it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  41%|▍| 264/641 [00:01<00:02, 171.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  46%|▍| 292/641 [00:01<00:01, 194.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  50%|▍| 319/641 [00:01<00:01, 212.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  55%|▌| 355/641 [00:01<00:01, 248.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  60%|▌| 384/641 [00:02<00:00, 257.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  64%|▋| 413/641 [00:02<00:01, 162.67it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  69%|▋| 443/641 [00:02<00:01, 188.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  73%|▋| 470/641 [00:02<00:00, 202.12it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  77%|▊| 495/641 [00:02<00:00, 210.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  81%|▊| 522/641 [00:02<00:00, 224.27it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  85%|▊| 547/641 [00:03<00:00, 144.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  89%|▉| 571/641 [00:03<00:00, 160.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  93%|▉| 595/641 [00:03<00:00, 176.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  98%|▉| 625/641 [00:03<00:00, 204.05it/\u001b[0m\u001b[A\n",
      "  7%|███▏                                        | 1/14 [00:05<01:06,  5.11s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:   0%|           | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:   5%| | 32/641 [00:00<00:01, 308.18it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  10%| | 63/641 [00:00<00:01, 303.64it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  15%|▏| 94/641 [00:00<00:02, 223.10it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  19%|▏| 119/641 [00:00<00:04, 119.32it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  24%|▏| 153/641 [00:00<00:03, 159.81it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  29%|▎| 189/641 [00:00<00:02, 200.68it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  34%|▎| 216/641 [00:01<00:02, 175.06it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  38%|▍| 245/641 [00:01<00:01, 198.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  44%|▍| 281/641 [00:01<00:01, 234.83it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  48%|▍| 310/641 [00:01<00:01, 246.80it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  54%|▌| 349/641 [00:01<00:01, 282.21it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  59%|▌| 381/641 [00:01<00:00, 288.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  64%|▋| 412/641 [00:02<00:01, 149.34it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  69%|▋| 445/641 [00:02<00:01, 179.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  74%|▋| 473/641 [00:02<00:00, 197.30it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  78%|▊| 500/641 [00:02<00:00, 209.84it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  83%|▊| 532/641 [00:02<00:00, 233.41it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  88%|▉| 562/641 [00:02<00:00, 246.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  92%|▉| 590/641 [00:02<00:00, 250.23it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  97%|▉| 619/641 [00:02<00:00, 259.55it/s\u001b[0m\u001b[A\n",
      " 14%|██████▎                                     | 2/14 [00:08<00:50,  4.18s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:   0%|          | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:   5%| | 33/641 [00:00<00:01, 324.66it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  10%| | 66/641 [00:00<00:01, 326.15it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  15%|▏| 99/641 [00:00<00:02, 230.46it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  20%|▏| 125/641 [00:00<00:04, 110.61it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  25%|▎| 161/641 [00:00<00:03, 152.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  31%|▎| 198/641 [00:01<00:02, 194.08it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  35%|▎| 227/641 [00:01<00:02, 178.91it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  40%|▍| 259/641 [00:01<00:01, 207.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  46%|▍| 294/641 [00:01<00:01, 237.36it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  52%|▌| 332/641 [00:01<00:01, 270.36it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  58%|▌| 372/641 [00:01<00:00, 301.09it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  63%|▋| 406/641 [00:01<00:00, 277.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  69%|▋| 444/641 [00:01<00:00, 302.45it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  74%|▋| 477/641 [00:02<00:00, 296.21it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  79%|▊| 509/641 [00:02<00:00, 294.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  85%|▊| 543/641 [00:02<00:00, 304.96it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  90%|▉| 575/641 [00:02<00:00, 301.61it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  95%|▉| 606/641 [00:02<00:00, 295.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  99%|▉| 636/641 [00:02<00:00, 296.53it/\u001b[0m\u001b[A\n",
      " 21%|█████████▍                                  | 3/14 [00:12<00:42,  3.90s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:   0%| | 0/641 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:   2%| | 13/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:   4%| | 27/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:   7%| | 43/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:   9%| | 58/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  11%| | 72/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  13%|▏| 85/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  15%|▏| 96/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  17%|▏| 106/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  18%|▏| 115/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  20%|▏| 130/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  23%|▏| 146/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  25%|▏| 159/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  28%|▎| 177/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  30%|▎| 191/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  32%|▎| 205/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  34%|▎| 215/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  36%|▎| 229/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  37%|▎| 239/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  40%|▍| 256/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  42%|▍| 272/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  44%|▍| 285/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  46%|▍| 298/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  49%|▍| 311/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  51%|▌| 328/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  54%|▌| 344/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  56%|▌| 360/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  59%|▌| 376/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  61%|▌| 391/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  63%|▋| 404/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  66%|▋| 420/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  68%|▋| 434/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  70%|▋| 448/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  72%|▋| 461/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  73%|▋| 471/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  75%|▊| 481/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  77%|▊| 494/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  79%|▊| 505/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  80%|▊| 516/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  83%|▊| 531/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  85%|▊| 543/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  87%|▊| 556/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  89%|▉| 568/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  90%|▉| 580/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  92%|▉| 592/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  94%|▉| 604/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  97%|▉| 620/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, negative r2g:  99%|▉| 633/641 [\u001b[0m\u001b[A\n",
      " 29%|████████████▌                               | 4/14 [00:19<00:52,  5.26s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   0%| | 0/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   2%| | 11/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   4%| | 23/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   6%| | 37/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   8%| | 50/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  10%| | 63/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  12%| | 75/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  13%|▏| 85/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  15%|▏| 94/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  16%|▏| 101/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  17%|▏| 108/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  18%|▏| 115/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  20%|▏| 127/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  22%|▏| 139/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  24%|▏| 152/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  26%|▎| 164/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  28%|▎| 179/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  30%|▎| 191/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  32%|▎| 203/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  33%|▎| 213/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  34%|▎| 221/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  36%|▎| 231/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  37%|▎| 240/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  40%|▍| 254/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  42%|▍| 268/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  44%|▍| 281/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  46%|▍| 293/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  48%|▍| 305/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  49%|▍| 317/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  52%|▌| 333/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  54%|▌| 346/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  56%|▌| 359/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  58%|▌| 372/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  60%|▌| 385/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  62%|▌| 398/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  64%|▋| 411/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  66%|▋| 422/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  68%|▋| 433/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  69%|▋| 444/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  71%|▋| 458/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  73%|▋| 469/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  75%|▋| 479/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  77%|▊| 491/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  78%|▊| 501/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  80%|▊| 513/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  82%|▊| 526/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  84%|▊| 537/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  86%|▊| 549/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  88%|▉| 561/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  89%|▉| 572/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  91%|▉| 582/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  92%|▉| 592/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  94%|▉| 602/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  96%|▉| 615/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  98%|▉| 627/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g: 100%|▉| 638/641 \u001b[0m\u001b[A\n",
      " 36%|███████████████▋                            | 5/14 [00:28<00:58,  6.55s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   0%| | 0/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   2%| | 11/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   3%| | 22/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   5%| | 34/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   7%| | 46/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   9%| | 57/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  11%| | 69/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  12%| | 80/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  14%|▏| 89/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  15%|▏| 97/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  16%|▏| 105/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  17%|▏| 112/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  19%|▏| 123/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  21%|▏| 133/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  23%|▏| 147/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  24%|▏| 157/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  26%|▎| 167/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  28%|▎| 180/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  30%|▎| 190/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  31%|▎| 200/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  33%|▎| 210/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  34%|▎| 218/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  36%|▎| 229/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  37%|▎| 237/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  39%|▍| 250/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  41%|▍| 264/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  43%|▍| 276/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  45%|▍| 287/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  46%|▍| 298/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  48%|▍| 309/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  50%|▌| 321/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  52%|▌| 336/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  54%|▌| 349/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  56%|▌| 361/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  59%|▌| 375/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  61%|▌| 388/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  62%|▌| 400/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  64%|▋| 412/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  66%|▋| 425/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  68%|▋| 436/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  70%|▋| 450/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  72%|▋| 462/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  74%|▋| 473/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  75%|▊| 483/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  77%|▊| 493/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  78%|▊| 503/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  80%|▊| 515/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  83%|▊| 529/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  84%|▊| 540/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  86%|▊| 550/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  88%|▉| 561/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  89%|▉| 571/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  91%|▉| 581/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  92%|▉| 592/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  94%|▉| 602/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  96%|▉| 615/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  98%|▉| 627/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene: 100%|▉| 638/641 \u001b[0m\u001b[A\n",
      " 43%|██████████████████▊                         | 6/14 [00:37<00:58,  7.26s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   0%|         | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   3%| | 19/641 [00:00<00:03, 186.67it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   7%| | 42/641 [00:00<00:02, 207.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  10%| | 63/641 [00:00<00:03, 192.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  13%|▏| 83/641 [00:01<00:18, 30.69it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  15%|▏| 96/641 [00:01<00:14, 37.49it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  17%|▏| 108/641 [00:02<00:11, 44.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  20%|▏| 125/641 [00:02<00:08, 59.50it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  23%|▏| 148/641 [00:02<00:05, 84.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  26%|▎| 168/641 [00:02<00:04, 104.11it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  29%|▎| 189/641 [00:02<00:03, 124.77it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  32%|▎| 208/641 [00:02<00:03, 113.94it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  35%|▎| 224/641 [00:02<00:03, 108.73it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  37%|▎| 239/641 [00:02<00:03, 116.73it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  41%|▍| 264/641 [00:03<00:02, 146.16it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  44%|▍| 285/641 [00:03<00:02, 160.59it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  47%|▍| 304/641 [00:03<00:02, 165.01it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  51%|▌| 328/641 [00:03<00:01, 183.33it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  55%|▌| 354/641 [00:03<00:01, 201.92it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  59%|▌| 376/641 [00:03<00:01, 202.71it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  62%|▌| 397/641 [00:03<00:01, 167.70it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  66%|▋| 421/641 [00:03<00:01, 183.34it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  69%|▋| 441/641 [00:04<00:01, 184.92it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  72%|▋| 464/641 [00:04<00:00, 193.99it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  76%|▊| 485/641 [00:04<00:00, 177.03it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  79%|▊| 504/641 [00:04<00:00, 170.11it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  82%|▊| 525/641 [00:04<00:00, 179.88it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  85%|▊| 544/641 [00:04<00:00, 174.62it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  88%|▉| 562/641 [00:04<00:00, 173.40it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  90%|▉| 580/641 [00:04<00:00, 166.31it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  93%|▉| 597/641 [00:04<00:00, 164.58it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  96%|▉| 617/641 [00:05<00:00, 173.89it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  99%|▉| 635/641 [00:05<00:00, 169.21it\u001b[0m\u001b[A\n",
      " 50%|██████████████████████                      | 7/14 [00:50<01:04,  9.24s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:   0%|          | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:   2%| | 10/641 [00:00<00:06, 91.51it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:   3%| | 20/641 [00:00<00:06, 94.45it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:   5%| | 30/641 [00:00<00:06, 96.51it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:   6%| | 41/641 [00:00<00:05, 101.35it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:   8%| | 53/641 [00:00<00:05, 104.82it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  10%| | 64/641 [00:00<00:05, 104.87it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  12%| | 75/641 [00:00<00:05, 101.45it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  13%|▏| 86/641 [00:00<00:05, 93.46it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  15%|▏| 96/641 [00:01<00:06, 83.17it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  16%|▏| 105/641 [00:01<00:07, 74.76it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  18%|▏| 113/641 [00:01<00:07, 72.24it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  19%|▏| 124/641 [00:01<00:06, 81.46it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  21%|▏| 134/641 [00:01<00:05, 85.94it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  23%|▏| 147/641 [00:01<00:05, 97.20it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  25%|▏| 158/641 [00:01<00:04, 100.04it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  27%|▎| 170/641 [00:01<00:04, 104.78it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  29%|▎| 183/641 [00:01<00:04, 111.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  30%|▎| 195/641 [00:02<00:04, 103.40it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  32%|▎| 206/641 [00:02<00:04, 89.78it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  34%|▎| 216/641 [00:02<00:05, 82.33it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  35%|▎| 227/641 [00:02<00:04, 88.69it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  37%|▎| 237/641 [00:02<00:04, 86.63it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  39%|▍| 249/641 [00:02<00:04, 95.16it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  41%|▍| 260/641 [00:02<00:03, 98.56it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  42%|▍| 271/641 [00:02<00:03, 99.27it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  44%|▍| 282/641 [00:03<00:03, 98.79it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  46%|▍| 295/641 [00:03<00:03, 104.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  48%|▍| 308/641 [00:03<00:03, 110.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  50%|▍| 320/641 [00:03<00:02, 110.54it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  52%|▌| 332/641 [00:03<00:02, 113.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  54%|▌| 344/641 [00:03<00:02, 110.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  56%|▌| 356/641 [00:03<00:02, 106.21it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  57%|▌| 368/641 [00:03<00:02, 109.69it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  59%|▌| 380/641 [00:03<00:02, 109.60it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  61%|▌| 392/641 [00:04<00:02, 102.13it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  63%|▋| 403/641 [00:04<00:02, 97.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  65%|▋| 414/641 [00:04<00:02, 100.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  66%|▋| 425/641 [00:04<00:02, 101.09it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  68%|▋| 436/641 [00:04<00:02, 99.10it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  70%|▋| 449/641 [00:04<00:01, 107.09it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  72%|▋| 461/641 [00:04<00:01, 109.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  74%|▋| 473/641 [00:04<00:01, 103.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  76%|▊| 484/641 [00:04<00:01, 105.13it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  77%|▊| 495/641 [00:05<00:01, 103.36it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  79%|▊| 506/641 [00:06<00:06, 20.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  81%|▊| 517/641 [00:06<00:04, 26.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  83%|▊| 531/641 [00:06<00:02, 36.96it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  84%|▊| 541/641 [00:06<00:02, 43.96it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  86%|▊| 553/641 [00:07<00:01, 54.52it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  88%|▉| 565/641 [00:07<00:01, 64.82it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  90%|▉| 576/641 [00:07<00:00, 73.15it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  92%|▉| 588/641 [00:07<00:00, 81.96it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  93%|▉| 599/641 [00:07<00:00, 86.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  95%|▉| 611/641 [00:07<00:00, 93.41it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  97%|▉| 623/641 [00:07<00:00, 98.85it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.85 quantile:  99%|▉| 635/641 [00:07<00:00, 102.02it/\u001b[0m\u001b[A\n",
      " 57%|█████████████████████████▏                  | 8/14 [00:59<00:54,  9.14s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   0%|           | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   2%| | 13/641 [00:00<00:05, 124.11it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   4%| | 26/641 [00:00<00:04, 125.01it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   6%| | 41/641 [00:00<00:04, 132.93it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   9%| | 56/641 [00:00<00:04, 135.54it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  11%| | 71/641 [00:00<00:04, 137.70it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  13%|▏| 85/641 [00:00<00:04, 129.01it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  15%|▏| 98/641 [00:00<00:04, 112.56it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  17%|▏| 110/641 [00:00<00:05, 99.35it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  19%|▏| 122/641 [00:01<00:05, 103.06it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  21%|▏| 137/641 [00:01<00:04, 114.33it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  24%|▏| 153/641 [00:01<00:03, 124.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  27%|▎| 170/641 [00:01<00:03, 134.74it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  29%|▎| 186/641 [00:01<00:03, 140.30it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  31%|▎| 201/641 [00:01<00:03, 129.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  34%|▎| 215/641 [00:01<00:03, 112.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  36%|▎| 229/641 [00:01<00:03, 117.57it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  38%|▍| 242/641 [00:01<00:03, 118.90it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  40%|▍| 257/641 [00:02<00:03, 126.78it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  42%|▍| 271/641 [00:02<00:02, 129.48it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  44%|▍| 285/641 [00:02<00:02, 130.42it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  47%|▍| 302/641 [00:02<00:02, 138.73it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  50%|▍| 318/641 [00:02<00:02, 142.91it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  52%|▌| 334/641 [00:02<00:02, 146.67it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  54%|▌| 349/641 [00:02<00:02, 142.16it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  57%|▌| 364/641 [00:02<00:01, 140.83it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  59%|▌| 380/641 [00:02<00:01, 144.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  62%|▌| 395/641 [00:03<00:01, 133.00it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  64%|▋| 409/641 [00:03<00:01, 132.31it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  66%|▋| 424/641 [00:03<00:01, 135.24it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  68%|▋| 438/641 [00:03<00:01, 133.74it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  71%|▋| 455/641 [00:03<00:01, 142.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  73%|▋| 470/641 [00:03<00:01, 138.83it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  76%|▊| 484/641 [00:03<00:01, 138.62it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  78%|▊| 498/641 [00:03<00:01, 135.60it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  80%|▊| 515/641 [00:03<00:00, 140.84it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  83%|▊| 532/641 [00:04<00:00, 148.25it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  85%|▊| 547/641 [00:04<00:00, 142.32it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  88%|▉| 563/641 [00:04<00:00, 146.41it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  90%|▉| 578/641 [00:04<00:00, 143.50it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  93%|▉| 593/641 [00:04<00:00, 142.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  95%|▉| 608/641 [00:04<00:00, 143.04it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  97%|▉| 623/641 [00:04<00:00, 144.85it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g: 100%|▉| 638/641 [00:04<00:00, 143.09it/s\u001b[0m\u001b[A\n",
      " 64%|████████████████████████████▎               | 9/14 [01:04<00:39,  7.99s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:   0%|          | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:   3%| | 19/641 [00:00<00:03, 187.67it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:   6%| | 41/641 [00:00<00:02, 203.52it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  10%| | 63/641 [00:00<00:02, 208.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  13%|▏| 84/641 [00:00<00:02, 201.45it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  16%|▏| 105/641 [00:00<00:03, 162.44it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  19%|▏| 123/641 [00:00<00:03, 163.10it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  23%|▏| 147/641 [00:00<00:02, 184.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  27%|▎| 170/641 [00:00<00:02, 197.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  30%|▎| 193/641 [00:01<00:02, 203.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  33%|▎| 214/641 [00:01<00:02, 176.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  36%|▎| 233/641 [00:01<00:02, 178.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  40%|▍| 256/641 [00:01<00:02, 190.71it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  43%|▍| 277/641 [00:01<00:01, 193.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  47%|▍| 302/641 [00:01<00:01, 207.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  51%|▌| 327/641 [00:01<00:01, 218.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  55%|▌| 350/641 [00:01<00:01, 217.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  58%|▌| 374/641 [00:01<00:01, 223.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  62%|▌| 397/641 [00:02<00:01, 205.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  66%|▋| 420/641 [00:02<00:01, 210.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  69%|▋| 442/641 [00:02<00:00, 210.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  73%|▋| 467/641 [00:02<00:00, 220.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  76%|▊| 490/641 [00:02<00:00, 212.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  80%|▊| 513/641 [00:02<00:00, 216.70it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  84%|▊| 537/641 [00:02<00:00, 220.56it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  88%|▉| 562/641 [00:02<00:00, 226.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  91%|▉| 585/641 [00:02<00:00, 226.33it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  95%|▉| 608/641 [00:02<00:00, 224.50it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  99%|▉| 632/641 [00:03<00:00, 227.27it/\u001b[0m\u001b[A\n",
      " 71%|██████████████████████████████▋            | 10/14 [01:08<00:26,  6.68s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   0%| | 0/641 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   2%| | 10/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   3%| | 20/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   5%| | 30/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   6%| | 41/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   8%| | 52/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  10%| | 63/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  11%| | 73/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  13%|▏| 83/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  15%|▏| 93/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  16%|▏| 102/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  17%|▏| 110/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  18%|▏| 118/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  20%|▏| 128/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  22%|▏| 140/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  24%|▏| 152/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  26%|▎| 164/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  28%|▎| 177/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  29%|▎| 188/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  31%|▎| 199/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  33%|▎| 209/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  34%|▎| 218/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  36%|▎| 229/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  37%|▎| 238/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  39%|▍| 250/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  41%|▍| 261/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  42%|▍| 272/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  44%|▍| 283/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  46%|▍| 295/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  48%|▍| 307/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  50%|▍| 319/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  52%|▌| 331/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  54%|▌| 343/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  55%|▌| 354/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  57%|▌| 365/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  59%|▌| 378/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  61%|▌| 389/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  62%|▌| 400/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  64%|▋| 410/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  66%|▋| 421/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  67%|▋| 431/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  69%|▋| 442/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  71%|▋| 454/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  73%|▋| 465/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  74%|▋| 476/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  76%|▊| 487/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  78%|▊| 498/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  80%|▊| 510/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  81%|▊| 522/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  83%|▊| 534/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  85%|▊| 545/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  87%|▊| 557/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  89%|▉| 568/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  90%|▉| 579/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  92%|▉| 591/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  94%|▉| 602/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  96%|▉| 614/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  98%|▉| 625/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  99%|▉| 636/641 [\u001b[0m\u001b[A\n",
      " 79%|█████████████████████████████████▊         | 11/14 [01:15<00:20,  6.85s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   0%| | 0/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   1%| | 6/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   2%| | 12/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   3%| | 18/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   4%| | 24/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   5%| | 31/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   6%| | 38/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   7%| | 45/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   8%| | 52/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   9%| | 59/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  10%| | 66/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  11%| | 73/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  12%| | 79/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  13%|▏| 85/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  14%|▏| 91/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  15%|▏| 96/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  16%|▏| 101/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  17%|▏| 106/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  17%|▏| 111/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  18%|▏| 116/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  19%|▏| 123/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  20%|▏| 129/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  21%|▏| 136/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  22%|▏| 144/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  24%|▏| 151/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  25%|▏| 158/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  26%|▎| 165/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  27%|▎| 173/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  28%|▎| 180/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  29%|▎| 187/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  30%|▎| 193/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  31%|▎| 199/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  32%|▎| 204/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  33%|▎| 209/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  33%|▎| 214/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  34%|▎| 221/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  35%|▎| 227/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  36%|▎| 233/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  37%|▎| 239/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  38%|▍| 246/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  39%|▍| 253/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  41%|▍| 260/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  42%|▍| 267/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  43%|▍| 274/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  44%|▍| 281/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  45%|▍| 289/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  46%|▍| 296/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  47%|▍| 303/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  48%|▍| 310/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  49%|▍| 317/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  51%|▌| 324/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  52%|▌| 332/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  53%|▌| 339/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  54%|▌| 346/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  55%|▌| 353/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  56%|▌| 360/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  58%|▌| 369/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  59%|▌| 376/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  60%|▌| 383/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  61%|▌| 390/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  62%|▌| 397/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  63%|▋| 404/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  64%|▋| 411/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  65%|▋| 418/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  66%|▋| 425/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  67%|▋| 432/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  68%|▋| 439/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  70%|▋| 447/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  71%|▋| 454/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  72%|▋| 461/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  73%|▋| 468/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  74%|▋| 475/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  75%|▊| 482/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  76%|▊| 489/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  77%|▊| 496/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  78%|▊| 503/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  80%|▊| 510/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  81%|▊| 517/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  82%|▊| 526/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  83%|▊| 534/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  84%|▊| 541/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  85%|▊| 548/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  87%|▊| 556/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  88%|▉| 563/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  89%|▉| 570/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  90%|▉| 577/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  91%|▉| 584/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  92%|▉| 592/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  93%|▉| 599/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  95%|▉| 606/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  96%|▉| 613/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  97%|▉| 620/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  98%|▉| 627/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  99%|▉| 634/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene: 100%|█| 641/641 \u001b[0m\u001b[A\n",
      " 86%|████████████████████████████████████▊      | 12/14 [01:29<00:17,  8.85s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   0%| | 0/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   1%| | 4/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   1%| | 9/641 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   2%| | 14/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   3%| | 19/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   4%| | 24/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   5%| | 30/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   6%| | 36/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   6%| | 41/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   7%| | 47/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   8%| | 52/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   9%| | 57/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  10%| | 63/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  11%| | 70/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  12%| | 76/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  13%|▏| 81/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  13%|▏| 86/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  14%|▏| 91/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  15%|▏| 95/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  15%|▏| 99/641 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  16%|▏| 103/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  17%|▏| 107/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  17%|▏| 111/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  18%|▏| 115/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  19%|▏| 120/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  20%|▏| 125/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  20%|▏| 130/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  21%|▏| 136/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  22%|▏| 142/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  23%|▏| 148/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  24%|▏| 154/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  25%|▏| 160/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  26%|▎| 166/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  27%|▎| 172/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  28%|▎| 179/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  29%|▎| 185/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  30%|▎| 191/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  31%|▎| 196/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  31%|▎| 201/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  32%|▎| 206/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  33%|▎| 210/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  33%|▎| 214/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  34%|▎| 219/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  35%|▎| 225/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  36%|▎| 230/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  37%|▎| 235/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  37%|▎| 240/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  38%|▍| 246/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  39%|▍| 252/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  40%|▍| 257/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  41%|▍| 263/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  42%|▍| 268/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  43%|▍| 273/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  43%|▍| 278/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  44%|▍| 284/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  45%|▍| 290/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  46%|▍| 296/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  47%|▍| 302/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  48%|▍| 308/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  49%|▍| 314/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  50%|▍| 320/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  51%|▌| 326/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  52%|▌| 332/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  53%|▌| 338/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  54%|▌| 344/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  55%|▌| 350/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  55%|▌| 355/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  56%|▌| 360/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  57%|▌| 367/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  58%|▌| 373/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  59%|▌| 379/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  60%|▌| 385/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  61%|▌| 391/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  62%|▌| 396/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  63%|▋| 401/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  63%|▋| 406/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  64%|▋| 411/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  65%|▋| 417/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  66%|▋| 422/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  67%|▋| 428/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  68%|▋| 433/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  68%|▋| 438/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  69%|▋| 444/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  70%|▋| 450/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  71%|▋| 456/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  72%|▋| 462/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  73%|▋| 468/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  74%|▋| 473/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  75%|▋| 478/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  76%|▊| 484/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  76%|▊| 489/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  77%|▊| 495/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  78%|▊| 501/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  79%|▊| 507/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  80%|▊| 513/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  81%|▊| 519/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  82%|▊| 525/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  83%|▊| 531/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  84%|▊| 537/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  85%|▊| 542/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  85%|▊| 547/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  86%|▊| 554/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  87%|▊| 560/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  88%|▉| 566/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  89%|▉| 572/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  90%|▉| 577/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  91%|▉| 583/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  92%|▉| 589/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  93%|▉| 595/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  94%|▉| 601/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  95%|▉| 607/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  96%|▉| 613/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  97%|▉| 619/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  98%|▉| 625/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  98%|▉| 631/641 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  99%|▉| 637/641 \u001b[0m\u001b[A\n",
      " 93%|███████████████████████████████████████▉   | 13/14 [01:43<00:10, 10.59s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   0%|         | 0/641 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   1%| | 7/641 [00:00<00:09, 63.75it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   2%| | 14/641 [00:00<00:10, 61.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   3%| | 21/641 [00:00<00:10, 60.45it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   4%| | 28/641 [00:00<00:09, 62.70it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   6%| | 36/641 [00:00<00:09, 66.54it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   7%| | 43/641 [00:00<00:08, 66.82it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   8%| | 50/641 [00:00<00:08, 67.71it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   9%| | 57/641 [00:00<00:08, 66.04it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  10%| | 65/641 [00:00<00:08, 68.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  11%| | 72/641 [00:01<00:08, 66.26it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  12%| | 79/641 [00:01<00:08, 63.70it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  13%|▏| 86/641 [00:01<00:09, 59.02it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  14%|▏| 92/641 [00:01<00:10, 51.77it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  15%|▏| 98/641 [00:04<01:12,  7.47it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  16%|▏| 102/641 [00:04<01:00,  8.93it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  17%|▏| 106/641 [00:04<00:49, 10.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  17%|▏| 111/641 [00:04<00:38, 13.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  18%|▏| 116/641 [00:04<00:29, 17.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  19%|▏| 124/641 [00:04<00:20, 25.26it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  20%|▏| 130/641 [00:04<00:16, 30.46it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  22%|▏| 138/641 [00:04<00:12, 38.84it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  23%|▏| 147/641 [00:04<00:10, 48.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  24%|▏| 154/641 [00:05<00:09, 51.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  25%|▎| 162/641 [00:05<00:08, 57.65it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  27%|▎| 170/641 [00:05<00:07, 62.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  28%|▎| 180/641 [00:05<00:06, 69.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  29%|▎| 188/641 [00:05<00:06, 66.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  31%|▎| 196/641 [00:05<00:06, 64.57it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  32%|▎| 203/641 [00:05<00:07, 56.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  33%|▎| 210/641 [00:06<00:08, 52.12it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  34%|▎| 216/641 [00:06<00:08, 50.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  35%|▎| 224/641 [00:06<00:07, 56.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  36%|▎| 230/641 [00:06<00:07, 55.50it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  37%|▎| 236/641 [00:06<00:07, 54.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  38%|▍| 244/641 [00:06<00:06, 60.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  39%|▍| 252/641 [00:06<00:06, 63.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  40%|▍| 259/641 [00:06<00:05, 65.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  41%|▍| 266/641 [00:06<00:05, 66.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  43%|▍| 273/641 [00:07<00:05, 63.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  44%|▍| 280/641 [00:07<00:05, 65.04it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  45%|▍| 289/641 [00:07<00:05, 68.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  46%|▍| 297/641 [00:07<00:04, 69.22it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  48%|▍| 305/641 [00:07<00:04, 72.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  49%|▍| 314/641 [00:07<00:04, 74.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  50%|▌| 322/641 [00:07<00:04, 72.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  51%|▌| 330/641 [00:07<00:04, 74.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  53%|▌| 338/641 [00:07<00:04, 70.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  54%|▌| 346/641 [00:08<00:04, 70.75it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  55%|▌| 354/641 [00:08<00:04, 68.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  56%|▌| 361/641 [00:08<00:04, 66.63it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  58%|▌| 371/641 [00:08<00:03, 74.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  59%|▌| 379/641 [00:08<00:03, 71.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  60%|▌| 387/641 [00:08<00:03, 72.06it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  62%|▌| 395/641 [00:08<00:04, 60.72it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  63%|▋| 402/641 [00:08<00:03, 60.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  64%|▋| 409/641 [00:08<00:03, 62.46it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  65%|▋| 417/641 [00:09<00:03, 66.00it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  66%|▋| 424/641 [00:09<00:03, 65.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  67%|▋| 431/641 [00:09<00:03, 64.75it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  68%|▋| 438/641 [00:09<00:03, 65.02it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  70%|▋| 447/641 [00:09<00:02, 70.75it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  71%|▋| 455/641 [00:09<00:02, 72.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  72%|▋| 463/641 [00:09<00:02, 73.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  73%|▋| 471/641 [00:09<00:02, 67.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  75%|▋| 478/641 [00:09<00:02, 65.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  76%|▊| 486/641 [00:10<00:02, 68.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  77%|▊| 493/641 [00:10<00:02, 67.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  78%|▊| 500/641 [00:10<00:02, 68.12it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  79%|▊| 508/641 [00:10<00:01, 69.02it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  80%|▊| 516/641 [00:10<00:01, 68.90it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  82%|▊| 525/641 [00:10<00:01, 74.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  83%|▊| 533/641 [00:10<00:01, 73.16it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  84%|▊| 541/641 [00:10<00:01, 69.08it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  85%|▊| 548/641 [00:10<00:01, 68.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  87%|▊| 557/641 [00:11<00:01, 73.23it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  88%|▉| 565/641 [00:11<00:01, 72.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  89%|▉| 573/641 [00:11<00:00, 71.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  91%|▉| 581/641 [00:11<00:00, 71.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  92%|▉| 589/641 [00:11<00:00, 72.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  93%|▉| 597/641 [00:11<00:00, 67.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  94%|▉| 605/641 [00:11<00:00, 69.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  96%|▉| 613/641 [00:11<00:00, 71.73it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  97%|▉| 621/641 [00:12<00:00, 71.00it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  98%|▉| 629/641 [00:12<00:00, 72.25it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  99%|▉| 637/641 [00:12<00:00, 68.90it/\u001b[0m\u001b[A\n",
      "100%|███████████████████████████████████████████| 14/14 [01:59<00:00,  8.52s/it]\u001b[A\n",
      "2025-04-10 14:55:51,955 GSEA         INFO     Subsetting TF2G adjacencies for TF with motif.\n",
      "2025-04-10 14:55:52,066 GSEA         INFO     Running GSEA...\n",
      "2025-04-10 14:55:52,066 GSEA         INFO     Generating rankings...\n",
      "100%|█████████████████████████████████████████| 638/638 [00:07<00:00, 88.46it/s]\n",
      "0it [00:00, ?it/s]\n",
      "Running for Positive TF to gene:   0%|        | 40/8330 [00:06<22:45,  6.07it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   3%|▏      | 280/8330 [00:22<02:16, 58.91it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   4%|▎      | 360/8330 [00:23<01:31, 87.05it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   6%|▎     | 480/8330 [00:24<01:08, 115.30it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene: 100%|██████| 8330/8330 [01:49<00:00, 76.37it/s]\n",
      "Running for Negative TF to gene: 100%|██| 8330/8330 [00:00<00:00, 198675.93it/s]\n",
      "2025-04-10 14:57:49,448 GSEA         INFO     Subsetting on adjusted pvalue: 1, minimal NES: 0 and minimal leading edge genes 10\n",
      "2025-04-10 14:57:49,828 GSEA         INFO     Merging eRegulons\n",
      "2025-04-10 14:57:53,788 SCENIC+      INFO     Formatting eGRN as table.\n",
      "2025-04-10 14:57:54,443 SCENIC+      INFO     Calculating triplet ranking.\n",
      "2025-04-10 14:57:54,630 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 14:58:07,981 SCENIC+      INFO     Saving network to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Apr 10 14:58:16 2025]\u001b[0m\n",
      "\u001b[32mFinished job 4.\u001b[0m\n",
      "\u001b[32m9 of 14 steps (64%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 14:58:16 2025]\u001b[0m\n",
      "\u001b[32mlocalrule AUCell_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "    jobid: 3\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 14:58:23,751 SCENIC+      INFO     Reading data.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 14:58:26,680 SCENIC+      INFO     Calculating enrichment scores.\n",
      "2025-04-10 15:00:06,463 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "\u001b[0m\u001b[32m[Thu Apr 10 15:00:08 2025]\u001b[0m\n",
      "\u001b[32mFinished job 3.\u001b[0m\n",
      "\u001b[32m10 of 14 steps (71%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 15:00:08 2025]\u001b[0m\n",
      "\u001b[32mlocalrule eGRN_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    jobid: 13\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 15:00:15,805 SCENIC+      INFO     Loading TF to gene adjacencies.\n",
      "2025-04-10 15:00:16,667 SCENIC+      INFO     Loading region to gene adjacencies.\n",
      "2025-04-10 15:00:17,555 SCENIC+      INFO     Loading cistromes.\n",
      "2025-04-10 15:00:17,601 GSEA         INFO     Thresholding region to gene relationships\n",
      "  0%|                                                    | 0/14 [00:00<?, ?it/s]\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   0%|          | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   9%| | 31/332 [00:00<00:00, 303.18it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  19%|▏| 62/332 [00:00<00:01, 157.31it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  25%|▏| 82/332 [00:00<00:01, 164.45it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  38%|▍| 125/332 [00:00<00:00, 240.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  46%|▍| 154/332 [00:00<00:01, 149.65it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  56%|▌| 187/332 [00:01<00:00, 184.56it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  67%|▋| 224/332 [00:01<00:00, 224.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  78%|▊| 260/332 [00:01<00:00, 255.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  88%|▉| 291/332 [00:01<00:00, 189.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  98%|▉| 326/332 [00:01<00:00, 221.58it/\u001b[0m\u001b[A\n",
      "  7%|███▏                                        | 1/14 [00:03<00:40,  3.15s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   0%|           | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  11%| | 36/332 [00:00<00:00, 355.13it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  22%|▏| 72/332 [00:00<00:00, 265.29it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  31%|▎| 102/332 [00:00<00:00, 277.60it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  41%|▍| 135/332 [00:00<00:01, 190.27it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  48%|▍| 158/332 [00:00<00:00, 180.27it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  58%|▌| 194/332 [00:00<00:00, 223.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  70%|▋| 233/332 [00:00<00:00, 265.35it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  79%|▊| 263/332 [00:01<00:00, 272.96it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  90%|▉| 300/332 [00:01<00:00, 298.24it/s\u001b[0m\u001b[A\n",
      " 14%|██████▎                                     | 2/14 [00:04<00:28,  2.36s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:   0%|          | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  11%| | 37/332 [00:00<00:02, 127.47it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  19%|▏| 62/332 [00:00<00:01, 167.74it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  25%|▎| 83/332 [00:00<00:01, 180.72it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  39%|▍| 131/332 [00:00<00:00, 270.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  49%|▍| 162/332 [00:00<00:00, 232.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  61%|▌| 202/332 [00:00<00:00, 276.05it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  73%|▋| 243/332 [00:00<00:00, 310.65it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  83%|▊| 277/332 [00:01<00:00, 301.08it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  97%|▉| 322/332 [00:01<00:00, 339.02it/\u001b[0m\u001b[A\n",
      " 21%|█████████▍                                  | 3/14 [00:06<00:24,  2.19s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   0%| | 0/332 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   4%| | 14/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   9%| | 31/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  14%|▏| 47/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  19%|▏| 62/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  23%|▏| 75/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  26%|▎| 86/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  32%|▎| 106/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  39%|▍| 128/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  43%|▍| 144/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  47%|▍| 157/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  53%|▌| 175/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  57%|▌| 190/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  62%|▌| 206/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  68%|▋| 226/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  73%|▋| 243/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  79%|▊| 261/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  83%|▊| 277/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  89%|▉| 297/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  94%|▉| 313/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  99%|▉| 328/332 [\u001b[0m\u001b[A\n",
      " 29%|████████████▌                               | 4/14 [00:10<00:28,  2.80s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   0%| | 0/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   4%| | 12/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   8%| | 26/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  12%| | 40/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  16%|▏| 54/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  20%|▏| 66/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  23%|▏| 76/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  26%|▎| 86/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  30%|▎| 101/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  36%|▎| 121/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  41%|▍| 135/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  44%|▍| 146/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  48%|▍| 161/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  53%|▌| 177/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  57%|▌| 190/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  61%|▌| 204/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  66%|▋| 219/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  70%|▋| 234/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  75%|▊| 249/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  79%|▊| 263/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  83%|▊| 275/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  88%|▉| 291/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  92%|▉| 306/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  96%|▉| 319/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene: 100%|█| 332/332 \u001b[0m\u001b[A\n",
      " 36%|███████████████▋                            | 5/14 [00:14<00:29,  3.24s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   0%| | 0/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   0%| | 1/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   4%| | 13/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   8%| | 26/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  12%| | 39/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  16%|▏| 52/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  19%|▏| 64/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  22%|▏| 74/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  25%|▎| 83/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  28%|▎| 94/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  33%|▎| 110/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  39%|▍| 128/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  42%|▍| 141/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  46%|▍| 152/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  50%|▌| 166/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  54%|▌| 179/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  58%|▌| 191/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  61%|▌| 204/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  66%|▋| 218/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  70%|▋| 232/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  74%|▋| 246/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  78%|▊| 259/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  81%|▊| 269/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  85%|▊| 283/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  90%|▉| 298/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  93%|▉| 310/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  97%|▉| 322/332 \u001b[0m\u001b[A\n",
      " 43%|██████████████████▊                         | 6/14 [00:19<00:30,  3.78s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   0%|         | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   6%| | 21/332 [00:00<00:01, 201.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  14%|▏| 45/332 [00:00<00:01, 220.54it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  20%|▏| 68/332 [00:00<00:01, 162.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  26%|▎| 86/332 [00:00<00:01, 150.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  35%|▎| 117/332 [00:00<00:01, 196.05it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  42%|▍| 139/332 [00:00<00:01, 164.11it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  48%|▍| 158/332 [00:00<00:01, 145.76it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  55%|▌| 182/332 [00:01<00:00, 165.10it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  62%|▌| 205/332 [00:01<00:00, 180.36it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  70%|▋| 232/332 [00:01<00:00, 201.31it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  77%|▊| 256/332 [00:01<00:00, 210.08it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  84%|▊| 278/332 [00:01<00:00, 183.63it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  93%|▉| 308/332 [00:01<00:00, 210.66it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g: 100%|▉| 331/332 [00:01<00:00, 205.27it\u001b[0m\u001b[A\n",
      " 50%|██████████████████████                      | 7/14 [00:30<00:42,  6.02s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   0%|          | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   3%| | 11/332 [00:00<00:03, 103.77it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   7%| | 22/332 [00:00<00:03, 101.30it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  10%| | 34/332 [00:00<00:02, 106.58it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  14%|▏| 46/332 [00:00<00:02, 108.12it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  17%|▏| 57/332 [00:00<00:02, 102.79it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  20%|▏| 68/332 [00:00<00:02, 98.23it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  23%|▏| 78/332 [00:00<00:02, 92.51it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  27%|▎| 88/332 [00:00<00:02, 90.04it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  30%|▎| 101/332 [00:01<00:02, 100.96it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  35%|▎| 115/332 [00:01<00:01, 111.04it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  39%|▍| 129/332 [00:01<00:01, 118.54it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  43%|▍| 142/332 [00:01<00:01, 96.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  46%|▍| 153/332 [00:01<00:01, 97.46it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  50%|▍| 165/332 [00:01<00:01, 101.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  53%|▌| 177/332 [00:01<00:01, 103.54it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  57%|▌| 189/332 [00:01<00:01, 106.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  61%|▌| 201/332 [00:01<00:01, 109.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  64%|▋| 214/332 [00:02<00:01, 113.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  68%|▋| 226/332 [00:02<00:00, 114.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  72%|▋| 238/332 [00:02<00:00, 113.05it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  75%|▊| 250/332 [00:02<00:00, 114.26it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  79%|▊| 262/332 [00:02<00:00, 111.46it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  83%|▊| 274/332 [00:02<00:00, 104.85it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  86%|▊| 285/332 [00:02<00:00, 105.83it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  90%|▉| 299/332 [00:02<00:00, 113.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  94%|▉| 312/332 [00:02<00:00, 116.78it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  98%|▉| 325/332 [00:03<00:00, 119.29it/\u001b[0m\u001b[A\n",
      " 57%|█████████████████████████▏                  | 8/14 [00:34<00:33,  5.64s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:   0%|           | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:   4%| | 14/332 [00:00<00:02, 138.03it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:   8%| | 28/332 [00:00<00:02, 138.73it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  13%|▏| 42/332 [00:00<00:02, 139.26it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  17%|▏| 56/332 [00:00<00:02, 131.68it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  21%|▏| 70/332 [00:00<00:02, 124.51it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  25%|▎| 83/332 [00:00<00:02, 116.99it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  29%|▎| 96/332 [00:00<00:01, 120.75it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  35%|▎| 115/332 [00:00<00:01, 140.51it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  39%|▍| 131/332 [00:00<00:01, 144.91it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  44%|▍| 146/332 [00:01<00:01, 120.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  49%|▍| 163/332 [00:01<00:01, 131.23it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  54%|▌| 178/332 [00:01<00:01, 133.86it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  58%|▌| 194/332 [00:01<00:00, 139.99it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  63%|▋| 210/332 [00:01<00:00, 144.54it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  68%|▋| 225/332 [00:01<00:00, 145.67it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  72%|▋| 240/332 [00:01<00:00, 144.39it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  77%|▊| 255/332 [00:01<00:00, 144.98it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  81%|▊| 270/332 [00:01<00:00, 136.19it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  86%|▊| 285/332 [00:02<00:00, 138.34it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  91%|▉| 303/332 [00:02<00:00, 147.81it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.9 quantile:  96%|▉| 320/332 [00:02<00:00, 152.76it/s\u001b[0m\u001b[A\n",
      " 64%|████████████████████████████▎               | 9/14 [00:38<00:24,  4.83s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:   0%|          | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:   7%| | 22/332 [00:00<00:01, 214.03it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  14%|▏| 46/332 [00:00<00:01, 223.11it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  21%|▏| 69/332 [00:00<00:01, 200.60it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  27%|▎| 90/332 [00:00<00:01, 186.00it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  37%|▎| 122/332 [00:00<00:00, 228.06it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  44%|▍| 146/332 [00:00<00:00, 191.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  52%|▌| 172/332 [00:00<00:00, 209.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  59%|▌| 196/332 [00:00<00:00, 214.96it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  67%|▋| 223/332 [00:01<00:00, 228.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  74%|▋| 247/332 [00:01<00:00, 230.02it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  82%|▊| 271/332 [00:01<00:00, 220.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  89%|▉| 296/332 [00:01<00:00, 228.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, 0.95 quantile:  97%|▉| 323/332 [00:01<00:00, 238.23it/\u001b[0m\u001b[A\n",
      " 71%|██████████████████████████████▋            | 10/14 [00:40<00:16,  4.00s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   0%| | 0/332 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   3%| | 10/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   6%| | 20/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   9%| | 31/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  13%|▏| 42/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  16%|▏| 53/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  19%|▏| 64/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  22%|▏| 74/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  25%|▎| 83/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  28%|▎| 92/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  32%|▎| 106/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  37%|▎| 122/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  40%|▍| 134/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  44%|▍| 145/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  47%|▍| 157/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  51%|▌| 168/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  54%|▌| 179/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  57%|▌| 190/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  61%|▌| 203/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  65%|▋| 215/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  68%|▋| 227/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  72%|▋| 238/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  75%|▊| 250/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  79%|▊| 261/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  82%|▊| 272/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  86%|▊| 284/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  89%|▉| 297/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  93%|▉| 309/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  97%|▉| 321/332 [\u001b[0m\u001b[A\n",
      " 79%|█████████████████████████████████▊         | 11/14 [00:44<00:12,  4.24s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   0%| | 0/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   2%| | 7/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   4%| | 14/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   6%| | 21/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   9%| | 29/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  11%| | 36/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  13%|▏| 43/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  15%|▏| 50/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  17%|▏| 57/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  19%|▏| 64/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  21%|▏| 70/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  23%|▏| 76/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  25%|▏| 82/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  27%|▎| 88/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  28%|▎| 94/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  31%|▎| 103/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  33%|▎| 111/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  37%|▎| 122/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  39%|▍| 130/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  42%|▍| 138/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  44%|▍| 145/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  46%|▍| 153/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  48%|▍| 160/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  51%|▌| 168/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  53%|▌| 175/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  55%|▌| 182/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  57%|▌| 190/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  59%|▌| 197/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  62%|▌| 205/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  64%|▋| 213/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  66%|▋| 220/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  68%|▋| 227/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  70%|▋| 234/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  73%|▋| 241/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  75%|▊| 249/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  77%|▊| 256/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  79%|▊| 263/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  81%|▊| 270/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  83%|▊| 277/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  86%|▊| 284/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  88%|▉| 291/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  90%|▉| 300/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  93%|▉| 308/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  95%|▉| 316/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  98%|▉| 324/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene: 100%|█| 332/332 \u001b[0m\u001b[A\n",
      " 86%|████████████████████████████████████▊      | 12/14 [00:50<00:09,  4.75s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   0%| | 0/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   2%| | 5/332 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   3%| | 10/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   5%| | 15/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   6%| | 20/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   8%| | 26/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   9%| | 31/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  11%| | 36/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  13%|▏| 42/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  14%|▏| 47/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  16%|▏| 53/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  17%|▏| 58/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  19%|▏| 63/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  20%|▏| 68/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  22%|▏| 73/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  23%|▏| 78/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  25%|▎| 83/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  26%|▎| 87/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  28%|▎| 92/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  30%|▎| 99/332 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  32%|▎| 105/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  33%|▎| 111/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  36%|▎| 119/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  38%|▍| 126/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  40%|▍| 132/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  42%|▍| 138/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  43%|▍| 143/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  45%|▍| 148/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  46%|▍| 154/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  48%|▍| 159/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  50%|▍| 165/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  52%|▌| 171/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  53%|▌| 177/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  55%|▌| 182/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  57%|▌| 188/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  58%|▌| 194/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  60%|▌| 200/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  62%|▌| 206/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  64%|▋| 212/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  66%|▋| 218/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  67%|▋| 224/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  69%|▋| 230/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  71%|▋| 236/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  73%|▋| 242/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  75%|▊| 249/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  77%|▊| 255/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  79%|▊| 261/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  80%|▊| 267/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  82%|▊| 273/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  84%|▊| 279/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  86%|▊| 285/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  88%|▉| 291/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  90%|▉| 298/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  92%|▉| 304/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  93%|▉| 310/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  95%|▉| 316/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  97%|▉| 322/332 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  99%|▉| 328/332 \u001b[0m\u001b[A\n",
      " 93%|███████████████████████████████████████▉   | 13/14 [00:59<00:05,  5.89s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   0%|         | 0/332 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   2%| | 8/332 [00:00<00:04, 71.41it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   5%| | 16/332 [00:00<00:04, 68.35it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   7%| | 23/332 [00:00<00:04, 66.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   9%| | 31/332 [00:00<00:04, 69.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  12%| | 39/332 [00:00<00:04, 69.75it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  14%|▏| 47/332 [00:00<00:04, 69.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  16%|▏| 54/332 [00:00<00:04, 69.01it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  18%|▏| 61/332 [00:00<00:04, 66.06it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  20%|▏| 68/332 [00:01<00:04, 62.44it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  23%|▏| 75/332 [00:01<00:04, 59.58it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  25%|▏| 82/332 [00:01<00:04, 58.39it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  27%|▎| 88/332 [00:01<00:04, 57.30it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  29%|▎| 96/332 [00:01<00:03, 62.13it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  32%|▎| 106/332 [00:01<00:03, 71.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  35%|▎| 115/332 [00:01<00:02, 76.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  38%|▍| 126/332 [00:01<00:02, 83.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  41%|▍| 135/332 [00:01<00:02, 69.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  43%|▍| 143/332 [00:02<00:03, 60.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  45%|▍| 150/332 [00:02<00:02, 61.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  48%|▍| 158/332 [00:02<00:02, 64.90it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  50%|▌| 166/332 [00:02<00:02, 67.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  52%|▌| 174/332 [00:02<00:02, 68.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  55%|▌| 182/332 [00:02<00:02, 68.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  58%|▌| 191/332 [00:02<00:01, 73.31it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  60%|▌| 199/332 [00:02<00:01, 74.05it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  62%|▌| 207/332 [00:03<00:01, 75.58it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  65%|▋| 215/332 [00:03<00:01, 76.31it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  67%|▋| 223/332 [00:03<00:01, 75.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  70%|▋| 231/332 [00:03<00:01, 74.85it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  72%|▋| 239/332 [00:03<00:01, 75.45it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  74%|▋| 247/332 [00:03<00:01, 74.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  77%|▊| 255/332 [00:03<00:01, 74.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  79%|▊| 263/332 [00:03<00:00, 71.60it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  82%|▊| 271/332 [00:03<00:00, 67.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  84%|▊| 279/332 [00:04<00:00, 69.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  86%|▊| 287/332 [00:04<00:00, 69.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  89%|▉| 297/332 [00:04<00:00, 75.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  92%|▉| 306/332 [00:04<00:00, 77.21it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  95%|▉| 314/332 [00:04<00:00, 77.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  97%|▉| 323/332 [00:04<00:00, 78.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g: 100%|▉| 331/332 [00:04<00:00, 76.92it/\u001b[0m\u001b[A\n",
      "100%|███████████████████████████████████████████| 14/14 [01:07<00:00,  4.80s/it]\u001b[A\n",
      "2025-04-10 15:01:24,852 GSEA         INFO     Subsetting TF2G adjacencies for TF with motif.\n",
      "2025-04-10 15:01:24,940 GSEA         INFO     Running GSEA...\n",
      "2025-04-10 15:01:24,940 GSEA         INFO     Generating rankings...\n",
      "100%|████████████████████████████████████████| 324/324 [00:01<00:00, 174.97it/s]\n",
      "0it [00:00, ?it/s]\n",
      "Running for Positive TF to gene:   1%|        | 40/4294 [00:06<10:50,  6.54it/s]\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   2%|▏       | 80/4294 [00:19<16:07,  4.36it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:  24%|█▏   | 1020/4294 [00:25<00:19, 165.98it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene: 100%|██████| 4294/4294 [00:58<00:00, 73.94it/s]\n",
      "Running for Negative TF to gene: 100%|██| 4294/4294 [00:00<00:00, 212639.36it/s]\n",
      "2025-04-10 15:02:26,067 GSEA         INFO     Subsetting on adjusted pvalue: 1, minimal NES: 0 and minimal leading edge genes 10\n",
      "2025-04-10 15:02:26,253 GSEA         INFO     Merging eRegulons\n",
      "2025-04-10 15:02:27,888 SCENIC+      INFO     Formatting eGRN as table.\n",
      "2025-04-10 15:02:28,485 SCENIC+      INFO     Calculating triplet ranking.\n",
      "2025-04-10 15:02:28,617 cisTarget    INFO     Reading cisTarget database\n",
      "2025-04-10 15:02:38,305 SCENIC+      INFO     Saving network to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Apr 10 15:02:40 2025]\u001b[0m\n",
      "\u001b[32mFinished job 13.\u001b[0m\n",
      "\u001b[32m11 of 14 steps (79%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 15:02:40 2025]\u001b[0m\n",
      "\u001b[32mlocalrule AUCell_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\n",
      "    jobid: 12\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 15:02:48,326 SCENIC+      INFO     Reading data.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 15:02:51,189 SCENIC+      INFO     Calculating enrichment scores.\n",
      "2025-04-10 15:04:29,607 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\n",
      "\u001b[0m\u001b[32m[Thu Apr 10 15:04:34 2025]\u001b[0m\n",
      "\u001b[32mFinished job 12.\u001b[0m\n",
      "\u001b[32m12 of 14 steps (86%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 15:04:34 2025]\u001b[0m\n",
      "\u001b[32mlocalrule scplus_mudata:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 1\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-04-10 15:04:41,996 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 15:04:44,800 SCENIC+      INFO     Reading AUC values.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-04-10 15:04:44,886 SCENIC+      INFO     Reading eRegulon metadata.\n",
      "2025-04-10 15:04:45,109 SCENIC+      INFO     Generating MuData object.\n",
      "2025-04-10 15:04:45,733 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "\u001b[0m\u001b[32m[Thu Apr 10 15:04:55 2025]\u001b[0m\n",
      "\u001b[32mFinished job 1.\u001b[0m\n",
      "\u001b[32m13 of 14 steps (93%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 15:04:55 2025]\u001b[0m\n",
      "\u001b[32mlocalrule all:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Apr 10 15:04:55 2025]\u001b[0m\n",
      "\u001b[32mFinished job 0.\u001b[0m\n",
      "\u001b[32m14 of 14 steps (100%) done\u001b[0m\n",
      "\u001b[33mComplete log: .snakemake/log/2025-04-10T135312.126428.snakemake.log\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! snakemake --cores 24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "a15c4fc1-ee03-45ce-9cf2-9720a5506716",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 24\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                            count\n",
      "---------------------------  -------\n",
      "AUCell_direct                      1\n",
      "AUCell_extended                    1\n",
      "all                                1\n",
      "download_genome_annotations        1\n",
      "eGRN_direct                        1\n",
      "eGRN_extended                      1\n",
      "get_search_space                   1\n",
      "motif_enrichment_cistarget         1\n",
      "motif_enrichment_dem               1\n",
      "prepare_GEX_ACC_multiome           1\n",
      "prepare_menr                       1\n",
      "region_to_gene                     1\n",
      "scplus_mudata                      1\n",
      "tf_to_gene                         1\n",
      "total                             14\n",
      "\n",
      "Select jobs to execute...\n",
      "Execute 3 jobs...\n",
      "\n",
      "[Thu Apr 10 13:53:12 2025]\n",
      "localrule motif_enrichment_cistarget:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\n",
      "    jobid: 9\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "\n",
      "[Thu Apr 10 13:53:12 2025]\n",
      "localrule download_genome_annotations:\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "    jobid: 8\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "\n",
      "[Thu Apr 10 13:53:12 2025]\n",
      "localrule prepare_GEX_ACC_multiome:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/cistopic_obj.pkl, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/concatenated_adata_log_pca_celltypist_23012025_with_raw_counts.BALL.h5ad\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    jobid: 2\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 13:53:51 2025]\n",
      "Finished job 2.\n",
      "1 of 14 steps (7%) done\n",
      "[Thu Apr 10 13:54:18 2025]\n",
      "Finished job 8.\n",
      "2 of 14 steps (14%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 13:54:18 2025]\n",
      "localrule get_search_space:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    jobid: 11\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 13:55:36 2025]\n",
      "Finished job 9.\n",
      "3 of 14 steps (21%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 13:55:36 2025]\n",
      "localrule motif_enrichment_dem:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\n",
      "    jobid: 7\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 13:57:09 2025]\n",
      "Finished job 11.\n",
      "4 of 14 steps (29%) done\n",
      "Select jobs to execute...\n",
      "[Thu Apr 10 13:57:37 2025]\n",
      "Finished job 7.\n",
      "5 of 14 steps (36%) done\n",
      "Execute 2 jobs...\n",
      "\n",
      "[Thu Apr 10 13:57:37 2025]\n",
      "localrule prepare_menr:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\n",
      "    jobid: 6\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "\n",
      "[Thu Apr 10 13:57:37 2025]\n",
      "localrule region_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    jobid: 10\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 13:58:24 2025]\n",
      "Finished job 6.\n",
      "6 of 14 steps (43%) done\n",
      "Select jobs to execute...\n",
      "[Thu Apr 10 14:14:15 2025]\n",
      "Finished job 10.\n",
      "7 of 14 steps (50%) done\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 14:14:15 2025]\n",
      "localrule tf_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    jobid: 5\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 14:53:42 2025]\n",
      "Finished job 5.\n",
      "8 of 14 steps (57%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 14:53:42 2025]\n",
      "localrule eGRN_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    jobid: 4\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 14:58:16 2025]\n",
      "Finished job 4.\n",
      "9 of 14 steps (64%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 14:58:16 2025]\n",
      "localrule AUCell_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "    jobid: 3\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 15:00:08 2025]\n",
      "Finished job 3.\n",
      "10 of 14 steps (71%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 15:00:08 2025]\n",
      "localrule eGRN_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    jobid: 13\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 15:02:40 2025]\n",
      "Finished job 13.\n",
      "11 of 14 steps (79%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 15:02:40 2025]\n",
      "localrule AUCell_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\n",
      "    jobid: 12\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 15:04:34 2025]\n",
      "Finished job 12.\n",
      "12 of 14 steps (86%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 15:04:34 2025]\n",
      "localrule scplus_mudata:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 1\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 15:04:55 2025]\n",
      "Finished job 1.\n",
      "13 of 14 steps (93%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Apr 10 15:04:55 2025]\n",
      "localrule all:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Apr 10 15:04:55 2025]\n",
      "Finished job 0.\n",
      "14 of 14 steps (100%) done\n",
      "Complete log: .snakemake/log/2025-04-10T135312.126428.snakemake.log\n"
     ]
    }
   ],
   "source": [
    "! cat .snakemake/log/2025-04-10T135312.126428.snakemake.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "74820415-4056-413c-b327-811a1a02679e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Package                              Version\n",
      "------------------------------------ -----------------\n",
      "adjustText                           1.0.4\n",
      "aiohttp                              3.9.3\n",
      "aiosignal                            1.3.1\n",
      "anndata                              0.10.5.post1\n",
      "annoy                                1.17.3\n",
      "appdirs                              1.4.4\n",
      "arboreto                             0.1.6\n",
      "argparse-dataclass                   2.0.0\n",
      "array_api_compat                     1.5.1\n",
      "asttokens                            2.4.1\n",
      "attr                                 0.3.2\n",
      "attrs                                23.2.0\n",
      "bbknn                                1.6.0\n",
      "beautifulsoup4                       4.12.3\n",
      "bidict                               0.23.1\n",
      "bioservices                          1.11.2\n",
      "blosc2                               2.5.1\n",
      "bokeh                                3.4.0\n",
      "boltons                              23.1.1\n",
      "bs4                                  0.0.2\n",
      "cattrs                               23.2.3\n",
      "certifi                              2024.2.2\n",
      "charset-normalizer                   3.3.2\n",
      "click                                8.1.7\n",
      "cloudpickle                          3.0.0\n",
      "colorama                             0.4.6\n",
      "colorlog                             6.8.2\n",
      "comm                                 0.2.2\n",
      "conda-inject                         1.3.1\n",
      "ConfigArgParse                       1.7\n",
      "connection_pool                      0.0.3\n",
      "contourpy                            1.2.0\n",
      "ctxcore                              0.2.0\n",
      "cycler                               0.12.1\n",
      "Cython                               0.29.37\n",
      "cytoolz                              0.12.3\n",
      "dask                                 2024.2.1\n",
      "dataclasses-json                     0.6.4\n",
      "datrie                               0.8.2\n",
      "debugpy                              1.8.1\n",
      "decorator                            5.1.1\n",
      "dill                                 0.3.8\n",
      "distributed                          2024.2.1\n",
      "docutils                             0.20.1\n",
      "dpath                                2.1.6\n",
      "easydev                              0.13.1\n",
      "et-xmlfile                           1.1.0\n",
      "executing                            2.0.1\n",
      "fastjsonschema                       2.19.1\n",
      "fbpca                                1.0\n",
      "filelock                             3.13.1\n",
      "fonttools                            4.50.0\n",
      "frozendict                           2.4.0\n",
      "frozenlist                           1.4.1\n",
      "fsspec                               2024.3.1\n",
      "future                               1.0.0\n",
      "gensim                               4.3.2\n",
      "geosketch                            1.2\n",
      "gevent                               24.2.1\n",
      "gitdb                                4.0.11\n",
      "GitPython                            3.1.42\n",
      "globre                               0.1.5\n",
      "greenlet                             3.0.3\n",
      "grequests                            0.7.0\n",
      "gseapy                               0.10.8\n",
      "h5py                                 3.10.0\n",
      "harmonypy                            0.0.9\n",
      "humanfriendly                        10.0\n",
      "idna                                 3.6\n",
      "igraph                               0.11.4\n",
      "imageio                              2.34.0\n",
      "immutables                           0.20\n",
      "importlib-metadata                   7.0.1\n",
      "importlib_resources                  6.1.2\n",
      "interlap                             0.2.7\n",
      "intervaltree                         3.1.0\n",
      "ipykernel                            6.29.3\n",
      "ipython                              8.22.2\n",
      "jedi                                 0.19.1\n",
      "Jinja2                               3.1.3\n",
      "joblib                               1.3.2\n",
      "jsonpickle                           3.0.3\n",
      "jsonschema                           4.21.1\n",
      "jsonschema-specifications            2023.12.1\n",
      "jupyter_client                       8.6.1\n",
      "jupyter_core                         5.7.2\n",
      "kaleido                              0.2.1\n",
      "kiwisolver                           1.4.5\n",
      "lazy_loader                          0.3\n",
      "lda                                  3.0.0\n",
      "leidenalg                            0.10.2\n",
      "line-profiler                        4.1.2\n",
      "llvmlite                             0.42.0\n",
      "locket                               1.0.0\n",
      "loompy                               3.0.7\n",
      "loomxpy                              0.4.2\n",
      "lxml                                 5.1.0\n",
      "lz4                                  4.3.3\n",
      "MACS2                                2.2.9.1\n",
      "markdown-it-py                       3.0.0\n",
      "MarkupSafe                           2.1.5\n",
      "marshmallow                          3.21.1\n",
      "matplotlib                           3.6.3\n",
      "matplotlib-inline                    0.1.6\n",
      "mdurl                                0.1.2\n",
      "mizani                               0.9.3\n",
      "msgpack                              1.0.8\n",
      "mudata                               0.2.3\n",
      "multidict                            6.0.5\n",
      "multiprocessing_on_dill              3.5.0a4\n",
      "mypy-extensions                      1.0.0\n",
      "natsort                              8.4.0\n",
      "nbformat                             5.10.3\n",
      "ncls                                 0.0.68\n",
      "ndindex                              1.8\n",
      "nest-asyncio                         1.6.0\n",
      "networkx                             3.2.1\n",
      "numba                                0.59.0\n",
      "numexpr                              2.9.0\n",
      "numpy                                1.26.4\n",
      "numpy-groupies                       0.10.2\n",
      "openpyxl                             3.1.2\n",
      "packaging                            24.0\n",
      "pandas                               1.5.0\n",
      "parso                                0.8.3\n",
      "partd                                1.4.1\n",
      "patsy                                0.5.6\n",
      "pexpect                              4.9.0\n",
      "pillow                               10.2.0\n",
      "pip                                  24.0\n",
      "plac                                 1.4.3\n",
      "platformdirs                         4.2.0\n",
      "plotly                               5.19.0\n",
      "plotnine                             0.12.4\n",
      "polars                               0.20.13\n",
      "progressbar2                         4.4.2\n",
      "prompt-toolkit                       3.0.43\n",
      "protobuf                             5.26.0\n",
      "psutil                               5.9.8\n",
      "ptyprocess                           0.7.0\n",
      "PuLP                                 2.8.0\n",
      "pure-eval                            0.2.2\n",
      "py-cpuinfo                           9.0.0\n",
      "pyarrow                              15.0.0\n",
      "pyarrow-hotfix                       0.6\n",
      "pybedtools                           0.9.1\n",
      "pybigtools                           0.1.2\n",
      "pyBigWig                             0.3.22\n",
      "pybiomart                            0.2.0\n",
      "pycistarget                          1.0a2\n",
      "pycisTopic                           2.0a0\n",
      "pyfasta                              0.5.2\n",
      "pygam                                0.9.0\n",
      "Pygments                             2.17.2\n",
      "pynndescent                          0.5.11\n",
      "pyparsing                            3.1.2\n",
      "pyranges                             0.0.111\n",
      "pyrle                                0.0.39\n",
      "pysam                                0.22.0\n",
      "pyscenic                             0.12.1+7.g1cd059f\n",
      "python-dateutil                      2.9.0.post0\n",
      "python-utils                         3.8.2\n",
      "pytz                                 2024.1\n",
      "pyvis                                0.3.2\n",
      "PyYAML                               6.0.1\n",
      "pyzmq                                25.1.2\n",
      "ray                                  2.9.3\n",
      "referencing                          0.34.0\n",
      "requests                             2.31.0\n",
      "requests-cache                       1.2.0\n",
      "reretry                              0.11.8\n",
      "rich                                 13.7.1\n",
      "rich-argparse                        1.4.0\n",
      "rpds-py                              0.18.0\n",
      "scanorama                            1.7.4\n",
      "scanpy                               1.8.2\n",
      "scatac_fragment_tools                0.1.0\n",
      "scenicplus                           1.0a1\n",
      "scikit-image                         0.22.0\n",
      "scikit-learn                         1.3.2\n",
      "scipy                                1.12.0\n",
      "scrublet                             0.2.3\n",
      "seaborn                              0.13.2\n",
      "setuptools                           65.5.0\n",
      "sinfo                                0.3.4\n",
      "six                                  1.16.0\n",
      "smart-open                           6.4.0\n",
      "smmap                                5.0.1\n",
      "snakemake                            8.5.5\n",
      "snakemake-interface-common           1.17.1\n",
      "snakemake-interface-executor-plugins 8.2.0\n",
      "snakemake-interface-report-plugins   1.0.0\n",
      "snakemake-interface-storage-plugins  3.1.1\n",
      "sorted-nearest                       0.0.39\n",
      "sortedcontainers                     2.4.0\n",
      "soupsieve                            2.5\n",
      "stack-data                           0.6.3\n",
      "statistics                           1.0.3.5\n",
      "statsmodels                          0.14.1\n",
      "stdlib-list                          0.10.0\n",
      "stopit                               1.1.2\n",
      "suds-community                       1.1.2\n",
      "tables                               3.9.2\n",
      "tabulate                             0.9.0\n",
      "tblib                                3.0.0\n",
      "tenacity                             8.2.3\n",
      "texttable                            1.7.0\n",
      "threadpoolctl                        3.4.0\n",
      "throttler                            1.2.2\n",
      "tifffile                             2024.2.12\n",
      "tmtoolkit                            0.12.0\n",
      "toolz                                0.12.1\n",
      "toposort                             1.10\n",
      "tornado                              6.4\n",
      "tqdm                                 4.66.2\n",
      "traitlets                            5.14.2\n",
      "tspex                                0.6.3\n",
      "typing                               3.7.4.3\n",
      "typing_extensions                    4.10.0\n",
      "typing-inspect                       0.9.0\n",
      "umap-learn                           0.5.5\n",
      "url-normalize                        1.4.3\n",
      "urllib3                              2.2.1\n",
      "wcwidth                              0.2.13\n",
      "wheel                                0.43.0\n",
      "wrapt                                1.16.0\n",
      "xlrd                                 2.0.1\n",
      "xmltodict                            0.13.0\n",
      "xyzservices                          2023.10.1\n",
      "yarl                                 1.9.4\n",
      "yte                                  1.5.4\n",
      "zict                                 3.0.0\n",
      "zipp                                 3.18.1\n",
      "zope.event                           5.0\n",
      "zope.interface                       6.2\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.0.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip list"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pycistopic_qc_may24",
   "language": "python",
   "name": "pycistopic_2a"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
