{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "79b46d6a-5972-4afa-a21d-a2195611500d",
   "metadata": {},
   "source": [
    "# SCENIC+"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9e74f852-c69e-4265-a760-0791dc7dc963",
   "metadata": {},
   "source": [
    "# 01 Database preparation\n",
    "\n",
    "All the steps up until `RNA preparation` were done in command line, but I copy them here for posterity\n",
    "\n",
    "The code is based on SCENIC+ tutorial; see: https://scenicplus.readthedocs.io/en/latest/tutorials.html"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "fc707e38-f076-4952-8573-8eff41d3c6cd",
   "metadata": {},
   "source": [
    "to create custom database, I create a conda environment: https://github.com/aertslab/create_cisTarget_databases\n",
    "\n",
    "with the following packages:\n",
    "```\n",
    "Package         Version\n",
    "--------------- -----------\n",
    "flatbuffers     25.2.10\n",
    "llvmlite        0.40.1\n",
    "numba           0.57.1\n",
    "numpy           1.24.4\n",
    "pandas          2.3.1\n",
    "pip             25.1.1\n",
    "pyarrow         20.0.0\n",
    "python-dateutil 2.9.0.post0\n",
    "pytz            2025.2\n",
    "setuptools      80.9.0\n",
    "six             1.17.0\n",
    "tzdata          2025.2\n",
    "wheel           0.45.1\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7edd61df-49e1-4e1e-8fd4-b9d19a0b1fc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "git clone https://github.com/aertslab/create_cisTarget_databases"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7aaa71e3-0614-4f4a-9ff7-8b0af3a3dfe1",
   "metadata": {},
   "source": [
    "## Cluster buster download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3586709d-f691-47ff-80a4-0f2207639e1c",
   "metadata": {},
   "outputs": [],
   "source": [
    "wget https://resources.aertslab.org/cistarget/programs/cbust\n",
    "chmod a+x cbust"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c38ca57d-52de-4d28-8be0-0e419afa53f1",
   "metadata": {},
   "source": [
    "## Motif collection download"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "40f0ecef-bbba-4764-82ea-64e7752cd774",
   "metadata": {},
   "outputs": [],
   "source": [
    "mkdir -p aertslab_motif_colleciton\n",
    "wget -O aertslab_motif_colleciton/v10nr_clust_public.zip https://resources.aertslab.org/cistarget/motif_collections/v10nr_clust_public/v10nr_clust_public.zip"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "066f5f33-c70b-4f18-b686-974d31942fb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd aertslab_motif_colleciton; unzip -q v10nr_clust_public.zip\n",
    "cd .."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9087ce69-258f-4991-a303-6d18086e1595",
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls aertslab_motif_colleciton/v10nr_clust_public/snapshots/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2afd35e-fcef-4c1d-8d78-d09b9356f00f",
   "metadata": {},
   "outputs": [],
   "source": [
    "!ls -l aertslab_motif_colleciton/v10nr_clust_public/singletons | head"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "161b2e80-2236-4339-b37e-e910a0dacf68",
   "metadata": {},
   "source": [
    "## Fasta from consensus regions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "912f3c88-f27c-4b07-8f70-dde5fc773cb9",
   "metadata": {},
   "outputs": [],
   "source": [
    "module load BEDTools/2.31.1-GCC-10.3.0\n",
    "\n",
    "REGION_BED=\"/preprocessing/atac/outs/consensus_peak_calling/consensus_regions.bed\"\n",
    "GENOME_FASTA=\"/genomes/homo_sapiens/hg38_ucsc/fasta/hg38.fa\"\n",
    "CHROMSIZES=\"/genomes/homo_sapiens/hg38_ucsc/fasta/hg38.chrom.sizes\"\n",
    "DATABASE_PREFIX=\"10x_ONT_BALL\"\n",
    "SCRIPT_DIR=\"/preprocessing/grn/create_cisTarget_databases\"\n",
    "\n",
    "${SCRIPT_DIR}/create_fasta_with_padded_bg_from_bed.sh \\\n",
    "        ${GENOME_FASTA} \\\n",
    "        ${CHROMSIZES} \\\n",
    "        ${REGION_BED} \\\n",
    "        hg38.BALL.with_1kb_bg_padding.fa \\\n",
    "        1000 \\\n",
    "        yes\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "e0b7dd7c-aa22-4e1c-b37b-e07f50dcee22",
   "metadata": {},
   "source": [
    "## Create cistarget database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2924057f-3dcc-4177-bece-226dbbbca9c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "mamba activate create_cistarget_databases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "8ac5ea09-e930-4f62-80ad-1e7922f0a1a7",
   "metadata": {},
   "outputs": [],
   "source": [
    "ls aertslab_motif_colleciton/v10nr_clust_public/singletons > motifs.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3085aadb-2a97-4946-877e-28899f089fc1",
   "metadata": {},
   "outputs": [],
   "source": [
    "OUT_DIR=\"/preprocessing/grn/\"\n",
    "CBDIR=\"${OUT_DIR}/aertslab_motif_colleciton/v10nr_clust_public/singletons\"\n",
    "FASTA_FILE=\"${OUT_DIR}/hg38.BALL.with_1kb_bg_padding.fa\"\n",
    "MOTIF_LIST=\"${OUT_DIR}/motifs.txt\"\n",
    "DATABASE_PREFIX=\"10x_ONT_BALL\"\n",
    "\n",
    "\"${OUT_DIR}/create_cisTarget_databases/create_cistarget_motif_databases.py\" \\\n",
    "    -f ${FASTA_FILE} \\\n",
    "    -M ${CBDIR} \\\n",
    "    -m ${MOTIF_LIST} \\\n",
    "    -o ${OUT_DIR}/${DATABASE_PREFIX} \\\n",
    "    --bgpadding 1000 \\\n",
    "    -t 20"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "093a0b3e-fae8-49a1-b80b-16ea0e87fef5",
   "metadata": {},
   "source": [
    "# 02 RNA preparation\n",
    "\n",
    " Save rna counts in adata.raw and change count matrix barcode to ATAC one"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "edd53f50-9492-4102-85f1-3129a7d25db3",
   "metadata": {},
   "source": [
    "From this step on, all is done in jupyter notebook"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e44f8610-986b-46d2-a7e9-4800c71e2a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "0fe07a7c-f455-4ef4-91a5-cbb81d612ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna = sc.read_h5ad(\"/lustre1/project/stg_00096/home/projects/2023_Cools_B-ALL/multiome_adata_04072025.h5ad\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "397d69ed-b4db-417d-a448-34eb9ab227b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Layers with keys: counts, log_counts"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.layers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1d6a2b44-a39b-4991-a815-8182c30f671c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>dna_total_counts</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>log1p_n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>log1p_total_counts</th>\n",
       "      <th>pct_counts_in_top_20_genes</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "      <th>pct_counts_ribo</th>\n",
       "      <th>pct_counts_hb</th>\n",
       "      <th>...</th>\n",
       "      <th>leiden_1.25</th>\n",
       "      <th>leiden_1.5</th>\n",
       "      <th>leiden_2.0</th>\n",
       "      <th>leiden_5.0</th>\n",
       "      <th>leiden_10.0</th>\n",
       "      <th>leiden_20.0</th>\n",
       "      <th>celltypist_cell_label_coarse</th>\n",
       "      <th>celltypist_conf_score_coarse</th>\n",
       "      <th>celltypist_cell_label_fine</th>\n",
       "      <th>celltypist_conf_score_fine</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACAGCCAGTTTGTG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>8056.0</td>\n",
       "      <td>1891</td>\n",
       "      <td>7.545390</td>\n",
       "      <td>7641.0</td>\n",
       "      <td>8.941415</td>\n",
       "      <td>21.823060</td>\n",
       "      <td>13.584609</td>\n",
       "      <td>7.642979</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>37</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.978114</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.950723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACAGCCATGTTTGG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>23901.0</td>\n",
       "      <td>2319</td>\n",
       "      <td>7.749322</td>\n",
       "      <td>10986.0</td>\n",
       "      <td>9.304468</td>\n",
       "      <td>21.354451</td>\n",
       "      <td>13.899508</td>\n",
       "      <td>9.111597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>63</td>\n",
       "      <td>36</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.603815</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.370262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACAGCCATTTAAGC-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>31317.0</td>\n",
       "      <td>1667</td>\n",
       "      <td>7.419381</td>\n",
       "      <td>6636.0</td>\n",
       "      <td>8.800415</td>\n",
       "      <td>21.503918</td>\n",
       "      <td>13.803495</td>\n",
       "      <td>6.268836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>134</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.951335</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.903238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACATGCATTGGGAG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1039.0</td>\n",
       "      <td>1599</td>\n",
       "      <td>7.377759</td>\n",
       "      <td>6888.0</td>\n",
       "      <td>8.837681</td>\n",
       "      <td>24.811266</td>\n",
       "      <td>18.626598</td>\n",
       "      <td>5.894309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>81</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.121671</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.183371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCAACAATAACCT-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>25718.0</td>\n",
       "      <td>1403</td>\n",
       "      <td>7.247081</td>\n",
       "      <td>5495.0</td>\n",
       "      <td>8.611775</td>\n",
       "      <td>24.913558</td>\n",
       "      <td>18.817106</td>\n",
       "      <td>5.150136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.991872</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.996793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGAAGCATGAAG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>13995.0</td>\n",
       "      <td>2061</td>\n",
       "      <td>7.631432</td>\n",
       "      <td>8523.0</td>\n",
       "      <td>9.050641</td>\n",
       "      <td>21.858501</td>\n",
       "      <td>14.009152</td>\n",
       "      <td>6.112871</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>38</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.930010</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.984357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGAAGCGAGTAA-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>34490.0</td>\n",
       "      <td>1361</td>\n",
       "      <td>7.216709</td>\n",
       "      <td>5059.0</td>\n",
       "      <td>8.529121</td>\n",
       "      <td>23.739870</td>\n",
       "      <td>17.019173</td>\n",
       "      <td>8.381103</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.403587</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.973642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGGCAGCCTGCA-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>1536</td>\n",
       "      <td>7.337588</td>\n",
       "      <td>6272.0</td>\n",
       "      <td>8.744010</td>\n",
       "      <td>29.432398</td>\n",
       "      <td>23.852041</td>\n",
       "      <td>6.505102</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>58</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.177973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTAAACAAG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>9679.0</td>\n",
       "      <td>1279</td>\n",
       "      <td>7.154615</td>\n",
       "      <td>5301.0</td>\n",
       "      <td>8.575839</td>\n",
       "      <td>35.653650</td>\n",
       "      <td>28.730429</td>\n",
       "      <td>2.886248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>45</td>\n",
       "      <td>50</td>\n",
       "      <td>31</td>\n",
       "      <td>Monocytes</td>\n",
       "      <td>0.745200</td>\n",
       "      <td>Non-classical monocytes</td>\n",
       "      <td>0.041574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTTAAGCTG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1756</td>\n",
       "      <td>7.471363</td>\n",
       "      <td>7045.0</td>\n",
       "      <td>8.860215</td>\n",
       "      <td>24.442867</td>\n",
       "      <td>18.424414</td>\n",
       "      <td>6.018453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>104</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.122617</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.947241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10106 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample  dna_total_counts  n_genes_by_counts  \\\n",
       "AAACAGCCAGTTTGTG-D0     D0            8056.0               1891   \n",
       "AAACAGCCATGTTTGG-D0     D0           23901.0               2319   \n",
       "AAACAGCCATTTAAGC-D0     D0           31317.0               1667   \n",
       "AAACATGCATTGGGAG-D0     D0            1039.0               1599   \n",
       "AAACCAACAATAACCT-D0     D0           25718.0               1403   \n",
       "...                    ...               ...                ...   \n",
       "TTTGTGAAGCATGAAG-Q3     Q3           13995.0               2061   \n",
       "TTTGTGAAGCGAGTAA-Q3     Q3           34490.0               1361   \n",
       "TTTGTGGCAGCCTGCA-Q3     Q3            3673.0               1536   \n",
       "TTTGTTGGTAAACAAG-Q3     Q3            9679.0               1279   \n",
       "TTTGTTGGTTAAGCTG-Q3     Q3               0.0               1756   \n",
       "\n",
       "                     log1p_n_genes_by_counts  total_counts  \\\n",
       "AAACAGCCAGTTTGTG-D0                 7.545390        7641.0   \n",
       "AAACAGCCATGTTTGG-D0                 7.749322       10986.0   \n",
       "AAACAGCCATTTAAGC-D0                 7.419381        6636.0   \n",
       "AAACATGCATTGGGAG-D0                 7.377759        6888.0   \n",
       "AAACCAACAATAACCT-D0                 7.247081        5495.0   \n",
       "...                                      ...           ...   \n",
       "TTTGTGAAGCATGAAG-Q3                 7.631432        8523.0   \n",
       "TTTGTGAAGCGAGTAA-Q3                 7.216709        5059.0   \n",
       "TTTGTGGCAGCCTGCA-Q3                 7.337588        6272.0   \n",
       "TTTGTTGGTAAACAAG-Q3                 7.154615        5301.0   \n",
       "TTTGTTGGTTAAGCTG-Q3                 7.471363        7045.0   \n",
       "\n",
       "                     log1p_total_counts  pct_counts_in_top_20_genes  \\\n",
       "AAACAGCCAGTTTGTG-D0            8.941415                   21.823060   \n",
       "AAACAGCCATGTTTGG-D0            9.304468                   21.354451   \n",
       "AAACAGCCATTTAAGC-D0            8.800415                   21.503918   \n",
       "AAACATGCATTGGGAG-D0            8.837681                   24.811266   \n",
       "AAACCAACAATAACCT-D0            8.611775                   24.913558   \n",
       "...                                 ...                         ...   \n",
       "TTTGTGAAGCATGAAG-Q3            9.050641                   21.858501   \n",
       "TTTGTGAAGCGAGTAA-Q3            8.529121                   23.739870   \n",
       "TTTGTGGCAGCCTGCA-Q3            8.744010                   29.432398   \n",
       "TTTGTTGGTAAACAAG-Q3            8.575839                   35.653650   \n",
       "TTTGTTGGTTAAGCTG-Q3            8.860215                   24.442867   \n",
       "\n",
       "                     pct_counts_mt  pct_counts_ribo  pct_counts_hb  ...  \\\n",
       "AAACAGCCAGTTTGTG-D0      13.584609         7.642979            0.0  ...   \n",
       "AAACAGCCATGTTTGG-D0      13.899508         9.111597            0.0  ...   \n",
       "AAACAGCCATTTAAGC-D0      13.803495         6.268836            0.0  ...   \n",
       "AAACATGCATTGGGAG-D0      18.626598         5.894309            0.0  ...   \n",
       "AAACCAACAATAACCT-D0      18.817106         5.150136            0.0  ...   \n",
       "...                            ...              ...            ...  ...   \n",
       "TTTGTGAAGCATGAAG-Q3      14.009152         6.112871            0.0  ...   \n",
       "TTTGTGAAGCGAGTAA-Q3      17.019173         8.381103            0.0  ...   \n",
       "TTTGTGGCAGCCTGCA-Q3      23.852041         6.505102            0.0  ...   \n",
       "TTTGTTGGTAAACAAG-Q3      28.730429         2.886248            0.0  ...   \n",
       "TTTGTTGGTTAAGCTG-Q3      18.424414         6.018453            0.0  ...   \n",
       "\n",
       "                     leiden_1.25  leiden_1.5  leiden_2.0  leiden_5.0  \\\n",
       "AAACAGCCAGTTTGTG-D0            4           4           4           1   \n",
       "AAACAGCCATGTTTGG-D0            8           8           7          13   \n",
       "AAACAGCCATTTAAGC-D0            4           4           4           1   \n",
       "AAACATGCATTGGGAG-D0            4           4           4          10   \n",
       "AAACCAACAATAACCT-D0            4           4           4          10   \n",
       "...                          ...         ...         ...         ...   \n",
       "TTTGTGAAGCATGAAG-Q3            5           7          18          38   \n",
       "TTTGTGAAGCGAGTAA-Q3           10          11           8          23   \n",
       "TTTGTGGCAGCCTGCA-Q3            1           0           0          16   \n",
       "TTTGTTGGTAAACAAG-Q3            9          10          11          45   \n",
       "TTTGTTGGTTAAGCTG-Q3            4           4           7          18   \n",
       "\n",
       "                    leiden_10.0 leiden_20.0 celltypist_cell_label_coarse  \\\n",
       "AAACAGCCAGTTTGTG-D0           9          37               B-cell lineage   \n",
       "AAACAGCCATGTTTGG-D0          63          36               B-cell lineage   \n",
       "AAACAGCCATTTAAGC-D0          26         134               B-cell lineage   \n",
       "AAACATGCATTGGGAG-D0          19          81               B-cell lineage   \n",
       "AAACCAACAATAACCT-D0           8          46               B-cell lineage   \n",
       "...                         ...         ...                          ...   \n",
       "TTTGTGAAGCATGAAG-Q3          30           9               B-cell lineage   \n",
       "TTTGTGAAGCGAGTAA-Q3          14           5               B-cell lineage   \n",
       "TTTGTGGCAGCCTGCA-Q3          22          58               B-cell lineage   \n",
       "TTTGTTGGTAAACAAG-Q3          50          31                    Monocytes   \n",
       "TTTGTTGGTTAAGCTG-Q3          36         104               B-cell lineage   \n",
       "\n",
       "                    celltypist_conf_score_coarse celltypist_cell_label_fine  \\\n",
       "AAACAGCCAGTTTGTG-D0                     0.978114                Pro-B cells   \n",
       "AAACAGCCATGTTTGG-D0                     0.603815                Pro-B cells   \n",
       "AAACAGCCATTTAAGC-D0                     0.951335                Pro-B cells   \n",
       "AAACATGCATTGGGAG-D0                     0.121671                Pro-B cells   \n",
       "AAACCAACAATAACCT-D0                     0.991872                Pro-B cells   \n",
       "...                                          ...                        ...   \n",
       "TTTGTGAAGCATGAAG-Q3                     0.930010                Pro-B cells   \n",
       "TTTGTGAAGCGAGTAA-Q3                     0.403587                Pro-B cells   \n",
       "TTTGTGGCAGCCTGCA-Q3                     0.484664                Pro-B cells   \n",
       "TTTGTTGGTAAACAAG-Q3                     0.745200    Non-classical monocytes   \n",
       "TTTGTTGGTTAAGCTG-Q3                     0.122617                Pro-B cells   \n",
       "\n",
       "                    celltypist_conf_score_fine  \n",
       "AAACAGCCAGTTTGTG-D0                   0.950723  \n",
       "AAACAGCCATGTTTGG-D0                   0.370262  \n",
       "AAACAGCCATTTAAGC-D0                   0.903238  \n",
       "AAACATGCATTGGGAG-D0                   0.183371  \n",
       "AAACCAACAATAACCT-D0                   0.996793  \n",
       "...                                        ...  \n",
       "TTTGTGAAGCATGAAG-Q3                   0.984357  \n",
       "TTTGTGAAGCGAGTAA-Q3                   0.973642  \n",
       "TTTGTGGCAGCCTGCA-Q3                   0.177973  \n",
       "TTTGTTGGTAAACAAG-Q3                   0.041574  \n",
       "TTTGTTGGTTAAGCTG-Q3                   0.947241  \n",
       "\n",
       "[10106 rows x 34 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14c445c1-27a7-4b84-9407-e751e1a16c40",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "atac_CB = pd.read_csv('/barcodes/cellranger_arc_atac.737K-arc-v1.txt.gz', header = None)\n",
    "rna_CB = pd.read_csv('/barcodes/cellranger_arc_rna.737K-arc-v1.txt.gz', header = None)\n",
    "rna_CB.rename(columns={0: 'rna_CB'}, inplace=True)\n",
    "atac_CB.rename(columns={0: 'atac_CB'}, inplace=True)\n",
    "df_merged =  pd.concat([rna_CB, atac_CB], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6a73088a-f226-4ce5-a76f-f698c66d8b37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>dna_total_counts</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>log1p_n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>log1p_total_counts</th>\n",
       "      <th>pct_counts_in_top_20_genes</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "      <th>pct_counts_ribo</th>\n",
       "      <th>pct_counts_hb</th>\n",
       "      <th>...</th>\n",
       "      <th>leiden_1.5</th>\n",
       "      <th>leiden_2.0</th>\n",
       "      <th>leiden_5.0</th>\n",
       "      <th>leiden_10.0</th>\n",
       "      <th>leiden_20.0</th>\n",
       "      <th>celltypist_cell_label_coarse</th>\n",
       "      <th>celltypist_conf_score_coarse</th>\n",
       "      <th>celltypist_cell_label_fine</th>\n",
       "      <th>celltypist_conf_score_fine</th>\n",
       "      <th>base_index</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>atac_CB</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTGTTACTA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>8056.0</td>\n",
       "      <td>1891</td>\n",
       "      <td>7.545390</td>\n",
       "      <td>7641.0</td>\n",
       "      <td>8.941415</td>\n",
       "      <td>21.823060</td>\n",
       "      <td>13.584609</td>\n",
       "      <td>7.642979</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>37</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.978114</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.950723</td>\n",
       "      <td>AAACAGCCAGTTTGTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTTTATCCT-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>23901.0</td>\n",
       "      <td>2319</td>\n",
       "      <td>7.749322</td>\n",
       "      <td>10986.0</td>\n",
       "      <td>9.304468</td>\n",
       "      <td>21.354451</td>\n",
       "      <td>13.899508</td>\n",
       "      <td>9.111597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>63</td>\n",
       "      <td>36</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.603815</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.370262</td>\n",
       "      <td>AAACAGCCATGTTTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTTCACTTT-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>31317.0</td>\n",
       "      <td>1667</td>\n",
       "      <td>7.419381</td>\n",
       "      <td>6636.0</td>\n",
       "      <td>8.800415</td>\n",
       "      <td>21.503918</td>\n",
       "      <td>13.803495</td>\n",
       "      <td>6.268836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>134</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.951335</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.903238</td>\n",
       "      <td>AAACAGCCATTTAAGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CATTTAGGTGTCCACA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1039.0</td>\n",
       "      <td>1599</td>\n",
       "      <td>7.377759</td>\n",
       "      <td>6888.0</td>\n",
       "      <td>8.837681</td>\n",
       "      <td>24.811266</td>\n",
       "      <td>18.626598</td>\n",
       "      <td>5.894309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>81</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.121671</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.183371</td>\n",
       "      <td>AAACATGCATTGGGAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTTTATCGTCGAGGCA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>25718.0</td>\n",
       "      <td>1403</td>\n",
       "      <td>7.247081</td>\n",
       "      <td>5495.0</td>\n",
       "      <td>8.611775</td>\n",
       "      <td>24.913558</td>\n",
       "      <td>18.817106</td>\n",
       "      <td>5.150136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.991872</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.996793</td>\n",
       "      <td>AAACCAACAATAACCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTAGGTGTCAGTGAAC-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>13995.0</td>\n",
       "      <td>2061</td>\n",
       "      <td>7.631432</td>\n",
       "      <td>8523.0</td>\n",
       "      <td>9.050641</td>\n",
       "      <td>21.858501</td>\n",
       "      <td>14.009152</td>\n",
       "      <td>6.112871</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>38</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.930010</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.984357</td>\n",
       "      <td>TTTGTGAAGCATGAAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTAGGTGTCGGTCCAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>34490.0</td>\n",
       "      <td>1361</td>\n",
       "      <td>7.216709</td>\n",
       "      <td>5059.0</td>\n",
       "      <td>8.529121</td>\n",
       "      <td>23.739870</td>\n",
       "      <td>17.019173</td>\n",
       "      <td>8.381103</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.403587</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.973642</td>\n",
       "      <td>TTTGTGAAGCGAGTAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTTTGGTGTTTCTAAG-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>1536</td>\n",
       "      <td>7.337588</td>\n",
       "      <td>6272.0</td>\n",
       "      <td>8.744010</td>\n",
       "      <td>29.432398</td>\n",
       "      <td>23.852041</td>\n",
       "      <td>6.505102</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>58</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.177973</td>\n",
       "      <td>TTTGTGGCAGCCTGCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTACTTCAAATATGC-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>9679.0</td>\n",
       "      <td>1279</td>\n",
       "      <td>7.154615</td>\n",
       "      <td>5301.0</td>\n",
       "      <td>8.575839</td>\n",
       "      <td>35.653650</td>\n",
       "      <td>28.730429</td>\n",
       "      <td>2.886248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>45</td>\n",
       "      <td>50</td>\n",
       "      <td>31</td>\n",
       "      <td>Monocytes</td>\n",
       "      <td>0.745200</td>\n",
       "      <td>Non-classical monocytes</td>\n",
       "      <td>0.041574</td>\n",
       "      <td>TTTGTTGGTAAACAAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTACTTCATGATGAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1756</td>\n",
       "      <td>7.471363</td>\n",
       "      <td>7045.0</td>\n",
       "      <td>8.860215</td>\n",
       "      <td>24.442867</td>\n",
       "      <td>18.424414</td>\n",
       "      <td>6.018453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>104</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.122617</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.947241</td>\n",
       "      <td>TTTGTTGGTTAAGCTG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10106 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         sample  dna_total_counts  n_genes_by_counts  \\\n",
       "atac_CB                                                                \n",
       "ACAGCGGGTGTTACTA-D0___D0     D0            8056.0               1891   \n",
       "ACAGCGGGTTTATCCT-D0___D0     D0           23901.0               2319   \n",
       "ACAGCGGGTTCACTTT-D0___D0     D0           31317.0               1667   \n",
       "CATTTAGGTGTCCACA-D0___D0     D0            1039.0               1599   \n",
       "CTTTATCGTCGAGGCA-D0___D0     D0           25718.0               1403   \n",
       "...                         ...               ...                ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3     Q3           13995.0               2061   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3     Q3           34490.0               1361   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3     Q3            3673.0               1536   \n",
       "CGTACTTCAAATATGC-Q3___Q3     Q3            9679.0               1279   \n",
       "CGTACTTCATGATGAT-Q3___Q3     Q3               0.0               1756   \n",
       "\n",
       "                          log1p_n_genes_by_counts  total_counts  \\\n",
       "atac_CB                                                           \n",
       "ACAGCGGGTGTTACTA-D0___D0                 7.545390        7641.0   \n",
       "ACAGCGGGTTTATCCT-D0___D0                 7.749322       10986.0   \n",
       "ACAGCGGGTTCACTTT-D0___D0                 7.419381        6636.0   \n",
       "CATTTAGGTGTCCACA-D0___D0                 7.377759        6888.0   \n",
       "CTTTATCGTCGAGGCA-D0___D0                 7.247081        5495.0   \n",
       "...                                           ...           ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                 7.631432        8523.0   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                 7.216709        5059.0   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                 7.337588        6272.0   \n",
       "CGTACTTCAAATATGC-Q3___Q3                 7.154615        5301.0   \n",
       "CGTACTTCATGATGAT-Q3___Q3                 7.471363        7045.0   \n",
       "\n",
       "                          log1p_total_counts  pct_counts_in_top_20_genes  \\\n",
       "atac_CB                                                                    \n",
       "ACAGCGGGTGTTACTA-D0___D0            8.941415                   21.823060   \n",
       "ACAGCGGGTTTATCCT-D0___D0            9.304468                   21.354451   \n",
       "ACAGCGGGTTCACTTT-D0___D0            8.800415                   21.503918   \n",
       "CATTTAGGTGTCCACA-D0___D0            8.837681                   24.811266   \n",
       "CTTTATCGTCGAGGCA-D0___D0            8.611775                   24.913558   \n",
       "...                                      ...                         ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3            9.050641                   21.858501   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3            8.529121                   23.739870   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3            8.744010                   29.432398   \n",
       "CGTACTTCAAATATGC-Q3___Q3            8.575839                   35.653650   \n",
       "CGTACTTCATGATGAT-Q3___Q3            8.860215                   24.442867   \n",
       "\n",
       "                          pct_counts_mt  pct_counts_ribo  pct_counts_hb  ...  \\\n",
       "atac_CB                                                                  ...   \n",
       "ACAGCGGGTGTTACTA-D0___D0      13.584609         7.642979            0.0  ...   \n",
       "ACAGCGGGTTTATCCT-D0___D0      13.899508         9.111597            0.0  ...   \n",
       "ACAGCGGGTTCACTTT-D0___D0      13.803495         6.268836            0.0  ...   \n",
       "CATTTAGGTGTCCACA-D0___D0      18.626598         5.894309            0.0  ...   \n",
       "CTTTATCGTCGAGGCA-D0___D0      18.817106         5.150136            0.0  ...   \n",
       "...                                 ...              ...            ...  ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3      14.009152         6.112871            0.0  ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3      17.019173         8.381103            0.0  ...   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3      23.852041         6.505102            0.0  ...   \n",
       "CGTACTTCAAATATGC-Q3___Q3      28.730429         2.886248            0.0  ...   \n",
       "CGTACTTCATGATGAT-Q3___Q3      18.424414         6.018453            0.0  ...   \n",
       "\n",
       "                          leiden_1.5  leiden_2.0  leiden_5.0  leiden_10.0  \\\n",
       "atac_CB                                                                     \n",
       "ACAGCGGGTGTTACTA-D0___D0           4           4           1            9   \n",
       "ACAGCGGGTTTATCCT-D0___D0           8           7          13           63   \n",
       "ACAGCGGGTTCACTTT-D0___D0           4           4           1           26   \n",
       "CATTTAGGTGTCCACA-D0___D0           4           4          10           19   \n",
       "CTTTATCGTCGAGGCA-D0___D0           4           4          10            8   \n",
       "...                              ...         ...         ...          ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3           7          18          38           30   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3          11           8          23           14   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3           0           0          16           22   \n",
       "CGTACTTCAAATATGC-Q3___Q3          10          11          45           50   \n",
       "CGTACTTCATGATGAT-Q3___Q3           4           7          18           36   \n",
       "\n",
       "                         leiden_20.0 celltypist_cell_label_coarse  \\\n",
       "atac_CB                                                             \n",
       "ACAGCGGGTGTTACTA-D0___D0          37               B-cell lineage   \n",
       "ACAGCGGGTTTATCCT-D0___D0          36               B-cell lineage   \n",
       "ACAGCGGGTTCACTTT-D0___D0         134               B-cell lineage   \n",
       "CATTTAGGTGTCCACA-D0___D0          81               B-cell lineage   \n",
       "CTTTATCGTCGAGGCA-D0___D0          46               B-cell lineage   \n",
       "...                              ...                          ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3           9               B-cell lineage   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3           5               B-cell lineage   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3          58               B-cell lineage   \n",
       "CGTACTTCAAATATGC-Q3___Q3          31                    Monocytes   \n",
       "CGTACTTCATGATGAT-Q3___Q3         104               B-cell lineage   \n",
       "\n",
       "                         celltypist_conf_score_coarse  \\\n",
       "atac_CB                                                 \n",
       "ACAGCGGGTGTTACTA-D0___D0                     0.978114   \n",
       "ACAGCGGGTTTATCCT-D0___D0                     0.603815   \n",
       "ACAGCGGGTTCACTTT-D0___D0                     0.951335   \n",
       "CATTTAGGTGTCCACA-D0___D0                     0.121671   \n",
       "CTTTATCGTCGAGGCA-D0___D0                     0.991872   \n",
       "...                                               ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                     0.930010   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                     0.403587   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                     0.484664   \n",
       "CGTACTTCAAATATGC-Q3___Q3                     0.745200   \n",
       "CGTACTTCATGATGAT-Q3___Q3                     0.122617   \n",
       "\n",
       "                         celltypist_cell_label_fine  \\\n",
       "atac_CB                                               \n",
       "ACAGCGGGTGTTACTA-D0___D0                Pro-B cells   \n",
       "ACAGCGGGTTTATCCT-D0___D0                Pro-B cells   \n",
       "ACAGCGGGTTCACTTT-D0___D0                Pro-B cells   \n",
       "CATTTAGGTGTCCACA-D0___D0                Pro-B cells   \n",
       "CTTTATCGTCGAGGCA-D0___D0                Pro-B cells   \n",
       "...                                             ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                Pro-B cells   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                Pro-B cells   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                Pro-B cells   \n",
       "CGTACTTCAAATATGC-Q3___Q3    Non-classical monocytes   \n",
       "CGTACTTCATGATGAT-Q3___Q3                Pro-B cells   \n",
       "\n",
       "                         celltypist_conf_score_fine        base_index  \n",
       "atac_CB                                                                \n",
       "ACAGCGGGTGTTACTA-D0___D0                   0.950723  AAACAGCCAGTTTGTG  \n",
       "ACAGCGGGTTTATCCT-D0___D0                   0.370262  AAACAGCCATGTTTGG  \n",
       "ACAGCGGGTTCACTTT-D0___D0                   0.903238  AAACAGCCATTTAAGC  \n",
       "CATTTAGGTGTCCACA-D0___D0                   0.183371  AAACATGCATTGGGAG  \n",
       "CTTTATCGTCGAGGCA-D0___D0                   0.996793  AAACCAACAATAACCT  \n",
       "...                                             ...               ...  \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                   0.984357  TTTGTGAAGCATGAAG  \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                   0.973642  TTTGTGAAGCGAGTAA  \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                   0.177973  TTTGTGGCAGCCTGCA  \n",
       "CGTACTTCAAATATGC-Q3___Q3                   0.041574  TTTGTTGGTAAACAAG  \n",
       "CGTACTTCATGATGAT-Q3___Q3                   0.947241  TTTGTTGGTTAAGCTG  \n",
       "\n",
       "[10106 rows x 35 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs['base_index'] = rna.obs.index.str.split('-').str[0]\n",
    "mapping_dict = pd.Series(df_merged['atac_CB'].values, index=df_merged['rna_CB']).to_dict()\n",
    "rna.obs['atac_CB'] = rna.obs['base_index'].map(mapping_dict)\n",
    "rna.obs['sample'] = rna.obs['sample'].str.split('_').str[0]\n",
    "rna.obs['atac_CB'] = rna.obs['atac_CB'].astype(str)+'-'+rna.obs['sample'].astype(str) + '___' + rna.obs['sample'].astype(str)\n",
    "rna.obs.set_index('atac_CB', inplace=True)\n",
    "rna.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "aa0c039c-9b2d-47d1-ba94-06a0915f7724",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna.X = rna.layers['counts'].copy()\n",
    "rna.raw = rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "344d8c9b-745d-46b0-8bfe-c7021f5be98d",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna.write_h5ad(\"/grn/multiome_adata_04072025.raw.BALL.h5ad\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d51006aa-fecc-45b8-8192-d9e78d598dc2",
   "metadata": {},
   "source": [
    "# 03 Run scenicplus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "3b7a0cbe-2e68-4295-b716-2f8ecb47ffa1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "   ____   ____ _____ _   _ ___ ____      \n",
      "  / ___| / ___| ____| \\ | |_ _/ ___|\u001b[31;1m _ \u001b[0m\n",
      "  \\___ \\| |   |  _| |  \\| || | |  \u001b[31;1m _|.|_\u001b[0m\n",
      "   ___) | |___| |___| |\\  || | |__\u001b[31;1m|_..._|\u001b[0m\n",
      "  |____/ \\____|_____|_| \\_|___\\____|\u001b[31;1m|_|\u001b[0m \n",
      "\n",
      "\n",
      "scenicplus verions: 1.0a1\n",
      "usage: scenicplus [-h] {init_snakemake,prepare_data,grn_inference} ...\n",
      "\n",
      "Single-Cell Enhancer-driven gene regulatory Network Inference and Clustering\n",
      "\n",
      "positional arguments:\n",
      "  {init_snakemake,prepare_data,grn_inference}\n",
      "\n",
      "options:\n",
      "  -h, --help            show this help message and exit\n"
     ]
    }
   ],
   "source": [
    "! scenicplus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "498ce8e6-899a-460a-946a-8ff58572ba5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2025-07-10 21:41:47,855 SCENIC+      INFO     Creating snakemake folder in: scplus_pipeline\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!scenicplus init_snakemake --out_dir scplus_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "20b10d7a-e9bb-421e-8c0b-063fada2c316",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: tree: command not found\n"
     ]
    }
   ],
   "source": [
    "!tree scplus_pipeline/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cedf0a69-eab9-44ec-bbfc-e27c95967884",
   "metadata": {},
   "source": [
    "```\n",
    "tree -d scplus_pipeline/\n",
    "scplus_pipeline/\n",
    "└── Snakemake\n",
    "    ├── config\n",
    "    └── workflow\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "08bf6d99-9ae5-4bdf-b317-707f101bb886",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p outs\n",
    "!mkdir -p tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ccecdc63-6b84-42f4-afe8-3084e8cb8023",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_SCENIC_plus.ipynb\n",
      "02_scenic_output-Copy1.ipynb\n",
      "02_scenic_output.ipynb\n",
      "10x_ONT_BALL.motifs_vs_regions.scores.feather\n",
      "10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "10x_ONT_BALL.regions_vs_motifs.scores.feather\n",
      "aertslab_motif_colleciton\n",
      "cbust\n",
      "cbust.1\n",
      "create_cisTarget_databases\n",
      "figures\n",
      "hg38.BALL.with_1kb_bg_padding.fa\n",
      "misc\n",
      "motif_cna_overlaps.ipynb\n",
      "motifs.txt\n",
      "multiome_adata_04072025.raw.BALL.h5ad\n",
      "old\n",
      "outs\n",
      "pseudobulk_exp_BALL.ipynb\n",
      "regulons\n",
      "scplus_pipeline\n",
      "subset\n",
      "tmp\n",
      "xx_visualisations.ipynb\n"
     ]
    }
   ],
   "source": [
    "! ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "67ec337d-7008-4b29-8749-e67932773e7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd scplus_pipeline/Snakemake/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f94aba4b",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat scplus_pipeline/Snakemake/config/config.yaml"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "77a77f9b",
   "metadata": {},
   "source": [
    "adjust `config.yaml` with the input from pycistopic and RNA output files;\n",
    "for exmaple:\n",
    "\n",
    "```\n",
    "input_data:\n",
    "  cisTopic_obj_fname: \"/preprocessing/atac/outs/cistopic_obj.pkl\"\n",
    "  GEX_anndata_fname: \"/preprocessing/grn/multiome_adata_04072025.raw.BALL.h5ad\"\n",
    "  region_set_folder: \"/preprocessing/atac/outs/region_sets\"\n",
    "  ctx_db_fname: \"/preprocessing/grn/10x_ONT_BALL.regions_vs_motifs.rankings.feather\"\n",
    "  dem_db_fname: \"/preprocessing/grn/10x_ONT_BALL.regions_vs_motifs.scores.feather\"\n",
    "  path_to_motif_annotations: \"/preprocessing/grn/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\"\n",
    "\n",
    "output_data:\n",
    "  # output for prepare_GEX_ACC .h5mu\n",
    "  combined_GEX_ACC_mudata: \"/preprocessing/grn/outs/ACC_GEX.h5mu\"\n",
    "  # output for motif enrichment results .hdf5\n",
    "  dem_result_fname: \"/preprocessing/grn/outs/dem_results.hdf5\"\n",
    "  ctx_result_fname: \"/preprocessing/grn/outs/ctx_results.hdf5\"\n",
    "  # output html for motif enrichment results .html\n",
    "  output_fname_dem_html: \"/preprocessing/grn/outs/dem_results.html\"\n",
    "  output_fname_ctx_html: \"/preprocessing/grn/outs/ctx_results.html\"\n",
    "  # output for prepare_menr .h5ad\n",
    "  cistromes_direct: \"/preprocessing/grn/outs/cistromes_direct.h5ad\"\n",
    "  cistromes_extended: \"/preprocessing/grn/outs/cistromes_extended.h5ad\"\n",
    "  # output tf names .txt\n",
    "  tf_names: \"/preprocessing/grn/outs/tf_names.txt\"\n",
    "  # output for download_genome_annotations .tsv\n",
    "  genome_annotation: \"/preprocessing/grn/outs/genome_annotation.tsv\"\n",
    "  chromsizes: \"/preprocessing/grn/outs/chromsizes.tsv\"\n",
    "  # output for search_space .tsb\n",
    "  search_space: \"/preprocessing/grn/outs/search_space.tsv\"\n",
    "  # output tf_to_gene .tsv\n",
    "  tf_to_gene_adjacencies: \"/preprocessing/grn/outs/tf_to_gene_adj.tsv\"\n",
    "  # output region_to_gene .tsv\n",
    "  region_to_gene_adjacencies: \"/preprocessing/grn/outs/region_to_gene_adj.tsv\"\n",
    "  # output eGRN .tsv\n",
    "  eRegulons_direct: \"/preprocessing/grn/outs/eRegulon_direct.tsv\"\n",
    "  eRegulons_extended: \"/preprocessing/grn/outs/eRegulons_extended.tsv\"\n",
    "  # output AUCell .h5mu\n",
    "  AUCell_direct: \"/preprocessing/grn/outs/AUCell_direct.h5mu\"\n",
    "  AUCell_extended: \"/preprocessing/grn/outs/AUCell_extended.h5mu\"\n",
    "  # output scplus mudata .h5mu\n",
    "  scplus_mdata: \"preprocessing/grn/outs/scplusmdata.h5mu\"\n",
    "\n",
    "params_general:\n",
    "  temp_dir: \"/preprocessing/grn//tmp\"\n",
    "  n_cpu: 40\n",
    "  seed: 666\n",
    "\n",
    "params_data_preparation:\n",
    "  # Params for prepare_GEX_ACC\n",
    "  bc_transform_func: \"\\\"lambda x: f'{x}'\\\"\"\n",
    "  is_multiome: True\n",
    "  key_to_group_by: \"\"\n",
    "  nr_cells_per_metacells: 10\n",
    "  # Params for prepare_menr\n",
    "  direct_annotation: \"Direct_annot\"\n",
    "  extended_annotation: \"Orthology_annot\"\n",
    "  # Params for download_genome_annotations\n",
    "  species: \"hsapiens\"\n",
    "  biomart_host: \"http://www.ensembl.org\"\n",
    "  # Params for search_space\n",
    "  search_space_upstream: \"1000 150000\"\n",
    "  search_space_downstream: \"1000 150000\"\n",
    "  search_space_extend_tss: \"10 10\"\n",
    "\n",
    "params_motif_enrichment:\n",
    "  species: \"homo_sapiens\"\n",
    "  annotation_version: \"v10nr_clust\"\n",
    "  motif_similarity_fdr: 0.001\n",
    "  orthologous_identity_threshold: 0.0\n",
    "  annotations_to_use: \"Direct_annot Orthology_annot\"\n",
    "  fraction_overlap_w_dem_database: 0.4\n",
    "  dem_max_bg_regions: 500\n",
    "  dem_balance_number_of_promoters: True\n",
    "  dem_promoter_space: 1_000\n",
    "  dem_adj_pval_thr: 0.05\n",
    "  dem_log2fc_thr: 1.0\n",
    "  dem_mean_fg_thr: 0.0\n",
    "  dem_motif_hit_thr: 3.0\n",
    "  fraction_overlap_w_ctx_database: 0.4\n",
    "  ctx_auc_threshold: 0.005\n",
    "  ctx_nes_threshold: 3.0\n",
    "  ctx_rank_threshold: 0.05\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "params_inference:\n",
    "  # Params for tf_to_gene\n",
    "  tf_to_gene_importance_method: \"GBM\"\n",
    "  # Params regions_to_gene\n",
    "  region_to_gene_importance_method: \"GBM\"\n",
    "  region_to_gene_correlation_method: \"SR\"\n",
    "  # Params for eGRN inference\n",
    "  order_regions_to_genes_by: \"importance\"\n",
    "  order_TFs_to_genes_by: \"importance\"\n",
    "  gsea_n_perm: 1000\n",
    "  quantile_thresholds_region_to_gene: \"0.85 0.90 0.95\"\n",
    "  top_n_regionTogenes_per_gene: \"5 10 15\"\n",
    "  top_n_regionTogenes_per_region: \"\"\n",
    "  min_regions_per_gene: 0\n",
    "  rho_threshold: 0.05\n",
    "  min_target_genes: 10\n",
    "\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9fddaab2",
   "metadata": {},
   "outputs": [],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bfce3542",
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /processing/grn/scplus_pipeline/Snakemake/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "95b901cd-b6fd-47c0-b19e-69064b0d784e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mAssuming unrestricted shared filesystem usage for local execution.\u001b[0m\n",
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
      "\u001b[33mProvided cores: 20\u001b[0m\n",
      "\u001b[33mRules claiming more threads will be scaled down.\u001b[0m\n",
      "\u001b[33mJob stats:\n",
      "job                           count\n",
      "--------------------------  -------\n",
      "AUCell_direct                     1\n",
      "AUCell_extended                   1\n",
      "all                               1\n",
      "eGRN_direct                       1\n",
      "eGRN_extended                     1\n",
      "get_search_space                  1\n",
      "motif_enrichment_cistarget        1\n",
      "motif_enrichment_dem              1\n",
      "prepare_menr                      1\n",
      "region_to_gene                    1\n",
      "scplus_mudata                     1\n",
      "tf_to_gene                        1\n",
      "total                            12\n",
      "\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Jul 10 22:49:28 2025]\u001b[0m\n",
      "\u001b[32mlocalrule get_search_space:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    jobid: 11\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-10 22:49:36,838 SCENIC+      INFO     Reading data\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-10 22:49:40,474 Get search space INFO     Extending promoter annotation to 10 bp upstream and 10 downstream\n",
      "2025-07-10 22:49:40,697 Get search space INFO     Extending search space to:\n",
      "            \t\t\t\t\t\t150000 bp downstream of the end of the gene.\n",
      "            \t\t\t\t\t\t150000 bp upstream of the start of the gene.\n",
      "2025-07-10 22:49:41,640 Get search space INFO     Intersecting with regions.\n",
      "2025-07-10 22:49:43,411 Get search space INFO     Calculating distances from region to gene\n",
      "2025-07-10 22:50:35,998 Get search space INFO     Imploding multiple entries per region and gene\n",
      "2025-07-10 22:52:39,589 SCENIC+      INFO     Writing search space to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "\u001b[0m\u001b[32m[Thu Jul 10 22:52:42 2025]\u001b[0m\n",
      "\u001b[32mFinished job 11.\u001b[0m\n",
      "\u001b[32m1 of 12 steps (8%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Jul 10 22:52:42 2025]\u001b[0m\n",
      "\u001b[32mlocalrule region_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    jobid: 10\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-10 22:52:51,563 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-10 22:52:54,463 SCENIC+      INFO     Reading search space\n",
      "2025-07-10 22:52:54,918 R2G          INFO     Calculating region to gene importances, using GBM method\n",
      "Running using 20 cores: 100%|█████████████| 16264/16264 [17:16<00:00, 15.69it/s]\n",
      "2025-07-10 23:10:27,573 R2G          INFO     Calculating region to gene correlation, using SR method\n",
      "Running using 20 cores:   0%|               | 14/16264 [00:00<01:56, 139.88it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   0%|               | 48/16264 [00:00<01:05, 249.34it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|              | 102/16264 [00:00<01:27, 184.96it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|              | 123/16264 [00:00<02:01, 133.16it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏              | 160/16264 [00:01<03:39, 73.35it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏             | 238/16264 [00:02<02:32, 105.43it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▏              | 258/16264 [00:02<03:17, 81.12it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▎              | 280/16264 [00:02<03:24, 78.08it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   3%|▌              | 560/16264 [00:05<02:42, 96.73it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   4%|▌              | 640/16264 [00:06<02:41, 96.95it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores: 100%|█████████████| 16264/16264 [02:45<00:00, 98.41it/s]\n",
      "2025-07-10 23:13:36,396 R2G          INFO     Done!\n",
      "2025-07-10 23:13:36,572 SCENIC+      INFO     Saving region to gene adjacencies to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Jul 10 23:13:44 2025]\u001b[0m\n",
      "\u001b[32mFinished job 10.\u001b[0m\n",
      "\u001b[32m2 of 12 steps (17%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Jul 10 23:13:44 2025]\u001b[0m\n",
      "\u001b[32mlocalrule motif_enrichment_cistarget:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\n",
      "    jobid: 9\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-10 23:13:53,168 SCENIC+      INFO     Reading region sets from: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\n",
      "2025-07-10 23:13:53,168 SCENIC+      INFO     Reading all .bed files in: Topics_top_3k\n",
      "2025-07-10 23:13:54,393 SCENIC+      INFO     Reading all .bed files in: DARs_cell_type\n",
      "2025-07-10 23:13:54,862 SCENIC+      INFO     Reading all .bed files in: Topics_otsu\n",
      "2025-07-10 23:14:06,266 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,351 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,371 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,405 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,433 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,479 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,492 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,524 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,539 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,558 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,563 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,564 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,573 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,578 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,595 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,598 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,601 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,622 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,654 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:06,658 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:09,608 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic1 which has 3000 regions\n",
      "2025-07-10 23:14:09,791 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic12 which has 3000 regions\n",
      "2025-07-10 23:14:10,038 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic23 which has 3000 regions\n",
      "2025-07-10 23:14:10,043 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic20 which has 3000 regions\n",
      "2025-07-10 23:14:10,124 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic14 which has 3000 regions\n",
      "2025-07-10 23:14:10,127 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic10 which has 3000 regions\n",
      "2025-07-10 23:14:10,164 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic19 which has 3000 regions\n",
      "2025-07-10 23:14:10,175 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic15 which has 3000 regions\n",
      "2025-07-10 23:14:10,196 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic17 which has 3000 regions\n",
      "2025-07-10 23:14:10,211 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic21 which has 3000 regions\n",
      "2025-07-10 23:14:10,292 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic22 which has 3000 regions\n",
      "2025-07-10 23:14:10,297 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic16 which has 3000 regions\n",
      "2025-07-10 23:14:10,343 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic25 which has 3000 regions\n",
      "2025-07-10 23:14:10,344 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic2 which has 3000 regions\n",
      "2025-07-10 23:14:10,352 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic18 which has 3000 regions\n",
      "2025-07-10 23:14:10,400 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic9 which has 3000 regions\n",
      "2025-07-10 23:14:10,411 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic11 which has 3000 regions\n",
      "2025-07-10 23:14:10,442 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic8 which has 3000 regions\n",
      "2025-07-10 23:14:10,472 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic5 which has 3000 regions\n",
      "2025-07-10 23:14:10,725 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic24 which has 3000 regions\n",
      "2025-07-10 23:14:15,906 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic12\n",
      "2025-07-10 23:14:15,946 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic21\n",
      "2025-07-10 23:14:16,066 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic20\n",
      "2025-07-10 23:14:16,184 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic14\n",
      "2025-07-10 23:14:16,248 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic18\n",
      "2025-07-10 23:14:16,334 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic9\n",
      "2025-07-10 23:14:16,370 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic15\n",
      "2025-07-10 23:14:16,419 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic19\n",
      "2025-07-10 23:14:16,424 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic16\n",
      "2025-07-10 23:14:16,435 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic11\n",
      "2025-07-10 23:14:16,454 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic10\n",
      "2025-07-10 23:14:16,468 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic8\n",
      "2025-07-10 23:14:16,520 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic2\n",
      "2025-07-10 23:14:16,609 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic23\n",
      "2025-07-10 23:14:16,657 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic17\n",
      "2025-07-10 23:14:16,743 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic25\n",
      "2025-07-10 23:14:16,993 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic24\n",
      "2025-07-10 23:14:17,032 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic22\n",
      "2025-07-10 23:14:17,489 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic5\n",
      "2025-07-10 23:14:17,514 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic1\n",
      "2025-07-10 23:14:18,203 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic21\n",
      "2025-07-10 23:14:18,364 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic12\n",
      "2025-07-10 23:14:18,403 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic20\n",
      "2025-07-10 23:14:18,473 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic14\n",
      "2025-07-10 23:14:18,521 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic9\n",
      "2025-07-10 23:14:18,590 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic18\n",
      "2025-07-10 23:14:18,704 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic11\n",
      "2025-07-10 23:14:18,728 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic8\n",
      "2025-07-10 23:14:18,841 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic19\n",
      "2025-07-10 23:14:18,866 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic2\n",
      "2025-07-10 23:14:18,880 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic17\n",
      "2025-07-10 23:14:18,909 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,073 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,166 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,176 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic25\n",
      "2025-07-10 23:14:19,218 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic10\n",
      "2025-07-10 23:14:19,242 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,381 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic16\n",
      "2025-07-10 23:14:19,402 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,465 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,523 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic15\n",
      "2025-07-10 23:14:19,552 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,719 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,739 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic24\n",
      "2025-07-10 23:14:19,762 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,788 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:19,892 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:20,013 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic22\n",
      "2025-07-10 23:14:20,014 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:20,089 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic5\n",
      "2025-07-10 23:14:20,245 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic23\n",
      "2025-07-10 23:14:20,413 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:20,543 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:20,962 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic1\n",
      "2025-07-10 23:14:20,974 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:21,203 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:21,333 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:21,633 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:22,119 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic3 which has 3000 regions\n",
      "2025-07-10 23:14:22,177 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic4 which has 3000 regions\n",
      "2025-07-10 23:14:22,240 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic7 which has 3000 regions\n",
      "2025-07-10 23:14:22,320 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic6 which has 3000 regions\n",
      "2025-07-10 23:14:22,326 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic13 which has 3000 regions\n",
      "2025-07-10 23:14:22,330 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:22,965 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:23,371 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q3 which has 4181 regions\n",
      "2025-07-10 23:14:23,510 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q1 which has 4880 regions\n",
      "2025-07-10 23:14:23,745 cisTarget    INFO     Running cisTarget for DARs_cell_type_Classical_monocytes which has 10248 regions\n",
      "2025-07-10 23:14:23,828 cisTarget    INFO     Running cisTarget for DARs_cell_type_CD16pos_NK_cells which has 10613 regions\n",
      "2025-07-10 23:14:24,086 cisTarget    INFO     Running cisTarget for DARs_cell_type_Non-classical_monocytes which has 9432 regions\n",
      "2025-07-10 23:14:24,447 cisTarget    INFO     Running cisTarget for DARs_cell_type_Naive_B_cells which has 8663 regions\n",
      "2025-07-10 23:14:24,447 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_D0 which has 4147 regions\n",
      "2025-07-10 23:14:24,518 cisTarget    INFO     Running cisTarget for DARs_cell_type_Tem_Trm_cytotoxic_T_cells which has 12196 regions\n",
      "2025-07-10 23:14:24,995 cisTarget    INFO     Running cisTarget for DARs_cell_type_Tcm_Naive_helper_T_cells which has 10715 regions\n",
      "2025-07-10 23:14:25,394 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q2 which has 9825 regions\n",
      "2025-07-10 23:14:25,425 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic4\n",
      "2025-07-10 23:14:25,715 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic3\n",
      "2025-07-10 23:14:25,802 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic6\n",
      "2025-07-10 23:14:25,922 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic21 which has 8762 regions\n",
      "2025-07-10 23:14:26,166 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic7\n",
      "2025-07-10 23:14:26,204 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic2 which has 10778 regions\n",
      "2025-07-10 23:14:26,330 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic15 which has 12462 regions\n",
      "2025-07-10 23:14:26,475 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q3\n",
      "2025-07-10 23:14:26,540 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic13\n",
      "2025-07-10 23:14:26,763 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic11 which has 9492 regions\n",
      "2025-07-10 23:14:27,270 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q1\n",
      "2025-07-10 23:14:27,299 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic5 which has 8640 regions\n",
      "2025-07-10 23:14:27,617 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_D0\n",
      "2025-07-10 23:14:27,862 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic4\n",
      "2025-07-10 23:14:28,045 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic3\n",
      "2025-07-10 23:14:28,451 cisTarget    INFO     Annotating motifs for DARs_cell_type_Classical_monocytes\n",
      "2025-07-10 23:14:28,545 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:28,697 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:28,699 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic6\n",
      "2025-07-10 23:14:28,777 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q3\n",
      "2025-07-10 23:14:28,807 cisTarget    INFO     Annotating motifs for DARs_cell_type_Non-classical_monocytes\n",
      "\u001b[0m2025-07-10 23:14:29,626 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic13\n",
      "2025-07-10 23:14:29,666 cisTarget    INFO     Annotating motifs for DARs_cell_type_Naive_B_cells\n",
      "2025-07-10 23:14:29,912 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q1\n",
      "2025-07-10 23:14:29,919 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic7\n",
      "2025-07-10 23:14:29,990 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_D0\n",
      "2025-07-10 23:14:30,009 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q2\n",
      "2025-07-10 23:14:30,368 cisTarget    INFO     Annotating motifs for DARs_cell_type_CD16pos_NK_cells\n",
      "2025-07-10 23:14:30,380 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic21\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "2025-07-10 23:14:30,467 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:30,981 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:31,083 cisTarget    INFO     Annotating motifs for DARs_cell_type_Tcm_Naive_helper_T_cells\n",
      "\u001b[0m2025-07-10 23:14:31,515 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:31,546 cisTarget    INFO     Getting cistromes for DARs_cell_type_Classical_monocytes\n",
      "2025-07-10 23:14:31,607 cisTarget    INFO     Annotating motifs for DARs_cell_type_Tem_Trm_cytotoxic_T_cells\n",
      "2025-07-10 23:14:31,843 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic5\n",
      "2025-07-10 23:14:32,251 cisTarget    INFO     Getting cistromes for DARs_cell_type_Non-classical_monocytes\n",
      "2025-07-10 23:14:32,503 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic1 which has 10461 regions\n",
      "2025-07-10 23:14:32,642 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:32,686 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic11\n",
      "2025-07-10 23:14:32,697 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic2\n",
      "2025-07-10 23:14:32,874 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q2\n",
      "2025-07-10 23:14:32,879 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic15\n",
      "2025-07-10 23:14:33,339 cisTarget    INFO     Getting cistromes for DARs_cell_type_Naive_B_cells\n",
      "2025-07-10 23:14:33,379 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic12 which has 13184 regions\n",
      "2025-07-10 23:14:33,436 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic21\n",
      "\u001b[0m2025-07-10 23:14:34,839 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic25 which has 15787 regions\n",
      "2025-07-10 23:14:35,053 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic5\n",
      "2025-07-10 23:14:35,132 cisTarget    INFO     Getting cistromes for DARs_cell_type_CD16pos_NK_cells\n",
      "2025-07-10 23:14:35,315 cisTarget    INFO     Getting cistromes for DARs_cell_type_Tcm_Naive_helper_T_cells\n",
      "2025-07-10 23:14:35,616 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic20 which has 13708 regions\n",
      "2025-07-10 23:14:35,802 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic17 which has 13442 regions\n",
      "\u001b[0m2025-07-10 23:14:35,894 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic11\n",
      "2025-07-10 23:14:36,253 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic24 which has 8988 regions\n",
      "2025-07-10 23:14:36,457 cisTarget    INFO     Getting cistromes for DARs_cell_type_Tem_Trm_cytotoxic_T_cells\n",
      "2025-07-10 23:14:36,558 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic15\n",
      "\u001b[0m2025-07-10 23:14:38,057 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic2\n",
      "\u001b[0m2025-07-10 23:14:38,468 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic1\n",
      "\u001b[0m2025-07-10 23:14:39,700 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic12\n",
      "2025-07-10 23:14:40,465 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:41,819 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic24\n",
      "2025-07-10 23:14:41,973 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-07-10 23:14:42,087 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic25\n",
      "2025-07-10 23:14:42,181 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic20\n",
      "2025-07-10 23:14:42,354 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic17\n",
      "2025-07-10 23:14:43,163 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:43,428 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic12\n",
      "\u001b[0m2025-07-10 23:14:43,549 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic1\n",
      "2025-07-10 23:14:44,228 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic9 which has 10776 regions\n",
      "\u001b[0m2025-07-10 23:14:44,675 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-07-10 23:14:46,153 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:46,259 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic25\n",
      "2025-07-10 23:14:46,271 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic20\n",
      "2025-07-10 23:14:46,439 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic17\n",
      "\u001b[0m2025-07-10 23:14:46,882 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic24\n",
      "2025-07-10 23:14:46,929 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:46,973 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic14 which has 22596 regions\n",
      "2025-07-10 23:14:47,174 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic23 which has 5784 regions\n",
      "\u001b[0m2025-07-10 23:14:48,472 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:49,285 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic10 which has 17023 regions\n",
      "2025-07-10 23:14:50,190 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic19 which has 3618 regions\n",
      "2025-07-10 23:14:50,442 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic8 which has 17060 regions\n",
      "2025-07-10 23:14:51,379 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-07-10 23:14:52,363 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic18 which has 10272 regions\n",
      "2025-07-10 23:14:53,043 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:53,822 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic9\n",
      "\u001b[0m2025-07-10 23:14:54,388 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:54,864 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:55,291 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic16 which has 22 regions\n",
      "2025-07-10 23:14:55,301 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:14:55,519 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic23\n",
      "2025-07-10 23:14:55,619 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic22 which has 12030 regions\n",
      "2025-07-10 23:14:56,382 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic19\n",
      "\u001b[0m2025-07-10 23:14:56,693 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic14\n",
      "2025-07-10 23:14:57,114 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic9\n",
      "\u001b[0m2025-07-10 23:14:57,800 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-07-10 23:14:58,866 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic3 which has 11393 regions\n",
      "2025-07-10 23:14:58,901 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic19\n",
      "2025-07-10 23:14:58,970 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic4 which has 9421 regions\n",
      "2025-07-10 23:14:59,111 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic14\n",
      "2025-07-10 23:14:59,195 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic7 which has 10752 regions\n",
      "2025-07-10 23:14:59,524 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic18\n",
      "2025-07-10 23:14:59,607 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic8\n",
      "\u001b[0m\u001b[0m2025-07-10 23:15:01,120 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic16\n",
      "2025-07-10 23:15:01,410 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic10\n",
      "2025-07-10 23:15:01,669 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic23\n",
      "2025-07-10 23:15:01,947 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic6 which has 15246 regions\n",
      "2025-07-10 23:15:02,327 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic18\n",
      "2025-07-10 23:15:02,631 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic8\n",
      "2025-07-10 23:15:03,320 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic16\n",
      "2025-07-10 23:15:06,325 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic10\n",
      "2025-07-10 23:15:06,375 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic22\n",
      "2025-07-10 23:15:07,748 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic4\n",
      "2025-07-10 23:15:08,118 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic3\n",
      "2025-07-10 23:15:09,978 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic7\n",
      "2025-07-10 23:15:10,882 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-10 23:15:11,422 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic4\n",
      "2025-07-10 23:15:11,975 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic3\n",
      "2025-07-10 23:15:12,034 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic22\n",
      "2025-07-10 23:15:12,295 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic6\n",
      "2025-07-10 23:15:14,593 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic13 which has 15288 regions\n",
      "2025-07-10 23:15:16,901 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic7\n",
      "2025-07-10 23:15:17,359 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic6\n",
      "2025-07-10 23:15:26,058 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic13\n",
      "2025-07-10 23:15:31,876 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic13\n",
      "2025-07-10 23:15:36,752 SCENIC+      INFO     Writing html to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\n",
      "2025-07-10 23:15:37,488 SCENIC+      INFO     Writing output to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Jul 10 23:16:34 2025]\u001b[0m\n",
      "\u001b[32mFinished job 9.\u001b[0m\n",
      "\u001b[32m3 of 12 steps (25%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Jul 10 23:16:34 2025]\u001b[0m\n",
      "\u001b[32mlocalrule motif_enrichment_dem:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\n",
      "    jobid: 7\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-10 23:16:43,763 SCENIC+      INFO     Reading region sets from: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\n",
      "2025-07-10 23:16:43,764 SCENIC+      INFO     Reading all .bed files in: Topics_top_3k\n",
      "2025-07-10 23:16:44,109 SCENIC+      INFO     Reading all .bed files in: DARs_cell_type\n",
      "2025-07-10 23:16:44,190 SCENIC+      INFO     Reading all .bed files in: Topics_otsu\n",
      "2025-07-10 23:16:56,427 DEM          INFO     Running DEM for Topics_top_3k_Topic12_vs_all\n",
      "2025-07-10 23:16:56,432 DEM          INFO     Running DEM for Topics_top_3k_Topic11_vs_all\n",
      "2025-07-10 23:16:56,735 DEM          INFO     Running DEM for Topics_top_3k_Topic15_vs_all\n",
      "2025-07-10 23:16:56,828 DEM          INFO     Running DEM for Topics_top_3k_Topic21_vs_all\n",
      "2025-07-10 23:16:56,858 DEM          INFO     Running DEM for Topics_top_3k_Topic1_vs_all\n",
      "2025-07-10 23:16:56,887 DEM          INFO     Running DEM for Topics_top_3k_Topic2_vs_all\n",
      "2025-07-10 23:16:56,938 DEM          INFO     Running DEM for Topics_top_3k_Topic5_vs_all\n",
      "2025-07-10 23:16:56,979 DEM          INFO     Running DEM for Topics_top_3k_Topic17_vs_all\n",
      "2025-07-10 23:16:57,014 DEM          INFO     Running DEM for Topics_top_3k_Topic10_vs_all\n",
      "2025-07-10 23:16:57,019 DEM          INFO     Running DEM for Topics_top_3k_Topic9_vs_all\n",
      "2025-07-10 23:16:57,098 DEM          INFO     Running DEM for Topics_top_3k_Topic25_vs_all\n",
      "2025-07-10 23:16:57,098 DEM          INFO     Running DEM for Topics_top_3k_Topic18_vs_all\n",
      "2025-07-10 23:16:57,146 DEM          INFO     Running DEM for Topics_top_3k_Topic14_vs_all\n",
      "2025-07-10 23:16:57,363 DEM          INFO     Running DEM for Topics_top_3k_Topic22_vs_all\n",
      "2025-07-10 23:16:57,419 DEM          INFO     Running DEM for Topics_top_3k_Topic23_vs_all\n",
      "2025-07-10 23:16:57,470 DEM          INFO     Running DEM for Topics_top_3k_Topic20_vs_all\n",
      "2025-07-10 23:16:57,477 DEM          INFO     Running DEM for Topics_top_3k_Topic24_vs_all\n",
      "2025-07-10 23:16:57,619 DEM          INFO     Running DEM for Topics_top_3k_Topic8_vs_all\n",
      "2025-07-10 23:16:57,792 DEM          INFO     Running DEM for Topics_top_3k_Topic19_vs_all\n",
      "2025-07-10 23:16:57,807 DEM          INFO     Running DEM for Topics_top_3k_Topic16_vs_all\n",
      "2025-07-10 23:17:22,528 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,532 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,532 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,644 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,645 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,663 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,688 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:22,920 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,168 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,196 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,205 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,221 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,233 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,248 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,319 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,391 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,412 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,452 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,459 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:23,463 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:27,174 DEM          INFO     Running DEM for Topics_top_3k_Topic7_vs_all\n",
      "2025-07-10 23:17:27,242 DEM          INFO     Running DEM for Topics_top_3k_Topic6_vs_all\n",
      "2025-07-10 23:17:27,247 DEM          INFO     Running DEM for Topics_top_3k_Topic3_vs_all\n",
      "2025-07-10 23:17:27,276 DEM          INFO     Running DEM for Topics_top_3k_Topic4_vs_all\n",
      "2025-07-10 23:17:27,337 DEM          INFO     Running DEM for Topics_top_3k_Topic13_vs_all\n",
      "2025-07-10 23:17:27,356 DEM          INFO     Running DEM for DARs_cell_type_Classical_monocytes_vs_all\n",
      "2025-07-10 23:17:27,374 DEM          INFO     Running DEM for DARs_cell_type_CD16pos_NK_cells_vs_all\n",
      "2025-07-10 23:17:27,717 DEM          INFO     Running DEM for DARs_cell_type_Non-classical_monocytes_vs_all\n",
      "2025-07-10 23:17:28,071 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q1_vs_all\n",
      "2025-07-10 23:17:28,143 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q3_vs_all\n",
      "2025-07-10 23:17:28,210 DEM          INFO     Running DEM for DARs_cell_type_Tem_Trm_cytotoxic_T_cells_vs_all\n",
      "2025-07-10 23:17:28,251 DEM          INFO     Running DEM for DARs_cell_type_Naive_B_cells_vs_all\n",
      "2025-07-10 23:17:28,298 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q2_vs_all\n",
      "2025-07-10 23:17:28,349 DEM          INFO     Running DEM for DARs_cell_type_Tcm_Naive_helper_T_cells_vs_all\n",
      "2025-07-10 23:17:28,366 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_D0_vs_all\n",
      "2025-07-10 23:17:29,416 DEM          INFO     Running DEM for Topics_otsu_Topic21_vs_all\n",
      "2025-07-10 23:17:29,552 DEM          INFO     Running DEM for Topics_otsu_Topic15_vs_all\n",
      "2025-07-10 23:17:29,579 DEM          INFO     Running DEM for Topics_otsu_Topic2_vs_all\n",
      "2025-07-10 23:17:29,784 DEM          INFO     Running DEM for Topics_otsu_Topic5_vs_all\n",
      "2025-07-10 23:17:29,839 DEM          INFO     Running DEM for Topics_otsu_Topic11_vs_all\n",
      "2025-07-10 23:17:36,225 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:38,114 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:38,912 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:39,511 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:39,609 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:41,366 DEM          INFO     Running DEM for Topics_otsu_Topic1_vs_all\n",
      "2025-07-10 23:17:41,542 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:42,021 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:42,698 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:43,157 DEM          INFO     Running DEM for Topics_otsu_Topic12_vs_all\n",
      "\u001b[0m2025-07-10 23:17:43,804 DEM          INFO     Running DEM for Topics_otsu_Topic25_vs_all\n",
      "2025-07-10 23:17:44,473 DEM          INFO     Running DEM for Topics_otsu_Topic17_vs_all\n",
      "2025-07-10 23:17:44,569 DEM          INFO     Running DEM for Topics_otsu_Topic9_vs_all\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m\u001b[0m2025-07-10 23:17:48,354 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:49,614 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:50,580 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-10 23:17:50,881 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:51,032 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:51,139 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:51,729 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:51,835 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-10 23:17:53,018 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:17:53,437 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m\u001b[0m2025-07-10 23:17:55,825 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-10 23:17:57,608 DEM          INFO     Running DEM for Topics_otsu_Topic20_vs_all\n",
      "2025-07-10 23:17:57,630 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m2025-07-10 23:17:59,600 DEM          INFO     Running DEM for Topics_otsu_Topic24_vs_all\n",
      "\u001b[0m2025-07-10 23:18:00,685 DEM          INFO     Running DEM for Topics_otsu_Topic14_vs_all\n",
      "2025-07-10 23:18:00,756 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m\u001b[0m2025-07-10 23:18:04,521 DEM          INFO     Running DEM for Topics_otsu_Topic23_vs_all\n",
      "\u001b[0m2025-07-10 23:18:05,606 DEM          INFO     Running DEM for Topics_otsu_Topic10_vs_all\n",
      "2025-07-10 23:18:06,689 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:07,328 DEM          INFO     Running DEM for Topics_otsu_Topic8_vs_all\n",
      "2025-07-10 23:18:07,496 DEM          INFO     Running DEM for Topics_otsu_Topic19_vs_all\n",
      "2025-07-10 23:18:08,716 DEM          INFO     Running DEM for Topics_otsu_Topic18_vs_all\n",
      "\u001b[0m2025-07-10 23:18:09,752 DEM          INFO     Running DEM for Topics_otsu_Topic22_vs_all\n",
      "2025-07-10 23:18:10,781 DEM          INFO     Running DEM for Topics_otsu_Topic16_vs_all\n",
      "2025-07-10 23:18:11,455 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:11,794 DEM          INFO     Running DEM for Topics_otsu_Topic3_vs_all\n",
      "2025-07-10 23:18:12,423 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:13,433 DEM          INFO     Running DEM for Topics_otsu_Topic4_vs_all\n",
      "\u001b[0m2025-07-10 23:18:14,046 DEM          INFO     Running DEM for Topics_otsu_Topic7_vs_all\n",
      "\u001b[0m2025-07-10 23:18:16,449 DEM          INFO     Running DEM for Topics_otsu_Topic6_vs_all\n",
      "2025-07-10 23:18:16,513 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:16,572 DEM          INFO     Running DEM for Topics_otsu_Topic13_vs_all\n",
      "\u001b[0m2025-07-10 23:18:22,870 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:25,050 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:25,262 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:26,721 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:36,089 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:36,126 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:40,206 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:42,235 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:42,587 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:42,855 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:50,578 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:50,601 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:52,664 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:53,102 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:18:59,692 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-10 23:19:01,735 SCENIC+      INFO     Writing html to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\n",
      "2025-07-10 23:19:01,868 SCENIC+      INFO     Writing output to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Thu Jul 10 23:19:07 2025]\u001b[0m\n",
      "\u001b[32mFinished job 7.\u001b[0m\n",
      "\u001b[32m4 of 12 steps (33%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Jul 10 23:19:07 2025]\u001b[0m\n",
      "\u001b[32mlocalrule prepare_menr:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\n",
      "    jobid: 6\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-10 23:19:16,129 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-10 23:19:20,715 SCENIC+      INFO     Getting cistromes.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-10 23:19:49,626 SCENIC+      INFO     Found 783 TFs.\n",
      "2025-07-10 23:19:49,626 SCENIC+      INFO     Saving TF names to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "2025-07-10 23:19:49,641 SCENIC+      INFO     Writing direct cistromes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad\n",
      "... storing 'motifs' as categorical\n",
      "2025-07-10 23:19:49,798 SCENIC+      INFO     Writing extended cistromes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\n",
      "... storing 'motifs' as categorical\n",
      "\u001b[0m\u001b[32m[Thu Jul 10 23:19:50 2025]\u001b[0m\n",
      "\u001b[32mFinished job 6.\u001b[0m\n",
      "\u001b[32m5 of 12 steps (42%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Thu Jul 10 23:19:50 2025]\u001b[0m\n",
      "\u001b[32mlocalrule tf_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    jobid: 5\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-10 23:20:00,115 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-10 23:20:02,821 SCENIC+      INFO     Using 784 TFs.\n",
      "2025-07-10 23:20:03,264 TF2G         INFO     Calculating TF-to-gene importance\n",
      "Running using 20 cores: 100%|█████████████| 36601/36601 [52:29<00:00, 11.62it/s]\n",
      "2025-07-11 00:12:44,493 TF2G         INFO     Adding correlation coefficients to adjacencies.\n",
      "2025-07-11 00:13:09,563 TF2G         INFO     Warning: adding TFs as their own target to adjecencies matrix. Importance values will be max + 1e-05\n",
      "2025-07-11 00:13:15,462 SCENIC+      INFO     Saving TF to gene adjacencies to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 00:13:25 2025]\u001b[0m\n",
      "\u001b[32mFinished job 5.\u001b[0m\n",
      "\u001b[32m6 of 12 steps (50%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:13:25 2025]\u001b[0m\n",
      "\u001b[32mlocalrule eGRN_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    jobid: 4\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 00:13:36,770 SCENIC+      INFO     Loading TF to gene adjacencies.\n",
      "2025-07-11 00:13:37,919 SCENIC+      INFO     Loading region to gene adjacencies.\n",
      "2025-07-11 00:13:39,173 SCENIC+      INFO     Loading cistromes.\n",
      "2025-07-11 00:13:39,246 GSEA         INFO     Thresholding region to gene relationships\n",
      "  0%|                                                    | 0/14 [00:00<?, ?it/s]\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:   0%|          | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:   4%| | 22/562 [00:00<00:02, 209.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:   8%| | 46/562 [00:00<00:02, 222.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  12%| | 69/562 [00:00<00:02, 208.48it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  16%|▏| 90/562 [00:00<00:04, 110.05it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  19%|▏| 106/562 [00:00<00:03, 115.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  22%|▏| 124/562 [00:00<00:03, 129.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  26%|▎| 148/562 [00:00<00:02, 155.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  31%|▎| 176/562 [00:01<00:02, 187.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  35%|▎| 198/562 [00:01<00:02, 168.45it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  39%|▍| 218/562 [00:01<00:01, 174.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  43%|▍| 240/562 [00:01<00:01, 184.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  47%|▍| 266/562 [00:01<00:01, 203.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  51%|▌| 288/562 [00:01<00:01, 141.78it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  56%|▌| 317/562 [00:01<00:01, 171.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  61%|▌| 345/562 [00:02<00:01, 195.02it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  65%|▋| 368/562 [00:02<00:01, 187.25it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  70%|▋| 395/562 [00:02<00:00, 205.63it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  75%|▋| 420/562 [00:02<00:00, 215.78it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  79%|▊| 444/562 [00:02<00:00, 211.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  83%|▊| 467/562 [00:02<00:00, 215.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  87%|▊| 490/562 [00:02<00:00, 217.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  91%|▉| 513/562 [00:02<00:00, 220.25it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  96%|▉| 537/562 [00:02<00:00, 225.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile: 100%|▉| 561/562 [00:03<00:00, 227.28it/\u001b[0m\u001b[A\n",
      "  7%|███▏                                        | 1/14 [00:05<01:07,  5.16s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   0%|           | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   4%|  | 23/562 [00:00<00:06, 79.49it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   9%| | 48/562 [00:00<00:03, 135.51it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  13%|▏| 71/562 [00:00<00:02, 164.29it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  16%|▏| 91/562 [00:00<00:02, 161.47it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  20%|▏| 110/562 [00:00<00:02, 154.78it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  25%|▏| 138/562 [00:00<00:02, 188.25it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  30%|▎| 171/562 [00:00<00:01, 227.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  35%|▎| 196/562 [00:01<00:01, 208.00it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  39%|▍| 219/562 [00:01<00:01, 205.43it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  44%|▍| 247/562 [00:01<00:01, 224.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  49%|▍| 274/562 [00:01<00:01, 236.62it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  54%|▌| 303/562 [00:01<00:01, 250.63it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  60%|▌| 335/562 [00:01<00:00, 266.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  65%|▋| 363/562 [00:01<00:00, 249.33it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  70%|▋| 393/562 [00:01<00:00, 260.32it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  75%|▋| 420/562 [00:01<00:00, 257.61it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  80%|▊| 447/562 [00:02<00:00, 144.80it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  84%|▊| 473/562 [00:02<00:00, 164.97it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  89%|▉| 498/562 [00:02<00:00, 182.44it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  93%|▉| 524/562 [00:02<00:00, 198.60it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  98%|▉| 553/562 [00:02<00:00, 219.00it/s\u001b[0m\u001b[A\n",
      " 14%|██████▎                                     | 2/14 [00:08<00:50,  4.20s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:   0%|          | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:   5%| | 26/562 [00:00<00:02, 252.70it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:   9%| | 53/562 [00:00<00:01, 258.83it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  14%|▏| 79/562 [00:00<00:02, 237.82it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  18%|▏| 103/562 [00:00<00:02, 206.71it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  22%|▏| 126/562 [00:00<00:02, 211.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  29%|▎| 162/562 [00:00<00:01, 257.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  34%|▎| 189/562 [00:00<00:01, 259.31it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  38%|▍| 216/562 [00:00<00:01, 234.13it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  44%|▍| 245/562 [00:01<00:01, 248.83it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  49%|▍| 275/562 [00:01<00:01, 262.56it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  55%|▌| 309/562 [00:01<00:00, 283.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  61%|▌| 341/562 [00:01<00:00, 293.48it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  66%|▋| 371/562 [00:01<00:00, 272.41it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  71%|▋| 401/562 [00:01<00:00, 279.10it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  77%|▊| 430/562 [00:01<00:00, 273.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  81%|▊| 458/562 [00:02<00:00, 154.85it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  86%|▊| 485/562 [00:02<00:00, 175.57it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  91%|▉| 513/562 [00:02<00:00, 197.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  97%|▉| 543/562 [00:02<00:00, 220.64it/\u001b[0m\u001b[A\n",
      " 21%|█████████▍                                  | 3/14 [00:11<00:40,  3.71s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   0%| | 0/562 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   2%| | 14/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   5%| | 28/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   8%| | 43/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  10%| | 56/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  12%| | 69/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  15%|▏| 82/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  17%|▏| 93/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  19%|▏| 104/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  20%|▏| 114/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  23%|▏| 129/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  25%|▎| 143/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  29%|▎| 162/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  32%|▎| 178/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  34%|▎| 193/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  36%|▎| 204/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  39%|▍| 218/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  41%|▍| 230/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  44%|▍| 247/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  47%|▍| 262/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  49%|▍| 276/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  52%|▌| 290/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  55%|▌| 307/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  57%|▌| 323/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  60%|▌| 338/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  63%|▋| 353/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  65%|▋| 368/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  69%|▋| 386/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  71%|▋| 401/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  74%|▋| 418/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  77%|▊| 433/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  80%|▊| 447/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  82%|▊| 461/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  85%|▊| 475/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  87%|▊| 489/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  90%|▉| 503/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  92%|▉| 517/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  94%|▉| 531/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  97%|▉| 545/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  99%|▉| 557/562 [\u001b[0m\u001b[A\n",
      " 29%|████████████▌                               | 4/14 [00:17<00:45,  4.60s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   0%| | 0/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   2%| | 10/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   4%| | 20/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   5%| | 29/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   7%| | 39/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   9%| | 48/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  10%| | 58/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  12%| | 67/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  14%|▏| 76/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  15%|▏| 84/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  16%|▏| 92/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  18%|▏| 100/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  19%|▏| 107/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  20%|▏| 114/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  22%|▏| 122/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  23%|▏| 131/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  25%|▏| 139/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  26%|▎| 148/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  28%|▎| 159/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  31%|▎| 172/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  32%|▎| 182/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  34%|▎| 192/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  36%|▎| 201/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  37%|▎| 210/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  39%|▍| 218/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  40%|▍| 226/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  42%|▍| 237/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  44%|▍| 246/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  46%|▍| 256/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  47%|▍| 265/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  49%|▍| 274/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  50%|▌| 282/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  52%|▌| 292/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  54%|▌| 302/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  56%|▌| 312/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  57%|▌| 322/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  59%|▌| 332/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  61%|▌| 341/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  62%|▌| 351/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  64%|▋| 360/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  65%|▋| 368/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  67%|▋| 378/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  69%|▋| 388/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  71%|▋| 397/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  72%|▋| 407/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  74%|▋| 417/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  76%|▊| 426/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  77%|▊| 434/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  79%|▊| 443/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  80%|▊| 451/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  82%|▊| 461/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  84%|▊| 471/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  85%|▊| 480/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  87%|▊| 489/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  89%|▉| 498/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  90%|▉| 507/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  92%|▉| 516/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  93%|▉| 525/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  95%|▉| 534/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  97%|▉| 544/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  98%|▉| 553/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene: 100%|█| 562/562 \u001b[0m\u001b[A\n",
      " 36%|███████████████▋                            | 5/14 [00:26<00:53,  5.91s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   0%| | 0/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   1%| | 8/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   3%| | 16/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   4%| | 23/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   5%| | 30/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   7%| | 38/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   8%| | 45/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   9%| | 52/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  11%| | 60/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  12%| | 68/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  13%|▏| 75/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  15%|▏| 82/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  16%|▏| 89/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  17%|▏| 96/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  18%|▏| 102/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  19%|▏| 107/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  20%|▏| 113/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  22%|▏| 121/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  23%|▏| 130/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  25%|▏| 138/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  26%|▎| 147/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  28%|▎| 156/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  30%|▎| 167/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  31%|▎| 176/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  33%|▎| 185/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  35%|▎| 194/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  36%|▎| 202/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  38%|▍| 212/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  39%|▍| 220/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  40%|▍| 227/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  42%|▍| 237/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  44%|▍| 246/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  45%|▍| 255/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  47%|▍| 264/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  49%|▍| 273/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  50%|▌| 282/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  52%|▌| 292/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  54%|▌| 303/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  56%|▌| 315/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  58%|▌| 325/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  60%|▌| 335/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  62%|▌| 346/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  63%|▋| 356/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  65%|▋| 366/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  67%|▋| 375/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  69%|▋| 386/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  70%|▋| 395/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  72%|▋| 404/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  74%|▋| 415/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  76%|▊| 425/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  77%|▊| 434/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  79%|▊| 443/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  80%|▊| 452/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  82%|▊| 461/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  84%|▊| 471/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  85%|▊| 480/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  87%|▊| 489/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  89%|▉| 498/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  90%|▉| 507/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  92%|▉| 516/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  93%|▉| 525/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  95%|▉| 535/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  97%|▉| 545/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  99%|▉| 554/562 \u001b[0m\u001b[A\n",
      " 43%|██████████████████▊                         | 6/14 [00:34<00:54,  6.85s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   0%|         | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   3%| | 16/562 [00:00<00:03, 152.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   6%| | 32/562 [00:00<00:03, 144.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   8%| | 47/562 [00:01<00:14, 34.82it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  11%| | 63/562 [00:01<00:09, 50.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  13%|▏| 75/562 [00:01<00:08, 58.28it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  15%|▏| 86/562 [00:01<00:07, 65.46it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  17%|▏| 97/562 [00:01<00:06, 74.04it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  19%|▏| 108/562 [00:01<00:05, 76.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  22%|▏| 123/562 [00:01<00:04, 92.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  25%|▏| 140/562 [00:01<00:03, 110.09it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  29%|▎| 162/562 [00:01<00:02, 137.35it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  32%|▎| 179/562 [00:02<00:02, 145.90it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  35%|▎| 195/562 [00:02<00:03, 121.63it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  37%|▎| 209/562 [00:02<00:03, 116.94it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  40%|▍| 222/562 [00:02<00:02, 116.56it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  43%|▍| 240/562 [00:02<00:02, 131.37it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  46%|▍| 260/562 [00:02<00:02, 147.03it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  49%|▍| 276/562 [00:02<00:01, 146.86it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  52%|▌| 292/562 [00:02<00:01, 149.13it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  56%|▌| 314/562 [00:03<00:01, 167.57it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  59%|▌| 332/562 [00:03<00:01, 170.94it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  62%|▌| 350/562 [00:03<00:01, 169.09it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  65%|▋| 368/562 [00:03<00:01, 143.29it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  69%|▋| 388/562 [00:03<00:01, 157.36it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  72%|▋| 405/562 [00:03<00:01, 153.41it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  75%|▊| 422/562 [00:03<00:00, 154.39it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  78%|▊| 438/562 [00:03<00:00, 145.36it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  81%|▊| 453/562 [00:03<00:00, 145.87it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  83%|▊| 468/562 [00:04<00:00, 146.56it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  86%|▊| 483/562 [00:04<00:00, 144.01it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  89%|▉| 499/562 [00:04<00:00, 147.20it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  91%|▉| 514/562 [00:04<00:00, 146.26it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  94%|▉| 529/562 [00:04<00:00, 146.12it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  98%|▉| 548/562 [00:04<00:00, 157.39it\u001b[0m\u001b[A\n",
      " 50%|██████████████████████                      | 7/14 [01:27<02:33, 21.88s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   0%|          | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   2%| | 13/562 [00:00<00:04, 129.70it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   5%| | 26/562 [00:00<00:04, 108.18it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   7%| | 39/562 [00:00<00:04, 114.03it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   9%| | 52/562 [00:00<00:04, 118.13it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  12%| | 67/562 [00:00<00:03, 124.54it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  14%|▏| 80/562 [00:00<00:04, 108.59it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  16%|▏| 92/562 [00:00<00:04, 98.56it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  18%|▏| 103/562 [00:01<00:05, 88.34it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  20%|▏| 113/562 [00:01<00:05, 82.34it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  22%|▏| 126/562 [00:01<00:04, 93.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  25%|▎| 141/562 [00:01<00:03, 107.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  29%|▎| 164/562 [00:01<00:02, 139.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  32%|▎| 179/562 [00:01<00:02, 137.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  35%|▎| 194/562 [00:01<00:03, 109.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  37%|▎| 207/562 [00:01<00:03, 104.40it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  39%|▍| 219/562 [00:02<00:03, 104.40it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  41%|▍| 231/562 [00:02<00:07, 42.11it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  44%|▍| 248/562 [00:02<00:05, 57.01it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  46%|▍| 261/562 [00:02<00:04, 67.21it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  49%|▍| 275/562 [00:03<00:03, 78.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  52%|▌| 290/562 [00:03<00:02, 92.12it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  55%|▌| 309/562 [00:03<00:02, 112.05it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  58%|▌| 325/562 [00:03<00:01, 122.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  60%|▌| 340/562 [00:03<00:01, 126.27it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  63%|▋| 355/562 [00:03<00:01, 131.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  66%|▋| 370/562 [00:03<00:01, 117.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  69%|▋| 387/562 [00:03<00:01, 127.41it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  71%|▋| 401/562 [00:04<00:01, 121.60it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  75%|▋| 419/562 [00:04<00:01, 133.31it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  77%|▊| 433/562 [00:04<00:01, 123.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  79%|▊| 446/562 [00:04<00:00, 124.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  82%|▊| 460/562 [00:04<00:00, 127.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  84%|▊| 474/562 [00:04<00:00, 126.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  87%|▊| 487/562 [00:04<00:00, 124.52it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  89%|▉| 501/562 [00:04<00:00, 126.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  91%|▉| 514/562 [00:04<00:00, 126.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  94%|▉| 527/562 [00:05<00:00, 125.52it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  97%|▉| 543/562 [00:05<00:00, 132.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  99%|▉| 557/562 [00:05<00:00, 130.05it/\u001b[0m\u001b[A\n",
      " 57%|█████████████████████████▏                  | 8/14 [01:33<01:41, 16.93s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   0%|           | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   3%| | 18/562 [00:00<00:03, 173.74it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   6%| | 36/562 [00:00<00:03, 150.91it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   9%| | 53/562 [00:00<00:03, 154.38it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  12%| | 70/562 [00:00<00:03, 156.71it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  15%|▏| 86/562 [00:00<00:03, 134.37it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  18%|▏| 100/562 [00:00<00:03, 125.52it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  20%|▏| 113/562 [00:00<00:04, 111.24it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  23%|▏| 132/562 [00:00<00:03, 130.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  27%|▎| 153/562 [00:01<00:02, 152.03it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  32%|▎| 178/562 [00:01<00:02, 177.17it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  35%|▎| 197/562 [00:01<00:02, 140.15it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  38%|▍| 213/562 [00:01<00:02, 141.70it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  41%|▍| 229/562 [00:01<00:02, 135.47it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  44%|▍| 250/562 [00:01<00:02, 151.10it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  47%|▍| 266/562 [00:01<00:01, 152.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  51%|▌| 284/562 [00:01<00:01, 159.48it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  55%|▌| 307/562 [00:02<00:01, 177.36it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  58%|▌| 328/562 [00:02<00:01, 183.53it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  62%|▌| 348/562 [00:02<00:01, 183.67it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  65%|▋| 367/562 [00:02<00:01, 161.85it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  69%|▋| 387/562 [00:02<00:01, 170.87it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  72%|▋| 405/562 [00:02<00:00, 164.69it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  75%|▊| 423/562 [00:02<00:00, 168.73it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  78%|▊| 441/562 [00:02<00:00, 162.35it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  81%|▊| 458/562 [00:02<00:00, 163.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  85%|▊| 475/562 [00:03<00:00, 162.05it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  88%|▉| 492/562 [00:03<00:00, 161.23it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  91%|▉| 509/562 [00:03<00:00, 163.39it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  94%|▉| 526/562 [00:03<00:00, 163.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  97%|▉| 545/562 [00:03<00:00, 169.57it/s\u001b[0m\u001b[A\n",
      " 64%|████████████████████████████▎               | 9/14 [01:38<01:05, 13.01s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   0%|          | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   4%| | 20/562 [00:00<00:02, 199.07it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   7%| | 40/562 [00:00<00:02, 195.72it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  12%| | 65/562 [00:00<00:02, 217.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  15%|▏| 87/562 [00:00<00:02, 168.77it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  19%|▏| 106/562 [00:00<00:02, 153.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  22%|▏| 123/562 [00:00<00:02, 157.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  26%|▎| 148/562 [00:00<00:02, 181.49it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  32%|▎| 180/562 [00:00<00:01, 219.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  36%|▎| 203/562 [00:01<00:02, 176.49it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  40%|▍| 223/562 [00:01<00:01, 180.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  44%|▍| 248/562 [00:01<00:01, 197.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  48%|▍| 269/562 [00:01<00:01, 194.59it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  52%|▌| 293/562 [00:01<00:01, 205.12it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  57%|▌| 321/562 [00:01<00:01, 223.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  62%|▌| 346/562 [00:01<00:00, 229.27it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  66%|▋| 370/562 [00:01<00:00, 200.52it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  70%|▋| 393/562 [00:02<00:00, 206.91it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  74%|▋| 415/562 [00:02<00:00, 209.13it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  78%|▊| 437/562 [00:02<00:00, 197.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  82%|▊| 460/562 [00:02<00:00, 204.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  86%|▊| 481/562 [00:02<00:00, 200.89it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  90%|▉| 504/562 [00:02<00:00, 207.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  94%|▉| 526/562 [00:02<00:00, 210.09it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  98%|▉| 551/562 [00:02<00:00, 220.25it/\u001b[0m\u001b[A\n",
      " 71%|██████████████████████████████▋            | 10/14 [01:41<00:40, 10.11s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   0%| | 0/562 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   2%| | 9/562 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   3%| | 18/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   5%| | 26/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   6%| | 34/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   7%| | 42/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:   9%| | 50/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  10%| | 58/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  12%| | 67/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  13%|▏| 75/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  15%|▏| 82/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  15%|▏| 87/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  17%|▏| 93/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  17%|▏| 98/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  18%|▏| 103/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  19%|▏| 108/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  20%|▏| 114/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  22%|▏| 122/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  23%|▏| 130/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  24%|▏| 137/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  26%|▎| 146/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  28%|▎| 157/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  30%|▎| 169/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  32%|▎| 178/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  33%|▎| 187/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  35%|▎| 195/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  36%|▎| 202/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  38%|▍| 212/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  39%|▍| 219/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  40%|▍| 226/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  42%|▍| 235/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  43%|▍| 242/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  44%|▍| 250/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  46%|▍| 258/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  47%|▍| 266/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  49%|▍| 274/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  50%|▌| 281/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  52%|▌| 290/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  53%|▌| 299/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  55%|▌| 308/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  57%|▌| 318/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  58%|▌| 327/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  60%|▌| 336/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  62%|▌| 346/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  63%|▋| 355/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  65%|▋| 363/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  66%|▋| 370/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  67%|▋| 378/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  69%|▋| 386/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  70%|▋| 394/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  71%|▋| 401/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  73%|▋| 410/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  75%|▋| 419/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  76%|▊| 427/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  77%|▊| 434/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  79%|▊| 442/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  80%|▊| 449/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  81%|▊| 456/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  82%|▊| 463/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  84%|▊| 471/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  85%|▊| 478/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  86%|▊| 486/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  88%|▉| 494/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  89%|▉| 501/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  90%|▉| 508/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  92%|▉| 516/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  93%|▉| 523/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  94%|▉| 530/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  96%|▉| 539/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  97%|▉| 547/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene:  99%|▉| 555/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 5 region-to-gene links per gene: 100%|█| 562/562 [\u001b[0m\u001b[A\n",
      " 79%|█████████████████████████████████▊         | 11/14 [01:51<00:30, 10.10s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   0%| | 0/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   1%| | 6/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   2%| | 12/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   3%| | 17/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   4%| | 22/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   5%| | 27/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   6%| | 32/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   7%| | 37/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   8%| | 43/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   9%| | 48/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   9%| | 53/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  10%| | 59/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  12%| | 66/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  13%|▏| 72/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  14%|▏| 77/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  15%|▏| 82/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  15%|▏| 87/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  16%|▏| 92/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  17%|▏| 96/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  18%|▏| 100/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  19%|▏| 104/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  19%|▏| 108/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  20%|▏| 112/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  21%|▏| 118/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  22%|▏| 123/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  23%|▏| 129/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  24%|▏| 134/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  25%|▏| 139/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  26%|▎| 145/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  27%|▎| 152/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  28%|▎| 160/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  30%|▎| 168/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  31%|▎| 175/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  32%|▎| 181/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  33%|▎| 187/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  34%|▎| 192/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  35%|▎| 197/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  36%|▎| 201/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  37%|▎| 206/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  38%|▍| 212/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  39%|▍| 217/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  39%|▍| 221/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  40%|▍| 225/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  41%|▍| 231/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  42%|▍| 237/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  43%|▍| 242/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  44%|▍| 247/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  45%|▍| 252/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  46%|▍| 257/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  47%|▍| 262/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  48%|▍| 267/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  48%|▍| 272/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  49%|▍| 277/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  50%|▌| 282/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  51%|▌| 289/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  52%|▌| 295/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  54%|▌| 301/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  55%|▌| 307/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  56%|▌| 314/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  57%|▌| 320/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  58%|▌| 326/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  59%|▌| 332/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  60%|▌| 338/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  62%|▌| 346/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  63%|▋| 352/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  64%|▋| 358/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  65%|▋| 363/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  65%|▋| 368/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  67%|▋| 374/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  68%|▋| 380/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  69%|▋| 386/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  70%|▋| 391/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  70%|▋| 396/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  71%|▋| 401/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  72%|▋| 406/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  73%|▋| 412/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  74%|▋| 418/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  75%|▊| 423/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  76%|▊| 428/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  77%|▊| 433/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  78%|▊| 438/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  79%|▊| 443/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  80%|▊| 448/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  81%|▊| 453/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  81%|▊| 458/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  82%|▊| 463/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  83%|▊| 469/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  84%|▊| 474/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  85%|▊| 479/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  86%|▊| 485/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  87%|▊| 490/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  88%|▉| 496/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  89%|▉| 501/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  90%|▉| 506/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  91%|▉| 511/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  92%|▉| 517/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  93%|▉| 522/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  94%|▉| 527/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  95%|▉| 533/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  96%|▉| 539/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  97%|▉| 544/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  98%|▉| 549/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  99%|▉| 554/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  99%|▉| 559/562 \u001b[0m\u001b[A\n",
      " 86%|████████████████████████████████████▊      | 12/14 [02:06<00:22, 11.42s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   0%| | 0/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   1%| | 5/562 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   2%| | 10/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   3%| | 15/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   3%| | 19/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   4%| | 23/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   5%| | 27/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   6%| | 31/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   6%| | 36/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   7%| | 40/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   8%| | 45/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   9%| | 49/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   9%| | 53/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  10%| | 58/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  11%| | 63/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  12%| | 68/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  13%|▏| 72/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  14%|▏| 76/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  14%|▏| 80/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  15%|▏| 84/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  16%|▏| 88/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  17%|▏| 93/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  17%|▏| 97/562 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  18%|▏| 101/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  19%|▏| 104/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  19%|▏| 107/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  20%|▏| 110/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  20%|▏| 114/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  21%|▏| 119/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  22%|▏| 124/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  23%|▏| 129/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  24%|▏| 134/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  25%|▏| 138/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  25%|▎| 142/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  26%|▎| 147/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  27%|▎| 153/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  28%|▎| 159/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  30%|▎| 166/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  31%|▎| 172/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  32%|▎| 178/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  33%|▎| 183/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  33%|▎| 188/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  34%|▎| 192/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  35%|▎| 196/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  36%|▎| 200/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  36%|▎| 203/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  37%|▎| 209/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  38%|▍| 213/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  39%|▍| 217/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  39%|▍| 221/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  40%|▍| 225/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  41%|▍| 231/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  42%|▍| 236/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  43%|▍| 241/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  44%|▍| 246/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  45%|▍| 251/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  46%|▍| 256/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  46%|▍| 261/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  47%|▍| 265/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  48%|▍| 269/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  49%|▍| 274/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  49%|▍| 278/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  50%|▌| 282/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  51%|▌| 288/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  52%|▌| 293/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  53%|▌| 298/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  54%|▌| 304/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  55%|▌| 309/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  56%|▌| 315/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  57%|▌| 320/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  58%|▌| 324/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  58%|▌| 328/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  59%|▌| 333/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  60%|▌| 337/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  61%|▌| 342/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  62%|▌| 348/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  63%|▋| 353/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  64%|▋| 358/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  64%|▋| 362/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  65%|▋| 366/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  66%|▋| 370/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  67%|▋| 375/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  68%|▋| 380/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  69%|▋| 385/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  69%|▋| 389/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  70%|▋| 393/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  71%|▋| 397/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  72%|▋| 402/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  72%|▋| 407/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  73%|▋| 412/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  74%|▋| 417/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  75%|▊| 422/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  76%|▊| 426/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  77%|▊| 430/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  77%|▊| 434/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  78%|▊| 439/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  79%|▊| 443/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  80%|▊| 447/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  80%|▊| 451/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  81%|▊| 456/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  82%|▊| 461/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  83%|▊| 465/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  84%|▊| 470/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  85%|▊| 475/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  85%|▊| 479/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  86%|▊| 484/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  87%|▊| 488/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  88%|▉| 493/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  89%|▉| 498/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  89%|▉| 502/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  90%|▉| 506/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  91%|▉| 511/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  92%|▉| 516/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  93%|▉| 521/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  93%|▉| 525/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  94%|▉| 529/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  95%|▉| 534/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  96%|▉| 539/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  97%|▉| 544/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  98%|▉| 549/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  98%|▉| 553/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  99%|▉| 557/562 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g: 100%|▉| 561/562 \u001b[0m\u001b[A\n",
      " 93%|███████████████████████████████████████▉   | 13/14 [02:23<00:13, 13.13s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   0%|         | 0/562 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   1%| | 8/562 [00:00<00:07, 78.38it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   3%| | 16/562 [00:00<00:08, 64.87it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   4%| | 23/562 [00:00<00:09, 59.66it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   5%| | 30/562 [00:00<00:08, 59.96it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   7%| | 37/562 [00:00<00:08, 62.35it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   8%| | 44/562 [00:00<00:08, 64.42it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   9%| | 51/562 [00:00<00:07, 65.51it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  10%| | 59/562 [00:00<00:07, 68.07it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  12%| | 67/562 [00:01<00:07, 69.20it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  13%|▏| 74/562 [00:01<00:08, 60.04it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  14%|▏| 81/562 [00:01<00:08, 59.42it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  16%|▏| 88/562 [00:01<00:09, 51.76it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  17%|▏| 94/562 [00:01<00:08, 52.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  18%|▏| 100/562 [00:01<00:09, 50.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  19%|▏| 106/562 [00:01<00:09, 47.04it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  20%|▏| 111/562 [00:01<00:10, 44.45it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  21%|▏| 119/562 [00:02<00:08, 51.52it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  22%|▏| 126/562 [00:02<00:07, 56.08it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  24%|▏| 135/562 [00:02<00:06, 64.23it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  25%|▎| 142/562 [00:02<00:06, 65.54it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  27%|▎| 153/562 [00:02<00:05, 77.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  30%|▎| 167/562 [00:02<00:04, 94.00it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  31%|▎| 177/562 [00:02<00:04, 83.80it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  33%|▎| 186/562 [00:02<00:05, 73.99it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  35%|▎| 194/562 [00:03<00:06, 60.06it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  36%|▎| 201/562 [00:03<00:06, 55.33it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  37%|▎| 210/562 [00:03<00:05, 62.21it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  39%|▍| 217/562 [00:03<00:05, 61.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  40%|▍| 224/562 [00:03<00:05, 58.36it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  42%|▍| 234/562 [00:03<00:04, 67.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  43%|▍| 242/562 [00:03<00:04, 69.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  44%|▍| 250/562 [00:03<00:04, 71.67it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  46%|▍| 258/562 [00:04<00:04, 73.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  47%|▍| 266/562 [00:04<00:04, 68.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  49%|▍| 274/562 [00:04<00:04, 71.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  50%|▌| 282/562 [00:04<00:04, 69.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  52%|▌| 292/562 [00:04<00:03, 75.03it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  54%|▌| 303/562 [00:04<00:03, 81.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  56%|▌| 314/562 [00:04<00:02, 88.27it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  57%|▌| 323/562 [00:04<00:02, 81.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  59%|▌| 332/562 [00:05<00:02, 79.02it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  61%|▌| 341/562 [00:05<00:02, 76.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  62%|▌| 350/562 [00:05<00:02, 78.63it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  64%|▋| 358/562 [00:05<00:02, 74.22it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  65%|▋| 366/562 [00:05<00:03, 63.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  67%|▋| 374/562 [00:05<00:02, 67.03it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  68%|▋| 382/562 [00:05<00:02, 70.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  69%|▋| 390/562 [00:05<00:02, 70.61it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  71%|▋| 398/562 [00:06<00:02, 63.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  72%|▋| 407/562 [00:06<00:02, 69.60it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  74%|▋| 415/562 [00:06<00:02, 72.26it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  75%|▊| 423/562 [00:06<00:02, 69.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  77%|▊| 431/562 [00:06<00:02, 64.08it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  78%|▊| 439/562 [00:06<00:01, 67.05it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  79%|▊| 446/562 [00:06<00:01, 66.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  81%|▊| 454/562 [00:06<00:01, 68.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  82%|▊| 461/562 [00:06<00:01, 67.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  83%|▊| 469/562 [00:07<00:01, 70.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  85%|▊| 477/562 [00:07<00:01, 65.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  86%|▊| 485/562 [00:07<00:01, 66.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  88%|▉| 493/562 [00:07<00:01, 67.96it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  89%|▉| 500/562 [00:07<00:00, 67.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  90%|▉| 507/562 [00:07<00:00, 67.12it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  91%|▉| 514/562 [00:07<00:00, 67.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  93%|▉| 521/562 [00:07<00:00, 67.57it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  94%|▉| 528/562 [00:07<00:00, 67.44it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  96%|▉| 537/562 [00:08<00:00, 71.72it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  97%|▉| 545/562 [00:08<00:00, 71.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  98%|▉| 553/562 [00:08<00:00, 72.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g: 100%|▉| 561/562 [00:08<00:00, 69.98it/\u001b[0m\u001b[A\n",
      "100%|███████████████████████████████████████████| 14/14 [03:01<00:00, 12.94s/it]\u001b[A\n",
      "2025-07-11 00:16:40,457 GSEA         INFO     Subsetting TF2G adjacencies for TF with motif.\n",
      "2025-07-11 00:16:40,595 GSEA         INFO     Running GSEA...\n",
      "2025-07-11 00:16:40,595 GSEA         INFO     Generating rankings...\n",
      "100%|████████████████████████████████████████| 562/562 [00:04<00:00, 123.16it/s]\n",
      "100%|███████████████████████████████████████████| 9/9 [00:00<00:00, 2078.56it/s]\n",
      "Running for Positive TF to gene:   1%|        | 40/7326 [00:08<25:43,  4.72it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   9%|▌     | 660/7326 [00:31<00:44, 149.64it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:  12%|▋     | 900/7326 [00:33<00:43, 148.35it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene: 100%|██████| 7326/7326 [01:36<00:00, 75.78it/s]\n",
      "Running for Negative TF to gene: 100%|███| 7326/7326 [00:00<00:00, 18108.33it/s]\n",
      "2025-07-11 00:18:23,593 GSEA         INFO     Subsetting on adjusted pvalue: 1, minimal NES: 0 and minimal leading edge genes 10\n",
      "2025-07-11 00:18:23,755 GSEA         INFO     Merging eRegulons\n",
      "2025-07-11 00:18:26,193 SCENIC+      INFO     Formatting eGRN as table.\n",
      "2025-07-11 00:18:27,004 SCENIC+      INFO     Calculating triplet ranking.\n",
      "2025-07-11 00:18:27,223 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 00:18:52,304 SCENIC+      INFO     Saving network to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 00:18:56 2025]\u001b[0m\n",
      "\u001b[32mFinished job 4.\u001b[0m\n",
      "\u001b[32m7 of 12 steps (58%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:18:56 2025]\u001b[0m\n",
      "\u001b[32mlocalrule AUCell_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "    jobid: 3\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 00:19:06,012 SCENIC+      INFO     Reading data.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 00:19:11,248 SCENIC+      INFO     Calculating enrichment scores.\n",
      "2025-07-11 00:20:58,472 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 00:20:59 2025]\u001b[0m\n",
      "\u001b[32mFinished job 3.\u001b[0m\n",
      "\u001b[32m8 of 12 steps (67%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:20:59 2025]\u001b[0m\n",
      "\u001b[32mlocalrule eGRN_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    jobid: 13\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 00:21:09,220 SCENIC+      INFO     Loading TF to gene adjacencies.\n",
      "2025-07-11 00:21:10,370 SCENIC+      INFO     Loading region to gene adjacencies.\n",
      "2025-07-11 00:21:11,547 SCENIC+      INFO     Loading cistromes.\n",
      "2025-07-11 00:21:11,608 GSEA         INFO     Thresholding region to gene relationships\n",
      "  0%|                                                    | 0/14 [00:00<?, ?it/s]\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   0%|          | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   8%| | 24/306 [00:00<00:01, 232.89it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  16%|▏| 50/306 [00:00<00:01, 246.50it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  25%|▏| 75/306 [00:00<00:01, 217.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  32%|▎| 98/306 [00:00<00:01, 125.50it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  41%|▍| 125/306 [00:00<00:01, 157.17it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  47%|▍| 145/306 [00:00<00:01, 152.72it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  57%|▌| 174/306 [00:00<00:00, 185.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  68%|▋| 207/306 [00:01<00:00, 220.47it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  78%|▊| 238/306 [00:01<00:00, 242.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  87%|▊| 265/306 [00:01<00:00, 234.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  96%|▉| 294/306 [00:01<00:00, 248.77it/\u001b[0m\u001b[A\n",
      "  7%|███▏                                        | 1/14 [00:03<00:47,  3.66s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:   0%|           | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:   9%| | 28/306 [00:00<00:01, 276.31it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  18%|▏| 56/306 [00:00<00:00, 251.66it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  27%|▎| 82/306 [00:00<00:01, 127.71it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  36%|▎| 109/306 [00:00<00:01, 160.08it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  42%|▍| 130/306 [00:00<00:01, 170.03it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  49%|▍| 151/306 [00:00<00:00, 177.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  60%|▌| 183/306 [00:00<00:00, 215.22it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  72%|▋| 220/306 [00:01<00:00, 255.99it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  82%|▊| 250/306 [00:01<00:00, 266.00it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, negative r2g:  91%|▉| 279/306 [00:01<00:00, 267.08it/s\u001b[0m\u001b[A\n",
      " 14%|██████▎                                     | 2/14 [00:05<00:32,  2.75s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:   0%|          | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  11%| | 33/306 [00:00<00:00, 323.88it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  22%|▏| 66/306 [00:00<00:00, 293.08it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  31%|▎| 96/306 [00:00<00:00, 270.21it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  42%|▍| 128/306 [00:00<00:00, 282.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  51%|▌| 157/306 [00:00<00:00, 173.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  63%|▋| 193/306 [00:00<00:00, 215.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  76%|▊| 234/306 [00:00<00:00, 261.80it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  87%|▊| 266/306 [00:01<00:00, 270.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  99%|▉| 303/306 [00:01<00:00, 294.28it/\u001b[0m\u001b[A\n",
      " 21%|█████████▍                                  | 3/14 [00:07<00:25,  2.36s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   0%| | 0/306 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   5%| | 14/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   9%| | 28/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  14%|▏| 42/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  18%|▏| 56/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  23%|▏| 69/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  26%|▎| 81/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  30%|▎| 93/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  36%|▎| 111/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  41%|▍| 124/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  45%|▍| 137/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  50%|▍| 152/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  56%|▌| 170/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  60%|▌| 184/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  66%|▋| 202/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  72%|▋| 219/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  77%|▊| 235/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  82%|▊| 251/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  87%|▊| 266/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  92%|▉| 282/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  97%|▉| 297/306 [\u001b[0m\u001b[A\n",
      " 29%|████████████▌                               | 4/14 [00:11<00:27,  2.78s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   0%| | 0/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   3%| | 10/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:   7%| | 20/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  10%| | 30/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  13%|▏| 39/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  16%|▏| 48/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  19%|▏| 57/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  21%|▏| 65/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  23%|▏| 71/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  26%|▎| 79/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  28%|▎| 86/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  30%|▎| 93/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  34%|▎| 103/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  37%|▎| 114/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  40%|▍| 123/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  43%|▍| 132/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  46%|▍| 140/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  49%|▍| 151/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  53%|▌| 162/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  56%|▌| 171/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  59%|▌| 180/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  63%|▋| 192/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  66%|▋| 203/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  70%|▋| 214/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  73%|▋| 224/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  76%|▊| 234/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  80%|▊| 244/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  83%|▊| 254/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  86%|▊| 264/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  90%|▉| 274/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  93%|▉| 284/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  96%|▉| 294/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, negative r2g:  99%|▉| 304/306 \u001b[0m\u001b[A\n",
      " 36%|███████████████▋                            | 5/14 [00:15<00:31,  3.48s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   0%| | 0/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   3%| | 9/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   6%| | 18/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:   8%| | 26/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  11%| | 34/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  14%|▏| 42/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  16%|▏| 50/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  19%|▏| 58/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  21%|▏| 65/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  24%|▏| 72/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  26%|▎| 79/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  28%|▎| 86/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  30%|▎| 93/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  34%|▎| 103/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  37%|▎| 112/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  40%|▍| 121/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  42%|▍| 129/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  44%|▍| 136/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  47%|▍| 143/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  50%|▍| 152/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  53%|▌| 162/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  56%|▌| 172/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  59%|▌| 181/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  62%|▌| 191/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  66%|▋| 201/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  69%|▋| 210/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  72%|▋| 220/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  75%|▋| 229/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  78%|▊| 238/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  81%|▊| 247/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  84%|▊| 256/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  87%|▊| 266/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  90%|▉| 275/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  93%|▉| 285/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  96%|▉| 294/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, negative r2g:  99%|▉| 302/306 \u001b[0m\u001b[A\n",
      " 43%|██████████████████▊                         | 6/14 [00:21<00:33,  4.18s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   0%|         | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:   6%| | 17/306 [00:00<00:01, 161.16it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  11%| | 35/306 [00:00<00:01, 171.50it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  17%|▏| 53/306 [00:00<00:01, 169.36it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  23%|▏| 70/306 [00:00<00:01, 146.56it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  28%|▎| 86/306 [00:00<00:01, 140.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  34%|▎| 104/306 [00:00<00:01, 151.01it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  40%|▍| 123/306 [00:00<00:01, 161.42it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  46%|▍| 140/306 [00:00<00:01, 122.59it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  53%|▌| 162/306 [00:01<00:00, 144.91it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  59%|▌| 182/306 [00:01<00:00, 157.43it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  67%|▋| 204/306 [00:01<00:00, 173.04it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  74%|▋| 226/306 [00:01<00:00, 185.20it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  81%|▊| 247/306 [00:01<00:00, 190.64it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  87%|▊| 267/306 [00:01<00:00, 173.94it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, negative r2g:  94%|▉| 288/306 [00:01<00:00, 183.24it\u001b[0m\u001b[A\n",
      " 50%|██████████████████████                      | 7/14 [01:12<02:15, 19.37s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   0%|          | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   5%| | 14/306 [00:00<00:02, 134.13it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   9%| | 28/306 [00:00<00:02, 126.41it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  13%|▏| 41/306 [00:00<00:02, 126.73it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  18%|▏| 56/306 [00:00<00:01, 129.87it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  23%|▏| 69/306 [00:00<00:01, 126.13it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  27%|▎| 82/306 [00:00<00:01, 119.90it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  31%|▎| 95/306 [00:00<00:01, 111.01it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  39%|▍| 118/306 [00:00<00:01, 142.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  43%|▍| 133/306 [00:01<00:03, 56.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  47%|▍| 145/306 [00:01<00:02, 65.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  53%|▌| 161/306 [00:01<00:01, 80.44it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  58%|▌| 177/306 [00:01<00:01, 95.33it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  65%|▋| 198/306 [00:01<00:00, 118.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  71%|▋| 216/306 [00:02<00:00, 132.75it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  76%|▊| 233/306 [00:02<00:00, 141.06it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  82%|▊| 251/306 [00:02<00:00, 145.84it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  88%|▉| 268/306 [00:02<00:00, 143.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  93%|▉| 284/306 [00:02<00:00, 146.13it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  98%|▉| 300/306 [00:02<00:00, 149.79it/\u001b[0m\u001b[A\n",
      " 57%|█████████████████████████▏                  | 8/14 [01:15<01:26, 14.39s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   0%|           | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   6%| | 19/306 [00:00<00:01, 187.02it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  12%| | 38/306 [00:00<00:01, 175.49it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  18%|▏| 56/306 [00:00<00:01, 175.61it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  24%|▏| 74/306 [00:00<00:01, 166.93it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  30%|▎| 91/306 [00:00<00:01, 144.36it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  38%|▍| 117/306 [00:00<00:01, 176.41it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  44%|▍| 136/306 [00:00<00:01, 142.82it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  51%|▌| 157/306 [00:00<00:00, 159.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  58%|▌| 177/306 [00:01<00:00, 169.23it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  66%|▋| 201/306 [00:01<00:00, 188.26it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  73%|▋| 223/306 [00:01<00:00, 194.17it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  81%|▊| 247/306 [00:01<00:00, 203.48it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  88%|▉| 268/306 [00:01<00:00, 189.24it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  94%|▉| 289/306 [00:01<00:00, 193.91it/s\u001b[0m\u001b[A\n",
      " 64%|████████████████████████████▎               | 9/14 [01:18<00:53, 10.68s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   0%|          | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   8%| | 23/306 [00:00<00:01, 227.15it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  15%|▏| 46/306 [00:00<00:01, 228.72it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  23%|▏| 69/306 [00:00<00:01, 217.68it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  30%|▎| 91/306 [00:00<00:01, 196.12it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  40%|▍| 122/306 [00:00<00:00, 227.44it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  48%|▍| 146/306 [00:00<00:00, 191.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  57%|▌| 173/306 [00:00<00:00, 210.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  66%|▋| 203/306 [00:00<00:00, 232.45it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  75%|▊| 230/306 [00:01<00:00, 242.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  83%|▊| 255/306 [00:01<00:00, 232.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  92%|▉| 282/306 [00:01<00:00, 242.38it/\u001b[0m\u001b[A\n",
      " 71%|██████████████████████████████▋            | 10/14 [01:20<00:32,  8.04s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   0%| | 0/306 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   3%| | 9/306 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   6%| | 18/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   8%| | 26/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  11%| | 34/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  14%|▏| 42/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  16%|▏| 50/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  19%|▏| 58/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  22%|▏| 66/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  24%|▏| 73/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  26%|▎| 79/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  28%|▎| 85/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  30%|▎| 91/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  33%|▎| 100/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  36%|▎| 110/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  39%|▍| 120/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  42%|▍| 128/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  44%|▍| 135/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  46%|▍| 142/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  49%|▍| 151/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  52%|▌| 160/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  56%|▌| 170/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  58%|▌| 179/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  62%|▌| 191/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  66%|▋| 201/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  69%|▋| 210/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  72%|▋| 219/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  75%|▋| 228/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  77%|▊| 237/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  81%|▊| 247/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  84%|▊| 256/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  87%|▊| 265/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  89%|▉| 273/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  92%|▉| 282/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  95%|▉| 291/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  98%|▉| 300/306 [\u001b[0m\u001b[A\n",
      " 79%|█████████████████████████████████▊         | 11/14 [01:25<00:21,  7.21s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   0%| | 0/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   2%| | 6/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   4%| | 12/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   6%| | 18/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   8%| | 23/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   9%| | 29/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  11%| | 34/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  13%|▏| 39/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  14%|▏| 44/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  16%|▏| 50/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  18%|▏| 55/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  20%|▏| 60/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  21%|▏| 65/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  23%|▏| 70/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  25%|▏| 76/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  26%|▎| 81/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  28%|▎| 86/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  30%|▎| 91/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  31%|▎| 96/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  34%|▎| 103/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  36%|▎| 109/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  38%|▍| 117/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  40%|▍| 123/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  42%|▍| 129/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  44%|▍| 134/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  45%|▍| 139/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  48%|▍| 147/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  50%|▍| 152/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  52%|▌| 158/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  54%|▌| 164/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  56%|▌| 171/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  58%|▌| 177/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  60%|▌| 183/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  62%|▌| 191/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  65%|▋| 198/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  67%|▋| 204/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  69%|▋| 210/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  71%|▋| 216/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  73%|▋| 222/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  74%|▋| 227/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  76%|▊| 233/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  78%|▊| 240/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  81%|▊| 247/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  83%|▊| 253/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  85%|▊| 259/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  87%|▊| 265/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  89%|▉| 271/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  91%|▉| 277/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  92%|▉| 283/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  95%|▉| 290/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  97%|▉| 296/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  99%|▉| 303/306 \u001b[0m\u001b[A\n",
      " 86%|████████████████████████████████████▊      | 12/14 [01:33<00:14,  7.33s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   0%| | 0/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   2%| | 5/306 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   3%| | 10/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   5%| | 15/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   6%| | 19/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   8%| | 23/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   9%| | 28/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  10%| | 32/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  12%| | 36/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  13%|▏| 41/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  15%|▏| 45/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  16%|▏| 50/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  18%|▏| 55/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  20%|▏| 60/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  21%|▏| 65/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  23%|▏| 69/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  24%|▏| 74/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  25%|▎| 78/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  27%|▎| 82/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  28%|▎| 86/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  29%|▎| 90/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  31%|▎| 94/306 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  33%|▎| 100/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  34%|▎| 105/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  36%|▎| 110/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  38%|▍| 117/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  40%|▍| 122/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  42%|▍| 127/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  43%|▍| 131/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  44%|▍| 135/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  45%|▍| 139/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  48%|▍| 146/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  49%|▍| 151/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  51%|▌| 156/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  53%|▌| 162/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  55%|▌| 168/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  57%|▌| 173/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  58%|▌| 178/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  60%|▌| 183/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  62%|▌| 189/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  63%|▋| 194/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  66%|▋| 201/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  67%|▋| 206/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  69%|▋| 211/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  71%|▋| 216/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  72%|▋| 221/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  74%|▋| 226/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  75%|▊| 231/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  77%|▊| 236/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  79%|▊| 242/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  81%|▊| 247/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  82%|▊| 252/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  84%|▊| 257/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  86%|▊| 262/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  87%|▊| 267/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  89%|▉| 272/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  91%|▉| 277/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  92%|▉| 282/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  94%|▉| 288/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  96%|▉| 293/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  97%|▉| 298/306 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  99%|▉| 303/306 \u001b[0m\u001b[A\n",
      " 93%|███████████████████████████████████████▉   | 13/14 [01:41<00:07,  7.62s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   0%|         | 0/306 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   3%| | 10/306 [00:00<00:03, 88.61it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   6%| | 19/306 [00:00<00:03, 75.99it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   9%| | 27/306 [00:00<00:03, 73.71it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  11%| | 35/306 [00:00<00:03, 71.97it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  14%|▏| 43/306 [00:00<00:03, 72.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  17%|▏| 53/306 [00:00<00:03, 77.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  20%|▏| 61/306 [00:01<00:10, 22.34it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  22%|▏| 68/306 [00:01<00:08, 27.07it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  25%|▏| 76/306 [00:01<00:06, 33.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  27%|▎| 83/306 [00:01<00:05, 37.28it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  29%|▎| 89/306 [00:02<00:05, 38.57it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  31%|▎| 96/306 [00:02<00:04, 44.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  35%|▎| 107/306 [00:02<00:03, 57.49it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  39%|▍| 118/306 [00:02<00:02, 67.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  41%|▍| 126/306 [00:02<00:02, 61.52it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  44%|▍| 134/306 [00:02<00:03, 54.89it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  46%|▍| 141/306 [00:02<00:02, 55.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  49%|▍| 151/306 [00:03<00:02, 64.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  52%|▌| 159/306 [00:03<00:02, 67.70it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  55%|▌| 169/306 [00:03<00:01, 75.32it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  58%|▌| 177/306 [00:03<00:01, 73.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  61%|▌| 186/306 [00:03<00:01, 77.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  65%|▋| 198/306 [00:03<00:01, 87.58it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  68%|▋| 208/306 [00:03<00:01, 87.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  71%|▋| 218/306 [00:03<00:00, 88.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  74%|▋| 227/306 [00:03<00:00, 88.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  77%|▊| 237/306 [00:03<00:00, 90.78it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  81%|▊| 248/306 [00:04<00:00, 94.60it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  84%|▊| 258/306 [00:04<00:00, 80.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  88%|▉| 269/306 [00:04<00:00, 85.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  91%|▉| 278/306 [00:04<00:00, 83.91it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  94%|▉| 289/306 [00:04<00:00, 89.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  98%|▉| 299/306 [00:04<00:00, 88.94it/\u001b[0m\u001b[A\n",
      "100%|███████████████████████████████████████████| 14/14 [02:15<00:00,  9.69s/it]\u001b[A\n",
      "2025-07-11 00:23:27,363 GSEA         INFO     Subsetting TF2G adjacencies for TF with motif.\n",
      "2025-07-11 00:23:27,476 GSEA         INFO     Running GSEA...\n",
      "2025-07-11 00:23:27,476 GSEA         INFO     Generating rankings...\n",
      "100%|████████████████████████████████████████| 306/306 [00:01<00:00, 218.74it/s]\n",
      "100%|███████████████████████████████████████████| 8/8 [00:00<00:00, 2143.37it/s]\n",
      "Running for Positive TF to gene:   1%|        | 40/3923 [00:08<13:26,  4.82it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   2%|        | 60/3923 [00:23<28:41,  2.24it/s]\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   3%|▏      | 120/3923 [00:28<10:16,  6.17it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene: 100%|██████| 3923/3923 [01:04<00:00, 61.04it/s]\n",
      "Running for Negative TF to gene: 100%|███| 3923/3923 [00:00<00:00, 31981.18it/s]\n",
      "2025-07-11 00:24:34,691 GSEA         INFO     Subsetting on adjusted pvalue: 1, minimal NES: 0 and minimal leading edge genes 10\n",
      "2025-07-11 00:24:34,793 GSEA         INFO     Merging eRegulons\n",
      "2025-07-11 00:24:35,840 SCENIC+      INFO     Formatting eGRN as table.\n",
      "2025-07-11 00:24:36,604 SCENIC+      INFO     Calculating triplet ranking.\n",
      "2025-07-11 00:24:36,762 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 00:24:48,343 SCENIC+      INFO     Saving network to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 00:24:52 2025]\u001b[0m\n",
      "\u001b[32mFinished job 13.\u001b[0m\n",
      "\u001b[32m9 of 12 steps (75%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:24:52 2025]\u001b[0m\n",
      "\u001b[32mlocalrule AUCell_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\n",
      "    jobid: 12\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 00:25:01,315 SCENIC+      INFO     Reading data.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 00:25:04,692 SCENIC+      INFO     Calculating enrichment scores.\n",
      "2025-07-11 00:26:53,028 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 00:26:54 2025]\u001b[0m\n",
      "\u001b[32mFinished job 12.\u001b[0m\n",
      "\u001b[32m10 of 12 steps (83%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:26:54 2025]\u001b[0m\n",
      "\u001b[32mlocalrule scplus_mudata:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 1\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 00:27:02,601 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 00:27:05,598 SCENIC+      INFO     Reading AUC values.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 00:27:05,715 SCENIC+      INFO     Reading eRegulon metadata.\n",
      "2025-07-11 00:27:05,806 SCENIC+      INFO     Generating MuData object.\n",
      "2025-07-11 00:27:06,623 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 00:27:13 2025]\u001b[0m\n",
      "\u001b[32mFinished job 1.\u001b[0m\n",
      "\u001b[32m11 of 12 steps (92%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:27:13 2025]\u001b[0m\n",
      "\u001b[32mlocalrule all:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 00:27:13 2025]\u001b[0m\n",
      "\u001b[32mFinished job 0.\u001b[0m\n",
      "\u001b[32m12 of 12 steps (100%) done\u001b[0m\n",
      "\u001b[33mComplete log: .snakemake/log/2025-07-10T224927.855785.snakemake.log\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! snakemake --cores 20"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "a15c4fc1-ee03-45ce-9cf2-9720a5506716",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 20\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                           count\n",
      "--------------------------  -------\n",
      "AUCell_direct                     1\n",
      "AUCell_extended                   1\n",
      "all                               1\n",
      "eGRN_direct                       1\n",
      "eGRN_extended                     1\n",
      "get_search_space                  1\n",
      "motif_enrichment_cistarget        1\n",
      "motif_enrichment_dem              1\n",
      "prepare_menr                      1\n",
      "region_to_gene                    1\n",
      "scplus_mudata                     1\n",
      "tf_to_gene                        1\n",
      "total                            12\n",
      "\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Jul 10 22:49:28 2025]\n",
      "localrule get_search_space:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/chromsizes.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    jobid: 11\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Jul 10 22:52:42 2025]\n",
      "Finished job 11.\n",
      "1 of 12 steps (8%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Jul 10 22:52:42 2025]\n",
      "localrule region_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    jobid: 10\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/search_space.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Jul 10 23:13:44 2025]\n",
      "Finished job 10.\n",
      "2 of 12 steps (17%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Jul 10 23:13:44 2025]\n",
      "localrule motif_enrichment_cistarget:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.html\n",
      "    jobid: 9\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Jul 10 23:16:34 2025]\n",
      "Finished job 9.\n",
      "3 of 12 steps (25%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Jul 10 23:16:34 2025]\n",
      "localrule motif_enrichment_dem:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.html\n",
      "    jobid: 7\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Jul 10 23:19:07 2025]\n",
      "Finished job 7.\n",
      "4 of 12 steps (33%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Jul 10 23:19:07 2025]\n",
      "localrule prepare_menr:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad\n",
      "    jobid: 6\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/dem_results.hdf5\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Thu Jul 10 23:19:50 2025]\n",
      "Finished job 6.\n",
      "5 of 12 steps (42%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Thu Jul 10 23:19:50 2025]\n",
      "localrule tf_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv\n",
      "    jobid: 5\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_names.txt\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:13:25 2025]\n",
      "Finished job 5.\n",
      "6 of 12 steps (50%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 00:13:25 2025]\n",
      "localrule eGRN_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    jobid: 4\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:18:56 2025]\n",
      "Finished job 4.\n",
      "7 of 12 steps (58%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 00:18:56 2025]\n",
      "localrule AUCell_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu\n",
      "    jobid: 3\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:20:59 2025]\n",
      "Finished job 3.\n",
      "8 of 12 steps (67%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 00:20:59 2025]\n",
      "localrule eGRN_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    jobid: 13\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/region_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:24:52 2025]\n",
      "Finished job 13.\n",
      "9 of 12 steps (75%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 00:24:52 2025]\n",
      "localrule AUCell_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu\n",
      "    jobid: 12\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:26:54 2025]\n",
      "Finished job 12.\n",
      "10 of 12 steps (83%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 00:26:54 2025]\n",
      "localrule scplus_mudata:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 1\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/eRegulon_direct.tsv\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:27:13 2025]\n",
      "Finished job 1.\n",
      "11 of 12 steps (92%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 00:27:13 2025]\n",
      "localrule all:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/outs/scplusmdata.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 00:27:13 2025]\n",
      "Finished job 0.\n",
      "12 of 12 steps (100%) done\n",
      "Complete log: .snakemake/log/2025-07-10T224927.855785.snakemake.log\n"
     ]
    }
   ],
   "source": [
    "! cat .snakemake/log/2025-07-10T224927.855785.snakemake.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "74820415-4056-413c-b327-811a1a02679e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Package                              Version\n",
      "------------------------------------ -----------------\n",
      "adjustText                           1.0.4\n",
      "aiohttp                              3.9.3\n",
      "aiosignal                            1.3.1\n",
      "anndata                              0.10.5.post1\n",
      "annoy                                1.17.3\n",
      "appdirs                              1.4.4\n",
      "arboreto                             0.1.6\n",
      "argparse-dataclass                   2.0.0\n",
      "array_api_compat                     1.5.1\n",
      "asttokens                            2.4.1\n",
      "attr                                 0.3.2\n",
      "attrs                                23.2.0\n",
      "bbknn                                1.6.0\n",
      "beautifulsoup4                       4.12.3\n",
      "bidict                               0.23.1\n",
      "bioservices                          1.11.2\n",
      "blosc2                               2.5.1\n",
      "bokeh                                3.4.0\n",
      "boltons                              23.1.1\n",
      "bs4                                  0.0.2\n",
      "cattrs                               23.2.3\n",
      "certifi                              2024.2.2\n",
      "charset-normalizer                   3.3.2\n",
      "click                                8.1.7\n",
      "cloudpickle                          3.0.0\n",
      "colorama                             0.4.6\n",
      "colorlog                             6.8.2\n",
      "comm                                 0.2.2\n",
      "conda-inject                         1.3.1\n",
      "ConfigArgParse                       1.7\n",
      "connection_pool                      0.0.3\n",
      "contourpy                            1.2.0\n",
      "ctxcore                              0.2.0\n",
      "cycler                               0.12.1\n",
      "Cython                               0.29.37\n",
      "cytoolz                              0.12.3\n",
      "dask                                 2024.2.1\n",
      "dataclasses-json                     0.6.4\n",
      "datrie                               0.8.2\n",
      "debugpy                              1.8.1\n",
      "decorator                            5.1.1\n",
      "dill                                 0.3.8\n",
      "distributed                          2024.2.1\n",
      "docutils                             0.20.1\n",
      "dpath                                2.1.6\n",
      "easydev                              0.13.1\n",
      "et-xmlfile                           1.1.0\n",
      "executing                            2.0.1\n",
      "fastjsonschema                       2.19.1\n",
      "fbpca                                1.0\n",
      "filelock                             3.13.1\n",
      "fonttools                            4.50.0\n",
      "frozendict                           2.4.0\n",
      "frozenlist                           1.4.1\n",
      "fsspec                               2024.3.1\n",
      "future                               1.0.0\n",
      "gensim                               4.3.2\n",
      "geosketch                            1.2\n",
      "gevent                               24.2.1\n",
      "gitdb                                4.0.11\n",
      "GitPython                            3.1.42\n",
      "globre                               0.1.5\n",
      "greenlet                             3.0.3\n",
      "grequests                            0.7.0\n",
      "gseapy                               0.10.8\n",
      "h5py                                 3.10.0\n",
      "harmonypy                            0.0.9\n",
      "humanfriendly                        10.0\n",
      "idna                                 3.6\n",
      "igraph                               0.11.4\n",
      "imageio                              2.34.0\n",
      "immutables                           0.20\n",
      "importlib-metadata                   7.0.1\n",
      "importlib_resources                  6.1.2\n",
      "interlap                             0.2.7\n",
      "intervaltree                         3.1.0\n",
      "ipykernel                            6.29.3\n",
      "ipython                              8.22.2\n",
      "jedi                                 0.19.1\n",
      "Jinja2                               3.1.3\n",
      "joblib                               1.3.2\n",
      "jsonpickle                           3.0.3\n",
      "jsonschema                           4.21.1\n",
      "jsonschema-specifications            2023.12.1\n",
      "jupyter_client                       8.6.1\n",
      "jupyter_core                         5.7.2\n",
      "kaleido                              0.2.1\n",
      "kiwisolver                           1.4.5\n",
      "lazy_loader                          0.3\n",
      "lda                                  3.0.0\n",
      "leidenalg                            0.10.2\n",
      "line-profiler                        4.1.2\n",
      "llvmlite                             0.42.0\n",
      "locket                               1.0.0\n",
      "loompy                               3.0.7\n",
      "loomxpy                              0.4.2\n",
      "lxml                                 5.1.0\n",
      "lz4                                  4.3.3\n",
      "MACS2                                2.2.9.1\n",
      "markdown-it-py                       3.0.0\n",
      "MarkupSafe                           2.1.5\n",
      "marshmallow                          3.21.1\n",
      "matplotlib                           3.6.3\n",
      "matplotlib-inline                    0.1.6\n",
      "mdurl                                0.1.2\n",
      "mizani                               0.9.3\n",
      "msgpack                              1.0.8\n",
      "mudata                               0.2.3\n",
      "multidict                            6.0.5\n",
      "multiprocessing_on_dill              3.5.0a4\n",
      "mypy-extensions                      1.0.0\n",
      "natsort                              8.4.0\n",
      "nbformat                             5.10.3\n",
      "ncls                                 0.0.68\n",
      "ndindex                              1.8\n",
      "nest-asyncio                         1.6.0\n",
      "networkx                             3.2.1\n",
      "numba                                0.59.0\n",
      "numexpr                              2.9.0\n",
      "numpy                                1.26.4\n",
      "numpy-groupies                       0.10.2\n",
      "openpyxl                             3.1.2\n",
      "packaging                            24.0\n",
      "pandas                               1.5.0\n",
      "parso                                0.8.3\n",
      "partd                                1.4.1\n",
      "patsy                                0.5.6\n",
      "pexpect                              4.9.0\n",
      "pillow                               10.2.0\n",
      "pip                                  24.0\n",
      "plac                                 1.4.3\n",
      "platformdirs                         4.2.0\n",
      "plotly                               5.19.0\n",
      "plotnine                             0.12.4\n",
      "polars                               0.20.13\n",
      "progressbar2                         4.4.2\n",
      "prompt-toolkit                       3.0.43\n",
      "protobuf                             5.26.0\n",
      "psutil                               5.9.8\n",
      "ptyprocess                           0.7.0\n",
      "PuLP                                 2.8.0\n",
      "pure-eval                            0.2.2\n",
      "py-cpuinfo                           9.0.0\n",
      "pyarrow                              15.0.0\n",
      "pyarrow-hotfix                       0.6\n",
      "pybedtools                           0.9.1\n",
      "pybigtools                           0.1.2\n",
      "pyBigWig                             0.3.22\n",
      "pybiomart                            0.2.0\n",
      "pycistarget                          1.0a2\n",
      "pycisTopic                           2.0a0\n",
      "pyfasta                              0.5.2\n",
      "pygam                                0.9.0\n",
      "Pygments                             2.17.2\n",
      "pynndescent                          0.5.11\n",
      "pyparsing                            3.1.2\n",
      "pyranges                             0.0.111\n",
      "pyrle                                0.0.39\n",
      "pysam                                0.22.0\n",
      "pyscenic                             0.12.1+7.g1cd059f\n",
      "python-dateutil                      2.9.0.post0\n",
      "python-utils                         3.8.2\n",
      "pytz                                 2024.1\n",
      "pyvis                                0.3.2\n",
      "PyYAML                               6.0.1\n",
      "pyzmq                                25.1.2\n",
      "ray                                  2.9.3\n",
      "referencing                          0.34.0\n",
      "requests                             2.31.0\n",
      "requests-cache                       1.2.0\n",
      "reretry                              0.11.8\n",
      "rich                                 13.7.1\n",
      "rich-argparse                        1.4.0\n",
      "rpds-py                              0.18.0\n",
      "scanorama                            1.7.4\n",
      "scanpy                               1.8.2\n",
      "scatac_fragment_tools                0.1.0\n",
      "scenicplus                           1.0a1\n",
      "scikit-image                         0.22.0\n",
      "scikit-learn                         1.3.2\n",
      "scipy                                1.12.0\n",
      "scrublet                             0.2.3\n",
      "seaborn                              0.13.2\n",
      "setuptools                           65.5.0\n",
      "sinfo                                0.3.4\n",
      "six                                  1.16.0\n",
      "smart-open                           6.4.0\n",
      "smmap                                5.0.1\n",
      "snakemake                            8.5.5\n",
      "snakemake-interface-common           1.17.1\n",
      "snakemake-interface-executor-plugins 8.2.0\n",
      "snakemake-interface-report-plugins   1.0.0\n",
      "snakemake-interface-storage-plugins  3.1.1\n",
      "sorted-nearest                       0.0.39\n",
      "sortedcontainers                     2.4.0\n",
      "soupsieve                            2.5\n",
      "stack-data                           0.6.3\n",
      "statistics                           1.0.3.5\n",
      "statsmodels                          0.14.1\n",
      "stdlib-list                          0.10.0\n",
      "stopit                               1.1.2\n",
      "suds-community                       1.1.2\n",
      "tables                               3.9.2\n",
      "tabulate                             0.9.0\n",
      "tblib                                3.0.0\n",
      "tenacity                             8.2.3\n",
      "texttable                            1.7.0\n",
      "threadpoolctl                        3.4.0\n",
      "throttler                            1.2.2\n",
      "tifffile                             2024.2.12\n",
      "tmtoolkit                            0.12.0\n",
      "toolz                                0.12.1\n",
      "toposort                             1.10\n",
      "tornado                              6.4\n",
      "tqdm                                 4.66.2\n",
      "traitlets                            5.14.2\n",
      "tspex                                0.6.3\n",
      "typing                               3.7.4.3\n",
      "typing_extensions                    4.10.0\n",
      "typing-inspect                       0.9.0\n",
      "umap-learn                           0.5.5\n",
      "url-normalize                        1.4.3\n",
      "urllib3                              2.2.1\n",
      "wcwidth                              0.2.13\n",
      "wheel                                0.43.0\n",
      "wrapt                                1.16.0\n",
      "xlrd                                 2.0.1\n",
      "xmltodict                            0.13.0\n",
      "xyzservices                          2023.10.1\n",
      "yarl                                 1.9.4\n",
      "yte                                  1.5.4\n",
      "zict                                 3.0.0\n",
      "zipp                                 3.18.1\n",
      "zope.event                           5.0\n",
      "zope.interface                       6.2\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.1.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n",
      "Note: you may need to restart the kernel to use updated packages.\n"
     ]
    }
   ],
   "source": [
    "pip list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0d33d6f-2836-47ed-bbfe-f1d51a3f7402",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pycistopic_qc_may24",
   "language": "python",
   "name": "pycistopic_2a"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
