{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "79b46d6a-5972-4afa-a21d-a2195611500d",
   "metadata": {},
   "source": [
    "# SCENIC+"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "093a0b3e-fae8-49a1-b80b-16ea0e87fef5",
   "metadata": {},
   "source": [
    "### save rna counts in adata.raw and change count matrix barcode to ATAC one"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "e44f8610-986b-46d2-a7e9-4800c71e2a06",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fe07a7c-f455-4ef4-91a5-cbb81d612ed0",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna = sc.read_h5ad(\"/multiome_adata_04072025.h5ad\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "397d69ed-b4db-417d-a448-34eb9ab227b5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Layers with keys: counts, log_counts"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.layers"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "1d6a2b44-a39b-4991-a815-8182c30f671c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>dna_total_counts</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>log1p_n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>log1p_total_counts</th>\n",
       "      <th>pct_counts_in_top_20_genes</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "      <th>pct_counts_ribo</th>\n",
       "      <th>pct_counts_hb</th>\n",
       "      <th>...</th>\n",
       "      <th>leiden_1.25</th>\n",
       "      <th>leiden_1.5</th>\n",
       "      <th>leiden_2.0</th>\n",
       "      <th>leiden_5.0</th>\n",
       "      <th>leiden_10.0</th>\n",
       "      <th>leiden_20.0</th>\n",
       "      <th>celltypist_cell_label_coarse</th>\n",
       "      <th>celltypist_conf_score_coarse</th>\n",
       "      <th>celltypist_cell_label_fine</th>\n",
       "      <th>celltypist_conf_score_fine</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AAACAGCCAGTTTGTG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>8056.0</td>\n",
       "      <td>1891</td>\n",
       "      <td>7.545390</td>\n",
       "      <td>7641.0</td>\n",
       "      <td>8.941415</td>\n",
       "      <td>21.823060</td>\n",
       "      <td>13.584609</td>\n",
       "      <td>7.642979</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>37</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.978114</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.950723</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACAGCCATGTTTGG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>23901.0</td>\n",
       "      <td>2319</td>\n",
       "      <td>7.749322</td>\n",
       "      <td>10986.0</td>\n",
       "      <td>9.304468</td>\n",
       "      <td>21.354451</td>\n",
       "      <td>13.899508</td>\n",
       "      <td>9.111597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>63</td>\n",
       "      <td>36</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.603815</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.370262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACAGCCATTTAAGC-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>31317.0</td>\n",
       "      <td>1667</td>\n",
       "      <td>7.419381</td>\n",
       "      <td>6636.0</td>\n",
       "      <td>8.800415</td>\n",
       "      <td>21.503918</td>\n",
       "      <td>13.803495</td>\n",
       "      <td>6.268836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>134</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.951335</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.903238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACATGCATTGGGAG-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1039.0</td>\n",
       "      <td>1599</td>\n",
       "      <td>7.377759</td>\n",
       "      <td>6888.0</td>\n",
       "      <td>8.837681</td>\n",
       "      <td>24.811266</td>\n",
       "      <td>18.626598</td>\n",
       "      <td>5.894309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>81</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.121671</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.183371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAACCAACAATAACCT-D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>25718.0</td>\n",
       "      <td>1403</td>\n",
       "      <td>7.247081</td>\n",
       "      <td>5495.0</td>\n",
       "      <td>8.611775</td>\n",
       "      <td>24.913558</td>\n",
       "      <td>18.817106</td>\n",
       "      <td>5.150136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.991872</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.996793</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGAAGCATGAAG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>13995.0</td>\n",
       "      <td>2061</td>\n",
       "      <td>7.631432</td>\n",
       "      <td>8523.0</td>\n",
       "      <td>9.050641</td>\n",
       "      <td>21.858501</td>\n",
       "      <td>14.009152</td>\n",
       "      <td>6.112871</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>38</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.930010</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.984357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGAAGCGAGTAA-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>34490.0</td>\n",
       "      <td>1361</td>\n",
       "      <td>7.216709</td>\n",
       "      <td>5059.0</td>\n",
       "      <td>8.529121</td>\n",
       "      <td>23.739870</td>\n",
       "      <td>17.019173</td>\n",
       "      <td>8.381103</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.403587</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.973642</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTGGCAGCCTGCA-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>1536</td>\n",
       "      <td>7.337588</td>\n",
       "      <td>6272.0</td>\n",
       "      <td>8.744010</td>\n",
       "      <td>29.432398</td>\n",
       "      <td>23.852041</td>\n",
       "      <td>6.505102</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>58</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.177973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTAAACAAG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>9679.0</td>\n",
       "      <td>1279</td>\n",
       "      <td>7.154615</td>\n",
       "      <td>5301.0</td>\n",
       "      <td>8.575839</td>\n",
       "      <td>35.653650</td>\n",
       "      <td>28.730429</td>\n",
       "      <td>2.886248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>45</td>\n",
       "      <td>50</td>\n",
       "      <td>31</td>\n",
       "      <td>Monocytes</td>\n",
       "      <td>0.745200</td>\n",
       "      <td>Non-classical monocytes</td>\n",
       "      <td>0.041574</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TTTGTTGGTTAAGCTG-Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1756</td>\n",
       "      <td>7.471363</td>\n",
       "      <td>7045.0</td>\n",
       "      <td>8.860215</td>\n",
       "      <td>24.442867</td>\n",
       "      <td>18.424414</td>\n",
       "      <td>6.018453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>104</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.122617</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.947241</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10106 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample  dna_total_counts  n_genes_by_counts  \\\n",
       "AAACAGCCAGTTTGTG-D0     D0            8056.0               1891   \n",
       "AAACAGCCATGTTTGG-D0     D0           23901.0               2319   \n",
       "AAACAGCCATTTAAGC-D0     D0           31317.0               1667   \n",
       "AAACATGCATTGGGAG-D0     D0            1039.0               1599   \n",
       "AAACCAACAATAACCT-D0     D0           25718.0               1403   \n",
       "...                    ...               ...                ...   \n",
       "TTTGTGAAGCATGAAG-Q3     Q3           13995.0               2061   \n",
       "TTTGTGAAGCGAGTAA-Q3     Q3           34490.0               1361   \n",
       "TTTGTGGCAGCCTGCA-Q3     Q3            3673.0               1536   \n",
       "TTTGTTGGTAAACAAG-Q3     Q3            9679.0               1279   \n",
       "TTTGTTGGTTAAGCTG-Q3     Q3               0.0               1756   \n",
       "\n",
       "                     log1p_n_genes_by_counts  total_counts  \\\n",
       "AAACAGCCAGTTTGTG-D0                 7.545390        7641.0   \n",
       "AAACAGCCATGTTTGG-D0                 7.749322       10986.0   \n",
       "AAACAGCCATTTAAGC-D0                 7.419381        6636.0   \n",
       "AAACATGCATTGGGAG-D0                 7.377759        6888.0   \n",
       "AAACCAACAATAACCT-D0                 7.247081        5495.0   \n",
       "...                                      ...           ...   \n",
       "TTTGTGAAGCATGAAG-Q3                 7.631432        8523.0   \n",
       "TTTGTGAAGCGAGTAA-Q3                 7.216709        5059.0   \n",
       "TTTGTGGCAGCCTGCA-Q3                 7.337588        6272.0   \n",
       "TTTGTTGGTAAACAAG-Q3                 7.154615        5301.0   \n",
       "TTTGTTGGTTAAGCTG-Q3                 7.471363        7045.0   \n",
       "\n",
       "                     log1p_total_counts  pct_counts_in_top_20_genes  \\\n",
       "AAACAGCCAGTTTGTG-D0            8.941415                   21.823060   \n",
       "AAACAGCCATGTTTGG-D0            9.304468                   21.354451   \n",
       "AAACAGCCATTTAAGC-D0            8.800415                   21.503918   \n",
       "AAACATGCATTGGGAG-D0            8.837681                   24.811266   \n",
       "AAACCAACAATAACCT-D0            8.611775                   24.913558   \n",
       "...                                 ...                         ...   \n",
       "TTTGTGAAGCATGAAG-Q3            9.050641                   21.858501   \n",
       "TTTGTGAAGCGAGTAA-Q3            8.529121                   23.739870   \n",
       "TTTGTGGCAGCCTGCA-Q3            8.744010                   29.432398   \n",
       "TTTGTTGGTAAACAAG-Q3            8.575839                   35.653650   \n",
       "TTTGTTGGTTAAGCTG-Q3            8.860215                   24.442867   \n",
       "\n",
       "                     pct_counts_mt  pct_counts_ribo  pct_counts_hb  ...  \\\n",
       "AAACAGCCAGTTTGTG-D0      13.584609         7.642979            0.0  ...   \n",
       "AAACAGCCATGTTTGG-D0      13.899508         9.111597            0.0  ...   \n",
       "AAACAGCCATTTAAGC-D0      13.803495         6.268836            0.0  ...   \n",
       "AAACATGCATTGGGAG-D0      18.626598         5.894309            0.0  ...   \n",
       "AAACCAACAATAACCT-D0      18.817106         5.150136            0.0  ...   \n",
       "...                            ...              ...            ...  ...   \n",
       "TTTGTGAAGCATGAAG-Q3      14.009152         6.112871            0.0  ...   \n",
       "TTTGTGAAGCGAGTAA-Q3      17.019173         8.381103            0.0  ...   \n",
       "TTTGTGGCAGCCTGCA-Q3      23.852041         6.505102            0.0  ...   \n",
       "TTTGTTGGTAAACAAG-Q3      28.730429         2.886248            0.0  ...   \n",
       "TTTGTTGGTTAAGCTG-Q3      18.424414         6.018453            0.0  ...   \n",
       "\n",
       "                     leiden_1.25  leiden_1.5  leiden_2.0  leiden_5.0  \\\n",
       "AAACAGCCAGTTTGTG-D0            4           4           4           1   \n",
       "AAACAGCCATGTTTGG-D0            8           8           7          13   \n",
       "AAACAGCCATTTAAGC-D0            4           4           4           1   \n",
       "AAACATGCATTGGGAG-D0            4           4           4          10   \n",
       "AAACCAACAATAACCT-D0            4           4           4          10   \n",
       "...                          ...         ...         ...         ...   \n",
       "TTTGTGAAGCATGAAG-Q3            5           7          18          38   \n",
       "TTTGTGAAGCGAGTAA-Q3           10          11           8          23   \n",
       "TTTGTGGCAGCCTGCA-Q3            1           0           0          16   \n",
       "TTTGTTGGTAAACAAG-Q3            9          10          11          45   \n",
       "TTTGTTGGTTAAGCTG-Q3            4           4           7          18   \n",
       "\n",
       "                    leiden_10.0 leiden_20.0 celltypist_cell_label_coarse  \\\n",
       "AAACAGCCAGTTTGTG-D0           9          37               B-cell lineage   \n",
       "AAACAGCCATGTTTGG-D0          63          36               B-cell lineage   \n",
       "AAACAGCCATTTAAGC-D0          26         134               B-cell lineage   \n",
       "AAACATGCATTGGGAG-D0          19          81               B-cell lineage   \n",
       "AAACCAACAATAACCT-D0           8          46               B-cell lineage   \n",
       "...                         ...         ...                          ...   \n",
       "TTTGTGAAGCATGAAG-Q3          30           9               B-cell lineage   \n",
       "TTTGTGAAGCGAGTAA-Q3          14           5               B-cell lineage   \n",
       "TTTGTGGCAGCCTGCA-Q3          22          58               B-cell lineage   \n",
       "TTTGTTGGTAAACAAG-Q3          50          31                    Monocytes   \n",
       "TTTGTTGGTTAAGCTG-Q3          36         104               B-cell lineage   \n",
       "\n",
       "                    celltypist_conf_score_coarse celltypist_cell_label_fine  \\\n",
       "AAACAGCCAGTTTGTG-D0                     0.978114                Pro-B cells   \n",
       "AAACAGCCATGTTTGG-D0                     0.603815                Pro-B cells   \n",
       "AAACAGCCATTTAAGC-D0                     0.951335                Pro-B cells   \n",
       "AAACATGCATTGGGAG-D0                     0.121671                Pro-B cells   \n",
       "AAACCAACAATAACCT-D0                     0.991872                Pro-B cells   \n",
       "...                                          ...                        ...   \n",
       "TTTGTGAAGCATGAAG-Q3                     0.930010                Pro-B cells   \n",
       "TTTGTGAAGCGAGTAA-Q3                     0.403587                Pro-B cells   \n",
       "TTTGTGGCAGCCTGCA-Q3                     0.484664                Pro-B cells   \n",
       "TTTGTTGGTAAACAAG-Q3                     0.745200    Non-classical monocytes   \n",
       "TTTGTTGGTTAAGCTG-Q3                     0.122617                Pro-B cells   \n",
       "\n",
       "                    celltypist_conf_score_fine  \n",
       "AAACAGCCAGTTTGTG-D0                   0.950723  \n",
       "AAACAGCCATGTTTGG-D0                   0.370262  \n",
       "AAACAGCCATTTAAGC-D0                   0.903238  \n",
       "AAACATGCATTGGGAG-D0                   0.183371  \n",
       "AAACCAACAATAACCT-D0                   0.996793  \n",
       "...                                        ...  \n",
       "TTTGTGAAGCATGAAG-Q3                   0.984357  \n",
       "TTTGTGAAGCGAGTAA-Q3                   0.973642  \n",
       "TTTGTGGCAGCCTGCA-Q3                   0.177973  \n",
       "TTTGTTGGTAAACAAG-Q3                   0.041574  \n",
       "TTTGTTGGTTAAGCTG-Q3                   0.947241  \n",
       "\n",
       "[10106 rows x 34 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14c445c1-27a7-4b84-9407-e751e1a16c40",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "atac_CB = pd.read_csv('/barcodes/cellranger_arc_atac.737K-arc-v1.txt.gz', header = None)\n",
    "rna_CB = pd.read_csv('/barcodes/cellranger_arc_rna.737K-arc-v1.txt.gz', header = None)\n",
    "rna_CB.rename(columns={0: 'rna_CB'}, inplace=True)\n",
    "atac_CB.rename(columns={0: 'atac_CB'}, inplace=True)\n",
    "df_merged =  pd.concat([rna_CB, atac_CB], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6a73088a-f226-4ce5-a76f-f698c66d8b37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>dna_total_counts</th>\n",
       "      <th>n_genes_by_counts</th>\n",
       "      <th>log1p_n_genes_by_counts</th>\n",
       "      <th>total_counts</th>\n",
       "      <th>log1p_total_counts</th>\n",
       "      <th>pct_counts_in_top_20_genes</th>\n",
       "      <th>pct_counts_mt</th>\n",
       "      <th>pct_counts_ribo</th>\n",
       "      <th>pct_counts_hb</th>\n",
       "      <th>...</th>\n",
       "      <th>leiden_1.5</th>\n",
       "      <th>leiden_2.0</th>\n",
       "      <th>leiden_5.0</th>\n",
       "      <th>leiden_10.0</th>\n",
       "      <th>leiden_20.0</th>\n",
       "      <th>celltypist_cell_label_coarse</th>\n",
       "      <th>celltypist_conf_score_coarse</th>\n",
       "      <th>celltypist_cell_label_fine</th>\n",
       "      <th>celltypist_conf_score_fine</th>\n",
       "      <th>base_index</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>atac_CB</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTGTTACTA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>8056.0</td>\n",
       "      <td>1891</td>\n",
       "      <td>7.545390</td>\n",
       "      <td>7641.0</td>\n",
       "      <td>8.941415</td>\n",
       "      <td>21.823060</td>\n",
       "      <td>13.584609</td>\n",
       "      <td>7.642979</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>9</td>\n",
       "      <td>37</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.978114</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.950723</td>\n",
       "      <td>AAACAGCCAGTTTGTG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTTTATCCT-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>23901.0</td>\n",
       "      <td>2319</td>\n",
       "      <td>7.749322</td>\n",
       "      <td>10986.0</td>\n",
       "      <td>9.304468</td>\n",
       "      <td>21.354451</td>\n",
       "      <td>13.899508</td>\n",
       "      <td>9.111597</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>8</td>\n",
       "      <td>7</td>\n",
       "      <td>13</td>\n",
       "      <td>63</td>\n",
       "      <td>36</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.603815</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.370262</td>\n",
       "      <td>AAACAGCCATGTTTGG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ACAGCGGGTTCACTTT-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>31317.0</td>\n",
       "      <td>1667</td>\n",
       "      <td>7.419381</td>\n",
       "      <td>6636.0</td>\n",
       "      <td>8.800415</td>\n",
       "      <td>21.503918</td>\n",
       "      <td>13.803495</td>\n",
       "      <td>6.268836</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>26</td>\n",
       "      <td>134</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.951335</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.903238</td>\n",
       "      <td>AAACAGCCATTTAAGC</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CATTTAGGTGTCCACA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>1039.0</td>\n",
       "      <td>1599</td>\n",
       "      <td>7.377759</td>\n",
       "      <td>6888.0</td>\n",
       "      <td>8.837681</td>\n",
       "      <td>24.811266</td>\n",
       "      <td>18.626598</td>\n",
       "      <td>5.894309</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>19</td>\n",
       "      <td>81</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.121671</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.183371</td>\n",
       "      <td>AAACATGCATTGGGAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CTTTATCGTCGAGGCA-D0___D0</th>\n",
       "      <td>D0</td>\n",
       "      <td>25718.0</td>\n",
       "      <td>1403</td>\n",
       "      <td>7.247081</td>\n",
       "      <td>5495.0</td>\n",
       "      <td>8.611775</td>\n",
       "      <td>24.913558</td>\n",
       "      <td>18.817106</td>\n",
       "      <td>5.150136</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>10</td>\n",
       "      <td>8</td>\n",
       "      <td>46</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.991872</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.996793</td>\n",
       "      <td>AAACCAACAATAACCT</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTAGGTGTCAGTGAAC-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>13995.0</td>\n",
       "      <td>2061</td>\n",
       "      <td>7.631432</td>\n",
       "      <td>8523.0</td>\n",
       "      <td>9.050641</td>\n",
       "      <td>21.858501</td>\n",
       "      <td>14.009152</td>\n",
       "      <td>6.112871</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>38</td>\n",
       "      <td>30</td>\n",
       "      <td>9</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.930010</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.984357</td>\n",
       "      <td>TTTGTGAAGCATGAAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTAGGTGTCGGTCCAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>34490.0</td>\n",
       "      <td>1361</td>\n",
       "      <td>7.216709</td>\n",
       "      <td>5059.0</td>\n",
       "      <td>8.529121</td>\n",
       "      <td>23.739870</td>\n",
       "      <td>17.019173</td>\n",
       "      <td>8.381103</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>8</td>\n",
       "      <td>23</td>\n",
       "      <td>14</td>\n",
       "      <td>5</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.403587</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.973642</td>\n",
       "      <td>TTTGTGAAGCGAGTAA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GTTTGGTGTTTCTAAG-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>3673.0</td>\n",
       "      <td>1536</td>\n",
       "      <td>7.337588</td>\n",
       "      <td>6272.0</td>\n",
       "      <td>8.744010</td>\n",
       "      <td>29.432398</td>\n",
       "      <td>23.852041</td>\n",
       "      <td>6.505102</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>16</td>\n",
       "      <td>22</td>\n",
       "      <td>58</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.484664</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.177973</td>\n",
       "      <td>TTTGTGGCAGCCTGCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTACTTCAAATATGC-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>9679.0</td>\n",
       "      <td>1279</td>\n",
       "      <td>7.154615</td>\n",
       "      <td>5301.0</td>\n",
       "      <td>8.575839</td>\n",
       "      <td>35.653650</td>\n",
       "      <td>28.730429</td>\n",
       "      <td>2.886248</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>11</td>\n",
       "      <td>45</td>\n",
       "      <td>50</td>\n",
       "      <td>31</td>\n",
       "      <td>Monocytes</td>\n",
       "      <td>0.745200</td>\n",
       "      <td>Non-classical monocytes</td>\n",
       "      <td>0.041574</td>\n",
       "      <td>TTTGTTGGTAAACAAG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CGTACTTCATGATGAT-Q3___Q3</th>\n",
       "      <td>Q3</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1756</td>\n",
       "      <td>7.471363</td>\n",
       "      <td>7045.0</td>\n",
       "      <td>8.860215</td>\n",
       "      <td>24.442867</td>\n",
       "      <td>18.424414</td>\n",
       "      <td>6.018453</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4</td>\n",
       "      <td>7</td>\n",
       "      <td>18</td>\n",
       "      <td>36</td>\n",
       "      <td>104</td>\n",
       "      <td>B-cell lineage</td>\n",
       "      <td>0.122617</td>\n",
       "      <td>Pro-B cells</td>\n",
       "      <td>0.947241</td>\n",
       "      <td>TTTGTTGGTTAAGCTG</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10106 rows × 35 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                         sample  dna_total_counts  n_genes_by_counts  \\\n",
       "atac_CB                                                                \n",
       "ACAGCGGGTGTTACTA-D0___D0     D0            8056.0               1891   \n",
       "ACAGCGGGTTTATCCT-D0___D0     D0           23901.0               2319   \n",
       "ACAGCGGGTTCACTTT-D0___D0     D0           31317.0               1667   \n",
       "CATTTAGGTGTCCACA-D0___D0     D0            1039.0               1599   \n",
       "CTTTATCGTCGAGGCA-D0___D0     D0           25718.0               1403   \n",
       "...                         ...               ...                ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3     Q3           13995.0               2061   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3     Q3           34490.0               1361   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3     Q3            3673.0               1536   \n",
       "CGTACTTCAAATATGC-Q3___Q3     Q3            9679.0               1279   \n",
       "CGTACTTCATGATGAT-Q3___Q3     Q3               0.0               1756   \n",
       "\n",
       "                          log1p_n_genes_by_counts  total_counts  \\\n",
       "atac_CB                                                           \n",
       "ACAGCGGGTGTTACTA-D0___D0                 7.545390        7641.0   \n",
       "ACAGCGGGTTTATCCT-D0___D0                 7.749322       10986.0   \n",
       "ACAGCGGGTTCACTTT-D0___D0                 7.419381        6636.0   \n",
       "CATTTAGGTGTCCACA-D0___D0                 7.377759        6888.0   \n",
       "CTTTATCGTCGAGGCA-D0___D0                 7.247081        5495.0   \n",
       "...                                           ...           ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                 7.631432        8523.0   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                 7.216709        5059.0   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                 7.337588        6272.0   \n",
       "CGTACTTCAAATATGC-Q3___Q3                 7.154615        5301.0   \n",
       "CGTACTTCATGATGAT-Q3___Q3                 7.471363        7045.0   \n",
       "\n",
       "                          log1p_total_counts  pct_counts_in_top_20_genes  \\\n",
       "atac_CB                                                                    \n",
       "ACAGCGGGTGTTACTA-D0___D0            8.941415                   21.823060   \n",
       "ACAGCGGGTTTATCCT-D0___D0            9.304468                   21.354451   \n",
       "ACAGCGGGTTCACTTT-D0___D0            8.800415                   21.503918   \n",
       "CATTTAGGTGTCCACA-D0___D0            8.837681                   24.811266   \n",
       "CTTTATCGTCGAGGCA-D0___D0            8.611775                   24.913558   \n",
       "...                                      ...                         ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3            9.050641                   21.858501   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3            8.529121                   23.739870   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3            8.744010                   29.432398   \n",
       "CGTACTTCAAATATGC-Q3___Q3            8.575839                   35.653650   \n",
       "CGTACTTCATGATGAT-Q3___Q3            8.860215                   24.442867   \n",
       "\n",
       "                          pct_counts_mt  pct_counts_ribo  pct_counts_hb  ...  \\\n",
       "atac_CB                                                                  ...   \n",
       "ACAGCGGGTGTTACTA-D0___D0      13.584609         7.642979            0.0  ...   \n",
       "ACAGCGGGTTTATCCT-D0___D0      13.899508         9.111597            0.0  ...   \n",
       "ACAGCGGGTTCACTTT-D0___D0      13.803495         6.268836            0.0  ...   \n",
       "CATTTAGGTGTCCACA-D0___D0      18.626598         5.894309            0.0  ...   \n",
       "CTTTATCGTCGAGGCA-D0___D0      18.817106         5.150136            0.0  ...   \n",
       "...                                 ...              ...            ...  ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3      14.009152         6.112871            0.0  ...   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3      17.019173         8.381103            0.0  ...   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3      23.852041         6.505102            0.0  ...   \n",
       "CGTACTTCAAATATGC-Q3___Q3      28.730429         2.886248            0.0  ...   \n",
       "CGTACTTCATGATGAT-Q3___Q3      18.424414         6.018453            0.0  ...   \n",
       "\n",
       "                          leiden_1.5  leiden_2.0  leiden_5.0  leiden_10.0  \\\n",
       "atac_CB                                                                     \n",
       "ACAGCGGGTGTTACTA-D0___D0           4           4           1            9   \n",
       "ACAGCGGGTTTATCCT-D0___D0           8           7          13           63   \n",
       "ACAGCGGGTTCACTTT-D0___D0           4           4           1           26   \n",
       "CATTTAGGTGTCCACA-D0___D0           4           4          10           19   \n",
       "CTTTATCGTCGAGGCA-D0___D0           4           4          10            8   \n",
       "...                              ...         ...         ...          ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3           7          18          38           30   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3          11           8          23           14   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3           0           0          16           22   \n",
       "CGTACTTCAAATATGC-Q3___Q3          10          11          45           50   \n",
       "CGTACTTCATGATGAT-Q3___Q3           4           7          18           36   \n",
       "\n",
       "                         leiden_20.0 celltypist_cell_label_coarse  \\\n",
       "atac_CB                                                             \n",
       "ACAGCGGGTGTTACTA-D0___D0          37               B-cell lineage   \n",
       "ACAGCGGGTTTATCCT-D0___D0          36               B-cell lineage   \n",
       "ACAGCGGGTTCACTTT-D0___D0         134               B-cell lineage   \n",
       "CATTTAGGTGTCCACA-D0___D0          81               B-cell lineage   \n",
       "CTTTATCGTCGAGGCA-D0___D0          46               B-cell lineage   \n",
       "...                              ...                          ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3           9               B-cell lineage   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3           5               B-cell lineage   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3          58               B-cell lineage   \n",
       "CGTACTTCAAATATGC-Q3___Q3          31                    Monocytes   \n",
       "CGTACTTCATGATGAT-Q3___Q3         104               B-cell lineage   \n",
       "\n",
       "                         celltypist_conf_score_coarse  \\\n",
       "atac_CB                                                 \n",
       "ACAGCGGGTGTTACTA-D0___D0                     0.978114   \n",
       "ACAGCGGGTTTATCCT-D0___D0                     0.603815   \n",
       "ACAGCGGGTTCACTTT-D0___D0                     0.951335   \n",
       "CATTTAGGTGTCCACA-D0___D0                     0.121671   \n",
       "CTTTATCGTCGAGGCA-D0___D0                     0.991872   \n",
       "...                                               ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                     0.930010   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                     0.403587   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                     0.484664   \n",
       "CGTACTTCAAATATGC-Q3___Q3                     0.745200   \n",
       "CGTACTTCATGATGAT-Q3___Q3                     0.122617   \n",
       "\n",
       "                         celltypist_cell_label_fine  \\\n",
       "atac_CB                                               \n",
       "ACAGCGGGTGTTACTA-D0___D0                Pro-B cells   \n",
       "ACAGCGGGTTTATCCT-D0___D0                Pro-B cells   \n",
       "ACAGCGGGTTCACTTT-D0___D0                Pro-B cells   \n",
       "CATTTAGGTGTCCACA-D0___D0                Pro-B cells   \n",
       "CTTTATCGTCGAGGCA-D0___D0                Pro-B cells   \n",
       "...                                             ...   \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                Pro-B cells   \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                Pro-B cells   \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                Pro-B cells   \n",
       "CGTACTTCAAATATGC-Q3___Q3    Non-classical monocytes   \n",
       "CGTACTTCATGATGAT-Q3___Q3                Pro-B cells   \n",
       "\n",
       "                         celltypist_conf_score_fine        base_index  \n",
       "atac_CB                                                                \n",
       "ACAGCGGGTGTTACTA-D0___D0                   0.950723  AAACAGCCAGTTTGTG  \n",
       "ACAGCGGGTTTATCCT-D0___D0                   0.370262  AAACAGCCATGTTTGG  \n",
       "ACAGCGGGTTCACTTT-D0___D0                   0.903238  AAACAGCCATTTAAGC  \n",
       "CATTTAGGTGTCCACA-D0___D0                   0.183371  AAACATGCATTGGGAG  \n",
       "CTTTATCGTCGAGGCA-D0___D0                   0.996793  AAACCAACAATAACCT  \n",
       "...                                             ...               ...  \n",
       "GTAGGTGTCAGTGAAC-Q3___Q3                   0.984357  TTTGTGAAGCATGAAG  \n",
       "GTAGGTGTCGGTCCAT-Q3___Q3                   0.973642  TTTGTGAAGCGAGTAA  \n",
       "GTTTGGTGTTTCTAAG-Q3___Q3                   0.177973  TTTGTGGCAGCCTGCA  \n",
       "CGTACTTCAAATATGC-Q3___Q3                   0.041574  TTTGTTGGTAAACAAG  \n",
       "CGTACTTCATGATGAT-Q3___Q3                   0.947241  TTTGTTGGTTAAGCTG  \n",
       "\n",
       "[10106 rows x 35 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs['base_index'] = rna.obs.index.str.split('-').str[0]\n",
    "mapping_dict = pd.Series(df_merged['atac_CB'].values, index=df_merged['rna_CB']).to_dict()\n",
    "rna.obs['atac_CB'] = rna.obs['base_index'].map(mapping_dict)\n",
    "rna.obs['sample'] = rna.obs['sample'].str.split('_').str[0]\n",
    "rna.obs['atac_CB'] = rna.obs['atac_CB'].astype(str)+'-'+rna.obs['sample'].astype(str) + '___' + rna.obs['sample'].astype(str)\n",
    "rna.obs.set_index('atac_CB', inplace=True)\n",
    "rna.obs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "db557c89-22d7-4bd3-a6c6-ff9269281372",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna = rna[rna.obs['celltypist_cell_label_fine'].isin(['Pro-B cells']) & rna.obs['sample'].isin(['D0','Q1'])].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "127bb1d2-143f-48f0-b346-22feb8564f64",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Pro-B cells    5361\n",
       "Name: celltypist_cell_label_fine, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs['celltypist_cell_label_fine'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "66fde54d-91a5-411e-bec2-5e6a34f342b0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Q1    3864\n",
       "D0    1497\n",
       "Name: sample, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rna.obs['sample'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "aa0c039c-9b2d-47d1-ba94-06a0915f7724",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna.X = rna.layers['counts'].copy()\n",
    "rna.raw = rna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "344d8c9b-745d-46b0-8bfe-c7021f5be98d",
   "metadata": {},
   "outputs": [],
   "source": [
    "rna.write_h5ad(\"multiome_adata_04072025_raw_counts.BALL.h5ad\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d51006aa-fecc-45b8-8192-d9e78d598dc2",
   "metadata": {},
   "source": [
    "# run scenicplus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3b7a0cbe-2e68-4295-b716-2f8ecb47ffa1",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "   ____   ____ _____ _   _ ___ ____      \n",
      "  / ___| / ___| ____| \\ | |_ _/ ___|\u001b[31;1m _ \u001b[0m\n",
      "  \\___ \\| |   |  _| |  \\| || | |  \u001b[31;1m _|.|_\u001b[0m\n",
      "   ___) | |___| |___| |\\  || | |__\u001b[31;1m|_..._|\u001b[0m\n",
      "  |____/ \\____|_____|_| \\_|___\\____|\u001b[31;1m|_|\u001b[0m \n",
      "\n",
      "\n",
      "scenicplus verions: 1.0a1\n",
      "usage: scenicplus [-h] {init_snakemake,prepare_data,grn_inference} ...\n",
      "\n",
      "Single-Cell Enhancer-driven gene regulatory Network Inference and Clustering\n",
      "\n",
      "positional arguments:\n",
      "  {init_snakemake,prepare_data,grn_inference}\n",
      "\n",
      "options:\n",
      "  -h, --help            show this help message and exit\n"
     ]
    }
   ],
   "source": [
    "! scenicplus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "498ce8e6-899a-460a-946a-8ff58572ba5c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2025-07-11 09:36:28,372 SCENIC+      INFO     Creating snakemake folder in: scplus_pipeline\n",
      "\u001b[0m"
     ]
    }
   ],
   "source": [
    "!scenicplus init_snakemake --out_dir scplus_pipeline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "20b10d7a-e9bb-421e-8c0b-063fada2c316",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/bash: line 1: tree: command not found\n"
     ]
    }
   ],
   "source": [
    "!tree scplus_pipeline/"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "cedf0a69-eab9-44ec-bbfc-e27c95967884",
   "metadata": {},
   "source": [
    "```\n",
    "tree -d scplus_pipeline/\n",
    "scplus_pipeline/\n",
    "└── Snakemake\n",
    "    ├── config\n",
    "    └── workflow\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "08bf6d99-9ae5-4bdf-b317-707f101bb886",
   "metadata": {},
   "outputs": [],
   "source": [
    "!mkdir -p outs\n",
    "!mkdir -p tmp"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ccecdc63-6b84-42f4-afe8-3084e8cb8023",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_SCENIC_plus.ipynb\n",
      "02_scenic_output.ipynb\n",
      "figures\n",
      "multiome_adata_04072025_raw_counts.BALL.h5ad\n",
      "old\n",
      "outs\n",
      "scplus_pipeline\n",
      "subset_concatenated_adata_log_pca_celltypist_23012025_with_raw_counts.BALL.h5ad\n",
      "tmp\n"
     ]
    }
   ],
   "source": [
    "! ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "67ec337d-7008-4b29-8749-e67932773e7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cd scplus_pipeline/Snakemake/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "3034cd5f-0b15-405c-952c-4ef434f78d86",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "01_SCENIC_plus.ipynb\n",
      "02_scenic_output.ipynb\n",
      "figures\n",
      "multiome_adata_04072025_raw_counts.BALL.h5ad\n",
      "old\n",
      "outs\n",
      "scplus_pipeline\n",
      "subset_concatenated_adata_log_pca_celltypist_23012025_with_raw_counts.BALL.h5ad\n",
      "tmp\n"
     ]
    }
   ],
   "source": [
    "!ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d70784ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "!cat /scplus_pipeline/Snakemake/config/config.yaml"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "754a2400-297c-414d-8491-b65f56b05dea",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/lustre1/project/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/scplus_pipeline/Snakemake\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/venv/lib/python3.11/site-packages/IPython/core/magics/osm.py:417: UserWarning: This is now an optional IPython functionality, setting dhist requires you to install the `pickleshare` library.\n",
      "  self.shell.db['dhist'] = compress_dhist(dhist)[-100:]\n"
     ]
    }
   ],
   "source": [
    "cd /scplus_pipeline/Snakemake/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "95b901cd-b6fd-47c0-b19e-69064b0d784e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[33mAssuming unrestricted shared filesystem usage for local execution.\u001b[0m\n",
      "\u001b[33mBuilding DAG of jobs...\u001b[0m\n",
      "\u001b[33mUsing shell: /usr/bin/bash\u001b[0m\n",
      "\u001b[33mProvided cores: 20\u001b[0m\n",
      "\u001b[33mRules claiming more threads will be scaled down.\u001b[0m\n",
      "\u001b[33mJob stats:\n",
      "job                           count\n",
      "--------------------------  -------\n",
      "AUCell_direct                     1\n",
      "AUCell_extended                   1\n",
      "all                               1\n",
      "eGRN_direct                       1\n",
      "eGRN_extended                     1\n",
      "get_search_space                  1\n",
      "motif_enrichment_cistarget        1\n",
      "motif_enrichment_dem              1\n",
      "prepare_GEX_ACC_multiome          1\n",
      "prepare_menr                      1\n",
      "region_to_gene                    1\n",
      "scplus_mudata                     1\n",
      "tf_to_gene                        1\n",
      "total                            13\n",
      "\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 11:20:58 2025]\u001b[0m\n",
      "\u001b[32mlocalrule motif_enrichment_cistarget:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.html\n",
      "    jobid: 9\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 11:21:06,972 SCENIC+      INFO     Reading region sets from: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\n",
      "2025-07-11 11:21:06,973 SCENIC+      INFO     Reading all .bed files in: Topics_top_3k\n",
      "2025-07-11 11:21:08,346 SCENIC+      INFO     Reading all .bed files in: DARs_cell_type\n",
      "2025-07-11 11:21:08,960 SCENIC+      INFO     Reading all .bed files in: Topics_otsu\n",
      "2025-07-11 11:21:44,599 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:44,926 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:45,380 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:45,399 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:45,895 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:46,092 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:46,154 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:46,376 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:46,469 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:46,731 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:47,173 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:47,450 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:47,468 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:47,574 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:48,168 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:51,310 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:51,733 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:52,968 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:53,001 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:54,718 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic1 which has 3000 regions\n",
      "2025-07-11 11:21:54,764 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic10 which has 3000 regions\n",
      "2025-07-11 11:21:55,125 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic8 which has 3000 regions\n",
      "2025-07-11 11:21:55,751 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:21:55,850 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic2 which has 3000 regions\n",
      "2025-07-11 11:21:56,233 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic23 which has 3000 regions\n",
      "2025-07-11 11:21:56,853 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic20 which has 3000 regions\n",
      "2025-07-11 11:21:58,014 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic14 which has 3000 regions\n",
      "2025-07-11 11:21:59,511 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic12 which has 3000 regions\n",
      "2025-07-11 11:22:02,557 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic19 which has 3000 regions\n",
      "2025-07-11 11:22:03,891 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic5 which has 3000 regions\n",
      "2025-07-11 11:22:05,288 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic15 which has 3000 regions\n",
      "2025-07-11 11:22:05,547 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic17 which has 3000 regions\n",
      "2025-07-11 11:22:06,859 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic9 which has 3000 regions\n",
      "2025-07-11 11:22:09,415 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic21 which has 3000 regions\n",
      "2025-07-11 11:22:09,843 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic8\n",
      "2025-07-11 11:22:09,867 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic16 which has 3000 regions\n",
      "2025-07-11 11:22:10,282 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic25 which has 3000 regions\n",
      "2025-07-11 11:22:11,700 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic18 which has 3000 regions\n",
      "2025-07-11 11:22:12,163 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic11 which has 3000 regions\n",
      "2025-07-11 11:22:14,632 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic24 which has 3000 regions\n",
      "2025-07-11 11:22:14,983 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic8\n",
      "2025-07-11 11:22:15,545 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic1\n",
      "2025-07-11 11:22:16,113 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:17,611 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic10\n",
      "2025-07-11 11:22:18,193 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic23\n",
      "2025-07-11 11:22:18,902 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic12\n",
      "2025-07-11 11:22:20,563 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic20\n",
      "2025-07-11 11:22:20,743 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic22 which has 3000 regions\n",
      "2025-07-11 11:22:21,224 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic17\n",
      "2025-07-11 11:22:22,450 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic3 which has 3000 regions\n",
      "2025-07-11 11:22:22,911 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic1\n",
      "2025-07-11 11:22:23,181 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic2\n",
      "2025-07-11 11:22:25,722 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic17\n",
      "2025-07-11 11:22:26,109 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic10\n",
      "2025-07-11 11:22:26,343 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:26,610 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic12\n",
      "2025-07-11 11:22:26,797 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:28,850 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic23\n",
      "2025-07-11 11:22:29,394 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic14\n",
      "2025-07-11 11:22:29,673 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:29,900 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:30,159 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic2\n",
      "2025-07-11 11:22:30,809 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic3\n",
      "2025-07-11 11:22:33,104 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:34,320 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic7 which has 3000 regions\n",
      "2025-07-11 11:22:35,295 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic4 which has 3000 regions\n",
      "2025-07-11 11:22:36,075 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic20\n",
      "2025-07-11 11:22:36,283 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic13 which has 3000 regions\n",
      "2025-07-11 11:22:36,831 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:37,431 cisTarget    INFO     Running cisTarget for Topics_top_3k_Topic6 which has 3000 regions\n",
      "2025-07-11 11:22:38,584 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic3\n",
      "2025-07-11 11:22:39,895 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic18\n",
      "2025-07-11 11:22:40,286 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:41,321 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic16\n",
      "2025-07-11 11:22:41,498 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic25\n",
      "2025-07-11 11:22:41,582 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:42,611 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic7\n",
      "2025-07-11 11:22:44,083 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic6\n",
      "2025-07-11 11:22:44,374 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic4\n",
      "2025-07-11 11:22:45,107 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic21\n",
      "2025-07-11 11:22:46,240 cisTarget    INFO     Running cisTarget for DARs_cell_type_CD16pos_NK_cells which has 10613 regions\n",
      "2025-07-11 11:22:48,268 cisTarget    INFO     Running cisTarget for DARs_cell_type_Classical_monocytes which has 10248 regions\n",
      "2025-07-11 11:22:48,421 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic14\n",
      "2025-07-11 11:22:49,918 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic6\n",
      "2025-07-11 11:22:51,568 cisTarget    INFO     Running cisTarget for DARs_cell_type_Non-classical_monocytes which has 9432 regions\n",
      "2025-07-11 11:22:51,952 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic18\n",
      "2025-07-11 11:22:52,125 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:52,316 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic4\n",
      "2025-07-11 11:22:52,589 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic5\n",
      "2025-07-11 11:22:52,747 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q1 which has 4880 regions\n",
      "2025-07-11 11:22:52,994 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic9\n",
      "2025-07-11 11:22:53,054 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic19\n",
      "2025-07-11 11:22:53,693 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic24\n",
      "2025-07-11 11:22:54,318 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic25\n",
      "2025-07-11 11:22:54,410 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic7\n",
      "2025-07-11 11:22:54,999 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:55,387 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:56,068 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic16\n",
      "2025-07-11 11:22:56,414 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic21\n",
      "2025-07-11 11:22:56,646 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:56,725 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic15\n",
      "2025-07-11 11:22:56,936 cisTarget    INFO     Annotating motifs for DARs_cell_type_Classical_monocytes\n",
      "2025-07-11 11:22:57,513 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic13\n",
      "2025-07-11 11:22:58,312 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:22:59,208 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q3 which has 4181 regions\n",
      "2025-07-11 11:22:59,347 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic11\n",
      "2025-07-11 11:22:59,568 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:00,053 cisTarget    INFO     Annotating motifs for DARs_cell_type_CD16pos_NK_cells\n",
      "2025-07-11 11:23:01,612 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:02,793 cisTarget    INFO     Getting cistromes for DARs_cell_type_Classical_monocytes\n",
      "2025-07-11 11:23:03,451 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q1\n",
      "2025-07-11 11:23:04,356 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic24\n",
      "\u001b[0m2025-07-11 11:23:06,462 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:06,520 cisTarget    INFO     Annotating motifs for DARs_cell_type_Non-classical_monocytes\n",
      "2025-07-11 11:23:06,577 cisTarget    INFO     Running cisTarget for DARs_cell_type_Tem_Trm_cytotoxic_T_cells which has 12196 regions\n",
      "2025-07-11 11:23:06,943 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q3\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "2025-07-11 11:23:09,705 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_D0 which has 4147 regions\n",
      "2025-07-11 11:23:09,929 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q1\n",
      "2025-07-11 11:23:10,481 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:10,811 cisTarget    INFO     Getting cistromes for DARs_cell_type_CD16pos_NK_cells\n",
      "2025-07-11 11:23:10,969 cisTarget    INFO     Running cisTarget for DARs_cell_type_Tcm_Naive_helper_T_cells which has 10715 regions\n",
      "2025-07-11 11:23:13,396 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic21 which has 8762 regions\n",
      "\u001b[0m2025-07-11 11:23:14,884 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic9\n",
      "2025-07-11 11:23:15,052 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q3\n",
      "2025-07-11 11:23:15,325 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic5\n",
      "2025-07-11 11:23:15,451 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic13\n",
      "2025-07-11 11:23:16,361 cisTarget    INFO     Getting cistromes for DARs_cell_type_Non-classical_monocytes\n",
      "2025-07-11 11:23:16,931 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic19\n",
      "\u001b[0m2025-07-11 11:23:19,743 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_D0\n",
      "2025-07-11 11:23:20,496 cisTarget    INFO     Annotating motifs for Topics_top_3k_Topic22\n",
      "2025-07-11 11:23:21,004 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic15\n",
      "2025-07-11 11:23:21,747 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:22,517 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic11\n",
      "2025-07-11 11:23:22,910 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:23,092 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic11 which has 9492 regions\n",
      "\u001b[0m2025-07-11 11:23:24,698 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic21\n",
      "2025-07-11 11:23:25,088 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:28,122 cisTarget    INFO     Running cisTarget for DARs_cell_type_Naive_B_cells which has 8663 regions\n",
      "\u001b[0m2025-07-11 11:23:29,802 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_D0\n",
      "2025-07-11 11:23:29,803 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:30,355 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:30,385 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:30,467 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:31,030 cisTarget    INFO     Running cisTarget for DARs_cell_type_Pro-B_cells_Q2 which has 9825 regions\n",
      "2025-07-11 11:23:32,310 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic21\n",
      "2025-07-11 11:23:33,269 cisTarget    INFO     Annotating motifs for DARs_cell_type_Tem_Trm_cytotoxic_T_cells\n",
      "2025-07-11 11:23:33,942 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:34,903 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic12 which has 13184 regions\n",
      "2025-07-11 11:23:36,149 cisTarget    INFO     Annotating motifs for DARs_cell_type_Tcm_Naive_helper_T_cells\n",
      "\u001b[0m2025-07-11 11:23:40,108 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic15 which has 12462 regions\n",
      "2025-07-11 11:23:41,976 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic2 which has 10778 regions\n",
      "2025-07-11 11:23:44,069 cisTarget    INFO     Getting cistromes for Topics_top_3k_Topic22\n",
      "2025-07-11 11:23:49,217 cisTarget    INFO     Getting cistromes for DARs_cell_type_Tem_Trm_cytotoxic_T_cells\n",
      "2025-07-11 11:23:49,788 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic1 which has 10461 regions\n",
      "2025-07-11 11:23:49,940 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic17 which has 13442 regions\n",
      "2025-07-11 11:23:50,923 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic11\n",
      "2025-07-11 11:23:52,285 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:23:57,114 cisTarget    INFO     Getting cistromes for DARs_cell_type_Tcm_Naive_helper_T_cells\n",
      "2025-07-11 11:24:01,019 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic12\n",
      "\u001b[0m2025-07-11 11:24:01,540 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic20 which has 13708 regions\n",
      "2025-07-11 11:24:02,716 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:02,989 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic23 which has 5784 regions\n",
      "2025-07-11 11:24:03,121 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:03,843 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic25 which has 15787 regions\n",
      "2025-07-11 11:24:04,193 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic24 which has 8988 regions\n",
      "2025-07-11 11:24:05,483 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic11\n",
      "2025-07-11 11:24:05,974 cisTarget    INFO     Annotating motifs for DARs_cell_type_Pro-B_cells_Q2\n",
      "2025-07-11 11:24:07,305 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic2\n",
      "2025-07-11 11:24:11,209 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:13,460 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic14 which has 22596 regions\n",
      "2025-07-11 11:24:14,337 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic12\n",
      "\u001b[0m2025-07-11 11:24:17,498 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:17,697 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic2\n",
      "2025-07-11 11:24:17,952 cisTarget    INFO     Annotating motifs for DARs_cell_type_Naive_B_cells\n",
      "\u001b[0m2025-07-11 11:24:23,997 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:24,163 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:25,158 cisTarget    INFO     Getting cistromes for DARs_cell_type_Pro-B_cells_Q2\n",
      "2025-07-11 11:24:26,142 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic17\n",
      "2025-07-11 11:24:27,850 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic8 which has 17060 regions\n",
      "2025-07-11 11:24:30,590 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic1\n",
      "2025-07-11 11:24:33,429 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic19 which has 3618 regions\n",
      "2025-07-11 11:24:35,304 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic15\n",
      "2025-07-11 11:24:35,511 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic5 which has 8640 regions\n",
      "2025-07-11 11:24:35,730 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic20\n",
      "2025-07-11 11:24:36,007 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-07-11 11:24:38,330 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic18 which has 10272 regions\n",
      "2025-07-11 11:24:38,653 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic17\n",
      "2025-07-11 11:24:40,801 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic9 which has 10776 regions\n",
      "\u001b[0m2025-07-11 11:24:41,322 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic16 which has 22 regions\n",
      "2025-07-11 11:24:42,747 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:24:46,962 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic20\n",
      "2025-07-11 11:24:47,951 cisTarget    INFO     Getting cistromes for DARs_cell_type_Naive_B_cells\n",
      "2025-07-11 11:24:48,423 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic16\n",
      "2025-07-11 11:24:50,984 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic23\n",
      "2025-07-11 11:24:52,077 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic19\n",
      "2025-07-11 11:24:52,481 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic8\n",
      "\u001b[0m2025-07-11 11:24:55,300 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic16\n",
      "2025-07-11 11:24:55,559 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic22 which has 12030 regions\n",
      "2025-07-11 11:24:56,102 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic1\n",
      "2025-07-11 11:24:56,104 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic18\n",
      "2025-07-11 11:24:56,267 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic24\n",
      "2025-07-11 11:24:56,811 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic15\n",
      "2025-07-11 11:24:58,226 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic10 which has 17023 regions\n",
      "2025-07-11 11:24:59,108 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic8\n",
      "2025-07-11 11:25:00,100 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic18\n",
      "2025-07-11 11:25:00,766 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic19\n",
      "2025-07-11 11:25:00,846 cisTarget    INFO     Reading cisTarget database\n",
      "\u001b[0m2025-07-11 11:25:05,726 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic5\n",
      "2025-07-11 11:25:07,752 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:25:09,063 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic14\n",
      "2025-07-11 11:25:10,103 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:25:12,935 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic13 which has 15288 regions\n",
      "\u001b[0m2025-07-11 11:25:14,638 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic5\n",
      "2025-07-11 11:25:16,749 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic3 which has 11393 regions\n",
      "2025-07-11 11:25:19,441 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic4 which has 9421 regions\n",
      "\u001b[0m2025-07-11 11:25:21,513 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:25:22,983 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic22\n",
      "2025-07-11 11:25:23,258 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 11:25:25,235 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic13\n",
      "\u001b[0m2025-07-11 11:25:27,619 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic14\n",
      "2025-07-11 11:25:29,636 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic23\n",
      "2025-07-11 11:25:29,776 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic25\n",
      "2025-07-11 11:25:30,461 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic6 which has 15246 regions\n",
      "2025-07-11 11:25:32,514 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic22\n",
      "\u001b[0m2025-07-11 11:25:32,836 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic13\n",
      "2025-07-11 11:25:35,675 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic24\n",
      "\u001b[0m2025-07-11 11:25:40,991 cisTarget    INFO     Running cisTarget for Topics_otsu_Topic7 which has 10752 regions\n",
      "2025-07-11 11:25:41,235 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic9\n",
      "2025-07-11 11:25:44,061 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic6\n",
      "2025-07-11 11:25:48,249 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic25\n",
      "2025-07-11 11:25:49,183 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic6\n",
      "\u001b[0m2025-07-11 11:25:55,501 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic3\n",
      "2025-07-11 11:25:56,074 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic9\n",
      "\u001b[0m2025-07-11 11:25:59,986 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic4\n",
      "2025-07-11 11:26:01,787 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic3\n",
      "2025-07-11 11:26:01,880 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic10\n",
      "2025-07-11 11:26:06,748 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic4\n",
      "2025-07-11 11:26:11,023 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic10\n",
      "2025-07-11 11:26:11,355 cisTarget    INFO     Annotating motifs for Topics_otsu_Topic7\n",
      "2025-07-11 11:26:18,196 cisTarget    INFO     Getting cistromes for Topics_otsu_Topic7\n",
      "2025-07-11 11:26:24,654 SCENIC+      INFO     Writing html to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.html\n",
      "2025-07-11 11:26:25,379 SCENIC+      INFO     Writing output to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 11:27:28 2025]\u001b[0m\n",
      "\u001b[32mFinished job 9.\u001b[0m\n",
      "\u001b[32m1 of 13 steps (8%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 11:27:28 2025]\u001b[0m\n",
      "\u001b[32mlocalrule motif_enrichment_dem:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.html\n",
      "    jobid: 7\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 11:27:36,644 SCENIC+      INFO     Reading region sets from: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets\n",
      "2025-07-11 11:27:36,645 SCENIC+      INFO     Reading all .bed files in: Topics_top_3k\n",
      "2025-07-11 11:27:36,910 SCENIC+      INFO     Reading all .bed files in: DARs_cell_type\n",
      "2025-07-11 11:27:36,987 SCENIC+      INFO     Reading all .bed files in: Topics_otsu\n",
      "2025-07-11 11:28:15,496 DEM          INFO     Running DEM for Topics_top_3k_Topic15_vs_all\n",
      "2025-07-11 11:28:16,381 DEM          INFO     Running DEM for Topics_top_3k_Topic9_vs_all\n",
      "2025-07-11 11:28:17,799 DEM          INFO     Running DEM for Topics_top_3k_Topic1_vs_all\n",
      "2025-07-11 11:28:18,861 DEM          INFO     Running DEM for Topics_top_3k_Topic11_vs_all\n",
      "2025-07-11 11:28:19,503 DEM          INFO     Running DEM for Topics_top_3k_Topic8_vs_all\n",
      "2025-07-11 11:28:19,710 DEM          INFO     Running DEM for Topics_top_3k_Topic10_vs_all\n",
      "2025-07-11 11:28:20,037 DEM          INFO     Running DEM for Topics_top_3k_Topic5_vs_all\n",
      "2025-07-11 11:28:20,447 DEM          INFO     Running DEM for Topics_top_3k_Topic16_vs_all\n",
      "2025-07-11 11:28:20,639 DEM          INFO     Running DEM for Topics_top_3k_Topic17_vs_all\n",
      "2025-07-11 11:28:21,160 DEM          INFO     Running DEM for Topics_top_3k_Topic12_vs_all\n",
      "2025-07-11 11:28:21,463 DEM          INFO     Running DEM for Topics_top_3k_Topic24_vs_all\n",
      "2025-07-11 11:28:22,635 DEM          INFO     Running DEM for Topics_top_3k_Topic21_vs_all\n",
      "2025-07-11 11:28:30,676 DEM          INFO     Running DEM for Topics_top_3k_Topic19_vs_all\n",
      "2025-07-11 11:28:35,916 DEM          INFO     Running DEM for Topics_top_3k_Topic25_vs_all\n",
      "2025-07-11 11:28:37,387 DEM          INFO     Running DEM for Topics_top_3k_Topic22_vs_all\n",
      "2025-07-11 11:28:38,321 DEM          INFO     Running DEM for Topics_top_3k_Topic2_vs_all\n",
      "2025-07-11 11:28:42,132 DEM          INFO     Running DEM for Topics_top_3k_Topic18_vs_all\n",
      "2025-07-11 11:28:44,012 DEM          INFO     Running DEM for Topics_top_3k_Topic20_vs_all\n",
      "2025-07-11 11:28:44,874 DEM          INFO     Running DEM for Topics_top_3k_Topic14_vs_all\n",
      "2025-07-11 11:28:45,646 DEM          INFO     Running DEM for Topics_top_3k_Topic23_vs_all\n",
      "2025-07-11 11:29:01,858 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:03,190 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:06,369 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:11,297 DEM          INFO     Running DEM for Topics_top_3k_Topic3_vs_all\n",
      "2025-07-11 11:29:18,240 DEM          INFO     Running DEM for Topics_top_3k_Topic7_vs_all\n",
      "2025-07-11 11:29:19,790 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:20,030 DEM          INFO     Running DEM for Topics_top_3k_Topic4_vs_all\n",
      "2025-07-11 11:29:20,504 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:23,697 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:28,577 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:28,778 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:32,419 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:33,617 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:33,788 DEM          INFO     Running DEM for Topics_top_3k_Topic6_vs_all\n",
      "2025-07-11 11:29:34,207 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:34,989 DEM          INFO     Running DEM for Topics_top_3k_Topic13_vs_all\n",
      "2025-07-11 11:29:36,550 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:39,218 DEM          INFO     Running DEM for DARs_cell_type_CD16pos_NK_cells_vs_all\n",
      "2025-07-11 11:29:40,659 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:42,042 DEM          INFO     Running DEM for DARs_cell_type_Classical_monocytes_vs_all\n",
      "2025-07-11 11:29:43,893 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q1_vs_all\n",
      "2025-07-11 11:29:46,972 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:29:47,297 DEM          INFO     Running DEM for DARs_cell_type_Non-classical_monocytes_vs_all\n",
      "2025-07-11 11:29:47,638 DEM          INFO     Running DEM for DARs_cell_type_Tem_Trm_cytotoxic_T_cells_vs_all\n",
      "2025-07-11 11:29:50,647 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q2_vs_all\n",
      "2025-07-11 11:29:52,190 DEM          INFO     Running DEM for DARs_cell_type_Naive_B_cells_vs_all\n",
      "2025-07-11 11:29:54,673 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_Q3_vs_all\n",
      "2025-07-11 11:29:57,075 DEM          INFO     Running DEM for DARs_cell_type_Tcm_Naive_helper_T_cells_vs_all\n",
      "2025-07-11 11:30:02,731 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:04,433 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:08,047 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "2025-07-11 11:30:14,390 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:14,947 DEM          INFO     Running DEM for DARs_cell_type_Pro-B_cells_D0_vs_all\n",
      "\u001b[0m2025-07-11 11:30:32,282 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:34,101 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:35,048 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:35,397 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:36,281 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:36,421 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m2025-07-11 11:30:41,054 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:30:41,697 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:30:43,595 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m2025-07-11 11:30:49,193 DEM          INFO     Running DEM for Topics_otsu_Topic2_vs_all\n",
      "\u001b[0m\u001b[0m2025-07-11 11:30:54,285 DEM          INFO     Running DEM for Topics_otsu_Topic15_vs_all\n",
      "2025-07-11 11:30:54,819 DEM          INFO     Running DEM for Topics_otsu_Topic21_vs_all\n",
      "2025-07-11 11:30:55,068 DEM          INFO     Running DEM for Topics_otsu_Topic1_vs_all\n",
      "2025-07-11 11:31:01,068 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:02,089 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:31:10,220 DEM          INFO     Running DEM for Topics_otsu_Topic11_vs_all\n",
      "2025-07-11 11:31:11,551 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:13,058 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:13,283 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:13,939 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:18,100 DEM          INFO     Running DEM for Topics_otsu_Topic5_vs_all\n",
      "2025-07-11 11:31:20,881 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:22,916 DEM          INFO     Running DEM for Topics_otsu_Topic25_vs_all\n",
      "2025-07-11 11:31:24,057 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:35,794 DEM          INFO     Running DEM for Topics_otsu_Topic14_vs_all\n",
      "2025-07-11 11:31:40,680 DEM          INFO     Running DEM for Topics_otsu_Topic24_vs_all\n",
      "2025-07-11 11:31:44,127 DEM          INFO     Running DEM for Topics_otsu_Topic10_vs_all\n",
      "2025-07-11 11:31:44,671 DEM          INFO     Running DEM for Topics_otsu_Topic23_vs_all\n",
      "2025-07-11 11:31:49,746 DEM          INFO     Running DEM for Topics_otsu_Topic8_vs_all\n",
      "2025-07-11 11:31:52,209 DEM          INFO     Running DEM for Topics_otsu_Topic19_vs_all\n",
      "2025-07-11 11:31:54,146 DEM          INFO     Running DEM for Topics_otsu_Topic17_vs_all\n",
      "2025-07-11 11:31:58,897 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:31:59,118 DEM          INFO     Running DEM for Topics_otsu_Topic18_vs_all\n",
      "2025-07-11 11:32:01,098 DEM          INFO     Running DEM for Topics_otsu_Topic22_vs_all\n",
      "2025-07-11 11:32:02,755 DEM          INFO     Running DEM for Topics_otsu_Topic12_vs_all\n",
      "2025-07-11 11:32:05,573 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:32:10,482 DEM          INFO     Running DEM for Topics_otsu_Topic3_vs_all\n",
      "2025-07-11 11:32:11,509 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:32:12,530 DEM          INFO     Running DEM for Topics_otsu_Topic9_vs_all\n",
      "2025-07-11 11:32:14,274 DEM          INFO     Running DEM for Topics_otsu_Topic16_vs_all\n",
      "2025-07-11 11:32:16,885 DEM          INFO     Running DEM for Topics_otsu_Topic4_vs_all\n",
      "2025-07-11 11:32:24,648 DEM          INFO     Running DEM for Topics_otsu_Topic7_vs_all\n",
      "2025-07-11 11:32:28,721 DEM          INFO     Running DEM for Topics_otsu_Topic20_vs_all\n",
      "2025-07-11 11:32:36,448 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:32:37,868 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:32:37,968 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:32:48,282 DEM          INFO     Running DEM for Topics_otsu_Topic6_vs_all\n",
      "2025-07-11 11:32:48,606 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:32:54,705 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:32:56,345 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m2025-07-11 11:33:06,205 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:33:07,188 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:33:17,147 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:33:20,850 DEM          INFO     Running DEM for Topics_otsu_Topic13_vs_all\n",
      "2025-07-11 11:33:22,174 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m\u001b[0m2025-07-11 11:33:31,788 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:33:57,394 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:34:10,048 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:34:27,853 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:34:29,071 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:34:41,282 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:34:57,177 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:34:59,434 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:35:05,336 DEM          INFO     Adding motif-to-TF annotation\n",
      "\u001b[0m2025-07-11 11:35:14,572 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:35:18,831 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:35:25,831 DEM          INFO     Adding motif-to-TF annotation\n",
      "2025-07-11 11:35:27,799 SCENIC+      INFO     Writing html to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.html\n",
      "2025-07-11 11:35:27,934 SCENIC+      INFO     Writing output to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 11:35:36 2025]\u001b[0m\n",
      "\u001b[32mFinished job 7.\u001b[0m\n",
      "\u001b[32m2 of 13 steps (15%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 11:35:36 2025]\u001b[0m\n",
      "\u001b[32mlocalrule prepare_GEX_ACC_multiome:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/D0_Q1_subset/outs/subset_cistopic_obj_all_info.pkl, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/multiome_adata_04072025_raw_counts.BALL.h5ad\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    jobid: 2\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 11:35:44,331 SCENIC+      INFO     Reading cisTopic object.\n",
      "2025-07-11 11:35:45,342 SCENIC+      INFO     Reading gene expression AnnData.\n",
      "2025-07-11 11:35:46,721 Ingesting multiome data INFO     Found 4178 multiome cells.\n",
      "2025-07-11 11:35:46,817 cisTopic     INFO     Imputing region accessibility\n",
      "2025-07-11 11:35:46,817 cisTopic     INFO     Impute region accessibility for regions 0-20000\n",
      "2025-07-11 11:35:47,230 cisTopic     INFO     Impute region accessibility for regions 20000-40000\n",
      "2025-07-11 11:35:47,644 cisTopic     INFO     Impute region accessibility for regions 40000-60000\n",
      "2025-07-11 11:35:48,057 cisTopic     INFO     Impute region accessibility for regions 60000-80000\n",
      "2025-07-11 11:35:48,472 cisTopic     INFO     Impute region accessibility for regions 80000-100000\n",
      "2025-07-11 11:35:48,886 cisTopic     INFO     Impute region accessibility for regions 100000-120000\n",
      "2025-07-11 11:35:49,300 cisTopic     INFO     Impute region accessibility for regions 120000-140000\n",
      "2025-07-11 11:35:49,713 cisTopic     INFO     Impute region accessibility for regions 140000-160000\n",
      "2025-07-11 11:35:50,127 cisTopic     INFO     Impute region accessibility for regions 160000-180000\n",
      "2025-07-11 11:35:50,541 cisTopic     INFO     Impute region accessibility for regions 180000-200000\n",
      "2025-07-11 11:35:50,751 cisTopic     INFO     Done!\n",
      "... storing 'pycisTopic_leiden_10_1.2' as categorical\n",
      "... storing 'barcode' as categorical\n",
      "... storing 'pycisTopic_leiden_10_0.6' as categorical\n",
      "... storing 'sample_id' as categorical\n",
      "... storing 'cell_type_all' as categorical\n",
      "... storing 'pycisTopic_leiden_10_0.2' as categorical\n",
      "... storing 'cell_type' as categorical\n",
      "... storing 'pycisTopic_leiden_10_3' as categorical\n",
      "... storing 'pycisTopic_leiden_10_0.3' as categorical\n",
      "... storing 'pycisTopic_leiden_10_6' as categorical\n",
      "... storing 'Chromosome' as categorical\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 11:35:58 2025]\u001b[0m\n",
      "\u001b[32mFinished job 2.\u001b[0m\n",
      "\u001b[32m3 of 13 steps (23%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 2 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 11:35:58 2025]\u001b[0m\n",
      "\u001b[32mlocalrule prepare_menr:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad\n",
      "    jobid: 6\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 11:35:58 2025]\u001b[0m\n",
      "\u001b[32mlocalrule get_search_space:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/chromsizes.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv\n",
      "    jobid: 11\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 11:36:05,971 SCENIC+      INFO     Reading multiome MuData.\n",
      "2025-07-11 11:36:06,202 SCENIC+      INFO     Reading data\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 11:36:07,833 SCENIC+      INFO     Getting cistromes.\n",
      "2025-07-11 11:36:08,887 Get search space INFO     Extending promoter annotation to 10 bp upstream and 10 downstream\n",
      "2025-07-11 11:36:09,118 Get search space INFO     Extending search space to:\n",
      "            \t\t\t\t\t\t150000 bp downstream of the end of the gene.\n",
      "            \t\t\t\t\t\t150000 bp upstream of the start of the gene.\n",
      "2025-07-11 11:36:10,095 Get search space INFO     Intersecting with regions.\n",
      "2025-07-11 11:36:11,782 Get search space INFO     Calculating distances from region to gene\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 11:36:38,607 SCENIC+      INFO     Found 773 TFs.\n",
      "2025-07-11 11:36:38,607 SCENIC+      INFO     Saving TF names to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt\n",
      "2025-07-11 11:36:38,609 SCENIC+      INFO     Writing direct cistromes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad\n",
      "... storing 'motifs' as categorical\n",
      "2025-07-11 11:36:38,727 SCENIC+      INFO     Writing extended cistromes to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad\n",
      "... storing 'motifs' as categorical\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 11:36:39 2025]\u001b[0m\n",
      "\u001b[32mFinished job 6.\u001b[0m\n",
      "\u001b[32m4 of 13 steps (31%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "2025-07-11 11:37:01,284 Get search space INFO     Imploding multiple entries per region and gene\n",
      "2025-07-11 11:38:54,730 SCENIC+      INFO     Writing search space to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 11:38:57 2025]\u001b[0m\n",
      "\u001b[32mFinished job 11.\u001b[0m\n",
      "\u001b[32m5 of 13 steps (38%) done\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 11:38:57 2025]\u001b[0m\n",
      "\u001b[32mlocalrule region_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv\n",
      "    jobid: 10\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 11:39:05,942 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 11:39:07,747 SCENIC+      INFO     Reading search space\n",
      "2025-07-11 11:39:08,181 R2G          INFO     Calculating region to gene importances, using GBM method\n",
      "Running using 20 cores: 100%|█████████████| 16198/16198 [45:41<00:00,  5.91it/s]\n",
      "2025-07-11 12:25:07,823 R2G          INFO     Calculating region to gene correlation, using SR method\n",
      "Running using 20 cores:   0%|                 | 1/16198 [00:00<36:17,  7.44it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   0%|               | 53/16198 [00:00<00:59, 270.57it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|               | 84/16198 [00:00<01:40, 160.23it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|              | 106/16198 [00:00<01:55, 139.66it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|              | 124/16198 [00:00<02:01, 131.80it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏              | 167/16198 [00:01<03:17, 81.19it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏              | 189/16198 [00:01<02:40, 99.84it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏              | 204/16198 [00:01<03:21, 79.50it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   1%|▏             | 233/16198 [00:02<02:27, 108.09it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▎              | 290/16198 [00:02<02:55, 90.42it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores:   2%|▎              | 369/16198 [00:03<02:44, 96.16it/s]/opt/venv/lib/python3.11/site-packages/scenicplus/enhancer_to_gene.py:158: ConstantInputWarning: An input array is constant; the correlation coefficient is not defined.\n",
      "  correlation_result = np.array([correlator(x, exp) for x in acc.T])\n",
      "Running using 20 cores: 100%|█████████████| 16198/16198 [02:44<00:00, 98.47it/s]\n",
      "2025-07-11 12:28:11,498 R2G          INFO     Done!\n",
      "2025-07-11 12:28:11,680 SCENIC+      INFO     Saving region to gene adjacencies to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 12:28:24 2025]\u001b[0m\n",
      "\u001b[32mFinished job 10.\u001b[0m\n",
      "\u001b[32m6 of 13 steps (46%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 12:28:24 2025]\u001b[0m\n",
      "\u001b[32mlocalrule tf_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv\n",
      "    jobid: 5\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 12:28:32,906 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 12:28:34,706 SCENIC+      INFO     Using 774 TFs.\n",
      "2025-07-11 12:28:35,124 TF2G         INFO     Calculating TF-to-gene importance\n",
      "Running using 20 cores: 100%|███████████| 36601/36601 [2:04:11<00:00,  4.91it/s]\n",
      "2025-07-11 14:33:05,330 TF2G         INFO     Adding correlation coefficients to adjacencies.\n",
      "2025-07-11 14:33:41,745 TF2G         INFO     Warning: adding TFs as their own target to adjecencies matrix. Importance values will be max + 1e-05\n",
      "2025-07-11 14:33:46,526 SCENIC+      INFO     Saving TF to gene adjacencies to: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 14:33:56 2025]\u001b[0m\n",
      "\u001b[32mFinished job 5.\u001b[0m\n",
      "\u001b[32m7 of 13 steps (54%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:33:56 2025]\u001b[0m\n",
      "\u001b[32mlocalrule eGRN_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv\n",
      "    jobid: 4\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 14:34:05,364 SCENIC+      INFO     Loading TF to gene adjacencies.\n",
      "2025-07-11 14:34:06,380 SCENIC+      INFO     Loading region to gene adjacencies.\n",
      "2025-07-11 14:34:07,498 SCENIC+      INFO     Loading cistromes.\n",
      "2025-07-11 14:34:07,815 GSEA         INFO     Thresholding region to gene relationships\n",
      "  0%|                                                    | 0/14 [00:00<?, ?it/s]\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:   0%|          | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:   4%| | 22/520 [00:00<00:02, 217.92it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:   9%| | 48/520 [00:00<00:01, 241.68it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  14%|▏| 73/520 [00:00<00:02, 202.60it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  18%|▏| 94/520 [00:00<00:03, 111.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  22%|▏| 115/520 [00:00<00:03, 131.58it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  28%|▎| 144/520 [00:00<00:02, 168.31it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  33%|▎| 170/520 [00:00<00:01, 189.57it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  37%|▎| 193/520 [00:01<00:01, 175.46it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  41%|▍| 215/520 [00:01<00:01, 185.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  46%|▍| 237/520 [00:01<00:02, 131.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  50%|▌| 261/520 [00:01<00:01, 153.51it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  56%|▌| 292/520 [00:01<00:01, 187.23it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  61%|▌| 318/520 [00:01<00:00, 203.85it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  66%|▋| 342/520 [00:01<00:00, 206.44it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  71%|▋| 368/520 [00:02<00:00, 219.89it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  75%|▊| 392/520 [00:02<00:00, 216.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  80%|▊| 417/520 [00:02<00:00, 225.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  85%|▊| 441/520 [00:02<00:00, 226.40it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  90%|▉| 466/520 [00:02<00:00, 231.46it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  94%|▉| 490/520 [00:02<00:00, 232.65it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.85 quantile:  99%|▉| 514/520 [00:02<00:00, 150.47it/\u001b[0m\u001b[A\n",
      "  7%|███▏                                        | 1/14 [00:09<02:07,  9.81s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   0%|           | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   5%| | 26/520 [00:00<00:01, 257.90it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  11%| | 56/520 [00:00<00:01, 277.69it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  16%|▏| 84/520 [00:00<00:01, 225.64it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  21%|▏| 108/520 [00:00<00:01, 207.53it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  27%|▎| 138/520 [00:00<00:01, 235.78it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  33%|▎| 170/520 [00:00<00:01, 257.51it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  38%|▍| 197/520 [00:00<00:01, 224.07it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  43%|▍| 226/520 [00:00<00:01, 239.68it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  49%|▍| 254/520 [00:01<00:01, 246.84it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  55%|▌| 287/520 [00:01<00:00, 269.78it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  61%|▌| 318/520 [00:01<00:00, 279.40it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  67%|▋| 347/520 [00:01<00:01, 156.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  72%|▋| 376/520 [00:01<00:00, 181.48it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  77%|▊| 401/520 [00:01<00:00, 191.99it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  83%|▊| 430/520 [00:01<00:00, 212.99it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  88%|▉| 456/520 [00:02<00:00, 222.93it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  93%|▉| 482/520 [00:02<00:00, 230.58it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  98%|▉| 511/520 [00:02<00:00, 245.06it/s\u001b[0m\u001b[A\n",
      " 14%|██████▎                                     | 2/14 [00:13<01:12,  6.03s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:   0%|          | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:   5%| | 28/520 [00:00<00:01, 278.75it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  12%| | 61/520 [00:00<00:01, 289.15it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  17%|▏| 90/520 [00:00<00:01, 235.26it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  22%|▏| 115/520 [00:00<00:01, 229.58it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  29%|▎| 153/520 [00:00<00:01, 274.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  35%|▎| 182/520 [00:00<00:01, 238.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  40%|▍| 208/520 [00:01<00:02, 150.28it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  47%|▍| 242/520 [00:01<00:01, 186.85it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  52%|▌| 273/520 [00:01<00:01, 212.83it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  59%|▌| 308/520 [00:01<00:00, 244.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  65%|▋| 337/520 [00:01<00:00, 247.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  71%|▋| 369/520 [00:01<00:00, 264.35it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  77%|▊| 398/520 [00:01<00:00, 262.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  82%|▊| 427/520 [00:01<00:00, 268.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  88%|▉| 456/520 [00:01<00:00, 272.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  93%|▉| 485/520 [00:02<00:00, 267.22it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, negative r2g:  99%|▉| 514/520 [00:02<00:00, 272.53it/\u001b[0m\u001b[A\n",
      " 21%|█████████▍                                  | 3/14 [00:16<00:51,  4.73s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   0%| | 0/520 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   3%| | 14/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   5%| | 28/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   8%| | 43/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  11%| | 59/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  14%|▏| 73/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  16%|▏| 85/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  18%|▏| 96/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  20%|▏| 105/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  23%|▏| 120/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  26%|▎| 135/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  29%|▎| 153/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  32%|▎| 167/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  35%|▎| 181/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  37%|▎| 193/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  39%|▍| 205/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  43%|▍| 222/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  46%|▍| 237/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  48%|▍| 251/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  51%|▌| 265/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  54%|▌| 283/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  57%|▌| 299/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  60%|▌| 314/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  63%|▋| 329/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  67%|▋| 346/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  69%|▋| 359/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  72%|▋| 376/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  75%|▊| 390/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  78%|▊| 404/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  80%|▊| 418/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  83%|▊| 432/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  86%|▊| 446/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  88%|▉| 459/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  91%|▉| 472/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  93%|▉| 486/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  96%|▉| 501/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  99%|▉| 515/520 [\u001b[0m\u001b[A\n",
      " 29%|████████████▌                               | 4/14 [00:22<00:51,  5.14s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   0%| | 0/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   2%| | 12/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   5%| | 24/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   7%| | 37/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   9%| | 49/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  12%| | 61/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  14%|▏| 72/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  16%|▏| 82/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  17%|▏| 90/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  19%|▏| 98/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  20%|▏| 106/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  23%|▏| 119/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  25%|▎| 130/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  28%|▎| 145/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  30%|▎| 158/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  33%|▎| 170/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  35%|▎| 181/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  37%|▎| 191/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  39%|▍| 201/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  41%|▍| 213/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  43%|▍| 226/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  46%|▍| 238/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  48%|▍| 250/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  50%|▌| 262/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  53%|▌| 275/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  55%|▌| 288/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  58%|▌| 302/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  61%|▌| 315/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  63%|▋| 327/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  66%|▋| 341/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  68%|▋| 353/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  70%|▋| 365/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  73%|▋| 378/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  75%|▊| 390/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  77%|▊| 401/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  79%|▊| 412/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  81%|▊| 422/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  83%|▊| 434/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  86%|▊| 445/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  88%|▉| 456/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  90%|▉| 467/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  92%|▉| 479/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  94%|▉| 490/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  97%|▉| 502/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  99%|▉| 513/520 \u001b[0m\u001b[A\n",
      " 36%|███████████████▋                            | 5/14 [00:29<00:52,  5.78s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   0%| | 0/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   2%| | 11/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   4%| | 22/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   7%| | 34/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   9%| | 45/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  11%| | 56/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  13%|▏| 67/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  15%|▏| 77/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  17%|▏| 86/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  18%|▏| 94/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  20%|▏| 102/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  22%|▏| 113/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  24%|▏| 126/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  27%|▎| 138/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  29%|▎| 151/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  31%|▎| 162/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  33%|▎| 173/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  35%|▎| 183/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  37%|▎| 191/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  38%|▍| 200/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  40%|▍| 209/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  43%|▍| 222/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  45%|▍| 235/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  47%|▍| 245/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  49%|▍| 255/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  51%|▌| 267/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  54%|▌| 281/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  56%|▌| 293/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  59%|▌| 305/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  61%|▌| 317/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  63%|▋| 329/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  66%|▋| 343/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  68%|▋| 354/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  70%|▋| 365/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  73%|▋| 378/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  75%|▋| 389/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  77%|▊| 399/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  79%|▊| 409/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  81%|▊| 420/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  83%|▊| 432/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  85%|▊| 443/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  87%|▊| 454/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  89%|▉| 465/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  92%|▉| 476/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  94%|▉| 487/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  96%|▉| 499/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  98%|▉| 510/520 \u001b[0m\u001b[A\n",
      " 43%|██████████████████▊                         | 6/14 [00:35<00:49,  6.17s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   0%|         | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   3%| | 17/520 [00:00<00:03, 164.31it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   7%| | 34/520 [00:00<00:03, 161.10it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  10%| | 51/520 [00:00<00:02, 158.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  13%|▏| 67/520 [00:00<00:03, 140.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  16%|▏| 82/520 [00:00<00:03, 126.99it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  18%|▏| 95/520 [00:00<00:03, 118.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  21%|▏| 108/520 [00:00<00:03, 119.24it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  24%|▏| 126/520 [00:00<00:02, 134.02it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  29%|▎| 149/520 [00:01<00:02, 159.37it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  32%|▎| 167/520 [00:01<00:02, 163.45it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  35%|▎| 184/520 [00:01<00:02, 120.00it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  39%|▍| 201/520 [00:01<00:02, 128.39it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  42%|▍| 216/520 [00:02<00:06, 43.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  45%|▍| 236/520 [00:02<00:04, 59.47it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  48%|▍| 252/520 [00:02<00:03, 72.04it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  52%|▌| 269/520 [00:02<00:02, 86.83it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  56%|▌| 290/520 [00:02<00:02, 108.05it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  59%|▌| 309/520 [00:02<00:01, 123.79it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  63%|▋| 327/520 [00:03<00:01, 121.36it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  67%|▋| 347/520 [00:03<00:01, 137.70it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  70%|▋| 364/520 [00:03<00:01, 141.99it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  73%|▋| 381/520 [00:03<00:00, 147.55it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  77%|▊| 398/520 [00:03<00:00, 143.41it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  80%|▊| 414/520 [00:03<00:00, 147.13it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  83%|▊| 431/520 [00:03<00:00, 150.68it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  86%|▊| 447/520 [00:03<00:00, 147.96it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  89%|▉| 463/520 [00:03<00:00, 147.80it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  92%|▉| 480/520 [00:04<00:00, 150.70it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  95%|▉| 496/520 [00:04<00:00, 151.24it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  99%|▉| 513/520 [00:04<00:00, 155.01it\u001b[0m\u001b[A\n",
      " 50%|██████████████████████                      | 7/14 [01:49<03:18, 28.29s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   0%|          | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   3%| | 15/520 [00:00<00:03, 142.72it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   6%| | 30/520 [00:00<00:03, 139.69it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   9%| | 45/520 [00:00<00:03, 143.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  12%| | 61/520 [00:00<00:03, 143.74it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  15%|▏| 76/520 [00:00<00:03, 121.45it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  17%|▏| 89/520 [00:00<00:03, 110.74it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  19%|▏| 101/520 [00:00<00:04, 99.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  22%|▏| 116/520 [00:00<00:03, 111.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  25%|▎| 132/520 [00:01<00:03, 123.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  30%|▎| 154/520 [00:01<00:02, 148.84it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  33%|▎| 170/520 [00:01<00:02, 141.03it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  36%|▎| 185/520 [00:01<00:02, 118.04it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  39%|▍| 201/520 [00:01<00:02, 127.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  42%|▍| 217/520 [00:01<00:02, 135.10it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  45%|▍| 234/520 [00:01<00:01, 143.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  48%|▍| 249/520 [00:01<00:01, 144.86it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  51%|▌| 264/520 [00:02<00:01, 145.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  54%|▌| 283/520 [00:02<00:01, 156.53it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  57%|▌| 299/520 [00:02<00:01, 154.61it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  61%|▌| 315/520 [00:02<00:01, 154.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  64%|▋| 331/520 [00:02<00:01, 140.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  67%|▋| 347/520 [00:02<00:01, 145.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  70%|▋| 362/520 [00:02<00:01, 141.40it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  73%|▋| 380/520 [00:02<00:00, 148.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  76%|▊| 395/520 [00:02<00:00, 137.45it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  79%|▊| 410/520 [00:03<00:00, 139.61it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  82%|▊| 426/520 [00:03<00:00, 143.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  85%|▊| 441/520 [00:03<00:00, 141.75it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  88%|▉| 457/520 [00:03<00:00, 146.12it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  91%|▉| 472/520 [00:03<00:00, 145.14it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  94%|▉| 487/520 [00:03<00:00, 144.89it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  97%|▉| 504/520 [00:03<00:00, 151.55it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g: 100%|█| 520/520 [00:03<00:00, 148.04it/\u001b[0m\u001b[A\n",
      " 57%|█████████████████████████▏                  | 8/14 [01:54<02:05, 20.92s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   0%|           | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   3%| | 18/520 [00:00<00:02, 175.07it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   7%| | 36/520 [00:00<00:02, 174.30it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  11%| | 55/520 [00:00<00:02, 178.17it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  14%|▏| 73/520 [00:00<00:02, 154.58it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  17%|▏| 89/520 [00:00<00:03, 135.94it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  20%|▏| 104/520 [00:00<00:03, 124.63it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  24%|▏| 124/520 [00:00<00:02, 143.62it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  27%|▎| 139/520 [00:01<00:09, 39.75it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  31%|▎| 162/520 [00:02<00:06, 58.15it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  34%|▎| 177/520 [00:02<00:05, 66.78it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  37%|▎| 193/520 [00:02<00:04, 79.92it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  40%|▍| 209/520 [00:02<00:03, 93.12it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  44%|▍| 230/520 [00:02<00:02, 115.73it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  48%|▍| 249/520 [00:02<00:02, 131.25it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  52%|▌| 268/520 [00:02<00:01, 143.22it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  56%|▌| 290/520 [00:02<00:01, 160.67it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  59%|▌| 309/520 [00:02<00:01, 168.17it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  63%|▋| 328/520 [00:02<00:01, 159.99it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  67%|▋| 349/520 [00:03<00:00, 171.80it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  71%|▋| 368/520 [00:03<00:00, 172.00it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  74%|▋| 386/520 [00:03<00:00, 168.08it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  78%|▊| 404/520 [00:03<00:00, 167.48it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  81%|▊| 422/520 [00:03<00:00, 169.32it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  85%|▊| 440/520 [00:03<00:00, 171.24it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  88%|▉| 460/520 [00:03<00:00, 177.39it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  92%|▉| 479/520 [00:03<00:00, 178.02it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  96%|▉| 499/520 [00:03<00:00, 180.37it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g: 100%|▉| 518/520 [00:04<00:00, 177.94it/s\u001b[0m\u001b[A\n",
      " 64%|████████████████████████████▎               | 9/14 [02:00<01:19, 15.99s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   0%|          | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   4%| | 21/520 [00:00<00:02, 208.44it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   8%| | 43/520 [00:00<00:02, 213.80it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  12%|▏| 65/520 [00:00<00:02, 207.19it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  17%|▏| 86/520 [00:00<00:02, 178.52it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  20%|▏| 105/520 [00:00<00:02, 156.52it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  25%|▎| 130/520 [00:00<00:02, 180.60it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  31%|▎| 160/520 [00:00<00:01, 213.67it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  35%|▎| 183/520 [00:00<00:01, 184.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  39%|▍| 205/520 [00:01<00:01, 191.63it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  45%|▍| 232/520 [00:01<00:01, 212.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  49%|▍| 255/520 [00:01<00:01, 210.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  55%|▌| 284/520 [00:01<00:01, 228.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  59%|▌| 308/520 [00:01<00:00, 230.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  64%|▋| 332/520 [00:01<00:00, 217.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  68%|▋| 355/520 [00:01<00:00, 220.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  73%|▋| 380/520 [00:01<00:00, 226.25it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  78%|▊| 403/520 [00:01<00:00, 215.41it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  82%|▊| 426/520 [00:02<00:00, 217.71it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  86%|▊| 448/520 [00:02<00:00, 216.29it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  91%|▉| 471/520 [00:02<00:00, 218.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  95%|▉| 495/520 [00:02<00:00, 223.90it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g: 100%|▉| 518/520 [00:02<00:00, 222.36it/\u001b[0m\u001b[A\n",
      " 71%|██████████████████████████████▋            | 10/14 [02:03<00:48, 12.18s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   0%| | 0/520 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   2%| | 11/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   4%| | 21/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   6%| | 31/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   8%| | 41/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  10%| | 51/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  12%| | 61/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  14%|▏| 71/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  15%|▏| 80/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  17%|▏| 88/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  18%|▏| 96/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  20%|▏| 103/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  22%|▏| 113/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  24%|▏| 124/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  26%|▎| 134/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  29%|▎| 149/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  31%|▎| 160/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  33%|▎| 171/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  35%|▎| 181/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  37%|▎| 192/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  39%|▍| 201/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  41%|▍| 211/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  43%|▍| 222/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  45%|▍| 233/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  47%|▍| 243/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  49%|▍| 253/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  51%|▌| 264/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  53%|▌| 276/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  55%|▌| 287/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  57%|▌| 298/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  59%|▌| 309/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  62%|▌| 320/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  64%|▋| 331/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  66%|▋| 342/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  68%|▋| 352/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  70%|▋| 362/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  72%|▋| 374/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  74%|▋| 385/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  76%|▊| 395/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  78%|▊| 405/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  80%|▊| 415/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  82%|▊| 425/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  84%|▊| 435/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  86%|▊| 445/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  88%|▉| 455/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  89%|▉| 465/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  92%|▉| 477/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  94%|▉| 487/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  96%|▉| 499/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  98%|▉| 509/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g: 100%|▉| 518/520 [\u001b[0m\u001b[A\n",
      " 79%|█████████████████████████████████▊         | 11/14 [02:11<00:32, 10.93s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   0%| | 0/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   1%| | 7/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   3%| | 14/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   4%| | 21/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   5%| | 27/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   7%| | 35/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   8%| | 42/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:   9%| | 49/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  11%| | 57/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  12%| | 64/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  13%|▏| 70/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  15%|▏| 76/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  16%|▏| 81/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  17%|▏| 86/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  18%|▏| 91/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  18%|▏| 96/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  19%|▏| 101/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  21%|▏| 107/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  22%|▏| 114/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  23%|▏| 122/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  25%|▏| 129/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  26%|▎| 137/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  28%|▎| 148/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  30%|▎| 156/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  32%|▎| 164/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  33%|▎| 171/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  34%|▎| 178/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  35%|▎| 184/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  37%|▎| 193/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  38%|▍| 200/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  40%|▍| 206/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  41%|▍| 215/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  43%|▍| 222/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  44%|▍| 230/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  46%|▍| 237/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  47%|▍| 244/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  48%|▍| 251/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  50%|▍| 258/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  51%|▌| 266/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  53%|▌| 274/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  54%|▌| 283/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  56%|▌| 291/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  57%|▌| 298/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  59%|▌| 305/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  60%|▌| 313/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  62%|▌| 321/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  63%|▋| 328/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  64%|▋| 335/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  66%|▋| 343/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  67%|▋| 350/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  69%|▋| 357/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  70%|▋| 365/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  72%|▋| 373/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  73%|▋| 380/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  74%|▋| 387/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  76%|▊| 394/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  77%|▊| 401/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  78%|▊| 408/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  80%|▊| 415/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  81%|▊| 422/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  83%|▊| 430/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  84%|▊| 437/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  85%|▊| 444/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  87%|▊| 451/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  88%|▉| 458/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  89%|▉| 465/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  91%|▉| 472/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  92%|▉| 480/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  94%|▉| 487/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  95%|▉| 495/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  97%|▉| 502/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  98%|▉| 509/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 10 region-to-gene links per gene:  99%|▉| 516/520 \u001b[0m\u001b[A\n",
      " 86%|████████████████████████████████████▊      | 12/14 [02:21<00:21, 10.57s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   0%| | 0/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   1%| | 6/520 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   2%| | 12/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   3%| | 18/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   4%| | 23/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   6%| | 29/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   7%| | 35/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   8%| | 41/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:   9%| | 47/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  10%| | 53/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  12%| | 60/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  13%|▏| 66/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  14%|▏| 71/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  15%|▏| 76/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  16%|▏| 81/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  17%|▏| 86/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  17%|▏| 90/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  18%|▏| 94/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  19%|▏| 98/520 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  20%|▏| 102/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  21%|▏| 108/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  22%|▏| 113/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  23%|▏| 119/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  24%|▏| 125/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  25%|▎| 131/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  27%|▎| 138/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  28%|▎| 147/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  29%|▎| 153/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  31%|▎| 159/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  32%|▎| 165/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  33%|▎| 171/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  34%|▎| 176/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  35%|▎| 181/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  36%|▎| 185/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  37%|▎| 193/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  38%|▍| 198/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  39%|▍| 203/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  40%|▍| 208/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  41%|▍| 215/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  42%|▍| 221/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  44%|▍| 227/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  45%|▍| 233/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  46%|▍| 239/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  47%|▍| 245/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  48%|▍| 251/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  49%|▍| 257/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  51%|▌| 263/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  52%|▌| 269/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  53%|▌| 276/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  54%|▌| 283/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  56%|▌| 289/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  57%|▌| 295/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  58%|▌| 301/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  59%|▌| 307/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  60%|▌| 313/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  61%|▌| 319/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  62%|▋| 325/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  63%|▋| 330/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  65%|▋| 337/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  66%|▋| 344/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  67%|▋| 350/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  68%|▋| 356/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  70%|▋| 362/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  71%|▋| 369/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  72%|▋| 375/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  73%|▋| 381/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  74%|▋| 387/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  75%|▊| 392/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  77%|▊| 398/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  78%|▊| 404/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  79%|▊| 410/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  80%|▊| 416/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  81%|▊| 422/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  82%|▊| 429/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  84%|▊| 435/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  85%|▊| 441/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  86%|▊| 448/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  88%|▉| 455/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  89%|▉| 461/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  90%|▉| 467/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  91%|▉| 473/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  92%|▉| 479/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  93%|▉| 485/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  94%|▉| 491/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  96%|▉| 498/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  97%|▉| 504/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  98%|▉| 510/520 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m positive r2g, Top 15 region-to-gene links per gene:  99%|▉| 516/520 \u001b[0m\u001b[A\n",
      " 93%|███████████████████████████████████████▉   | 13/14 [02:34<00:11, 11.40s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   0%|         | 0/520 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   2%| | 10/520 [00:00<00:05, 91.97it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   4%| | 20/520 [00:00<00:05, 83.49it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   6%| | 29/520 [00:00<00:05, 84.64it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   8%| | 39/520 [00:00<00:05, 89.91it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   9%| | 49/520 [00:00<00:05, 87.84it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  12%| | 61/520 [00:00<00:05, 88.60it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  13%|▏| 70/520 [00:00<00:05, 77.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  15%|▏| 78/520 [00:00<00:06, 72.24it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  17%|▏| 86/520 [00:01<00:06, 63.89it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  18%|▏| 93/520 [00:01<00:07, 60.17it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  19%|▏| 100/520 [00:01<00:07, 56.58it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  21%|▏| 109/520 [00:01<00:06, 64.11it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  23%|▏| 118/520 [00:01<00:05, 70.19it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  25%|▏| 128/520 [00:01<00:05, 76.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  27%|▎| 139/520 [00:01<00:04, 84.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  29%|▎| 153/520 [00:01<00:03, 98.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  32%|▎| 164/520 [00:02<00:03, 96.74it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  33%|▎| 174/520 [00:02<00:04, 81.98it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  35%|▎| 183/520 [00:02<00:04, 71.80it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  37%|▎| 193/520 [00:02<00:04, 77.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  39%|▍| 202/520 [00:02<00:04, 76.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  41%|▍| 212/520 [00:02<00:03, 81.40it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  43%|▍| 222/520 [00:02<00:03, 85.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  45%|▍| 233/520 [00:02<00:03, 90.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  47%|▍| 243/520 [00:03<00:03, 90.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  49%|▍| 253/520 [00:03<00:02, 89.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  51%|▌| 263/520 [00:03<00:02, 91.85it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  53%|▌| 275/520 [00:03<00:02, 98.93it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  55%|▌| 286/520 [00:03<00:02, 99.93it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  57%|▌| 297/520 [00:03<00:02, 98.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  59%|▌| 307/520 [00:03<00:02, 97.07it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  61%|▌| 317/520 [00:03<00:02, 96.82it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  63%|▋| 327/520 [00:03<00:02, 83.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  65%|▋| 337/520 [00:04<00:02, 87.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  67%|▋| 348/520 [00:04<00:01, 91.15it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  69%|▋| 358/520 [00:04<00:01, 84.41it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  71%|▋| 370/520 [00:04<00:01, 92.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  73%|▋| 380/520 [00:04<00:01, 93.38it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  75%|▊| 390/520 [00:04<00:01, 84.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  77%|▊| 400/520 [00:04<00:01, 85.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  79%|▊| 409/520 [00:04<00:01, 85.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  81%|▊| 420/520 [00:04<00:01, 88.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  83%|▊| 431/520 [00:05<00:00, 92.41it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  85%|▊| 441/520 [00:05<00:00, 87.22it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  87%|▊| 451/520 [00:05<00:00, 90.29it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  89%|▉| 461/520 [00:05<00:00, 91.56it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  91%|▉| 471/520 [00:05<00:00, 87.27it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  92%|▉| 481/520 [00:05<00:00, 88.94it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  94%|▉| 491/520 [00:05<00:00, 89.64it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  97%|▉| 502/520 [00:05<00:00, 93.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  98%|▉| 512/520 [00:06<00:00, 90.75it/\u001b[0m\u001b[A\n",
      "100%|███████████████████████████████████████████| 14/14 [03:10<00:00, 13.58s/it]\u001b[A\n",
      "2025-07-11 14:37:17,945 GSEA         INFO     Subsetting TF2G adjacencies for TF with motif.\n",
      "2025-07-11 14:37:18,066 GSEA         INFO     Running GSEA...\n",
      "2025-07-11 14:37:18,067 GSEA         INFO     Generating rankings...\n",
      "100%|█████████████████████████████████████████| 520/520 [00:05<00:00, 93.28it/s]\n",
      "0it [00:00, ?it/s]\n",
      "Running for Positive TF to gene:   1%|        | 40/6859 [00:11<31:28,  3.61it/s]\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "Running for Positive TF to gene:   2%|▏      | 160/6859 [00:35<10:45, 10.37it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:  11%|▊      | 740/6859 [00:46<01:43, 58.93it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene: 100%|██████| 6859/6859 [02:27<00:00, 46.64it/s]\n",
      "Running for Negative TF to gene: 100%|██| 6859/6859 [00:00<00:00, 128944.74it/s]\n",
      "2025-07-11 14:39:51,802 GSEA         INFO     Subsetting on adjusted pvalue: 1, minimal NES: 0 and minimal leading edge genes 10\n",
      "2025-07-11 14:39:51,947 GSEA         INFO     Merging eRegulons\n",
      "2025-07-11 14:39:54,785 SCENIC+      INFO     Formatting eGRN as table.\n",
      "2025-07-11 14:39:55,404 SCENIC+      INFO     Calculating triplet ranking.\n",
      "2025-07-11 14:39:55,604 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 14:40:21,136 SCENIC+      INFO     Saving network to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 14:40:25 2025]\u001b[0m\n",
      "\u001b[32mFinished job 4.\u001b[0m\n",
      "\u001b[32m8 of 13 steps (62%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:40:25 2025]\u001b[0m\n",
      "\u001b[32mlocalrule eGRN_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv\n",
      "    jobid: 13\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 14:40:34,697 SCENIC+      INFO     Loading TF to gene adjacencies.\n",
      "2025-07-11 14:40:35,720 SCENIC+      INFO     Loading region to gene adjacencies.\n",
      "2025-07-11 14:40:36,727 SCENIC+      INFO     Loading cistromes.\n",
      "2025-07-11 14:40:36,872 GSEA         INFO     Thresholding region to gene relationships\n",
      "  0%|                                                    | 0/14 [00:00<?, ?it/s]\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:   0%|          | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  10%| | 27/272 [00:00<00:00, 266.95it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  20%|▏| 54/272 [00:00<00:00, 253.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  29%|▎| 80/272 [00:00<00:00, 235.40it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  38%|▍| 104/272 [00:00<00:01, 132.83it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  45%|▍| 122/272 [00:00<00:01, 140.05it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  53%|▌| 144/272 [00:00<00:00, 158.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  64%|▋| 175/272 [00:00<00:00, 196.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  77%|▊| 209/272 [00:01<00:00, 233.30it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  86%|▊| 235/272 [00:01<00:00, 235.26it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, negative r2g:  97%|▉| 265/272 [00:01<00:00, 252.29it/\u001b[0m\u001b[A\n",
      "  7%|███▏                                        | 1/14 [00:04<00:58,  4.51s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:   0%|           | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  12%| | 32/272 [00:00<00:00, 313.16it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  24%|▏| 64/272 [00:00<00:01, 149.02it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  32%|▎| 87/272 [00:00<00:01, 169.40it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  43%|▍| 118/272 [00:00<00:00, 205.33it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  52%|▌| 142/272 [00:00<00:00, 205.20it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  66%|▋| 180/272 [00:00<00:00, 251.89it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  80%|▊| 218/272 [00:00<00:00, 287.05it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.9 quantile:  92%|▉| 249/272 [00:01<00:00, 280.50it/s\u001b[0m\u001b[A\n",
      " 14%|██████▎                                     | 2/14 [00:06<00:37,  3.13s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:   0%|          | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  12%|▏| 34/272 [00:00<00:00, 331.58it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  25%|▎| 68/272 [00:00<00:00, 284.69it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  36%|▎| 97/272 [00:00<00:01, 146.83it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  44%|▍| 121/272 [00:00<00:00, 167.96it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  54%|▌| 148/272 [00:00<00:00, 192.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  69%|▋| 187/272 [00:00<00:00, 242.83it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  82%|▊| 223/272 [00:00<00:00, 269.88it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, 0.95 quantile:  94%|▉| 257/272 [00:01<00:00, 288.27it/\u001b[0m\u001b[A\n",
      " 21%|█████████▍                                  | 3/14 [00:08<00:29,  2.69s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   0%| | 0/272 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:   6%| | 15/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  11%| | 31/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  17%|▏| 47/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  23%|▏| 63/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  28%|▎| 77/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  33%|▎| 90/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  40%|▍| 109/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  45%|▍| 122/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  49%|▍| 133/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  56%|▌| 151/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  61%|▌| 166/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  68%|▋| 184/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  74%|▋| 202/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  80%|▊| 218/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  86%|▊| 234/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  92%|▉| 251/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 5 region-to-gene links per gene:  98%|▉| 267/272 [\u001b[0m\u001b[A\n",
      " 29%|████████████▌                               | 4/14 [00:11<00:28,  2.86s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   0%| | 0/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:   5%| | 13/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  10%| | 26/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  15%|▏| 40/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  19%|▏| 53/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  24%|▏| 65/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  28%|▎| 76/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  32%|▎| 87/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  38%|▍| 102/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  42%|▍| 114/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  46%|▍| 126/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  50%|▌| 137/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  54%|▌| 147/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  59%|▌| 161/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  64%|▋| 175/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  69%|▋| 189/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  75%|▊| 205/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  81%|▊| 219/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  85%|▊| 232/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  90%|▉| 246/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 10 region-to-gene links per gene:  96%|▉| 260/272 \u001b[0m\u001b[A\n",
      " 36%|███████████████▋                            | 5/14 [00:15<00:28,  3.21s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   0%| | 0/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   4%| | 12/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:   9%| | 24/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  14%|▏| 37/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  18%|▏| 49/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  22%|▏| 61/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  26%|▎| 72/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  30%|▎| 82/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  34%|▎| 93/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  40%|▍| 108/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  44%|▍| 119/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  47%|▍| 128/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  52%|▌| 141/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  57%|▌| 155/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  61%|▌| 167/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  66%|▋| 180/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  71%|▋| 194/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  76%|▊| 208/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  81%|▊| 221/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  86%|▊| 233/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  90%|▉| 246/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene:  95%|▉| 259/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, Top 15 region-to-gene links per gene: 100%|▉| 271/272 \u001b[0m\u001b[A\n",
      " 43%|██████████████████▊                         | 6/14 [00:19<00:27,  3.42s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   0%|         | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:   6%| | 16/272 [00:00<00:01, 158.73it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  13%|▏| 35/272 [00:00<00:01, 175.80it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  19%|▏| 53/272 [00:00<00:01, 165.48it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  26%|▎| 70/272 [00:00<00:01, 153.68it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  32%|▎| 86/272 [00:00<00:01, 144.65it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  40%|▍| 109/272 [00:00<00:00, 168.89it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  47%|▍| 127/272 [00:00<00:01, 128.62it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  54%|▌| 147/272 [00:00<00:00, 145.66it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  61%|▌| 167/272 [00:01<00:00, 158.80it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  68%|▋| 185/272 [00:01<00:01, 71.27it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  76%|▊| 208/272 [00:01<00:00, 93.44it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  82%|▊| 224/272 [00:01<00:00, 99.71it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  90%|▉| 244/272 [00:02<00:00, 118.11it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m negative r2g, BASC binarized:  97%|▉| 265/272 [00:02<00:00, 137.11it\u001b[0m\u001b[A\n",
      " 50%|██████████████████████                      | 7/14 [00:41<01:06,  9.50s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   0%|          | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:   6%| | 17/272 [00:00<00:01, 154.23it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  12%| | 33/272 [00:00<00:01, 156.27it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  19%|▏| 51/272 [00:00<00:01, 160.02it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  25%|▏| 67/272 [00:00<00:01, 142.27it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  30%|▎| 82/272 [00:00<00:01, 130.54it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  37%|▎| 100/272 [00:00<00:01, 143.69it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  43%|▍| 116/272 [00:00<00:01, 143.81it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  48%|▍| 131/272 [00:00<00:01, 125.77it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  56%|▌| 151/272 [00:01<00:00, 144.70it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  62%|▌| 168/272 [00:01<00:00, 151.47it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  69%|▋| 189/272 [00:01<00:00, 165.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  76%|▊| 207/272 [00:01<00:00, 164.98it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  82%|▊| 224/272 [00:01<00:00, 157.92it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  89%|▉| 241/272 [00:01<00:00, 159.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.85 quantile, positive r2g:  96%|▉| 260/272 [00:01<00:00, 167.19it/\u001b[0m\u001b[A\n",
      " 57%|█████████████████████████▏                  | 8/14 [00:44<00:44,  7.48s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   0%|           | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:   8%| | 21/272 [00:00<00:01, 199.77it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  15%|▏| 41/272 [00:00<00:01, 198.72it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  22%|▏| 61/272 [00:00<00:01, 192.90it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  30%|▎| 81/272 [00:00<00:01, 159.86it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  38%|▍| 102/272 [00:00<00:00, 175.07it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  44%|▍| 121/272 [00:00<00:00, 162.08it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  51%|▌| 138/272 [00:00<00:00, 160.88it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  59%|▌| 161/272 [00:00<00:00, 178.34it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  68%|▋| 186/272 [00:01<00:00, 197.96it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  76%|▊| 208/272 [00:01<00:00, 202.41it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  84%|▊| 229/272 [00:01<00:00, 195.14it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.9 quantile, positive r2g:  93%|▉| 252/272 [00:01<00:00, 203.36it/s\u001b[0m\u001b[A\n",
      " 64%|████████████████████████████▎               | 9/14 [00:47<00:29,  5.93s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:   0%|          | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  10%| | 26/272 [00:00<00:00, 255.00it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  19%|▏| 52/272 [00:00<00:00, 249.13it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  28%|▎| 77/272 [00:00<00:00, 214.02it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  38%|▍| 102/272 [00:00<00:00, 225.59it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  46%|▍| 125/272 [00:00<00:00, 203.34it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  57%|▌| 154/272 [00:00<00:00, 226.18it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  67%|▋| 183/272 [00:00<00:00, 244.76it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  78%|▊| 212/272 [00:00<00:00, 254.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  88%|▉| 238/272 [00:01<00:00, 240.97it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m 0.95 quantile, positive r2g:  98%|▉| 266/272 [00:01<00:00, 251.82it/\u001b[0m\u001b[A\n",
      " 71%|██████████████████████████████▋            | 10/14 [00:50<00:19,  4.98s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   0%| | 0/272 [00\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   4%| | 11/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:   8%| | 22/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  12%| | 33/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  16%|▏| 44/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  20%|▏| 55/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  24%|▏| 65/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  28%|▎| 75/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  31%|▎| 84/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  35%|▎| 96/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  40%|▍| 109/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  44%|▍| 120/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  47%|▍| 129/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  52%|▌| 141/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  57%|▌| 155/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  61%|▌| 166/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  66%|▋| 180/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  71%|▋| 192/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  75%|▊| 204/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  80%|▊| 217/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  84%|▊| 229/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  88%|▉| 240/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  93%|▉| 253/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 5 region-to-gene links per gene, positive r2g:  97%|▉| 265/272 [\u001b[0m\u001b[A\n",
      " 79%|█████████████████████████████████▊         | 11/14 [00:53<00:13,  4.62s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   0%| | 0/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   3%| | 9/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:   7%| | 18/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  10%| | 26/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  12%|▏| 34/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  15%|▏| 42/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  18%|▏| 50/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  21%|▏| 58/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  24%|▏| 65/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  26%|▎| 72/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  29%|▎| 78/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  31%|▎| 84/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  33%|▎| 91/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  37%|▎| 101/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  40%|▍| 109/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  43%|▍| 117/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  46%|▍| 124/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  47%|▍| 129/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  51%|▌| 138/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  54%|▌| 146/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  57%|▌| 155/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  60%|▌| 163/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  64%|▋| 173/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  67%|▋| 181/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  69%|▋| 189/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  72%|▋| 197/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  75%|▊| 205/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  79%|▊| 214/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  82%|▊| 222/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  85%|▊| 230/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  88%|▉| 238/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  90%|▉| 245/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  93%|▉| 254/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  96%|▉| 262/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 10 region-to-gene links per gene, positive r2g:  99%|▉| 270/272 \u001b[0m\u001b[A\n",
      " 86%|████████████████████████████████████▊      | 12/14 [00:59<00:09,  4.99s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   0%| | 0/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   3%| | 7/272 [0\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   5%| | 14/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:   7%| | 20/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  10%| | 26/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  12%| | 32/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  14%|▏| 38/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  16%|▏| 44/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  19%|▏| 51/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  21%|▏| 57/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  23%|▏| 63/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  25%|▎| 69/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  27%|▎| 74/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  29%|▎| 79/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  31%|▎| 84/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  33%|▎| 90/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  36%|▎| 98/272 [\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  39%|▍| 105/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  41%|▍| 111/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  43%|▍| 117/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  45%|▍| 123/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  47%|▍| 128/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  50%|▌| 136/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  52%|▌| 142/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  55%|▌| 150/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  58%|▌| 157/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  60%|▌| 163/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  63%|▋| 172/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  66%|▋| 179/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  68%|▋| 186/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  71%|▋| 193/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  74%|▋| 200/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  76%|▊| 207/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  79%|▊| 215/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  82%|▊| 222/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  84%|▊| 228/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  86%|▊| 235/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  89%|▉| 241/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  91%|▉| 248/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  93%|▉| 254/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  96%|▉| 260/272 \u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m Top 15 region-to-gene links per gene, positive r2g:  98%|▉| 267/272 \u001b[0m\u001b[A\n",
      " 93%|███████████████████████████████████████▉   | 13/14 [01:05<00:05,  5.33s/it]\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   0%|         | 0/272 [00:00<?, ?it/s]\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   4%| | 11/272 [00:00<00:02, 103.20it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:   8%| | 22/272 [00:00<00:02, 94.09it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  12%| | 33/272 [00:00<00:02, 97.71it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  16%|▏| 43/272 [00:00<00:02, 96.07it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  19%|▏| 53/272 [00:00<00:02, 93.33it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  23%|▏| 63/272 [00:00<00:02, 90.36it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  27%|▎| 73/272 [00:00<00:02, 81.77it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  30%|▎| 82/272 [00:01<00:07, 25.65it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  33%|▎| 91/272 [00:01<00:05, 32.29it/s\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  38%|▍| 104/272 [00:01<00:03, 44.79it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  42%|▍| 114/272 [00:02<00:03, 52.39it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  45%|▍| 123/272 [00:02<00:02, 52.87it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  48%|▍| 131/272 [00:02<00:02, 56.37it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  52%|▌| 142/272 [00:02<00:01, 66.66it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  57%|▌| 155/272 [00:02<00:01, 79.69it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  61%|▌| 165/272 [00:02<00:01, 82.80it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  65%|▋| 178/272 [00:02<00:00, 94.43it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  69%|▋| 189/272 [00:02<00:00, 98.42it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  74%|▋| 200/272 [00:02<00:00, 98.95it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  78%|▊| 212/272 [00:03<00:00, 102.59it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  82%|▊| 223/272 [00:03<00:00, 97.65it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  86%|▊| 234/272 [00:03<00:00, 96.24it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  90%|▉| 244/272 [00:03<00:00, 96.62it/\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  94%|▉| 256/272 [00:03<00:00, 101.16it\u001b[0m\u001b[A\n",
      "\u001b[32;1mProcessing:\u001b[0m BASC binarized, positive r2g:  98%|▉| 267/272 [00:03<00:00, 103.24it\u001b[0m\u001b[A\n",
      "100%|███████████████████████████████████████████| 14/14 [01:38<00:00,  7.05s/it]\u001b[A\n",
      "2025-07-11 14:42:15,599 GSEA         INFO     Subsetting TF2G adjacencies for TF with motif.\n",
      "2025-07-11 14:42:15,694 GSEA         INFO     Running GSEA...\n",
      "2025-07-11 14:42:15,694 GSEA         INFO     Generating rankings...\n",
      "100%|████████████████████████████████████████| 272/272 [00:01<00:00, 171.14it/s]\n",
      "0it [00:00, ?it/s]\n",
      "Running for Positive TF to gene:   1%|        | 40/3551 [00:12<17:36,  3.32it/s]\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "/opt/venv/lib/python3.11/site-packages/joblib/externals/loky/process_executor.py:752: UserWarning: A worker stopped while some jobs were given to the executor. This can be caused by a too short worker timeout or by a memory leak.\n",
      "  warnings.warn(\n",
      "\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   4%|▎      | 140/3551 [00:34<07:32,  7.54it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:   6%|▍      | 220/3551 [00:36<02:12, 25.18it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:  29%|█▊    | 1040/3551 [00:46<00:32, 78.12it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene:  37%|██▏   | 1300/3551 [00:52<00:33, 66.21it/s]/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:71: RuntimeWarning: divide by zero encountered in divide\n",
      "  norm_tag =  1.0/sum_correl_tag\n",
      "/opt/venv/lib/python3.11/site-packages/gseapy/algorithm.py:74: RuntimeWarning: invalid value encountered in multiply\n",
      "  RES = np.cumsum(tag_indicator * correl_vector * norm_tag - no_tag_indicator * norm_no_tag, axis=axis)\n",
      "Running for Positive TF to gene: 100%|██████| 3551/3551 [01:28<00:00, 39.93it/s]\n",
      "Running for Negative TF to gene: 100%|██| 3551/3551 [00:00<00:00, 135926.16it/s]\n",
      "2025-07-11 14:43:47,180 GSEA         INFO     Subsetting on adjusted pvalue: 1, minimal NES: 0 and minimal leading edge genes 10\n",
      "2025-07-11 14:43:47,266 GSEA         INFO     Merging eRegulons\n",
      "2025-07-11 14:43:48,318 SCENIC+      INFO     Formatting eGRN as table.\n",
      "2025-07-11 14:43:48,922 SCENIC+      INFO     Calculating triplet ranking.\n",
      "2025-07-11 14:43:49,075 cisTarget    INFO     Reading cisTarget database\n",
      "2025-07-11 14:44:01,274 SCENIC+      INFO     Saving network to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv\n",
      "\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[0m\u001b[32m[Fri Jul 11 14:44:06 2025]\u001b[0m\n",
      "\u001b[32mFinished job 13.\u001b[0m\n",
      "\u001b[32m9 of 13 steps (69%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:44:06 2025]\u001b[0m\n",
      "\u001b[32mlocalrule AUCell_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu\n",
      "    jobid: 3\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 14:44:14,999 SCENIC+      INFO     Reading data.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 14:44:19,582 SCENIC+      INFO     Calculating enrichment scores.\n",
      "2025-07-11 14:45:29,500 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 14:45:30 2025]\u001b[0m\n",
      "\u001b[32mFinished job 3.\u001b[0m\n",
      "\u001b[32m10 of 13 steps (77%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:45:30 2025]\u001b[0m\n",
      "\u001b[32mlocalrule AUCell_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu\n",
      "    jobid: 12\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 14:45:39,263 SCENIC+      INFO     Reading data.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 14:45:41,313 SCENIC+      INFO     Calculating enrichment scores.\n",
      "2025-07-11 14:46:54,417 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 14:46:55 2025]\u001b[0m\n",
      "\u001b[32mFinished job 12.\u001b[0m\n",
      "\u001b[32m11 of 13 steps (85%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:46:55 2025]\u001b[0m\n",
      "\u001b[32mlocalrule scplus_mudata:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "    jobid: 1\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "2025-07-11 14:47:03,592 SCENIC+      INFO     Reading multiome MuData.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 14:47:05,440 SCENIC+      INFO     Reading AUC values.\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "/opt/venv/lib/python3.11/site-packages/anndata/_core/anndata.py:522: FutureWarning: The dtype argument is deprecated and will be removed in late 2024.\n",
      "  warnings.warn(\n",
      "2025-07-11 14:47:05,524 SCENIC+      INFO     Reading eRegulon metadata.\n",
      "2025-07-11 14:47:05,615 SCENIC+      INFO     Generating MuData object.\n",
      "2025-07-11 14:47:06,310 SCENIC+      INFO     Writing file to /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "\u001b[0m\u001b[32m[Fri Jul 11 14:47:09 2025]\u001b[0m\n",
      "\u001b[32mFinished job 1.\u001b[0m\n",
      "\u001b[32m12 of 13 steps (92%) done\u001b[0m\n",
      "\u001b[33mSelect jobs to execute...\u001b[0m\n",
      "\u001b[33mExecute 1 jobs...\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:47:09 2025]\u001b[0m\n",
      "\u001b[32mlocalrule all:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "    resources: tmpdir=/tmp\u001b[0m\n",
      "\u001b[32m\u001b[0m\n",
      "\u001b[32m[Fri Jul 11 14:47:09 2025]\u001b[0m\n",
      "\u001b[32mFinished job 0.\u001b[0m\n",
      "\u001b[32m13 of 13 steps (100%) done\u001b[0m\n",
      "\u001b[33mComplete log: .snakemake/log/2025-07-11T112057.951245.snakemake.log\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "! snakemake --cores 20 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a15c4fc1-ee03-45ce-9cf2-9720a5506716",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Building DAG of jobs...\n",
      "Using shell: /usr/bin/bash\n",
      "Provided cores: 20\n",
      "Rules claiming more threads will be scaled down.\n",
      "Job stats:\n",
      "job                           count\n",
      "--------------------------  -------\n",
      "AUCell_direct                     1\n",
      "AUCell_extended                   1\n",
      "all                               1\n",
      "eGRN_direct                       1\n",
      "eGRN_extended                     1\n",
      "get_search_space                  1\n",
      "motif_enrichment_cistarget        1\n",
      "motif_enrichment_dem              1\n",
      "prepare_GEX_ACC_multiome          1\n",
      "prepare_menr                      1\n",
      "region_to_gene                    1\n",
      "scplus_mudata                     1\n",
      "tf_to_gene                        1\n",
      "total                            13\n",
      "\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 11:20:58 2025]\n",
      "localrule motif_enrichment_cistarget:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.html\n",
      "    jobid: 9\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 11:27:28 2025]\n",
      "Finished job 9.\n",
      "1 of 13 steps (8%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 11:27:28 2025]\n",
      "localrule motif_enrichment_dem:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/outs/region_sets, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.scores.feather, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/aertslab_motif_colleciton/v10nr_clust_public/snapshots/motifs-v10-nr.hgnc-m0.00001-o0.0.tbl\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.html\n",
      "    jobid: 7\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 11:35:36 2025]\n",
      "Finished job 7.\n",
      "2 of 13 steps (15%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 11:35:36 2025]\n",
      "localrule prepare_GEX_ACC_multiome:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/atac/D0_Q1_subset/outs/subset_cistopic_obj_all_info.pkl, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/multiome_adata_04072025_raw_counts.BALL.h5ad\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    jobid: 2\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 11:35:58 2025]\n",
      "Finished job 2.\n",
      "3 of 13 steps (23%) done\n",
      "Select jobs to execute...\n",
      "Execute 2 jobs...\n",
      "\n",
      "[Fri Jul 11 11:35:58 2025]\n",
      "localrule prepare_menr:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad\n",
      "    jobid: 6\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ctx_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/dem_results.hdf5, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "\n",
      "[Fri Jul 11 11:35:58 2025]\n",
      "localrule get_search_space:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/genome_annotation.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/chromsizes.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv\n",
      "    jobid: 11\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 11:36:39 2025]\n",
      "Finished job 6.\n",
      "4 of 13 steps (31%) done\n",
      "Select jobs to execute...\n",
      "[Fri Jul 11 11:38:57 2025]\n",
      "Finished job 11.\n",
      "5 of 13 steps (38%) done\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 11:38:57 2025]\n",
      "localrule region_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv\n",
      "    jobid: 10\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/search_space.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 12:28:24 2025]\n",
      "Finished job 10.\n",
      "6 of 13 steps (46%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 12:28:24 2025]\n",
      "localrule tf_to_gene:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv\n",
      "    jobid: 5\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_names.txt, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:33:56 2025]\n",
      "Finished job 5.\n",
      "7 of 13 steps (54%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 14:33:56 2025]\n",
      "localrule eGRN_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv\n",
      "    jobid: 4\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_direct.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:40:25 2025]\n",
      "Finished job 4.\n",
      "8 of 13 steps (62%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 14:40:25 2025]\n",
      "localrule eGRN_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/10x_ONT_BALL.regions_vs_motifs.rankings.feather\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv\n",
      "    jobid: 13\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/region_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/tf_to_gene_adj.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/cistromes_extended.h5ad\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:44:06 2025]\n",
      "Finished job 13.\n",
      "9 of 13 steps (69%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 14:44:06 2025]\n",
      "localrule AUCell_direct:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu\n",
      "    jobid: 3\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:45:30 2025]\n",
      "Finished job 3.\n",
      "10 of 13 steps (77%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 14:45:30 2025]\n",
      "localrule AUCell_extended:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu\n",
      "    jobid: 12\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    threads: 20\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:46:55 2025]\n",
      "Finished job 12.\n",
      "11 of 13 steps (85%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 14:46:55 2025]\n",
      "localrule scplus_mudata:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv\n",
      "    output: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "    jobid: 1\n",
      "    reason: Missing output files: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu; Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulon_direct.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/eRegulons_extended.tsv, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_direct.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/AUCell_extended.h5mu, /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/ACC_GEX.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:47:09 2025]\n",
      "Finished job 1.\n",
      "12 of 13 steps (92%) done\n",
      "Select jobs to execute...\n",
      "Execute 1 jobs...\n",
      "\n",
      "[Fri Jul 11 14:47:09 2025]\n",
      "localrule all:\n",
      "    input: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "    jobid: 0\n",
      "    reason: Input files updated by another job: /staging/leuven/stg_00096/home/projects/2023_Cools_B-ALL/scenic_plus/subset/outs/scplusmdata.h5mu\n",
      "    resources: tmpdir=/tmp\n",
      "\n",
      "[Fri Jul 11 14:47:09 2025]\n",
      "Finished job 0.\n",
      "13 of 13 steps (100%) done\n",
      "Complete log: .snakemake/log/2025-07-11T112057.951245.snakemake.log\n"
     ]
    }
   ],
   "source": [
    "! cat .snakemake/log/2025-07-11T112057.951245.snakemake.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "74820415-4056-413c-b327-811a1a02679e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Package                              Version\n",
      "------------------------------------ -----------------\n",
      "adjustText                           1.0.4\n",
      "aiohttp                              3.9.3\n",
      "aiosignal                            1.3.1\n",
      "anndata                              0.10.5.post1\n",
      "annoy                                1.17.3\n",
      "appdirs                              1.4.4\n",
      "arboreto                             0.1.6\n",
      "argparse-dataclass                   2.0.0\n",
      "array_api_compat                     1.5.1\n",
      "asttokens                            2.4.1\n",
      "attr                                 0.3.2\n",
      "attrs                                23.2.0\n",
      "bbknn                                1.6.0\n",
      "beautifulsoup4                       4.12.3\n",
      "bidict                               0.23.1\n",
      "bioservices                          1.11.2\n",
      "blosc2                               2.5.1\n",
      "bokeh                                3.4.0\n",
      "boltons                              23.1.1\n",
      "bs4                                  0.0.2\n",
      "cattrs                               23.2.3\n",
      "certifi                              2024.2.2\n",
      "charset-normalizer                   3.3.2\n",
      "click                                8.1.7\n",
      "cloudpickle                          3.0.0\n",
      "colorama                             0.4.6\n",
      "colorlog                             6.8.2\n",
      "comm                                 0.2.2\n",
      "conda-inject                         1.3.1\n",
      "ConfigArgParse                       1.7\n",
      "connection_pool                      0.0.3\n",
      "contourpy                            1.2.0\n",
      "ctxcore                              0.2.0\n",
      "cycler                               0.12.1\n",
      "Cython                               0.29.37\n",
      "cytoolz                              0.12.3\n",
      "dask                                 2024.2.1\n",
      "dataclasses-json                     0.6.4\n",
      "datrie                               0.8.2\n",
      "debugpy                              1.8.1\n",
      "decorator                            5.1.1\n",
      "dill                                 0.3.8\n",
      "distributed                          2024.2.1\n",
      "docutils                             0.20.1\n",
      "dpath                                2.1.6\n",
      "easydev                              0.13.1\n",
      "et-xmlfile                           1.1.0\n",
      "executing                            2.0.1\n",
      "fastjsonschema                       2.19.1\n",
      "fbpca                                1.0\n",
      "filelock                             3.13.1\n",
      "fonttools                            4.50.0\n",
      "frozendict                           2.4.0\n",
      "frozenlist                           1.4.1\n",
      "fsspec                               2024.3.1\n",
      "future                               1.0.0\n",
      "gensim                               4.3.2\n",
      "geosketch                            1.2\n",
      "gevent                               24.2.1\n",
      "gitdb                                4.0.11\n",
      "GitPython                            3.1.42\n",
      "globre                               0.1.5\n",
      "greenlet                             3.0.3\n",
      "grequests                            0.7.0\n",
      "gseapy                               0.10.8\n",
      "h5py                                 3.10.0\n",
      "harmonypy                            0.0.9\n",
      "humanfriendly                        10.0\n",
      "idna                                 3.6\n",
      "igraph                               0.11.4\n",
      "imageio                              2.34.0\n",
      "immutables                           0.20\n",
      "importlib-metadata                   7.0.1\n",
      "importlib_resources                  6.1.2\n",
      "interlap                             0.2.7\n",
      "intervaltree                         3.1.0\n",
      "ipykernel                            6.29.3\n",
      "ipython                              8.22.2\n",
      "jedi                                 0.19.1\n",
      "Jinja2                               3.1.3\n",
      "joblib                               1.3.2\n",
      "jsonpickle                           3.0.3\n",
      "jsonschema                           4.21.1\n",
      "jsonschema-specifications            2023.12.1\n",
      "jupyter_client                       8.6.1\n",
      "jupyter_core                         5.7.2\n",
      "kaleido                              0.2.1\n",
      "kiwisolver                           1.4.5\n",
      "lazy_loader                          0.3\n",
      "lda                                  3.0.0\n",
      "leidenalg                            0.10.2\n",
      "line-profiler                        4.1.2\n",
      "llvmlite                             0.42.0\n",
      "locket                               1.0.0\n",
      "loompy                               3.0.7\n",
      "loomxpy                              0.4.2\n",
      "lxml                                 5.1.0\n",
      "lz4                                  4.3.3\n",
      "MACS2                                2.2.9.1\n",
      "markdown-it-py                       3.0.0\n",
      "MarkupSafe                           2.1.5\n",
      "marshmallow                          3.21.1\n",
      "matplotlib                           3.6.3\n",
      "matplotlib-inline                    0.1.6\n",
      "mdurl                                0.1.2\n",
      "mizani                               0.9.3\n",
      "msgpack                              1.0.8\n",
      "mudata                               0.2.3\n",
      "multidict                            6.0.5\n",
      "multiprocessing_on_dill              3.5.0a4\n",
      "mypy-extensions                      1.0.0\n",
      "natsort                              8.4.0\n",
      "nbformat                             5.10.3\n",
      "ncls                                 0.0.68\n",
      "ndindex                              1.8\n",
      "nest-asyncio                         1.6.0\n",
      "networkx                             3.2.1\n",
      "numba                                0.59.0\n",
      "numexpr                              2.9.0\n",
      "numpy                                1.26.4\n",
      "numpy-groupies                       0.10.2\n",
      "openpyxl                             3.1.2\n",
      "packaging                            24.0\n",
      "pandas                               1.5.0\n",
      "parso                                0.8.3\n",
      "partd                                1.4.1\n",
      "patsy                                0.5.6\n",
      "pexpect                              4.9.0\n",
      "pillow                               10.2.0\n",
      "pip                                  24.0\n",
      "plac                                 1.4.3\n",
      "platformdirs                         4.2.0\n",
      "plotly                               5.19.0\n",
      "plotnine                             0.12.4\n",
      "polars                               0.20.13\n",
      "progressbar2                         4.4.2\n",
      "prompt-toolkit                       3.0.43\n",
      "protobuf                             5.26.0\n",
      "psutil                               5.9.8\n",
      "ptyprocess                           0.7.0\n",
      "PuLP                                 2.8.0\n",
      "pure-eval                            0.2.2\n",
      "py-cpuinfo                           9.0.0\n",
      "pyarrow                              15.0.0\n",
      "pyarrow-hotfix                       0.6\n",
      "pybedtools                           0.9.1\n",
      "pybigtools                           0.1.2\n",
      "pyBigWig                             0.3.22\n",
      "pybiomart                            0.2.0\n",
      "pycistarget                          1.0a2\n",
      "pycisTopic                           2.0a0\n",
      "pyfasta                              0.5.2\n",
      "pygam                                0.9.0\n",
      "Pygments                             2.17.2\n",
      "pynndescent                          0.5.11\n",
      "pyparsing                            3.1.2\n",
      "pyranges                             0.0.111\n",
      "pyrle                                0.0.39\n",
      "pysam                                0.22.0\n",
      "pyscenic                             0.12.1+7.g1cd059f\n",
      "python-dateutil                      2.9.0.post0\n",
      "python-utils                         3.8.2\n",
      "pytz                                 2024.1\n",
      "pyvis                                0.3.2\n",
      "PyYAML                               6.0.1\n",
      "pyzmq                                25.1.2\n",
      "ray                                  2.9.3\n",
      "referencing                          0.34.0\n",
      "requests                             2.31.0\n",
      "requests-cache                       1.2.0\n",
      "reretry                              0.11.8\n",
      "rich                                 13.7.1\n",
      "rich-argparse                        1.4.0\n",
      "rpds-py                              0.18.0\n",
      "scanorama                            1.7.4\n",
      "scanpy                               1.8.2\n",
      "scatac_fragment_tools                0.1.0\n",
      "scenicplus                           1.0a1\n",
      "scikit-image                         0.22.0\n",
      "scikit-learn                         1.3.2\n",
      "scipy                                1.12.0\n",
      "scrublet                             0.2.3\n",
      "seaborn                              0.13.2\n",
      "setuptools                           65.5.0\n",
      "sinfo                                0.3.4\n",
      "six                                  1.16.0\n",
      "smart-open                           6.4.0\n",
      "smmap                                5.0.1\n",
      "snakemake                            8.5.5\n",
      "snakemake-interface-common           1.17.1\n",
      "snakemake-interface-executor-plugins 8.2.0\n",
      "snakemake-interface-report-plugins   1.0.0\n",
      "snakemake-interface-storage-plugins  3.1.1\n",
      "sorted-nearest                       0.0.39\n",
      "sortedcontainers                     2.4.0\n",
      "soupsieve                            2.5\n",
      "stack-data                           0.6.3\n",
      "statistics                           1.0.3.5\n",
      "statsmodels                          0.14.1\n",
      "stdlib-list                          0.10.0\n",
      "stopit                               1.1.2\n",
      "suds-community                       1.1.2\n",
      "tables                               3.9.2\n",
      "tabulate                             0.9.0\n",
      "tblib                                3.0.0\n",
      "tenacity                             8.2.3\n",
      "texttable                            1.7.0\n",
      "threadpoolctl                        3.4.0\n",
      "throttler                            1.2.2\n",
      "tifffile                             2024.2.12\n",
      "tmtoolkit                            0.12.0\n",
      "toolz                                0.12.1\n",
      "toposort                             1.10\n",
      "tornado                              6.4\n",
      "tqdm                                 4.66.2\n",
      "traitlets                            5.14.2\n",
      "tspex                                0.6.3\n",
      "typing                               3.7.4.3\n",
      "typing_extensions                    4.10.0\n",
      "typing-inspect                       0.9.0\n",
      "umap-learn                           0.5.5\n",
      "url-normalize                        1.4.3\n",
      "urllib3                              2.2.1\n",
      "wcwidth                              0.2.13\n",
      "wheel                                0.43.0\n",
      "wrapt                                1.16.0\n",
      "xlrd                                 2.0.1\n",
      "xmltodict                            0.13.0\n",
      "xyzservices                          2023.10.1\n",
      "yarl                                 1.9.4\n",
      "yte                                  1.5.4\n",
      "zict                                 3.0.0\n",
      "zipp                                 3.18.1\n",
      "zope.event                           5.0\n",
      "zope.interface                       6.2\n",
      "\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m A new release of pip is available: \u001b[0m\u001b[31;49m24.0\u001b[0m\u001b[39;49m -> \u001b[0m\u001b[32;49m25.1.1\u001b[0m\n",
      "\u001b[1m[\u001b[0m\u001b[34;49mnotice\u001b[0m\u001b[1;39;49m]\u001b[0m\u001b[39;49m To update, run: \u001b[0m\u001b[32;49mpip install --upgrade pip\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "pip list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b0d33d6f-2836-47ed-bbfe-f1d51a3f7402",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "pycistopic_qc_may24",
   "language": "python",
   "name": "pycistopic_2a"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
